{"cells":[{"cell_type":"markdown","metadata":{"id":"w_pilFRTQ-Ry"},"source":["# IGVF MPRA workshop MPRAsnakeflow tutorial\n","\n","In this tutorial we will go over the results of MPRAsnakeflow on the example data provided [in this repository](https://github.com/kircherlab/MPRAsnakeflow_tutorial/tree/main)\n","It is not necessary to run MPRAsnakeflow in advance as the resulting files are shared in the Drive."]},{"cell_type":"markdown","metadata":{"id":"W9Rx9D0CSauI"},"source":["To start working on this tutorial in Colab, save a copy on your Google Drive (`File` $\\rightarrow$ `Save a copy in Drive`).\n","\n","To work on the tutorial locally, you can download this assignment as an IPython Notebook (`File` $\\rightarrow$ `Download` $\\rightarrow$ `Download .ipynb`). To work locally, you need to have Jupyter installed (ideally in a conda environment as discribed [here](https://test-jupyter.readthedocs.io/en/latest/install.html#id3))."]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":20844,"status":"ok","timestamp":1709212502932,"user":{"displayName":"Jonathan Rosen","userId":"05103865168852203023"},"user_tz":-60},"id":"iQMtBNQ3RQTk","outputId":"838a848b-abd0-449e-a228-4c43a6f919ca"},"outputs":[{"name":"stdout","output_type":"stream","text":["Mounted at /content/gdrive\n"]}],"source":["import matplotlib.pyplot as plt\n","import os\n","import cv2\n","from google.colab.patches import cv2_imshow\n","from google.colab import drive\n","drive.mount('/content/gdrive', force_remount=True)\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"5ixPxO-41Ug5"},"outputs":[],"source":["data_dir = \"/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data\""]},{"cell_type":"markdown","metadata":{"id":"fImhaLcbXnnK"},"source":["First, let's take a look at the structure of our input files:"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":1836,"status":"ok","timestamp":1709212510121,"user":{"displayName":"Jonathan Rosen","userId":"05103865168852203023"},"user_tz":-60},"id":"zXe3NjazVjal","outputId":"da624a45-48f2-456e-b56e-3f5f7601e68e"},"outputs":[{"name":"stdout","output_type":"stream","text":["/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data:\n","assignment  counts  design  experiment.csv  workshop_config.yaml\n","\n","/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/assignment:\n","assoc_bc.fastq.gz  assoc_fwd.fastq.gz  assoc_rev.fastq.gz\n","\n","/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/counts:\n","dna-1_S1_R1_001.fastq.gz  dna-3_S3_R2_001.fastq.gz  rna-2_S5_R2_001.fastq.gz\n","dna-1_S1_R2_001.fastq.gz  dna-3_S3_R3_001.fastq.gz  rna-2_S5_R3_001.fastq.gz\n","dna-1_S1_R3_001.fastq.gz  rna-1_S4_R1_001.fastq.gz  rna-3_S6_R1_001.fastq.gz\n","dna-2_S2_R1_001.fastq.gz  rna-1_S4_R2_001.fastq.gz  rna-3_S6_R2_001.fastq.gz\n","dna-2_S2_R2_001.fastq.gz  rna-1_S4_R3_001.fastq.gz  rna-3_S6_R3_001.fastq.gz\n","dna-2_S2_R3_001.fastq.gz  rna-1_S6_R3_001.fastq.gz\n","dna-3_S3_R1_001.fastq.gz  rna-2_S5_R1_001.fastq.gz\n","\n","/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/design:\n","workshop_design.fa  workshop_labels.tsv\n"]}],"source":["!ls -R \"{data_dir}\""]},{"cell_type":"markdown","metadata":{"id":"RUAgXK3djNHA"},"source":["----\n","## Assignment step (optional)"]},{"cell_type":"markdown","metadata":{"id":"vKiQfJ_fYcT4"},"source":["To eventually analyze your MPRA data, you need to know which barcode goes with which tested sequence. This is called assignment and typically associated multiple barcodes (i.e. the sequences that you will later count in the MPRA experiment) to a target sequence. It is possible that you already designed your assignment in advance (e.g. have barcodes synthezied with the targeted sequences) and already have a file which holds for each of the designed sequences the assignment to the barcodes. If this is the case you do not need to run the assignment step again and can run the experiment workflow with the file you already have.\\\n","For the assignment step, we will need the files in the assignment folder (sequencing of barcodes and of the target sequences), as well as the config file and the design fasta (target sequences) in the design folder. We will take a look at these files now. The structure of a fastq file is as follows:  \n","\n","\n","```\n","@Sequence ID 1  \n","Sequence 1  \n","+(Sequence ID 1)  \n","Quality score string of Sequence 1  \n","@Sequence ID 2  \n","Sequence 2  \n","+(Sequence ID 2)\n","Quality score string of sequence 2  \n","...\n","```\n"]},{"cell_type":"code","execution_count":1,"metadata":{"id":"rLnT-YxoYQCq"},"outputs":[{"name":"stdout","output_type":"stream","text":["First two barcodes in our barcode sequencing file:\n","\n","gzip: {data_dir}/assignment/assoc_bc.fastq.gz: No such file or directory\n","\n","First two sequences in our forward sequencing file:\n","\n","gzip: {data_dir}/assignment/assoc_fwd.fastq.gz: No such file or directory\n"]}],"source":["print(\"First two barcodes in our barcode sequencing file:\\n\")\n","!zcat \"{data_dir}/assignment/assoc_bc.fastq.gz\" | head -n 8\n","\n","print(\"\\nFirst two sequences in our forward sequencing file:\\n\")\n","!zcat \"{data_dir}/assignment/assoc_fwd.fastq.gz\" | head -n 8"]},{"cell_type":"markdown","metadata":{"id":"X3WlUsyYaC-I"},"source":["If you take a good look at these files, you can see the sequence IDs in lines starting with the \"@\". By definition the sequence ID ends at the first white space (remainder is considered a comment). This sequence ID is identical between the different files, i.e. the forward, reverse and barcode reads from the same molecule are in the same order. From the combination of these files, we can see which barcode is associated with which sequenced target (forward/reverse reads). These sequenced forward/reverse reads should therefore correspond to the sequences that we designed. The designed sequences can be found in our design fasta, which contains the following information:  \n","\n","```\n",">oligo ID 1  \n","Designed sequence 1\n",">oligo ID 2  \n","Designed sequence 2\n","...\n","```\n","\n","\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"agU1VZaDbbET"},"outputs":[],"source":["!head \"{data_dir}/design/workshop_design.fa\""]},{"cell_type":"markdown","metadata":{"id":"jfpzTs8ieFTp"},"source":["\"Oligos\" are the target sequences that we designed in advance, i.e. the sequences we would like to see in our assignment sequencing files (fastq files). In the association step, our goal is to assign the sequences to the oligos, since as you can see the oligo ID and sequence ID are not matching. By matching the reads to our designed oligos, we also automatically link the sequenced barcodes to our designed oligos, which is essential for us later."]},{"cell_type":"markdown","metadata":{"id":"8_3ls_TTjdKy"},"source":["Let's take a look at the part of the config file that concerns the assignment part."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"d6aNL38dkNKf"},"outputs":[],"source":["!sed '/experiments/Q' \"{data_dir}/workshop_config.yaml\""]},{"cell_type":"markdown","metadata":{"id":"giNFGgB62jY8"},"source":["As you can see this part of the config has three different positions for fastq files. Namely, `FW`, `BC` and `REV`, the abbreviations for forward, barcode and reverse reads. Thus, the corresponding files should be added here.\n","\n"]},{"cell_type":"markdown","metadata":{"id":"Fk03hXfAIyqf"},"source":["As a sanity check and to understand where these values come from, we will take a look at if the information matches our input files. We used 15 bp adapters for this MPRA experiment, which were added at both ends of each designed oligo. The ``alignment_start`` configuration refers to these adapters as our actual target sequences only starts after the 15 bp of adapter sequence."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"xzG6eoZ5HG_L"},"outputs":[],"source":["adapter_length=15\n","\n","print(\"\\nAverage length of a barcode:\")\n","!zcat \"{data_dir}/assignment/assoc_bc.fastq.gz\" | awk 'NR % 4 == 2 {{sum += length($$0)}} END {{print sum*4/NR}}'\n","\n","print(\"\\nAverage length of a target sequence (oligo length - 2 x adapter):\")\n","seq_length_with_bc = !cat \"{data_dir}/design/workshop_design.fa\" | awk 'NR % 4 == 2 {{sum +=length($$0)}} END {{print sum*4/NR}}'\n","seq_length_with_bc = int(seq_length_with_bc[0])\n","seq_length = seq_length_with_bc - 2*adapter_length\n","print(seq_length)"]},{"cell_type":"markdown","metadata":{"id":"HgmPA67ZC02X"},"source":["Finally, the ``min_support`` and ``fraction`` are necessary to resolve what to do in the situation where the same barcode sequence is observed with multiple oligos. In this scenario and using the given ``standardConfig`` setting, the barcode will be assigned to its most frequently observed oligo if the barcode was observed at least three times and if the most frequent oligo explained at least 70% of the occurances of this barcode."]},{"cell_type":"markdown","metadata":{"id":"15n1pacbIgPX"},"source":["### Assignment step results\n","\n","Since running MPRAsnakeflow takes longer than the time scheduled for this tutorial, we uploaded some of the relevant output files to this drive. Just for the curious people among you, for the example data the pipeline takes 1h using just one core for both tasks, assignment and experiment. The results folder structure should be the same as after running the assignment step using our example config file (explained above)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"7VyJn-GmI1so"},"outputs":[],"source":["# Let's take a look at the files in the assignment results folder:\n","results_dir = \"/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_results/assignment\"\n","!ls -R \"{results_dir}\"\n"]},{"cell_type":"markdown","metadata":{"id":"GUf58DsuKpno"},"source":["The statistic folder contains run statistics for the assignment step, such as the total number of associated barcounts, the number of oligos with at least 15 barcodes, and more. We will now just take a look at the generated count histogram of barcodes per oligo."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"8wx2Y8ygKnxT"},"outputs":[],"source":["img = plt.imread(os.path.join(results_dir, \"assignMPRAworkshop/statistic/assignment.standardConfig.png\"))\n","plt.imshow(img)\n","plt.axis(\"off\")"]},{"cell_type":"markdown","metadata":{"id":"Ofe2u1eEMUyA"},"source":["The median number of barcodes is 24. We usually require each tested sequence to have at least 10 barcodes associated.\n","\n","We will now take a look at the actual output file of the assignment run. Which is a sorted table of `barcodes`, their assigned `sequence IDs`, information about the `quality` of the alignment (fwd/rev reads to designed oligos) and how often this barcode was assigned to this sequence compared to the total number this barcode was observed in the experiment. The information in the quality field is derived from the SAM/BAM output (https://samtools.github.io/hts-specs/SAMv1.pdf) and represents a concatenation of the SAM position, cigar, NM, MD and mapQ."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"7GewWfZvMsTm"},"outputs":[],"source":["assignment = os.path.join(results_dir, \"assignMPRAworkshop/assignment_barcodes.standardConfig.sorted.tsv.gz\")\n","!zcat {assignment} | head"]},{"cell_type":"markdown","metadata":{"id":"LvxcwwaW8HYW"},"source":["We would like to know how many different oligos have been included in the final assignment.\\\n","When we compare this number to the number of sequences in our design, we can get a feeling of how many sequences were able to get assigned with their barcodes. This has implications for the proportion of targets that we can eventually analyse and whether each target is supported with sufficiently many barcodes. This information is important for the experiment overall and analyzing the counts of each individual replicate."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"e8ljhyVLCzVF"},"outputs":[],"source":["assigned_sequences = !zcat /content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_results/assignment/assignMPRAworkshop/assignment_barcodes.standardConfig.sorted.tsv.gz | awk -F'\\t' '{names[$2]++} END {print length(names)}'\n","\n","assigned_sequences = int(assigned_sequences[0])\n","print(\"Number of sequences we were able to assign: %s\"%(assigned_sequences))\n","\n","designed_sequences = !cat \"{data_dir}/design/workshop_design.fa\" | grep -v \">\" | wc -l\n","designed_sequences = int(designed_sequences[0])\n","print(\"Number of sequences in the initial design: %s\"%(designed_sequences))\n","\n","print(r\"Conclusion: %s%% sequences of our initial design can be used in the experimental/counting step\"%(round(assigned_sequences/designed_sequences * 100, 1)))"]},{"cell_type":"markdown","metadata":{"id":"f24iQgU3EZ3W"},"source":["-----\n","## Experiment step\n","\n","For the experiment step, where the counts per oligo are calculated, we need all the files in the MPRAsnakeflow_data/counts folder, as well as the config file and the barcode assignment file, which is the output of the assignment step.\n","\n","We investigated the assignment file in the end of the previous section. This table is now used to count the number of observed barcodes for each target sequence.\n","\n","First we will cover the config file and afterwards we will look at interesting results like the correlation of the DNA and RNA counts among the replicates. This is useful in practice to check if the data preparation went well, because if the resulting data is not correlating between replicates, problems during the experiment will introduce a bias in the results of our analysis that we would like to know about."]},{"cell_type":"markdown","metadata":{"id":"ilitAMXmcsoa"},"source":["The following block shows the part of the config file that concerns the experiment part. Since assignment and experiment part can be run independently the `bc_length` needs to be set again.\n","\n","The `experiment_file` holds the path to the file of the experiment table (i.e. condition, replicate and name of read files given)."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"INKcJ46qGMMc"},"outputs":[],"source":["!sed -n '/experiments:/,$p' \"{data_dir}/workshop_config.yaml\""]},{"cell_type":"markdown","metadata":{"id":"U2sYKkmOtFH8"},"source":["We can once again take a look at the input files, to understand where these cnofigurations come from. Each count experiment, in our case, is described using three files (the paths of which are in the experiment.csv file): forward reads, reverse reads, and UMIs. Note that the UMIs are optional, and serve to deduplicate barcodes with overdispersed counts due to preferential amplification related to GC content or library preparation steps. From one of the UMI read files, we can see that indeed our UMI length is 16:"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"lf-Xm-rCs_hL"},"outputs":[],"source":["print(\"\\nAverage length of a UMI:\")\n","!zcat \"{data_dir}/counts/dna-1_S1_R2_001.fastq.gz\" | awk 'NR % 4 == 2 {{sum += length($$0)}} END {{print sum*4/NR}}'"]},{"cell_type":"markdown","metadata":{"id":"nxbmlm1baUHY"},"source":["We can take a look at the amount of barcodes that have been removed due to UMI collision, by checking how many unique BC x UMI combinations we have in our data. (We are now checking the RNA counts in replicate 1, but we should look at both DNA and RNA counts in all three replicates.)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"E-S7BxgiLyoX"},"outputs":[],"source":["results_dir = \"/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_results/experiments\"\n","print(\"First few lines of a count file, where column 1 contains the barcode, 2 the UMI and 3 the count.\")\n","!zcat \"{results_dir}\"/workshop_data/counts/HepG2_1_RNA_filtered_counts.tsv.gz | head\n","\n","print(\"\\nAmount of total BC x UMI combinations:\")\n","!zcat \"{results_dir}\"/workshop_data/counts/HepG2_1_RNA_filtered_counts.tsv.gz | wc -l\n","\n","print(\"\\nAmount of duplicate BC x UMI combinations (the file is sorted):\")\n","!zcat \"{results_dir}\"/workshop_data/counts/HepG2_1_RNA_filtered_counts.tsv.gz | uniq -d | wc -l"]},{"cell_type":"markdown","metadata":{"id":"KqvIJ7piaDgv"},"source":["There are no collisions, indicating that this subset of the data is of high quality. Great!"]},{"cell_type":"markdown","metadata":{"id":"staLIjklrpgN"},"source":["The count files are then joined with the assignment file (the output of the assignment step) to assign counts to oligos. The barcode counts are then aggregated over the oligos. This is done within MPRAsnakeflow with additional filter steps, but to give you a feeling of what's happening we give you some simplified code below:"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"G1A9kb3Yr8SC"},"outputs":[],"source":["counts_rep1 = os.path.join(results_dir, \"workshop_data/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz\")\n","print(\"Joining the assignment and the counts of replicate 1:\\n\\nBarcode\\toligo\\tDNA\\tRNA\")\n","!join <(zcat {assignment}) <(zcat {counts_rep1}) -t $'\\t' | cut -f 1,2,5-6 | head\n","\n","print(\"\\nAggregating the barcodes over oligos and the counts of replicate 1:\\n\\noligo\\tDNA\\tRNA\")\n","!join <(zcat {assignment}) <(zcat {counts_rep1}) -t $'\\t' | cut -f 1,2,5-6 | sort -k2,2 | \\\n"," awk -v OFS=\"\\t\" '{{if ($$2 == prev) {{dna += $$3; rna += $$4}} else {{ if (NR > 1) print prev,dna,rna; prev=$$2; dna=$$3;rna=$$4}}}}' | head"]},{"cell_type":"markdown","metadata":{"id":"b8eZ096druwB"},"source":["In the next steps, barcodes are already aggregated to their oligos/targets.  \n","  \n","We will have a look at the correlation plots between the replicates for DNA and RNA now. Typically, we will sequence RNA counts deeper than DNA -- because of the larger dynamic range of RNA. We therefore expect that the correlation of RNA counts is higher than that of the DNA. Our example data shows this trend as well.\n","\n","We have a correlation of DNA counts per insert between the replicates of 0.27 and 0.31. This correlation is for a subset of the complete MPRA experiment. The shown output uses the user set minThreshold for removing barcodes with too few counts. In order to see the influence of the parameter another plot is generated without this threshold."]},{"cell_type":"markdown","metadata":{"id":"oVgp5AyhjLlM"},"source":["With the threshold of 5 barcodes per oligo and at least one DNA and RNA count\n","- DNA correlation: 0.28 - 0.31\n","- RNA correlation: 0.68 - 0.70"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"gKRtX0Tzuovc"},"outputs":[],"source":["dna_pairwise = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png')\n","\n","dna_pairwise_resize = cv2.resize(dna_pairwise, (600, 1000))\n","\n","cv2_imshow(dna_pairwise_resize)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"HYgnMN_yj0Tc"},"outputs":[],"source":["rna_pairwise = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png')\n","\n","rna_pairwise_resize = cv2.resize(rna_pairwise, (600, 1000))\n","\n","cv2_imshow(rna_pairwise_resize)"]},{"cell_type":"markdown","metadata":{"id":"MfouM3Z1j03u"},"source":["And without these thresholds:\n","- DNA correlation: 0.25 - 0.27\n","- RNA correlation: 0.58 - 0.61"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"eUr5yMsmjWms"},"outputs":[],"source":["dna_pairwise_no_thresh = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png')\n","\n","dna_pairwise_no_thresh_resize = cv2.resize(dna_pairwise_no_thresh, (600, 1000))\n","\n","cv2_imshow(dna_pairwise_no_thresh_resize)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"KvvbPctVLwpp"},"outputs":[],"source":["rna_pairwise_no_thresh = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png')\n","\n","rna_pairwise_no_thresh_resize = cv2.resize(rna_pairwise_no_thresh, (600, 1000))\n","\n","cv2_imshow(rna_pairwise_no_thresh_resize)"]},{"cell_type":"markdown","metadata":{"id":"bVbVEhcfQOgt"},"source":["In the following two figures you can see that this trend is continues for a threshold of 10 barcodes per insert.\n","- DNA correlation: 0.31 - 0.39\n","- RNA correlation: 0.79 - 0.80"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"pfzO2adJOCRO"},"outputs":[],"source":["dna_pairwise_no_thresh = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig_10bc/HepG2_DNA_pairwise_minThreshold.png')\n","\n","dna_pairwise_no_thresh_resize = cv2.resize(dna_pairwise_no_thresh, (600, 1000))\n","\n","cv2_imshow(dna_pairwise_no_thresh_resize)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SDkSrN_PNwqY"},"outputs":[],"source":["rna_pairwise_no_thresh = cv2.imread(f'{results_dir}/workshop_data/statistic/assigned_counts/MPRAworkshop/tutorialConfig_10bc/HepG2_RNA_pairwise_minThreshold.png')\n","\n","rna_pairwise_no_thresh_resize = cv2.resize(rna_pairwise_no_thresh, (600, 1000))\n","\n","cv2_imshow(rna_pairwise_no_thresh_resize)"]},{"cell_type":"markdown","metadata":{"id":"mOqUIe3ec49b"},"source":["Even though it is a small example, the minimal barcode count threshold shows a clear effect on correlation. However, we also note that we are losing a proprotion of oligos with these strict filters and thereby limit the conclusions that we can draw from the data."]},{"cell_type":"markdown","metadata":{"id":"r5GjpJc8mgjw"},"source":["Now we want go through the final output of MPRAsnakeflow: the count table of sequences and barcodes together with their aggregated log2 values. Note that there are no statistics calculated at this step yet, the table doesn't say anything about certainty/variance of the activity. The log2 value refers to the log2 ratio of RNA counts over DNA counts for each oligo, which is a measurement of the activity of the target sequence. More on this in the statistical analysis tutorial."]},{"cell_type":"code","execution_count":null,"metadata":{"id":"WpXNmTZdy43C"},"outputs":[],"source":[" !zcat {results_dir}/workshop_data/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz | head"]},{"cell_type":"markdown","metadata":{"id":"sgXawcNS2vyM"},"source":["This is also the file that you will be using in the next 45 minutes. Don't hesitate to ask questions, and we hope you enjoyed this small introduction to MPRAsnakeflow."]}],"metadata":{"colab":{"provenance":[]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.12.4"}},"nbformat":4,"nbformat_minor":0}
