{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/kircherlab/MPRAsnakeflow_tutorial/blob/development/tutorial_association.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# MPRAsnakeflow association tutorial"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "import matplotlib.pyplot as plt\n",
        "import os\n",
        "import cv2\n",
        "from google.colab.patches import cv2_imshow\n",
        "from google.colab import drive\n",
        "drive.mount('/content/gdrive', force_remount=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "gdrive_dir = \"/content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data\""
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "First, let's take a look at the structure of our input files:"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "!ls -R \"{gdrive_dir}\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%bash\n",
        "mkdir -p /content/work\n",
        "cd /content/work\n",
        "ln -s /content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/assignment_config.yaml\n",
        "ln -s /content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/design\n",
        "ln -s /content/gdrive/MyDrive/MPRA-IGVF-Workshop-2024/Tutorials/T2_MPRA_stats/MPRAsnakeflow_data/assignment"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uAAYJQqv9zIk"
      },
      "source": [
        "## Pre-requirements"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dOsEkOFu-MbM"
      },
      "source": [
        "### Docker\n",
        "Docker using udocker in colab from github.com/drengskapur/docker-in-colab."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NSTDSHTN4JJk",
        "outputId": "2917873e-6493-43d8-8124-cbb787e0ae8c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Docker-in-Colab 1.1.0\n",
            "\n",
            "Usage:     udocker(\"--help\")\n",
            "Examples:  https://github.com/indigo-dc/udocker?tab=readme-ov-file#examples\n"
          ]
        }
      ],
      "source": [
        "# Copyright 2024 Drengskapur\n",
        "#\n",
        "# Licensed under the Apache License, Version 2.0 (the \"License\");\n",
        "# you may not use this file except in compliance with the License.\n",
        "# You may obtain a copy of the License at\n",
        "#\n",
        "#     http://www.apache.org/licenses/LICENSE-2.0\n",
        "#\n",
        "# Unless required by applicable law or agreed to in writing, software\n",
        "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
        "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
        "# See the License for the specific language governing permissions and\n",
        "# limitations under the License.\n",
        "#\n",
        "# @title {display-mode:\"form\"}\n",
        "# @markdown <br/><br/><center><img src=\"https://cdn.jsdelivr.net/gh/drengskapur/docker-in-colab/assets/docker.svg\" height=\"150\"><img src=\"https://cdn.jsdelivr.net/gh/drengskapur/docker-in-colab/assets/colab.svg\" height=\"150\"></center><br/>\n",
        "# @markdown <center><h1>Docker in Colab</h1></center><center>github.com/drengskapur/docker-in-colab<br/><br/><br/><b>udocker(\"run hello-world\")</b></center><br/>\n",
        "def udocker_init():\n",
        "    import os\n",
        "    if not os.path.exists(\"/home/user\"):\n",
        "        !pip install udocker > /dev/null\n",
        "        !udocker --allow-root install > /dev/null\n",
        "        !useradd -m user > /dev/null\n",
        "    print(f'Docker-in-Colab 1.1.0\\n')\n",
        "    print(f'Usage:     udocker(\"--help\")')\n",
        "    print(f'Examples:  https://github.com/indigo-dc/udocker?tab=readme-ov-file#examples')\n",
        "\n",
        "    def execute(command: str):\n",
        "        user_prompt = \"\\033[1;32muser@pc\\033[0m\"\n",
        "        print(f\"{user_prompt}$ udocker {command}\")\n",
        "        !su - user -c \"udocker $command\"\n",
        "\n",
        "    return execute\n",
        "\n",
        "udocker = udocker_init()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4WYuqwGr-fEp"
      },
      "source": [
        "### snakemake\n",
        "\n",
        "snakemake v8.16.0 using the official docker image."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZZv9f_eu4bB9",
        "outputId": "677e86d8-e508-45db-9153-031cf0004740"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ udocker pull visze/snakemake_mprasnakeflow:0.1.0\n",
            "Info: downloading layer sha256:ca895cfad24f83bcf15712d53a79a7ca7afd230c1044c3e46e59b2a6ee8fd4e5\n",
            "Info: downloading layer sha256:6a1ace6fb4a7b40b0f1019fa1a3225d2970f62518b41c272b3b462acc0e72d4f\n",
            "Info: downloading layer sha256:030e57b99b3b4f88a0d85eea59fcef5b8d7f7301a9992c88f9cab73191ad554f\n",
            "Info: downloading layer sha256:d7e123fc54cd9ae8ee4d671f0f7b134486532e5f07de84d09cb3e4a521f2b848\n",
            "Info: downloading layer sha256:2241bf831ea24afc27d2f4974c716a3e8bd7dee9f213c89fb93e8bb25d4f14cd\n",
            "Info: downloading layer sha256:c6e539d6d31625c3c6ec4170ceeaa960c9779c60d4839e3ef0d5cd5251331943\n",
            "Info: downloading layer sha256:e71e00cb9af4e3c09ee580b9c031c586c30fa46d4934445a20c36e070d37023a\n",
            "Info: downloading layer sha256:fee224ae51e3aff735790e41da3a59005e455f38f97062961a57e193ff19de11\n",
            "Info: downloading layer sha256:5466efd28f652843211c2f22d358d79cd7fc127dcaae1d2de5a5cbc09e9bfd24\n",
            "Info: downloading layer sha256:0ebeb0a01cdfb95005934857d1270693f20340315f7f7294f144bd4bb818f723\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:2cfc827e1b091e4798313adbfff1198ebdaaa6a964983be8f7178320383ce046\n",
            "Info: downloading layer sha256:1ea0c37c519139520d65d3f877e039d2eec4b1464a4d9699960b8ec2d67d89c1\n",
            "Info: downloading layer sha256:002030c168895ab3d4fccccf8a6a2acd4912899542f323c749d12d4f0a678c33\n",
            "Info: downloading layer sha256:d1223905c3279266f2e5e0baca4e47377e231ab87a695a8cedea811ba85033e5\n",
            "Info: downloading layer sha256:37f75da0042b20eb2076c6c4e10b47b1d87758899293b5d1aab45be4c5955b91\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:19215b5611be3052e1a6a68d92a9752a7b9b4df5b3ebb7dec3aa818e4e9af617\n",
            "Info: downloading layer sha256:b5381d1371f0f88dbd1886992e5c66a49a1347034cc483c768d1ef5a6fe80da6\n",
            "Info: downloading layer sha256:90ca98f3e719e87efca54fd8fc08b1f2f02ddf8f7d84ccd9364482ba61ec54b9\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:b349bcb0e87e5bf44760044cbce8f10943218f8f193de7ecd89703b3204b2dd8\n",
            "Info: downloading layer sha256:2930e754fbf462279f5e7c4ec8958e17bc48cf973c783ae4b4d58f05b82912a6\n",
            "Info: downloading layer sha256:f11c1adaa26e078479ccdd45312ea3b88476441b91be0ec898a7e07bfd05badc\n"
          ]
        }
      ],
      "source": [
        "udocker(\"pull visze/snakemake_mprasnakeflow:0.1.0\")\n",
        "udocker(\"create --name=mprasnakeflow visze/snakemake_mprasnakeflow:0.1.0\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "zYhKnwlV-orz"
      },
      "source": [
        "Try to run `snakemake --version` in docker:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "jQi5e3634sDP",
        "outputId": "c59fd326-1457-4c85-afb7-a98453d7ed80"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ udocker run visze/snakemake_mprasnakeflow:0.1.0 snakemake --version\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 5840c75d-aabb-3605-ad0f-2eccc7bae286                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: _entrypoint.sh\n",
            "8.16.0+0.g838b2fce.dirty\n"
          ]
        }
      ],
      "source": [
        "udocker(\"run mprasnakeflow snakemake --version\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Mb0rhjfo8dbA"
      },
      "source": [
        "## Assignment input data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "BGLsBxpyst9V"
      },
      "outputs": [],
      "source": [
        "%%bash\n",
        "mkdir -p assoc_basic/data\n",
        "cd assoc_basic/data\n",
        "curl -s ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4237nnn/GSM4237954/suppl/GSM4237954_9MPRA_elements.fa.gz | \\\n",
        "zcat - |awk '{ count+=1; if (count == 1) { print } else { print substr($1,1,171)}; if (count == 2) { count=0 } }' | awk '{gsub(/[\\]\\[]/,\"_\")} $0' > design.fa\n",
        "\n",
        "wget -q -O SRR10800986_1.fastq.gz https://kircherlab.bihealth.org/download/ali/SRR10800986_1.200K.fastq.gz\n",
        "wget -q -O SRR10800986_2.fastq.gz https://kircherlab.bihealth.org/download/ali/SRR10800986_2.200K.fastq.gz\n",
        "wget -q -O SRR10800986_3.fastq.gz https://kircherlab.bihealth.org/download/ali/SRR10800986_3.200K.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6GilAsqZ4t9r"
      },
      "source": [
        "#### Complete data"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9BLFl8GF_Lgr"
      },
      "source": [
        "Install `sra-tools` for fetchicng assignment data from SRA."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XJd25fFG-zBl",
        "outputId": "e8d7bfca-c35d-4f4a-c03b-4c5f4834f1d1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Preparing transaction: ...working... done\n",
            "Verifying transaction: ...working... done\n",
            "Executing transaction: ...working... done\n"
          ]
        }
      ],
      "source": [
        "!mamba install -y -qq -c bioconda -c conda-forge sra-tools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Iyhf2Z5ms1sN",
        "outputId": "5fb44139-98cf-40e4-b570-bcd8f61be2fe"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Process is interrupted.\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "cd assoc_basic/data\n",
        "prefetch SRR10800986\n",
        "fastq-dump --gzip --split-files SRR10800986"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "893Sh49a9u7M"
      },
      "source": [
        "### MPRAsnakeflow workflow"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "oFHG7CLihweM",
        "outputId": "2be96230-6641-48cc-eecc-61f0291afbbc"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Branch 'development' set up to track remote branch 'development' from 'origin'.\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Cloning into 'MPRAsnakeflow'...\n",
            "Switched to a new branch 'development'\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "cd /content\n",
        "git clone https://github.com/kircherlab/MPRAsnakeflow.git\n",
        "cd MPRAsnakeflow && git checkout development"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "nVLiDi7eCW-r"
      },
      "outputs": [],
      "source": [
        "!mkdir -p /data\n",
        "!mkdir -p /content/apptainer\n",
        "!ln -s /content/MPRAsnakeflow /data/MPRAsnakeflow"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0KqdtmOb_gHg"
      },
      "source": [
        "## Run MPRAsnakeflow\n",
        "\n",
        "### Assignment data\n",
        "\n",
        "/content/MPRAsnakeflow/resources/assoc_basic/config.yml\n",
        "\n",
        "First a dry run `-n` just showing the overview of excecuted rules `--quiet rules` using one core `-c 1`.\n",
        "\n",
        "We have to let snakemake know the main `Snakefile` of MPRAsnakeflow using `--snakefile /data/MPRAsnakeflow/workflow/Snakefile`.\n",
        "\n",
        "Also the workflow with input files in configured in a YAML file and we give it as input with `--configfile /data/MPRAsnakeflow/resources/assoc_basic/config.yml`\n",
        "\n",
        " Please note that we map `/content/MPRAsnakeflow` to `/data/MPRAsnakeflow/` inside the snakemake container.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JDCndozT6FJA",
        "outputId": "f56d2be1-c424-415d-8cd7-a4975621beee"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "total 16K\n",
            "drwxr-xr-x  2 root root 4.0K Jul 31 08:25 apptainer\n",
            "drwxrwxrwx  3 root root 4.0K Jul 31 08:12 assoc_basic\n",
            "drwxr-xr-x 10 root root 4.0K Jul 31 08:30 MPRAsnakeflow\n",
            "drwxr-xr-x  1 root root 4.0K Jul 29 13:22 sample_data\n"
          ]
        }
      ],
      "source": [
        "!chmod og+w assoc_basic\n",
        "!ls -lh"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TXlOomWL5Am2",
        "outputId": "bf6d34f9-b3bc-4ed8-9f61-97ecbcdbfd2a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ udocker run   -v=/content/assoc_basic:/data/run   --workdir /data/run   visze/snakemake_mprasnakeflow:0.1.0   snakemake --sdm conda --conda-prefix /data/conda_envs   --snakefile /data/MPRAsnakeflow/workflow/Snakefile   --configfile /data/MPRAsnakeflow/resources/assoc_basic/config.yml   -c 1 -n --quiet rules\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 22247a11-3131-393c-b3b9-821915dd57a0                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: _entrypoint.sh\n",
            "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
            "\u001b[35mYour conda installation is not configured to use strict channel priorities. This is however crucial for having robust and correct environments (for details, see https://conda-forge.org/docs/user/tipsandtricks.html). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job                                    count\n",
            "-----------------------------------  -------\n",
            "all                                        1\n",
            "assignment_attach_idx                     60\n",
            "assignment_bwa_ref                         1\n",
            "assignment_check_design                    1\n",
            "assignment_collect                         1\n",
            "assignment_collectBCs                      1\n",
            "assignment_fastq_split                     3\n",
            "assignment_filter                          1\n",
            "assignment_flagstat                        1\n",
            "assignment_getBCs                         30\n",
            "assignment_idx_bam                         1\n",
            "assignment_mapping_bwa                    30\n",
            "assignment_merge                          30\n",
            "assignment_statistic_assignedCounts        1\n",
            "assignment_statistic_assignment            1\n",
            "assignment_statistic_totalCounts           1\n",
            "total                                    164\n",
            "\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "udocker(\"run \\\n",
        "  -v=${PWD}/assoc_basic:/data/run \\\n",
        "  --workdir /data/run \\\n",
        "  visze/snakemake_mprasnakeflow:0.1.0 \\\n",
        "  snakemake --sdm conda --conda-prefix /data/conda_envs \\\n",
        "  --snakefile /data/MPRAsnakeflow/workflow/Snakefile \\\n",
        "  --configfile /data/MPRAsnakeflow/resources/assoc_basic/config.yml \\\n",
        "  -c 1 -n --quiet rules\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yqaXIqIM_5uw",
        "outputId": "5f67cd1b-1ed1-4731-9984-b66416e19e81"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ udocker run   -v=${PWD}/assoc_basic:/data/run   --workdir /data/run   visze/snakemake_mprasnakeflow:0.1.0   snakemake --sdm conda --conda-prefix /data/conda_envs   --snakefile /data/MPRAsnakeflow/workflow/Snakefile   --configfile /data/MPRAsnakeflow/resources/assoc_basic/config.yml   -c 1\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING f4f85828-4751-387c-9256-fd5b4b85bfcb                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: _entrypoint.sh\n",
            "\u001b[33mAssuming unrestricted shared filesystem usage.\u001b[0m\n",
            "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
            "\u001b[35mYour conda installation is not configured to use strict channel priorities. This is however crucial for having robust and correct environments (for details, see https://conda-forge.org/docs/user/tipsandtricks.html). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.\u001b[0m\n",
            "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
            "\u001b[33mProvided cores: 1 (use --cores to define parallelism)\u001b[0m\n",
            "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
            "\u001b[33mSingularity containers: ignored\u001b[0m\n",
            "\u001b[33mJob stats:\n",
            "job                                    count\n",
            "-----------------------------------  -------\n",
            "all                                        1\n",
            "assignment_attach_idx                     60\n",
            "assignment_bwa_ref                         1\n",
            "assignment_check_design                    1\n",
            "assignment_collect                         1\n",
            "assignment_collectBCs                      1\n",
            "assignment_fastq_split                     3\n",
            "assignment_filter                          1\n",
            "assignment_flagstat                        1\n",
            "assignment_getBCs                         30\n",
            "assignment_idx_bam                         1\n",
            "assignment_mapping_bwa                    30\n",
            "assignment_merge                          30\n",
            "assignment_statistic_assignedCounts        1\n",
            "assignment_statistic_assignment            1\n",
            "assignment_statistic_totalCounts           1\n",
            "total                                    164\n",
            "\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:17:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_check_design:\n",
            "    input: data/design.fa, /data/MPRAsnakeflow/workflow/rules/../scripts/assignment/check_design_file.py (cached)\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fxi, results/assignment/assocBasic/design_check.done, results/assignment/assocBasic/reference/reference.fa\n",
            "    log: results/logs/assignment/check_design.assocBasic.log, results/assignment/assocBasic/design_check.err\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/653b2f2da62f9e29538eeaf42dc2a489_\u001b[0m\n",
            "\u001b[33mTouching output file results/assignment/assocBasic/design_check.done.\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:17:57 2024]\u001b[0m\n",
            "\u001b[32mFinished job 8.\u001b[0m\n",
            "\u001b[32m1 of 164 steps (1%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/reference/reference.fa.fxi.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:17:57 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_1.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.FW.log\n",
            "    jobid: 7\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=FW\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/65318a22b1fa4bfcea38ef4cbe5cac2b_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:01 2024]\u001b[0m\n",
            "\u001b[32mFinished job 7.\u001b[0m\n",
            "\u001b[32m2 of 164 steps (1%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:01 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_2.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.BC.log\n",
            "    jobid: 9\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=BC\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/65318a22b1fa4bfcea38ef4cbe5cac2b_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:04 2024]\u001b[0m\n",
            "\u001b[32mFinished job 9.\u001b[0m\n",
            "\u001b[32m3 of 164 steps (2%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:04 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.3.FW.log\n",
            "    jobid: 26\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=3\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:06 2024]\u001b[0m\n",
            "\u001b[32mFinished job 26.\u001b[0m\n",
            "\u001b[32m4 of 164 steps (2%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:06 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.4.FW.log\n",
            "    jobid: 31\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:07 2024]\u001b[0m\n",
            "\u001b[32mFinished job 31.\u001b[0m\n",
            "\u001b[32m5 of 164 steps (3%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:07 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.8.FW.log\n",
            "    jobid: 51\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:07 2024]\u001b[0m\n",
            "\u001b[32mFinished job 51.\u001b[0m\n",
            "\u001b[32m6 of 164 steps (4%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.15.FW.log\n",
            "    jobid: 86\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=15\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:08 2024]\u001b[0m\n",
            "\u001b[32mFinished job 86.\u001b[0m\n",
            "\u001b[32m7 of 164 steps (4%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.17.FW.log\n",
            "    jobid: 96\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=17\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:09 2024]\u001b[0m\n",
            "\u001b[32mFinished job 96.\u001b[0m\n",
            "\u001b[32m8 of 164 steps (5%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:09 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.5.FW.log\n",
            "    jobid: 36\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=5\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:10 2024]\u001b[0m\n",
            "\u001b[32mFinished job 36.\u001b[0m\n",
            "\u001b[32m9 of 164 steps (5%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:10 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.9.FW.log\n",
            "    jobid: 56\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=9\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:11 2024]\u001b[0m\n",
            "\u001b[32mFinished job 56.\u001b[0m\n",
            "\u001b[32m10 of 164 steps (6%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:11 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.14.FW.log\n",
            "    jobid: 81\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=14\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:12 2024]\u001b[0m\n",
            "\u001b[32mFinished job 81.\u001b[0m\n",
            "\u001b[32m11 of 164 steps (7%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:12 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.7.FW.log\n",
            "    jobid: 46\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=7\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:13 2024]\u001b[0m\n",
            "\u001b[32mFinished job 46.\u001b[0m\n",
            "\u001b[32m12 of 164 steps (7%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:13 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.0.FW.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=0\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:14 2024]\u001b[0m\n",
            "\u001b[32mFinished job 6.\u001b[0m\n",
            "\u001b[32m13 of 164 steps (8%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:15 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.16.FW.log\n",
            "    jobid: 91\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=16\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 91.\u001b[0m\n",
            "\u001b[32m14 of 164 steps (9%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:16 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.13.FW.log\n",
            "    jobid: 76\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=13\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:17 2024]\u001b[0m\n",
            "\u001b[32mFinished job 76.\u001b[0m\n",
            "\u001b[32m15 of 164 steps (9%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:17 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.2.FW.log\n",
            "    jobid: 21\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=2\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:18 2024]\u001b[0m\n",
            "\u001b[32mFinished job 21.\u001b[0m\n",
            "\u001b[32m16 of 164 steps (10%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:18 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.6.FW.log\n",
            "    jobid: 41\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=6\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:18 2024]\u001b[0m\n",
            "\u001b[32mFinished job 41.\u001b[0m\n",
            "\u001b[32m17 of 164 steps (10%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:19 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.10.FW.log\n",
            "    jobid: 61\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=10\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:19 2024]\u001b[0m\n",
            "\u001b[32mFinished job 61.\u001b[0m\n",
            "\u001b[32m18 of 164 steps (11%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:19 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.18.FW.log\n",
            "    jobid: 101\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=18\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:20 2024]\u001b[0m\n",
            "\u001b[32mFinished job 101.\u001b[0m\n",
            "\u001b[32m19 of 164 steps (12%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.11.FW.log\n",
            "    jobid: 66\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=11\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:21 2024]\u001b[0m\n",
            "\u001b[32mFinished job 66.\u001b[0m\n",
            "\u001b[32m20 of 164 steps (12%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:21 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.1.FW.log\n",
            "    jobid: 16\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=1\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:22 2024]\u001b[0m\n",
            "\u001b[32mFinished job 16.\u001b[0m\n",
            "\u001b[32m21 of 164 steps (13%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:22 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.19.FW.log\n",
            "    jobid: 106\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=19\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:23 2024]\u001b[0m\n",
            "\u001b[32mFinished job 106.\u001b[0m\n",
            "\u001b[32m22 of 164 steps (13%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:23 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.12.FW.log\n",
            "    jobid: 71\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=12\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:24 2024]\u001b[0m\n",
            "\u001b[32mFinished job 71.\u001b[0m\n",
            "\u001b[32m23 of 164 steps (14%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:24 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.24.FW.log\n",
            "    jobid: 131\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=24\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:24 2024]\u001b[0m\n",
            "\u001b[32mFinished job 131.\u001b[0m\n",
            "\u001b[32m24 of 164 steps (15%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:25 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.22.FW.log\n",
            "    jobid: 121\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=22\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:25 2024]\u001b[0m\n",
            "\u001b[32mFinished job 121.\u001b[0m\n",
            "\u001b[32m25 of 164 steps (15%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:25 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.29.FW.log\n",
            "    jobid: 156\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=29\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:26 2024]\u001b[0m\n",
            "\u001b[32mFinished job 156.\u001b[0m\n",
            "\u001b[32m26 of 164 steps (16%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:26 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.25.FW.log\n",
            "    jobid: 136\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=25\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:27 2024]\u001b[0m\n",
            "\u001b[32mFinished job 136.\u001b[0m\n",
            "\u001b[32m27 of 164 steps (16%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:27 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.26.FW.log\n",
            "    jobid: 141\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=26\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:28 2024]\u001b[0m\n",
            "\u001b[32mFinished job 141.\u001b[0m\n",
            "\u001b[32m28 of 164 steps (17%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:28 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.21.FW.log\n",
            "    jobid: 116\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=21\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:29 2024]\u001b[0m\n",
            "\u001b[32mFinished job 116.\u001b[0m\n",
            "\u001b[32m29 of 164 steps (18%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:29 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.20.FW.log\n",
            "    jobid: 111\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=20\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:30 2024]\u001b[0m\n",
            "\u001b[32mFinished job 111.\u001b[0m\n",
            "\u001b[32m30 of 164 steps (18%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:30 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.23.FW.log\n",
            "    jobid: 126\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=23\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:31 2024]\u001b[0m\n",
            "\u001b[32mFinished job 126.\u001b[0m\n",
            "\u001b[32m31 of 164 steps (19%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:31 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.27.FW.log\n",
            "    jobid: 146\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=27\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:32 2024]\u001b[0m\n",
            "\u001b[32mFinished job 146.\u001b[0m\n",
            "\u001b[32m32 of 164 steps (20%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:32 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.28.FW.log\n",
            "    jobid: 151\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=28\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:33 2024]\u001b[0m\n",
            "\u001b[32mFinished job 151.\u001b[0m\n",
            "\u001b[32m33 of 164 steps (20%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:33 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_3.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.REV.log\n",
            "    jobid: 11\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=REV\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/65318a22b1fa4bfcea38ef4cbe5cac2b_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:37 2024]\u001b[0m\n",
            "\u001b[32mFinished job 11.\u001b[0m\n",
            "\u001b[32m34 of 164 steps (21%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:37 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.8.REV.log\n",
            "    jobid: 52\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:38 2024]\u001b[0m\n",
            "\u001b[32mFinished job 52.\u001b[0m\n",
            "\u001b[32m35 of 164 steps (21%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:38 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split8.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.8.log.gz\n",
            "    jobid: 50\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:39 2024]\u001b[0m\n",
            "\u001b[32mFinished job 50.\u001b[0m\n",
            "\u001b[32m36 of 164 steps (22%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split8.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:39 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.17.REV.log\n",
            "    jobid: 97\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=17\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:40 2024]\u001b[0m\n",
            "\u001b[32mFinished job 97.\u001b[0m\n",
            "\u001b[32m37 of 164 steps (23%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:40 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split17.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.17.log.gz\n",
            "    jobid: 95\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:41 2024]\u001b[0m\n",
            "\u001b[32mFinished job 95.\u001b[0m\n",
            "\u001b[32m38 of 164 steps (23%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split17.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:41 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.0.REV.log\n",
            "    jobid: 10\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=0\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:42 2024]\u001b[0m\n",
            "\u001b[32mFinished job 10.\u001b[0m\n",
            "\u001b[32m39 of 164 steps (24%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:42 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split0.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.0.log.gz\n",
            "    jobid: 5\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:43 2024]\u001b[0m\n",
            "\u001b[32mFinished job 5.\u001b[0m\n",
            "\u001b[32m40 of 164 steps (24%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split0.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:43 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.5.REV.log\n",
            "    jobid: 37\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=5\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:44 2024]\u001b[0m\n",
            "\u001b[32mFinished job 37.\u001b[0m\n",
            "\u001b[32m41 of 164 steps (25%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:44 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split5.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.5.log.gz\n",
            "    jobid: 35\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:46 2024]\u001b[0m\n",
            "\u001b[32mFinished job 35.\u001b[0m\n",
            "\u001b[32m42 of 164 steps (26%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split5.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:46 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.19.REV.log\n",
            "    jobid: 107\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=19\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:47 2024]\u001b[0m\n",
            "\u001b[32mFinished job 107.\u001b[0m\n",
            "\u001b[32m43 of 164 steps (26%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:47 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split19.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.19.log.gz\n",
            "    jobid: 105\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:48 2024]\u001b[0m\n",
            "\u001b[32mFinished job 105.\u001b[0m\n",
            "\u001b[32m44 of 164 steps (27%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split19.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:48 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.3.REV.log\n",
            "    jobid: 27\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=3\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:48 2024]\u001b[0m\n",
            "\u001b[32mFinished job 27.\u001b[0m\n",
            "\u001b[32m45 of 164 steps (27%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:49 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split3.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.3.log.gz\n",
            "    jobid: 25\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:50 2024]\u001b[0m\n",
            "\u001b[32mFinished job 25.\u001b[0m\n",
            "\u001b[32m46 of 164 steps (28%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split3.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:50 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.9.REV.log\n",
            "    jobid: 57\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=9\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:50 2024]\u001b[0m\n",
            "\u001b[32mFinished job 57.\u001b[0m\n",
            "\u001b[32m47 of 164 steps (29%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:51 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split9.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.9.log.gz\n",
            "    jobid: 55\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:52 2024]\u001b[0m\n",
            "\u001b[32mFinished job 55.\u001b[0m\n",
            "\u001b[32m48 of 164 steps (29%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split9.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.4.REV.log\n",
            "    jobid: 32\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:52 2024]\u001b[0m\n",
            "\u001b[32mFinished job 32.\u001b[0m\n",
            "\u001b[32m49 of 164 steps (30%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split4.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.4.log.gz\n",
            "    jobid: 30\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:53 2024]\u001b[0m\n",
            "\u001b[32mFinished job 30.\u001b[0m\n",
            "\u001b[32m50 of 164 steps (30%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split4.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:54 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.14.REV.log\n",
            "    jobid: 82\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=14\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:54 2024]\u001b[0m\n",
            "\u001b[32mFinished job 82.\u001b[0m\n",
            "\u001b[32m51 of 164 steps (31%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:54 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split14.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.14.log.gz\n",
            "    jobid: 80\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:55 2024]\u001b[0m\n",
            "\u001b[32mFinished job 80.\u001b[0m\n",
            "\u001b[32m52 of 164 steps (32%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split14.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:55 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.10.REV.log\n",
            "    jobid: 62\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=10\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:56 2024]\u001b[0m\n",
            "\u001b[32mFinished job 62.\u001b[0m\n",
            "\u001b[32m53 of 164 steps (32%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:56 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split10.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.10.log.gz\n",
            "    jobid: 60\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:58 2024]\u001b[0m\n",
            "\u001b[32mFinished job 60.\u001b[0m\n",
            "\u001b[32m54 of 164 steps (33%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split10.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:58 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.18.REV.log\n",
            "    jobid: 102\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=18\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:59 2024]\u001b[0m\n",
            "\u001b[32mFinished job 102.\u001b[0m\n",
            "\u001b[32m55 of 164 steps (34%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:18:59 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split18.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.18.log.gz\n",
            "    jobid: 100\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 100.\u001b[0m\n",
            "\u001b[32m56 of 164 steps (34%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split18.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.1.REV.log\n",
            "    jobid: 17\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=1\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:01 2024]\u001b[0m\n",
            "\u001b[32mFinished job 17.\u001b[0m\n",
            "\u001b[32m57 of 164 steps (35%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:01 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split1.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.1.log.gz\n",
            "    jobid: 15\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:02 2024]\u001b[0m\n",
            "\u001b[32mFinished job 15.\u001b[0m\n",
            "\u001b[32m58 of 164 steps (35%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split1.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:02 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.15.REV.log\n",
            "    jobid: 87\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=15\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:03 2024]\u001b[0m\n",
            "\u001b[32mFinished job 87.\u001b[0m\n",
            "\u001b[32m59 of 164 steps (36%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:03 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split15.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.15.log.gz\n",
            "    jobid: 85\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:04 2024]\u001b[0m\n",
            "\u001b[32mFinished job 85.\u001b[0m\n",
            "\u001b[32m60 of 164 steps (37%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split15.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:04 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.13.REV.log\n",
            "    jobid: 77\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=13\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:05 2024]\u001b[0m\n",
            "\u001b[32mFinished job 77.\u001b[0m\n",
            "\u001b[32m61 of 164 steps (37%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:05 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split13.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.13.log.gz\n",
            "    jobid: 75\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:06 2024]\u001b[0m\n",
            "\u001b[32mFinished job 75.\u001b[0m\n",
            "\u001b[32m62 of 164 steps (38%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split13.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:06 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.7.REV.log\n",
            "    jobid: 47\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=7\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:07 2024]\u001b[0m\n",
            "\u001b[32mFinished job 47.\u001b[0m\n",
            "\u001b[32m63 of 164 steps (38%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:07 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split7.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.7.log.gz\n",
            "    jobid: 45\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:08 2024]\u001b[0m\n",
            "\u001b[32mFinished job 45.\u001b[0m\n",
            "\u001b[32m64 of 164 steps (39%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split7.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.2.REV.log\n",
            "    jobid: 22\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=2\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:09 2024]\u001b[0m\n",
            "\u001b[32mFinished job 22.\u001b[0m\n",
            "\u001b[32m65 of 164 steps (40%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:09 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split2.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.2.log.gz\n",
            "    jobid: 20\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:10 2024]\u001b[0m\n",
            "\u001b[32mFinished job 20.\u001b[0m\n",
            "\u001b[32m66 of 164 steps (40%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split2.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:10 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.16.REV.log\n",
            "    jobid: 92\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=16\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:11 2024]\u001b[0m\n",
            "\u001b[32mFinished job 92.\u001b[0m\n",
            "\u001b[32m67 of 164 steps (41%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:11 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split16.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.16.log.gz\n",
            "    jobid: 90\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:12 2024]\u001b[0m\n",
            "\u001b[32mFinished job 90.\u001b[0m\n",
            "\u001b[32m68 of 164 steps (41%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split16.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:12 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.6.REV.log\n",
            "    jobid: 42\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=6\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:13 2024]\u001b[0m\n",
            "\u001b[32mFinished job 42.\u001b[0m\n",
            "\u001b[32m69 of 164 steps (42%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:13 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split6.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.6.log.gz\n",
            "    jobid: 40\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:14 2024]\u001b[0m\n",
            "\u001b[32mFinished job 40.\u001b[0m\n",
            "\u001b[32m70 of 164 steps (43%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split6.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:14 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.12.REV.log\n",
            "    jobid: 72\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=12\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:15 2024]\u001b[0m\n",
            "\u001b[32mFinished job 72.\u001b[0m\n",
            "\u001b[32m71 of 164 steps (43%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:15 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split12.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.12.log.gz\n",
            "    jobid: 70\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 70.\u001b[0m\n",
            "\u001b[32m72 of 164 steps (44%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split12.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:17 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.11.REV.log\n",
            "    jobid: 67\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=11\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:17 2024]\u001b[0m\n",
            "\u001b[32mFinished job 67.\u001b[0m\n",
            "\u001b[32m73 of 164 steps (45%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:17 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split11.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.11.log.gz\n",
            "    jobid: 65\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:18 2024]\u001b[0m\n",
            "\u001b[32mFinished job 65.\u001b[0m\n",
            "\u001b[32m74 of 164 steps (45%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split11.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:18 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.22.REV.log\n",
            "    jobid: 122\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=22\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:19 2024]\u001b[0m\n",
            "\u001b[32mFinished job 122.\u001b[0m\n",
            "\u001b[32m75 of 164 steps (46%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:19 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split22.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.22.log.gz\n",
            "    jobid: 120\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:20 2024]\u001b[0m\n",
            "\u001b[32mFinished job 120.\u001b[0m\n",
            "\u001b[32m76 of 164 steps (46%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split22.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.23.REV.log\n",
            "    jobid: 127\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=23\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:21 2024]\u001b[0m\n",
            "\u001b[32mFinished job 127.\u001b[0m\n",
            "\u001b[32m77 of 164 steps (47%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:21 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split23.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.23.log.gz\n",
            "    jobid: 125\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:22 2024]\u001b[0m\n",
            "\u001b[32mFinished job 125.\u001b[0m\n",
            "\u001b[32m78 of 164 steps (48%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split23.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:22 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.29.REV.log\n",
            "    jobid: 157\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=29\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:23 2024]\u001b[0m\n",
            "\u001b[32mFinished job 157.\u001b[0m\n",
            "\u001b[32m79 of 164 steps (48%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:23 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split29.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.29.log.gz\n",
            "    jobid: 155\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:24 2024]\u001b[0m\n",
            "\u001b[32mFinished job 155.\u001b[0m\n",
            "\u001b[32m80 of 164 steps (49%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split29.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:24 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.25.REV.log\n",
            "    jobid: 137\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=25\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:25 2024]\u001b[0m\n",
            "\u001b[32mFinished job 137.\u001b[0m\n",
            "\u001b[32m81 of 164 steps (49%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:25 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split25.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.25.log.gz\n",
            "    jobid: 135\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:26 2024]\u001b[0m\n",
            "\u001b[32mFinished job 135.\u001b[0m\n",
            "\u001b[32m82 of 164 steps (50%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split25.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:26 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.26.REV.log\n",
            "    jobid: 142\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=26\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:27 2024]\u001b[0m\n",
            "\u001b[32mFinished job 142.\u001b[0m\n",
            "\u001b[32m83 of 164 steps (51%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:27 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split26.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.26.log.gz\n",
            "    jobid: 140\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:28 2024]\u001b[0m\n",
            "\u001b[32mFinished job 140.\u001b[0m\n",
            "\u001b[32m84 of 164 steps (51%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split26.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:28 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.20.REV.log\n",
            "    jobid: 112\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=20\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:29 2024]\u001b[0m\n",
            "\u001b[32mFinished job 112.\u001b[0m\n",
            "\u001b[32m85 of 164 steps (52%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:29 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split20.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.20.log.gz\n",
            "    jobid: 110\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:31 2024]\u001b[0m\n",
            "\u001b[32mFinished job 110.\u001b[0m\n",
            "\u001b[32m86 of 164 steps (52%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split20.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:31 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.27.REV.log\n",
            "    jobid: 147\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=27\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:31 2024]\u001b[0m\n",
            "\u001b[32mFinished job 147.\u001b[0m\n",
            "\u001b[32m87 of 164 steps (53%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:32 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split27.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.27.log.gz\n",
            "    jobid: 145\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:33 2024]\u001b[0m\n",
            "\u001b[32mFinished job 145.\u001b[0m\n",
            "\u001b[32m88 of 164 steps (54%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split27.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:33 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.24.REV.log\n",
            "    jobid: 132\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=24\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:33 2024]\u001b[0m\n",
            "\u001b[32mFinished job 132.\u001b[0m\n",
            "\u001b[32m89 of 164 steps (54%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:33 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split24.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.24.log.gz\n",
            "    jobid: 130\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:34 2024]\u001b[0m\n",
            "\u001b[32mFinished job 130.\u001b[0m\n",
            "\u001b[32m90 of 164 steps (55%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split24.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:34 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.28.REV.log\n",
            "    jobid: 152\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=28\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:35 2024]\u001b[0m\n",
            "\u001b[32mFinished job 152.\u001b[0m\n",
            "\u001b[32m91 of 164 steps (55%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:35 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split28.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.28.log.gz\n",
            "    jobid: 150\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:36 2024]\u001b[0m\n",
            "\u001b[32mFinished job 150.\u001b[0m\n",
            "\u001b[32m92 of 164 steps (56%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split28.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:36 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.21.REV.log\n",
            "    jobid: 117\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=21\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:37 2024]\u001b[0m\n",
            "\u001b[32mFinished job 117.\u001b[0m\n",
            "\u001b[32m93 of 164 steps (57%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:37 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split21.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.21.log.gz\n",
            "    jobid: 115\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/bb8949380f7226ba92e4a563406e4b7d_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:38 2024]\u001b[0m\n",
            "\u001b[32mFinished job 115.\u001b[0m\n",
            "\u001b[32m94 of 164 steps (57%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split21.un.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:38 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_bwa_ref:\n",
            "    input: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.dict\n",
            "    log: results/logs/assignment/bwa_ref.assocBasic.log\n",
            "    jobid: 12\n",
            "    reason: Missing output files: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:47 2024]\u001b[0m\n",
            "\u001b[32mFinished job 12.\u001b[0m\n",
            "\u001b[32m95 of 164 steps (58%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:47 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.0.log\n",
            "    jobid: 4\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split0.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6302 sequences (1054466 bp)...\n",
            "[M::mem_process_seqs] Processed 6302 reads in 0.449 CPU sec, 0.452 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.496 sec; CPU: 0.475 sec\n",
            "\u001b[32m[Wed Jul 31 13:19:51 2024]\u001b[0m\n",
            "\u001b[32mFinished job 4.\u001b[0m\n",
            "\u001b[32m96 of 164 steps (59%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:51 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.0.log\n",
            "    jobid: 3\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:55 2024]\u001b[0m\n",
            "\u001b[32mFinished job 3.\u001b[0m\n",
            "\u001b[32m97 of 164 steps (59%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:19:55 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.12.log\n",
            "    jobid: 69\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split12.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6319 sequences (1056348 bp)...\n",
            "[M::mem_process_seqs] Processed 6319 reads in 0.743 CPU sec, 0.791 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.878 sec; CPU: 0.771 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 69.\u001b[0m\n",
            "\u001b[32m98 of 164 steps (60%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.12.log\n",
            "    jobid: 68\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:04 2024]\u001b[0m\n",
            "\u001b[32mFinished job 68.\u001b[0m\n",
            "\u001b[32m99 of 164 steps (60%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:04 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.15.log\n",
            "    jobid: 84\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split15.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6294 sequences (1052791 bp)...\n",
            "[M::mem_process_seqs] Processed 6294 reads in 0.426 CPU sec, 0.428 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.472 sec; CPU: 0.452 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:08 2024]\u001b[0m\n",
            "\u001b[32mFinished job 84.\u001b[0m\n",
            "\u001b[32m100 of 164 steps (61%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.15.log\n",
            "    jobid: 83\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:11 2024]\u001b[0m\n",
            "\u001b[32mFinished job 83.\u001b[0m\n",
            "\u001b[32m101 of 164 steps (62%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:11 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.17.log\n",
            "    jobid: 94\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split17.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6273 sequences (1051883 bp)...\n",
            "[M::mem_process_seqs] Processed 6273 reads in 0.438 CPU sec, 0.440 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.481 sec; CPU: 0.464 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 94.\u001b[0m\n",
            "\u001b[32m102 of 164 steps (62%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:16 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.17.log\n",
            "    jobid: 93\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:20 2024]\u001b[0m\n",
            "\u001b[32mFinished job 93.\u001b[0m\n",
            "\u001b[32m103 of 164 steps (63%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.3.log\n",
            "    jobid: 24\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split3.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6273 sequences (1051065 bp)...\n",
            "[M::mem_process_seqs] Processed 6273 reads in 0.465 CPU sec, 0.468 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.512 sec; CPU: 0.491 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:24 2024]\u001b[0m\n",
            "\u001b[32mFinished job 24.\u001b[0m\n",
            "\u001b[32m104 of 164 steps (63%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:24 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.3.log\n",
            "    jobid: 23\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:28 2024]\u001b[0m\n",
            "\u001b[32mFinished job 23.\u001b[0m\n",
            "\u001b[32m105 of 164 steps (64%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:28 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.8.log\n",
            "    jobid: 49\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split8.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6276 sequences (1051880 bp)...\n",
            "[M::mem_process_seqs] Processed 6276 reads in 0.445 CPU sec, 0.449 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.491 sec; CPU: 0.471 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:32 2024]\u001b[0m\n",
            "\u001b[32mFinished job 49.\u001b[0m\n",
            "\u001b[32m106 of 164 steps (65%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:32 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.8.log\n",
            "    jobid: 48\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:36 2024]\u001b[0m\n",
            "\u001b[32mFinished job 48.\u001b[0m\n",
            "\u001b[32m107 of 164 steps (65%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:36 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.7.log\n",
            "    jobid: 44\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split7.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6301 sequences (1055437 bp)...\n",
            "[M::mem_process_seqs] Processed 6301 reads in 0.433 CPU sec, 0.436 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.478 sec; CPU: 0.460 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:40 2024]\u001b[0m\n",
            "\u001b[32mFinished job 44.\u001b[0m\n",
            "\u001b[32m108 of 164 steps (66%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:40 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.7.log\n",
            "    jobid: 43\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:44 2024]\u001b[0m\n",
            "\u001b[32mFinished job 43.\u001b[0m\n",
            "\u001b[32m109 of 164 steps (66%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:44 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.16.log\n",
            "    jobid: 89\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split16.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6266 sequences (1050174 bp)...\n",
            "[M::mem_process_seqs] Processed 6266 reads in 0.448 CPU sec, 0.451 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.495 sec; CPU: 0.475 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:48 2024]\u001b[0m\n",
            "\u001b[32mFinished job 89.\u001b[0m\n",
            "\u001b[32m110 of 164 steps (67%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:48 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.16.log\n",
            "    jobid: 88\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:51 2024]\u001b[0m\n",
            "\u001b[32mFinished job 88.\u001b[0m\n",
            "\u001b[32m111 of 164 steps (68%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:51 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.5.log\n",
            "    jobid: 34\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split5.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6255 sequences (1047043 bp)...\n",
            "[M::mem_process_seqs] Processed 6255 reads in 0.534 CPU sec, 0.544 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.593 sec; CPU: 0.560 sec\n",
            "\u001b[32m[Wed Jul 31 13:20:55 2024]\u001b[0m\n",
            "\u001b[32mFinished job 34.\u001b[0m\n",
            "\u001b[32m112 of 164 steps (68%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:20:55 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.5.log\n",
            "    jobid: 33\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 33.\u001b[0m\n",
            "\u001b[32m113 of 164 steps (69%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.1.log\n",
            "    jobid: 14\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split1.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6297 sequences (1054342 bp)...\n",
            "[M::mem_process_seqs] Processed 6297 reads in 0.433 CPU sec, 0.434 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.476 sec; CPU: 0.459 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:04 2024]\u001b[0m\n",
            "\u001b[32mFinished job 14.\u001b[0m\n",
            "\u001b[32m114 of 164 steps (70%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:04 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.1.log\n",
            "    jobid: 13\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:07 2024]\u001b[0m\n",
            "\u001b[32mFinished job 13.\u001b[0m\n",
            "\u001b[32m115 of 164 steps (70%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:07 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.2.log\n",
            "    jobid: 19\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split2.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6263 sequences (1047478 bp)...\n",
            "[M::mem_process_seqs] Processed 6263 reads in 0.681 CPU sec, 0.709 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.773 sec; CPU: 0.707 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:12 2024]\u001b[0m\n",
            "\u001b[32mFinished job 19.\u001b[0m\n",
            "\u001b[32m116 of 164 steps (71%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:12 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.2.log\n",
            "    jobid: 18\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 18.\u001b[0m\n",
            "\u001b[32m117 of 164 steps (71%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:16 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.10.log\n",
            "    jobid: 59\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split10.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6277 sequences (1049511 bp)...\n",
            "[M::mem_process_seqs] Processed 6277 reads in 0.444 CPU sec, 0.447 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.490 sec; CPU: 0.471 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:20 2024]\u001b[0m\n",
            "\u001b[32mFinished job 59.\u001b[0m\n",
            "\u001b[32m118 of 164 steps (72%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.10.log\n",
            "    jobid: 58\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:23 2024]\u001b[0m\n",
            "\u001b[32mFinished job 58.\u001b[0m\n",
            "\u001b[32m119 of 164 steps (73%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:24 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.6.log\n",
            "    jobid: 39\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split6.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6273 sequences (1050284 bp)...\n",
            "[M::mem_process_seqs] Processed 6273 reads in 0.696 CPU sec, 0.858 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.934 sec; CPU: 0.723 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:29 2024]\u001b[0m\n",
            "\u001b[32mFinished job 39.\u001b[0m\n",
            "\u001b[32m120 of 164 steps (73%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:29 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.6.log\n",
            "    jobid: 38\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:32 2024]\u001b[0m\n",
            "\u001b[32mFinished job 38.\u001b[0m\n",
            "\u001b[32m121 of 164 steps (74%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:32 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.9.log\n",
            "    jobid: 54\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split9.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6260 sequences (1047456 bp)...\n",
            "[M::mem_process_seqs] Processed 6260 reads in 0.423 CPU sec, 0.424 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.468 sec; CPU: 0.450 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:36 2024]\u001b[0m\n",
            "\u001b[32mFinished job 54.\u001b[0m\n",
            "\u001b[32m122 of 164 steps (74%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:36 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.9.log\n",
            "    jobid: 53\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:40 2024]\u001b[0m\n",
            "\u001b[32mFinished job 53.\u001b[0m\n",
            "\u001b[32m123 of 164 steps (75%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:40 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.4.log\n",
            "    jobid: 29\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split4.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6248 sequences (1044088 bp)...\n",
            "[M::mem_process_seqs] Processed 6248 reads in 0.681 CPU sec, 0.719 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.778 sec; CPU: 0.709 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:44 2024]\u001b[0m\n",
            "\u001b[32mFinished job 29.\u001b[0m\n",
            "\u001b[32m124 of 164 steps (76%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:45 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.4.log\n",
            "    jobid: 28\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:48 2024]\u001b[0m\n",
            "\u001b[32mFinished job 28.\u001b[0m\n",
            "\u001b[32m125 of 164 steps (76%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:48 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.19.log\n",
            "    jobid: 104\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split19.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6253 sequences (1045669 bp)...\n",
            "[M::mem_process_seqs] Processed 6253 reads in 0.423 CPU sec, 0.425 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.469 sec; CPU: 0.450 sec\n",
            "\u001b[32m[Wed Jul 31 13:21:52 2024]\u001b[0m\n",
            "\u001b[32mFinished job 104.\u001b[0m\n",
            "\u001b[32m126 of 164 steps (77%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.19.log\n",
            "    jobid: 103\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:55 2024]\u001b[0m\n",
            "\u001b[32mFinished job 103.\u001b[0m\n",
            "\u001b[32m127 of 164 steps (77%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:21:55 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.14.log\n",
            "    jobid: 79\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split14.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6249 sequences (1047032 bp)...\n",
            "[M::mem_process_seqs] Processed 6249 reads in 0.445 CPU sec, 0.449 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.508 sec; CPU: 0.473 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 79.\u001b[0m\n",
            "\u001b[32m128 of 164 steps (78%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.14.log\n",
            "    jobid: 78\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:03 2024]\u001b[0m\n",
            "\u001b[32mFinished job 78.\u001b[0m\n",
            "\u001b[32m129 of 164 steps (79%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:04 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.13.log\n",
            "    jobid: 74\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split13.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6256 sequences (1046393 bp)...\n",
            "[M::mem_process_seqs] Processed 6256 reads in 0.445 CPU sec, 0.447 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.491 sec; CPU: 0.471 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:07 2024]\u001b[0m\n",
            "\u001b[32mFinished job 74.\u001b[0m\n",
            "\u001b[32m130 of 164 steps (79%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.13.log\n",
            "    jobid: 73\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:11 2024]\u001b[0m\n",
            "\u001b[32mFinished job 73.\u001b[0m\n",
            "\u001b[32m131 of 164 steps (80%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:11 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.11.log\n",
            "    jobid: 64\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split11.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6285 sequences (1050624 bp)...\n",
            "[M::mem_process_seqs] Processed 6285 reads in 0.453 CPU sec, 0.456 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.501 sec; CPU: 0.481 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 64.\u001b[0m\n",
            "\u001b[32m132 of 164 steps (80%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:16 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.11.log\n",
            "    jobid: 63\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:19 2024]\u001b[0m\n",
            "\u001b[32mFinished job 63.\u001b[0m\n",
            "\u001b[32m133 of 164 steps (81%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.18.log\n",
            "    jobid: 99\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split18.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6248 sequences (1044211 bp)...\n",
            "[M::mem_process_seqs] Processed 6248 reads in 0.451 CPU sec, 0.454 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.497 sec; CPU: 0.478 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:23 2024]\u001b[0m\n",
            "\u001b[32mFinished job 99.\u001b[0m\n",
            "\u001b[32m134 of 164 steps (82%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:24 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.18.log\n",
            "    jobid: 98\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:28 2024]\u001b[0m\n",
            "\u001b[32mFinished job 98.\u001b[0m\n",
            "\u001b[32m135 of 164 steps (82%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:28 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.23.log\n",
            "    jobid: 124\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split23.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6203 sequences (1038886 bp)...\n",
            "[M::mem_process_seqs] Processed 6203 reads in 0.440 CPU sec, 0.442 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.486 sec; CPU: 0.467 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:32 2024]\u001b[0m\n",
            "\u001b[32mFinished job 124.\u001b[0m\n",
            "\u001b[32m136 of 164 steps (83%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:32 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.23.log\n",
            "    jobid: 123\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:36 2024]\u001b[0m\n",
            "\u001b[32mFinished job 123.\u001b[0m\n",
            "\u001b[32m137 of 164 steps (84%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:36 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.22.log\n",
            "    jobid: 119\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split22.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6193 sequences (1035955 bp)...\n",
            "[M::mem_process_seqs] Processed 6193 reads in 0.643 CPU sec, 0.670 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.724 sec; CPU: 0.670 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:40 2024]\u001b[0m\n",
            "\u001b[32mFinished job 119.\u001b[0m\n",
            "\u001b[32m138 of 164 steps (84%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:40 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.22.log\n",
            "    jobid: 118\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:44 2024]\u001b[0m\n",
            "\u001b[32mFinished job 118.\u001b[0m\n",
            "\u001b[32m139 of 164 steps (85%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:44 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.26.log\n",
            "    jobid: 139\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split26.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6180 sequences (1036011 bp)...\n",
            "[M::mem_process_seqs] Processed 6180 reads in 0.420 CPU sec, 0.423 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.471 sec; CPU: 0.448 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:48 2024]\u001b[0m\n",
            "\u001b[32mFinished job 139.\u001b[0m\n",
            "\u001b[32m140 of 164 steps (85%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:48 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.26.log\n",
            "    jobid: 138\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:52 2024]\u001b[0m\n",
            "\u001b[32mFinished job 138.\u001b[0m\n",
            "\u001b[32m141 of 164 steps (86%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.28.log\n",
            "    jobid: 149\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split28.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6183 sequences (1036207 bp)...\n",
            "[M::mem_process_seqs] Processed 6183 reads in 0.654 CPU sec, 0.686 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.765 sec; CPU: 0.680 sec\n",
            "\u001b[32m[Wed Jul 31 13:22:56 2024]\u001b[0m\n",
            "\u001b[32mFinished job 149.\u001b[0m\n",
            "\u001b[32m142 of 164 steps (87%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:22:56 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.28.log\n",
            "    jobid: 148\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 148.\u001b[0m\n",
            "\u001b[32m143 of 164 steps (87%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.24.log\n",
            "    jobid: 129\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split24.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6173 sequences (1033293 bp)...\n",
            "[M::mem_process_seqs] Processed 6173 reads in 0.460 CPU sec, 0.465 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.509 sec; CPU: 0.486 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:04 2024]\u001b[0m\n",
            "\u001b[32mFinished job 129.\u001b[0m\n",
            "\u001b[32m144 of 164 steps (88%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:05 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.24.log\n",
            "    jobid: 128\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:08 2024]\u001b[0m\n",
            "\u001b[32mFinished job 128.\u001b[0m\n",
            "\u001b[32m145 of 164 steps (88%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:08 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.25.log\n",
            "    jobid: 134\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split25.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6174 sequences (1033986 bp)...\n",
            "[M::mem_process_seqs] Processed 6174 reads in 0.698 CPU sec, 0.710 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.778 sec; CPU: 0.727 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:13 2024]\u001b[0m\n",
            "\u001b[32mFinished job 134.\u001b[0m\n",
            "\u001b[32m146 of 164 steps (89%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:13 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.25.log\n",
            "    jobid: 133\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:17 2024]\u001b[0m\n",
            "\u001b[32mFinished job 133.\u001b[0m\n",
            "\u001b[32m147 of 164 steps (90%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:17 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.20.log\n",
            "    jobid: 109\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split20.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6184 sequences (1033607 bp)...\n",
            "[M::mem_process_seqs] Processed 6184 reads in 0.432 CPU sec, 0.434 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.479 sec; CPU: 0.458 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:21 2024]\u001b[0m\n",
            "\u001b[32mFinished job 109.\u001b[0m\n",
            "\u001b[32m148 of 164 steps (90%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:21 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.20.log\n",
            "    jobid: 108\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:25 2024]\u001b[0m\n",
            "\u001b[32mFinished job 108.\u001b[0m\n",
            "\u001b[32m149 of 164 steps (91%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:25 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.21.log\n",
            "    jobid: 114\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split21.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6163 sequences (1032768 bp)...\n",
            "[M::mem_process_seqs] Processed 6163 reads in 0.477 CPU sec, 0.489 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.540 sec; CPU: 0.505 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:30 2024]\u001b[0m\n",
            "\u001b[32mFinished job 114.\u001b[0m\n",
            "\u001b[32m150 of 164 steps (91%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:30 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.21.log\n",
            "    jobid: 113\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:34 2024]\u001b[0m\n",
            "\u001b[32mFinished job 113.\u001b[0m\n",
            "\u001b[32m151 of 164 steps (92%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:34 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.27.log\n",
            "    jobid: 144\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split27.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6157 sequences (1030658 bp)...\n",
            "[M::mem_process_seqs] Processed 6157 reads in 0.427 CPU sec, 0.429 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.471 sec; CPU: 0.453 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:38 2024]\u001b[0m\n",
            "\u001b[32mFinished job 144.\u001b[0m\n",
            "\u001b[32m152 of 164 steps (93%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:38 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.27.log\n",
            "    jobid: 143\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:42 2024]\u001b[0m\n",
            "\u001b[32mFinished job 143.\u001b[0m\n",
            "\u001b[32m153 of 164 steps (93%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:42 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.29.log\n",
            "    jobid: 154\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split29.mapped.bam; Input files updated by another job: results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.sa\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 6139 sequences (1028528 bp)...\n",
            "[M::mem_process_seqs] Processed 6139 reads in 0.437 CPU sec, 0.441 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.482 sec; CPU: 0.463 sec\n",
            "\u001b[32m[Wed Jul 31 13:23:46 2024]\u001b[0m\n",
            "\u001b[32mFinished job 154.\u001b[0m\n",
            "\u001b[32m154 of 164 steps (94%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:46 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_collect:\n",
            "    input: results/assignment/assocBasic/bam/merge_split0.mapped.bam, results/assignment/assocBasic/bam/merge_split1.mapped.bam, results/assignment/assocBasic/bam/merge_split2.mapped.bam, results/assignment/assocBasic/bam/merge_split3.mapped.bam, results/assignment/assocBasic/bam/merge_split4.mapped.bam, results/assignment/assocBasic/bam/merge_split5.mapped.bam, results/assignment/assocBasic/bam/merge_split6.mapped.bam, results/assignment/assocBasic/bam/merge_split7.mapped.bam, results/assignment/assocBasic/bam/merge_split8.mapped.bam, results/assignment/assocBasic/bam/merge_split9.mapped.bam, results/assignment/assocBasic/bam/merge_split10.mapped.bam, results/assignment/assocBasic/bam/merge_split11.mapped.bam, results/assignment/assocBasic/bam/merge_split12.mapped.bam, results/assignment/assocBasic/bam/merge_split13.mapped.bam, results/assignment/assocBasic/bam/merge_split14.mapped.bam, results/assignment/assocBasic/bam/merge_split15.mapped.bam, results/assignment/assocBasic/bam/merge_split16.mapped.bam, results/assignment/assocBasic/bam/merge_split17.mapped.bam, results/assignment/assocBasic/bam/merge_split18.mapped.bam, results/assignment/assocBasic/bam/merge_split19.mapped.bam, results/assignment/assocBasic/bam/merge_split20.mapped.bam, results/assignment/assocBasic/bam/merge_split21.mapped.bam, results/assignment/assocBasic/bam/merge_split22.mapped.bam, results/assignment/assocBasic/bam/merge_split23.mapped.bam, results/assignment/assocBasic/bam/merge_split24.mapped.bam, results/assignment/assocBasic/bam/merge_split25.mapped.bam, results/assignment/assocBasic/bam/merge_split26.mapped.bam, results/assignment/assocBasic/bam/merge_split27.mapped.bam, results/assignment/assocBasic/bam/merge_split28.mapped.bam, results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    output: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    log: results/logs/assignment/collect.assocBasic.log\n",
            "    jobid: 162\n",
            "    reason: Missing output files: results/assignment/assocBasic/aligned_merged_reads.bam; Input files updated by another job: results/assignment/assocBasic/bam/merge_split1.mapped.bam, results/assignment/assocBasic/bam/merge_split7.mapped.bam, results/assignment/assocBasic/bam/merge_split29.mapped.bam, results/assignment/assocBasic/bam/merge_split0.mapped.bam, results/assignment/assocBasic/bam/merge_split15.mapped.bam, results/assignment/assocBasic/bam/merge_split13.mapped.bam, results/assignment/assocBasic/bam/merge_split9.mapped.bam, results/assignment/assocBasic/bam/merge_split17.mapped.bam, results/assignment/assocBasic/bam/merge_split23.mapped.bam, results/assignment/assocBasic/bam/merge_split28.mapped.bam, results/assignment/assocBasic/bam/merge_split11.mapped.bam, results/assignment/assocBasic/bam/merge_split4.mapped.bam, results/assignment/assocBasic/bam/merge_split27.mapped.bam, results/assignment/assocBasic/bam/merge_split16.mapped.bam, results/assignment/assocBasic/bam/merge_split10.mapped.bam, results/assignment/assocBasic/bam/merge_split8.mapped.bam, results/assignment/assocBasic/bam/merge_split2.mapped.bam, results/assignment/assocBasic/bam/merge_split6.mapped.bam, results/assignment/assocBasic/bam/merge_split18.mapped.bam, results/assignment/assocBasic/bam/merge_split24.mapped.bam, results/assignment/assocBasic/bam/merge_split25.mapped.bam, results/assignment/assocBasic/bam/merge_split3.mapped.bam, results/assignment/assocBasic/bam/merge_split21.mapped.bam, results/assignment/assocBasic/bam/merge_split26.mapped.bam, results/assignment/assocBasic/bam/merge_split20.mapped.bam, results/assignment/assocBasic/bam/merge_split22.mapped.bam, results/assignment/assocBasic/bam/merge_split19.mapped.bam, results/assignment/assocBasic/bam/merge_split14.mapped.bam, results/assignment/assocBasic/bam/merge_split5.mapped.bam, results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:52 2024]\u001b[0m\n",
            "\u001b[32mFinished job 162.\u001b[0m\n",
            "\u001b[32m155 of 164 steps (95%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split0.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split1.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split2.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split3.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split4.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split5.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split6.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split7.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split8.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split9.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split10.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split11.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split12.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split13.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split14.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split15.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split16.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split17.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split18.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split19.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split20.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split21.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split22.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split23.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split24.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split25.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split26.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split27.mapped.bam.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split28.mapped.bam.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:52 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.29.log\n",
            "    jobid: 153\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:56 2024]\u001b[0m\n",
            "\u001b[32mFinished job 153.\u001b[0m\n",
            "\u001b[32m156 of 164 steps (95%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/bam/merge_split29.mapped.bam.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:56 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_collectBCs:\n",
            "    input: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv\n",
            "    output: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    log: results/logs/assignment/collectBCs.assocBasic.log\n",
            "    jobid: 2\n",
            "    reason: Missing output files: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz; Input files updated by another job: results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/ec92106e14f6232977fd0918d24cb216_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:58 2024]\u001b[0m\n",
            "\u001b[32mFinished job 2.\u001b[0m\n",
            "\u001b[32m157 of 164 steps (96%) done\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv.\u001b[0m\n",
            "\u001b[33mRemoving temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv.\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:23:58 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_filter:\n",
            "    input: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/assignment/filterAssignmentTsv.py (cached)\n",
            "    output: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    log: results/logs/assignment/filter.assocBasic.default.log\n",
            "    jobid: 1\n",
            "    reason: Missing output files: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz; Input files updated by another job: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/653b2f2da62f9e29538eeaf42dc2a489_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:00 2024]\u001b[0m\n",
            "\u001b[32mFinished job 1.\u001b[0m\n",
            "\u001b[32m158 of 164 steps (96%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:00 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_statistic_assignment:\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_assignment.R (cached)\n",
            "    output: results/assignment/assocBasic/statistic/assignment.default.tsv.gz, results/assignment/assocBasic/statistic/assignment.default.png\n",
            "    log: results/logs/assignment/statistic_assignment.assocBasic.default.log\n",
            "    jobid: 159\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assignment.default.png; Input files updated by another job: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/a97672b6846360f00f0e85ee53a46950_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:06 2024]\u001b[0m\n",
            "\u001b[32mFinished job 159.\u001b[0m\n",
            "\u001b[32m159 of 164 steps (97%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:06 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_idx_bam:\n",
            "    input: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    output: results/assignment/assocBasic/aligned_merged_reads.bam.bai\n",
            "    log: results/logs/assignment/assocBasic/assignment_idx_bam.log\n",
            "    jobid: 163\n",
            "    reason: Missing output files: results/assignment/assocBasic/aligned_merged_reads.bam.bai; Input files updated by another job: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:11 2024]\u001b[0m\n",
            "\u001b[32mFinished job 163.\u001b[0m\n",
            "\u001b[32m160 of 164 steps (98%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:11 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_flagstat:\n",
            "    input: results/assignment/assocBasic/aligned_merged_reads.bam, results/assignment/assocBasic/aligned_merged_reads.bam.bai\n",
            "    output: results/assignment/assocBasic/statistic/assignment/bam_stats.txt\n",
            "    log: results/logs/assignment/flagstat.assocBasic.log\n",
            "    jobid: 161\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assignment/bam_stats.txt; Input files updated by another job: results/assignment/assocBasic/aligned_merged_reads.bam.bai, results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/50b61aef61c0c65fd933378d537fa1ad_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:16 2024]\u001b[0m\n",
            "\u001b[32mFinished job 161.\u001b[0m\n",
            "\u001b[32m161 of 164 steps (98%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:16 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_statistic_totalCounts:\n",
            "    input: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_total_counts.py (cached)\n",
            "    output: results/assignment/assocBasic/statistic/total_counts.tsv.gz\n",
            "    log: results/logs/assignment/statistic_totalCounts.assocBasic.log\n",
            "    jobid: 160\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/total_counts.tsv.gz; Input files updated by another job: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/653b2f2da62f9e29538eeaf42dc2a489_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:20 2024]\u001b[0m\n",
            "\u001b[32mFinished job 160.\u001b[0m\n",
            "\u001b[32m162 of 164 steps (99%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:20 2024]\u001b[0m\n",
            "\u001b[32mlocalrule assignment_statistic_assignedCounts:\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_total_counts.py (cached)\n",
            "    output: results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz\n",
            "    log: results/logs/assignment/statistic_assignedCounts.assocBasic.default.log\n",
            "    jobid: 158\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz; Input files updated by another job: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[33mActivating conda environment: ../conda_envs/653b2f2da62f9e29538eeaf42dc2a489_\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:22 2024]\u001b[0m\n",
            "\u001b[32mFinished job 158.\u001b[0m\n",
            "\u001b[32m163 of 164 steps (99%) done\u001b[0m\n",
            "\u001b[33mSelect jobs to execute...\u001b[0m\n",
            "\u001b[33mExecute 1 jobs...\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:22 2024]\u001b[0m\n",
            "\u001b[32mlocalrule all:\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz, results/assignment/assocBasic/statistic/assignment.default.png, results/assignment/assocBasic/statistic/total_counts.tsv.gz, results/assignment/assocBasic/statistic/assignment/bam_stats.txt\n",
            "    jobid: 0\n",
            "    reason: Input files updated by another job: results/assignment/assocBasic/statistic/assignment.default.png, results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz, results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, results/assignment/assocBasic/statistic/assignment/bam_stats.txt, results/assignment/assocBasic/statistic/total_counts.tsv.gz\n",
            "    resources: tmpdir=/tmp\u001b[0m\n",
            "\u001b[32m\u001b[0m\n",
            "\u001b[32m[Wed Jul 31 13:24:22 2024]\u001b[0m\n",
            "\u001b[32mFinished job 0.\u001b[0m\n",
            "\u001b[32m164 of 164 steps (100%) done\u001b[0m\n",
            "\u001b[33mComplete log: .snakemake/log/2024-07-31T131747.282200.snakemake.log\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "udocker(\"run \\\n",
        "  -v=${PWD}/assoc_basic:/data/run \\\n",
        "  --workdir /data/run \\\n",
        "  visze/snakemake_mprasnakeflow:0.1.0 \\\n",
        "  snakemake --sdm conda --conda-prefix /data/conda_envs \\\n",
        "  --snakefile /data/MPRAsnakeflow/workflow/Snakefile \\\n",
        "  --configfile /data/MPRAsnakeflow/resources/assoc_basic/config.yml \\\n",
        "  -c 1\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "oReNOt6AgJt8",
        "outputId": "b4564e32-3898-4bf5-98cb-bea11bc3919e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "software-properties-common is already the newest version (0.99.22.9).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 45 not upgraded.\n",
            "Repository: 'deb https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu/ jammy main'\n",
            "Description:\n",
            "This PPA contains stable releases of Apptainer.\n",
            "\n",
            "Documentation https://apptainer.org/docs/\n",
            "Support https://apptainer.org/help/\n",
            "Bug report https://github.com/apptainer/apptainer/issues\n",
            "More info: https://launchpad.net/~apptainer/+archive/ubuntu/ppa\n",
            "Adding repository.\n",
            "Adding deb entry to /etc/apt/sources.list.d/apptainer-ubuntu-ppa-jammy.list\n",
            "Adding disabled deb-src entry to /etc/apt/sources.list.d/apptainer-ubuntu-ppa-jammy.list\n",
            "Adding key to /etc/apt/trusted.gpg.d/apptainer-ubuntu-ppa.gpg with fingerprint F6B0F5193D4F3301EF491FF0AFE36534FC6218AE\n",
            "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Hit:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Ign:6 https://r2u.stat.illinois.edu/ubuntu jammy InRelease\n",
            "Get:7 https://r2u.stat.illinois.edu/ubuntu jammy Release [5,713 B]\n",
            "Get:8 https://r2u.stat.illinois.edu/ubuntu jammy Release.gpg [793 B]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:10 https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu jammy InRelease [18.1 kB]\n",
            "Get:11 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,112 kB]\n",
            "Hit:12 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Get:13 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,208 kB]\n",
            "Get:14 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease [24.3 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,421 kB]\n",
            "Get:16 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,130 kB]\n",
            "Get:17 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [2,780 kB]\n",
            "Hit:18 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [2,882 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,393 kB]\n",
            "Get:21 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,549 kB]\n",
            "Get:22 https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu jammy/main amd64 Packages [716 B]\n",
            "Get:23 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy/main amd64 Packages [49.2 kB]\n",
            "Fetched 24.0 MB in 3s (7,051 kB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Hit:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease\n",
            "Hit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:6 http://security.ubuntu.com/ubuntu jammy-security InRelease\n",
            "Ign:7 https://r2u.stat.illinois.edu/ubuntu jammy InRelease\n",
            "Hit:8 https://r2u.stat.illinois.edu/ubuntu jammy Release\n",
            "Hit:9 https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:12 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:13 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "47 packages can be upgraded. Run 'apt list --upgradable' to see them.\n",
            "\u001b[1;33mW: \u001b[0mSkipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\u001b[0m\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  fakeroot libfakeroot libfuse3-3 liblzo2-2 squashfs-tools uidmap\n",
            "Suggested packages:\n",
            "  fuse3\n",
            "The following NEW packages will be installed:\n",
            "  apptainer fakeroot libfakeroot libfuse3-3 liblzo2-2 squashfs-tools uidmap\n",
            "0 upgraded, 7 newly installed, 0 to remove and 47 not upgraded.\n",
            "Need to get 29.4 MB of archives.\n",
            "After this operation, 125 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfuse3-3 amd64 3.10.5-1build1 [81.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 uidmap amd64 1:4.8.1-2ubuntu2.2 [22.4 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 liblzo2-2 amd64 2.10-2build3 [53.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 squashfs-tools amd64 1:4.5-3build1 [159 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfakeroot amd64 1.28-1ubuntu1 [31.5 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 fakeroot amd64 1.28-1ubuntu1 [60.4 kB]\n",
            "Get:7 https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu jammy/main amd64 apptainer amd64 1.3.2-1~ubuntu22.04.0+stable1 [29.0 MB]\n",
            "Fetched 29.4 MB in 36s (823 kB/s)\n",
            "Selecting previously unselected package libfuse3-3:amd64.\n",
            "(Reading database ... 123589 files and directories currently installed.)\n",
            "Preparing to unpack .../0-libfuse3-3_3.10.5-1build1_amd64.deb ...\n",
            "Unpacking libfuse3-3:amd64 (3.10.5-1build1) ...\n",
            "Selecting previously unselected package uidmap.\n",
            "Preparing to unpack .../1-uidmap_1%3a4.8.1-2ubuntu2.2_amd64.deb ...\n",
            "Unpacking uidmap (1:4.8.1-2ubuntu2.2) ...\n",
            "Selecting previously unselected package liblzo2-2:amd64.\n",
            "Preparing to unpack .../2-liblzo2-2_2.10-2build3_amd64.deb ...\n",
            "Unpacking liblzo2-2:amd64 (2.10-2build3) ...\n",
            "Selecting previously unselected package squashfs-tools.\n",
            "Preparing to unpack .../3-squashfs-tools_1%3a4.5-3build1_amd64.deb ...\n",
            "Unpacking squashfs-tools (1:4.5-3build1) ...\n",
            "Selecting previously unselected package libfakeroot:amd64.\n",
            "Preparing to unpack .../4-libfakeroot_1.28-1ubuntu1_amd64.deb ...\n",
            "Unpacking libfakeroot:amd64 (1.28-1ubuntu1) ...\n",
            "Selecting previously unselected package fakeroot.\n",
            "Preparing to unpack .../5-fakeroot_1.28-1ubuntu1_amd64.deb ...\n",
            "Unpacking fakeroot (1.28-1ubuntu1) ...\n",
            "Selecting previously unselected package apptainer.\n",
            "Preparing to unpack .../6-apptainer_1.3.2-1~ubuntu22.04.0+stable1_amd64.deb ...\n",
            "Unpacking apptainer (1.3.2-1~ubuntu22.04.0+stable1) ...\n",
            "Setting up uidmap (1:4.8.1-2ubuntu2.2) ...\n",
            "Setting up liblzo2-2:amd64 (2.10-2build3) ...\n",
            "Setting up libfakeroot:amd64 (1.28-1ubuntu1) ...\n",
            "Setting up fakeroot (1.28-1ubuntu1) ...\n",
            "update-alternatives: using /usr/bin/fakeroot-sysv to provide /usr/bin/fakeroot (fakeroot) in auto mode\n",
            "Setting up libfuse3-3:amd64 (3.10.5-1build1) ...\n",
            "Setting up squashfs-tools (1:4.5-3build1) ...\n",
            "Setting up apptainer (1.3.2-1~ubuntu22.04.0+stable1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "Usage:\n",
            "  apptainer [global options...] <command>\n",
            "\n",
            "Available Commands:\n",
            "  build       Build an Apptainer image\n",
            "  cache       Manage the local cache\n",
            "  capability  Manage Linux capabilities for users and groups\n",
            "  checkpoint  Manage container checkpoint state (experimental)\n",
            "  completion  Generate the autocompletion script for the specified shell\n",
            "  config      Manage various apptainer configuration (root user only)\n",
            "  delete      Deletes requested image from the library\n",
            "  exec        Run a command within a container\n",
            "  inspect     Show metadata for an image\n",
            "  instance    Manage containers running as services\n",
            "  key         Manage OpenPGP keys\n",
            "  keyserver   Manage apptainer keyservers\n",
            "  oci         Manage OCI containers\n",
            "  overlay     Manage an EXT3 writable overlay image\n",
            "  plugin      Manage Apptainer plugins\n",
            "  pull        Pull an image from a URI\n",
            "  push        Upload image to the provided URI\n",
            "  registry    Manage authentication to OCI/Docker registries\n",
            "  remote      Manage apptainer remote endpoints\n",
            "  run         Run the user-defined default command within a container\n",
            "  run-help    Show the user-defined help for an image\n",
            "  search      Search a Container Library for images\n",
            "  shell       Run a shell within a container\n",
            "  sif         Manipulate Singularity Image Format (SIF) images\n",
            "  sign        Add digital signature(s) to an image\n",
            "  test        Run the user-defined tests within a container\n",
            "  verify      Verify digital signature(s) within an image\n",
            "  version     Show the version for Apptainer\n",
            "\n",
            "Run 'apptainer --help' for more detailed usage information.\n"
          ]
        }
      ],
      "source": [
        "!apt install -y software-properties-common\n",
        "!add-apt-repository -y ppa:apptainer/ppa\n",
        "!apt update\n",
        "!apt install -y apptainer"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "DJGF3wBCpcni",
        "outputId": "31bbeceb-2bdb-463f-a52f-ef25fbed2aa0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  apptainer-suid\n",
            "0 upgraded, 1 newly installed, 0 to remove and 47 not upgraded.\n",
            "Need to get 5,915 kB of archives.\n",
            "After this operation, 18.5 MB of additional disk space will be used.\n",
            "Get:1 https://ppa.launchpadcontent.net/apptainer/ppa/ubuntu jammy/main amd64 apptainer-suid amd64 1.3.2-1~ubuntu22.04.0+stable1 [5,915 kB]\n",
            "Fetched 5,915 kB in 11s (518 kB/s)\n",
            "Selecting previously unselected package apptainer-suid.\n",
            "(Reading database ... 124865 files and directories currently installed.)\n",
            "Preparing to unpack .../apptainer-suid_1.3.2-1~ubuntu22.04.0+stable1_amd64.deb ...\n",
            "Unpacking apptainer-suid (1.3.2-1~ubuntu22.04.0+stable1) ...\n",
            "Setting up apptainer-suid (1.3.2-1~ubuntu22.04.0+stable1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt install -y apptainer-suid"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "S9JVnCXfgXO2",
        "outputId": "ffc09f70-b6bf-4984-b160-7ba7a87672de"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Error for command \"build\": unknown flag: --security\n",
            "\n",
            "Options for build command:\n",
            "\n",
            "  -B, --bind stringArray         a user-bind path specification. spec has the format\n",
            "                                 src[:dest[:opts]],where src and dest are outside and inside\n",
            "                                 paths. If dest is not given,it is set equal to src. Mount\n",
            "                                 options ('opts') may be specified as 'ro'(read-only) or 'rw'\n",
            "                                 (read/write, which is the default).Multiple bind paths can be\n",
            "                                 given by a comma separated list.\n",
            "      --build-arg strings        defines variable=value to replace {{ variable }} entries in\n",
            "                                 build definition file\n",
            "      --build-arg-file string    specifies a file containing variable=value lines to replace\n",
            "                                 '{{ variable }}' with value in build definition files\n",
            "      --disable-cache            do not use cache or create cache\n",
            "      --docker-host string       specify a custom Docker daemon host\n",
            "      --docker-login             login to a Docker Repository interactively\n",
            "  -e, --encrypt                  build an image with an encrypted file system\n",
            "  -f, --fakeroot                 build with the appearance of running as root (default when\n",
            "                                 building from a definition file unprivileged)\n",
            "      --fix-perms                ensure owner has rwX permissions on all container content for\n",
            "                                 oci/docker sources\n",
            "  -F, --force                    overwrite an image file if it exists\n",
            "  -h, --help                     help for build\n",
            "      --json                     interpret build definition as JSON\n",
            "      --library string           container Library URL\n",
            "      --mount stringArray        a mount specification e.g.\n",
            "                                 'type=bind,source=/opt,destination=/hostopt'.\n",
            "      --no-cleanup               do NOT clean up bundle after failed build, can be helpful for\n",
            "                                 debugging\n",
            "      --no-https                 use http instead of https for docker:// oras:// and\n",
            "                                 library://<hostname>/... URIs\n",
            "  -T, --notest                   build without running tests in %test section\n",
            "      --nv                       inject host Nvidia libraries during build for post and test\n",
            "                                 sections\n",
            "      --nvccli                   use nvidia-container-cli for GPU setup (experimental)\n",
            "      --passphrase               prompt for an encryption passphrase\n",
            "      --pem-path string          enter an path to a PEM formatted RSA key for an encrypted\n",
            "                                 container\n",
            "      --rocm                     inject host Rocm libraries during build for post and test sections\n",
            "  -s, --sandbox                  build image as sandbox format (chroot directory structure)\n",
            "      --section strings          only run specific section(s) of deffile (setup, post, files,\n",
            "                                 environment, test, labels, none) (default [all])\n",
            "  -u, --update                   run definition over existing container (skips header)\n",
            "      --userns                   build without using setuid even if available\n",
            "      --warn-unused-build-args   shows warning instead of fatal message when build args are\n",
            "                                 not exact matched\n",
            "      --writable-tmpfs           during the %test section, makes the file system accessible as\n",
            "                                 read-write with non persistent data (with overlay support only)\n",
            "\n",
            "Run 'apptainer --help' for more detailed usage information.\n"
          ]
        }
      ],
      "source": [
        "!apptainer build snakemake.sif docker://snakemake/snakemake:latest"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dx2-FjkphdQD",
        "outputId": "6002707d-8989-487d-ab82-75fdbfb0acc8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[31mFATAL:  \u001b[0m container creation failed: mount hook function failure: mount overlay->/var/lib/apptainer/mnt/session/final error: while mounting overlay: can't mount overlay filesystem to /var/lib/apptainer/mnt/session/final: while setting effective capabilities: CAP_DAC_READ_SEARCH is not in the permitted capability set\n"
          ]
        }
      ],
      "source": [
        "!apptainer exec --security seccomp=unconfined --security systempaths=unconfined --security no-new-privileges snakemake.sif snakemake --help"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z-xqhZYn6ZEi",
        "outputId": "063edac3-cde3-45d4-90a3-200532288472"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "INSTANCE NAME    PID    IP    IMAGE\n"
          ]
        }
      ],
      "source": [
        "!apptainer instance list"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "IgkrlgRYjt_K",
        "outputId": "282d379d-5bb2-49e7-ad13-a4dce76e3586"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
            "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
            "100 78.9M  100 78.9M    0     0  70.7M      0  0:00:01  0:00:01 --:--:--  231M\n",
            "PREFIX=/root/miniforge3\n",
            "Unpacking payload ...\n",
            "Extracting _libgcc_mutex-0.1-conda_forge.tar.bz2\n",
            "Extracting ca-certificates-2024.2.2-hbcca054_0.conda\n",
            "Extracting ld_impl_linux-64-2.40-h41732ed_0.conda\n",
            "Extracting libstdcxx-ng-13.2.0-h7e041cc_5.conda\n",
            "Extracting pybind11-abi-4-hd8ed1ab_3.tar.bz2\n",
            "Extracting python_abi-3.10-4_cp310.conda\n",
            "Extracting tzdata-2024a-h0c530f3_0.conda\n",
            "Extracting libgomp-13.2.0-h807b86a_5.conda\n",
            "Extracting _openmp_mutex-4.5-2_gnu.tar.bz2\n",
            "Extracting libgcc-ng-13.2.0-h807b86a_5.conda\n",
            "Extracting bzip2-1.0.8-hd590300_5.conda\n",
            "Extracting c-ares-1.28.1-hd590300_0.conda\n",
            "Extracting fmt-10.2.1-h00ab1b0_0.conda\n",
            "Extracting icu-73.2-h59595ed_0.conda\n",
            "Extracting keyutils-1.6.1-h166bdaf_0.tar.bz2\n",
            "Extracting libev-4.33-hd590300_2.conda\n",
            "Extracting libffi-3.4.2-h7f98852_5.tar.bz2\n",
            "Extracting libiconv-1.17-hd590300_2.conda\n",
            "Extracting libnsl-2.0.1-hd590300_0.conda\n",
            "Extracting libuuid-2.38.1-h0b41bf4_0.conda\n",
            "Extracting libxcrypt-4.4.36-hd590300_1.conda\n",
            "Extracting libzlib-1.2.13-hd590300_5.conda\n",
            "Extracting lz4-c-1.9.4-hcb278e6_0.conda\n",
            "Extracting lzo-2.10-h516909a_1000.tar.bz2\n",
            "Extracting ncurses-6.4.20240210-h59595ed_0.conda\n",
            "Extracting openssl-3.2.1-hd590300_1.conda\n",
            "Extracting reproc-14.2.4.post0-hd590300_1.conda\n",
            "Extracting xz-5.2.6-h166bdaf_0.tar.bz2\n",
            "Extracting yaml-cpp-0.8.0-h59595ed_0.conda\n",
            "Extracting libedit-3.1.20191231-he28a2e2_2.tar.bz2\n",
            "Extracting libnghttp2-1.58.0-h47da74e_1.conda\n",
            "Extracting libsolv-0.7.28-hfc55251_2.conda\n",
            "Extracting libsqlite-3.45.2-h2797004_0.conda\n",
            "Extracting libssh2-1.11.0-h0841786_0.conda\n",
            "Extracting libxml2-2.12.6-h232c23b_1.conda\n",
            "Extracting readline-8.2-h8228510_1.conda\n",
            "Extracting reproc-cpp-14.2.4.post0-h59595ed_1.conda\n",
            "Extracting tk-8.6.13-noxft_h4845f30_101.conda\n",
            "Extracting zstd-1.5.5-hfc55251_0.conda\n",
            "Extracting krb5-1.21.2-h659d440_0.conda\n",
            "Extracting libarchive-3.7.2-h2aa1ff5_1.conda\n",
            "Extracting python-3.10.14-hd12c33a_0_cpython.conda\n",
            "Extracting libcurl-8.7.1-hca28451_0.conda\n",
            "Extracting menuinst-2.0.2-py310hff52083_0.conda\n",
            "Extracting archspec-0.2.3-pyhd8ed1ab_0.conda\n",
            "Extracting boltons-24.0.0-pyhd8ed1ab_0.conda\n",
            "Extracting brotli-python-1.1.0-py310hc6cd4ac_1.conda\n",
            "Extracting certifi-2024.2.2-pyhd8ed1ab_0.conda\n",
            "Extracting charset-normalizer-3.3.2-pyhd8ed1ab_0.conda\n",
            "Extracting colorama-0.4.6-pyhd8ed1ab_0.tar.bz2\n",
            "Extracting distro-1.9.0-pyhd8ed1ab_0.conda\n",
            "Extracting idna-3.6-pyhd8ed1ab_0.conda\n",
            "Extracting jsonpointer-2.4-py310hff52083_3.conda\n",
            "Extracting libmamba-1.5.8-had39da4_0.conda\n",
            "Extracting packaging-24.0-pyhd8ed1ab_0.conda\n",
            "Extracting platformdirs-4.2.0-pyhd8ed1ab_0.conda\n",
            "Extracting pluggy-1.4.0-pyhd8ed1ab_0.conda\n",
            "Extracting pycosat-0.6.6-py310h2372a71_0.conda\n",
            "Extracting pycparser-2.22-pyhd8ed1ab_0.conda\n",
            "Extracting pysocks-1.7.1-pyha2e5f31_6.tar.bz2\n",
            "Extracting ruamel.yaml.clib-0.2.8-py310h2372a71_0.conda\n",
            "Extracting setuptools-69.5.1-pyhd8ed1ab_0.conda\n",
            "Extracting truststore-0.8.0-pyhd8ed1ab_0.conda\n",
            "Extracting wheel-0.43.0-pyhd8ed1ab_1.conda\n",
            "Extracting cffi-1.16.0-py310h2fee648_0.conda\n",
            "Extracting jsonpatch-1.33-pyhd8ed1ab_0.conda\n",
            "Extracting libmambapy-1.5.8-py310h39ff949_0.conda\n",
            "Extracting pip-24.0-pyhd8ed1ab_0.conda\n",
            "Extracting ruamel.yaml-0.18.6-py310h2372a71_0.conda\n",
            "Extracting tqdm-4.66.2-pyhd8ed1ab_0.conda\n",
            "Extracting urllib3-2.2.1-pyhd8ed1ab_0.conda\n",
            "Extracting requests-2.31.0-pyhd8ed1ab_0.conda\n",
            "Extracting zstandard-0.22.0-py310h1275a96_0.conda\n",
            "Extracting conda-package-streaming-0.9.0-pyhd8ed1ab_0.conda\n",
            "Extracting conda-package-handling-2.2.0-pyh38be061_0.conda\n",
            "Extracting conda-24.3.0-py310hff52083_0.conda\n",
            "Extracting conda-libmamba-solver-24.1.0-pyhd8ed1ab_0.conda\n",
            "Extracting mamba-1.5.8-py310h51d5547_0.conda\n",
            "\n",
            "Installing base environment...\n",
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /root/miniforge3\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - conda-forge/linux-64::_libgcc_mutex==0.1=conda_forge[md5=d7c89558ba9fa0495403155b64376d81]\n",
            "   - conda-forge/linux-64::ca-certificates==2024.2.2=hbcca054_0[md5=2f4327a1cbe7f022401b236e915a5fef]\n",
            "   - conda-forge/linux-64::ld_impl_linux-64==2.40=h41732ed_0[md5=7aca3059a1729aa76c597603f10b0dd3]\n",
            "   - conda-forge/linux-64::libstdcxx-ng==13.2.0=h7e041cc_5[md5=f6f6600d18a4047b54f803cf708b868a]\n",
            "   - conda-forge/noarch::pybind11-abi==4=hd8ed1ab_3[md5=878f923dd6acc8aeb47a75da6c4098be]\n",
            "   - conda-forge/linux-64::python_abi==3.10=4_cp310[md5=26322ec5d7712c3ded99dd656142b8ce]\n",
            "   - conda-forge/noarch::tzdata==2024a=h0c530f3_0[md5=161081fc7cec0bfda0d86d7cb595f8d8]\n",
            "   - conda-forge/linux-64::libgomp==13.2.0=h807b86a_5[md5=d211c42b9ce49aee3734fdc828731689]\n",
            "   - conda-forge/linux-64::_openmp_mutex==4.5=2_gnu[md5=73aaf86a425cc6e73fcf236a5a46396d]\n",
            "   - conda-forge/linux-64::libgcc-ng==13.2.0=h807b86a_5[md5=d4ff227c46917d3b4565302a2bbb276b]\n",
            "   - conda-forge/linux-64::bzip2==1.0.8=hd590300_5[md5=69b8b6202a07720f448be700e300ccf4]\n",
            "   - conda-forge/linux-64::c-ares==1.28.1=hd590300_0[md5=dcde58ff9a1f30b0037a2315d1846d1f]\n",
            "   - conda-forge/linux-64::fmt==10.2.1=h00ab1b0_0[md5=35ef8bc24bd34074ebae3c943d551728]\n",
            "   - conda-forge/linux-64::icu==73.2=h59595ed_0[md5=cc47e1facc155f91abd89b11e48e72ff]\n",
            "   - conda-forge/linux-64::keyutils==1.6.1=h166bdaf_0[md5=30186d27e2c9fa62b45fb1476b7200e3]\n",
            "   - conda-forge/linux-64::libev==4.33=hd590300_2[md5=172bf1cd1ff8629f2b1179945ed45055]\n",
            "   - conda-forge/linux-64::libffi==3.4.2=h7f98852_5[md5=d645c6d2ac96843a2bfaccd2d62b3ac3]\n",
            "   - conda-forge/linux-64::libiconv==1.17=hd590300_2[md5=d66573916ffcf376178462f1b61c941e]\n",
            "   - conda-forge/linux-64::libnsl==2.0.1=hd590300_0[md5=30fd6e37fe21f86f4bd26d6ee73eeec7]\n",
            "   - conda-forge/linux-64::libuuid==2.38.1=h0b41bf4_0[md5=40b61aab5c7ba9ff276c41cfffe6b80b]\n",
            "   - conda-forge/linux-64::libxcrypt==4.4.36=hd590300_1[md5=5aa797f8787fe7a17d1b0821485b5adc]\n",
            "   - conda-forge/linux-64::libzlib==1.2.13=hd590300_5[md5=f36c115f1ee199da648e0597ec2047ad]\n",
            "   - conda-forge/linux-64::lz4-c==1.9.4=hcb278e6_0[md5=318b08df404f9c9be5712aaa5a6f0bb0]\n",
            "   - conda-forge/linux-64::lzo==2.10=h516909a_1000[md5=bb14fcb13341b81d5eb386423b9d2bac]\n",
            "   - conda-forge/linux-64::ncurses==6.4.20240210=h59595ed_0[md5=97da8860a0da5413c7c98a3b3838a645]\n",
            "   - conda-forge/linux-64::openssl==3.2.1=hd590300_1[md5=9d731343cff6ee2e5a25c4a091bf8e2a]\n",
            "   - conda-forge/linux-64::reproc==14.2.4.post0=hd590300_1[md5=82ca53502dfd5a64a80dee76dae14685]\n",
            "   - conda-forge/linux-64::xz==5.2.6=h166bdaf_0[md5=2161070d867d1b1204ea749c8eec4ef0]\n",
            "   - conda-forge/linux-64::yaml-cpp==0.8.0=h59595ed_0[md5=965eaacd7c18eb8361fd12bb9e7a57d7]\n",
            "   - conda-forge/linux-64::libedit==3.1.20191231=he28a2e2_2[md5=4d331e44109e3f0e19b4cb8f9b82f3e1]\n",
            "   - conda-forge/linux-64::libnghttp2==1.58.0=h47da74e_1[md5=700ac6ea6d53d5510591c4344d5c989a]\n",
            "   - conda-forge/linux-64::libsolv==0.7.28=hfc55251_2[md5=535bafe1ed0a5bdd3f4c125ca05d378c]\n",
            "   - conda-forge/linux-64::libsqlite==3.45.2=h2797004_0[md5=866983a220e27a80cb75e85cb30466a1]\n",
            "   - conda-forge/linux-64::libssh2==1.11.0=h0841786_0[md5=1f5a58e686b13bcfde88b93f547d23fe]\n",
            "   - conda-forge/linux-64::libxml2==2.12.6=h232c23b_1[md5=6853448e9ca1cfd5f15382afd2a6d123]\n",
            "   - conda-forge/linux-64::readline==8.2=h8228510_1[md5=47d31b792659ce70f470b5c82fdfb7a4]\n",
            "   - conda-forge/linux-64::reproc-cpp==14.2.4.post0=h59595ed_1[md5=715e1d720ec1a03715bebd237972fca5]\n",
            "   - conda-forge/linux-64::tk==8.6.13=noxft_h4845f30_101[md5=d453b98d9c83e71da0741bb0ff4d76bc]\n",
            "   - conda-forge/linux-64::zstd==1.5.5=hfc55251_0[md5=04b88013080254850d6c01ed54810589]\n",
            "   - conda-forge/linux-64::krb5==1.21.2=h659d440_0[md5=cd95826dbd331ed1be26bdf401432844]\n",
            "   - conda-forge/linux-64::libarchive==3.7.2=h2aa1ff5_1[md5=3bf887827d1968275978361a6e405e4f]\n",
            "   - conda-forge/linux-64::python==3.10.14=hd12c33a_0_cpython[md5=2b4ba962994e8bd4be9ff5b64b75aff2]\n",
            "   - conda-forge/linux-64::libcurl==8.7.1=hca28451_0[md5=755c7f876815003337d2c61ff5d047e5]\n",
            "   - conda-forge/linux-64::menuinst==2.0.2=py310hff52083_0[md5=4837faab0d3e665df57fef662148c6a3]\n",
            "   - conda-forge/noarch::archspec==0.2.3=pyhd8ed1ab_0[md5=192278292e20704f663b9c766909d67b]\n",
            "   - conda-forge/noarch::boltons==24.0.0=pyhd8ed1ab_0[md5=61de176bd62041f9cd5bd4fcd09eb0ff]\n",
            "   - conda-forge/linux-64::brotli-python==1.1.0=py310hc6cd4ac_1[md5=1f95722c94f00b69af69a066c7433714]\n",
            "   - conda-forge/noarch::certifi==2024.2.2=pyhd8ed1ab_0[md5=0876280e409658fc6f9e75d035960333]\n",
            "   - conda-forge/noarch::charset-normalizer==3.3.2=pyhd8ed1ab_0[md5=7f4a9e3fcff3f6356ae99244a014da6a]\n",
            "   - conda-forge/noarch::colorama==0.4.6=pyhd8ed1ab_0[md5=3faab06a954c2a04039983f2c4a50d99]\n",
            "   - conda-forge/noarch::distro==1.9.0=pyhd8ed1ab_0[md5=bbdb409974cd6cb30071b1d978302726]\n",
            "   - conda-forge/noarch::idna==3.6=pyhd8ed1ab_0[md5=1a76f09108576397c41c0b0c5bd84134]\n",
            "   - conda-forge/linux-64::jsonpointer==2.4=py310hff52083_3[md5=08ec1463dbc5c806a32fc431874032ca]\n",
            "   - conda-forge/linux-64::libmamba==1.5.8=had39da4_0[md5=def669885dc103d8acb7ac2ac35e0b2f]\n",
            "   - conda-forge/noarch::packaging==24.0=pyhd8ed1ab_0[md5=248f521b64ce055e7feae3105e7abeb8]\n",
            "   - conda-forge/noarch::platformdirs==4.2.0=pyhd8ed1ab_0[md5=a0bc3eec34b0fab84be6b2da94e98e20]\n",
            "   - conda-forge/noarch::pluggy==1.4.0=pyhd8ed1ab_0[md5=139e9feb65187e916162917bb2484976]\n",
            "   - conda-forge/linux-64::pycosat==0.6.6=py310h2372a71_0[md5=0adaac9a86d59adae2bc86b3cdef2df1]\n",
            "   - conda-forge/noarch::pycparser==2.22=pyhd8ed1ab_0[md5=844d9eb3b43095b031874477f7d70088]\n",
            "   - conda-forge/noarch::pysocks==1.7.1=pyha2e5f31_6[md5=2a7de29fb590ca14b5243c4c812c8025]\n",
            "   - conda-forge/linux-64::ruamel.yaml.clib==0.2.8=py310h2372a71_0[md5=dcf6d2535586c77b31425ed835610c54]\n",
            "   - conda-forge/noarch::setuptools==69.5.1=pyhd8ed1ab_0[md5=7462280d81f639363e6e63c81276bd9e]\n",
            "   - conda-forge/noarch::truststore==0.8.0=pyhd8ed1ab_0[md5=08316d001eca8854392cf2837828ea11]\n",
            "   - conda-forge/noarch::wheel==0.43.0=pyhd8ed1ab_1[md5=0b5293a157c2b5cd513dd1b03d8d3aae]\n",
            "   - conda-forge/linux-64::cffi==1.16.0=py310h2fee648_0[md5=45846a970e71ac98fd327da5d40a0a2c]\n",
            "   - conda-forge/noarch::jsonpatch==1.33=pyhd8ed1ab_0[md5=bfdb7c5c6ad1077c82a69a8642c87aff]\n",
            "   - conda-forge/linux-64::libmambapy==1.5.8=py310h39ff949_0[md5=37f8aa15b73c4691eeec15caf45aab25]\n",
            "   - conda-forge/noarch::pip==24.0=pyhd8ed1ab_0[md5=f586ac1e56c8638b64f9c8122a7b8a67]\n",
            "   - conda-forge/linux-64::ruamel.yaml==0.18.6=py310h2372a71_0[md5=50b7d9b39099cdbabf65bf27df73a793]\n",
            "   - conda-forge/noarch::tqdm==4.66.2=pyhd8ed1ab_0[md5=2b8dfb969f984497f3f98409a9545776]\n",
            "   - conda-forge/noarch::urllib3==2.2.1=pyhd8ed1ab_0[md5=08807a87fa7af10754d46f63b368e016]\n",
            "   - conda-forge/noarch::requests==2.31.0=pyhd8ed1ab_0[md5=a30144e4156cdbb236f99ebb49828f8b]\n",
            "   - conda-forge/linux-64::zstandard==0.22.0=py310h1275a96_0[md5=54698ba13cd3494547b289cd86a2176a]\n",
            "   - conda-forge/noarch::conda-package-streaming==0.9.0=pyhd8ed1ab_0[md5=38253361efb303deead3eab39ae9269b]\n",
            "   - conda-forge/noarch::conda-package-handling==2.2.0=pyh38be061_0[md5=8a3ae7f6318376aa08ea753367bb7dd6]\n",
            "   - conda-forge/linux-64::conda==24.3.0=py310hff52083_0[md5=4187d17c4b75d8f7757820e835c507c9]\n",
            "   - conda-forge/noarch::conda-libmamba-solver==24.1.0=pyhd8ed1ab_0[md5=304dc78ad6e52e0fd663df1d484c1531]\n",
            "   - conda-forge/linux-64::mamba==1.5.8=py310h51d5547_0[md5=3b335eaa4894cbb5379a75f83a4d6b40]\n",
            "\n",
            "\n",
            "  Package                         Version  Build               Channel         Size\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  \u001b[32m+ _libgcc_mutex          \u001b[0m           0.1  conda_forge         conda-forge         \n",
            "  \u001b[32m+ ca-certificates        \u001b[0m      2024.2.2  hbcca054_0          conda-forge         \n",
            "  \u001b[32m+ ld_impl_linux-64       \u001b[0m          2.40  h41732ed_0          conda-forge         \n",
            "  \u001b[32m+ libstdcxx-ng           \u001b[0m        13.2.0  h7e041cc_5          conda-forge         \n",
            "  \u001b[32m+ pybind11-abi           \u001b[0m             4  hd8ed1ab_3          conda-forge         \n",
            "  \u001b[32m+ python_abi             \u001b[0m          3.10  4_cp310             conda-forge         \n",
            "  \u001b[32m+ tzdata                 \u001b[0m         2024a  h0c530f3_0          conda-forge         \n",
            "  \u001b[32m+ libgomp                \u001b[0m        13.2.0  h807b86a_5          conda-forge         \n",
            "  \u001b[32m+ _openmp_mutex          \u001b[0m           4.5  2_gnu               conda-forge         \n",
            "  \u001b[32m+ libgcc-ng              \u001b[0m        13.2.0  h807b86a_5          conda-forge         \n",
            "  \u001b[32m+ bzip2                  \u001b[0m         1.0.8  hd590300_5          conda-forge         \n",
            "  \u001b[32m+ c-ares                 \u001b[0m        1.28.1  hd590300_0          conda-forge         \n",
            "  \u001b[32m+ fmt                    \u001b[0m        10.2.1  h00ab1b0_0          conda-forge         \n",
            "  \u001b[32m+ icu                    \u001b[0m          73.2  h59595ed_0          conda-forge         \n",
            "  \u001b[32m+ keyutils               \u001b[0m         1.6.1  h166bdaf_0          conda-forge         \n",
            "  \u001b[32m+ libev                  \u001b[0m          4.33  hd590300_2          conda-forge         \n",
            "  \u001b[32m+ libffi                 \u001b[0m         3.4.2  h7f98852_5          conda-forge         \n",
            "  \u001b[32m+ libiconv               \u001b[0m          1.17  hd590300_2          conda-forge         \n",
            "  \u001b[32m+ libnsl                 \u001b[0m         2.0.1  hd590300_0          conda-forge         \n",
            "  \u001b[32m+ libuuid                \u001b[0m        2.38.1  h0b41bf4_0          conda-forge         \n",
            "  \u001b[32m+ libxcrypt              \u001b[0m        4.4.36  hd590300_1          conda-forge         \n",
            "  \u001b[32m+ libzlib                \u001b[0m        1.2.13  hd590300_5          conda-forge         \n",
            "  \u001b[32m+ lz4-c                  \u001b[0m         1.9.4  hcb278e6_0          conda-forge         \n",
            "  \u001b[32m+ lzo                    \u001b[0m          2.10  h516909a_1000       conda-forge         \n",
            "  \u001b[32m+ ncurses                \u001b[0m  6.4.20240210  h59595ed_0          conda-forge         \n",
            "  \u001b[32m+ openssl                \u001b[0m         3.2.1  hd590300_1          conda-forge         \n",
            "  \u001b[32m+ reproc                 \u001b[0m  14.2.4.post0  hd590300_1          conda-forge         \n",
            "  \u001b[32m+ xz                     \u001b[0m         5.2.6  h166bdaf_0          conda-forge         \n",
            "  \u001b[32m+ yaml-cpp               \u001b[0m         0.8.0  h59595ed_0          conda-forge         \n",
            "  \u001b[32m+ libedit                \u001b[0m  3.1.20191231  he28a2e2_2          conda-forge         \n",
            "  \u001b[32m+ libnghttp2             \u001b[0m        1.58.0  h47da74e_1          conda-forge         \n",
            "  \u001b[32m+ libsolv                \u001b[0m        0.7.28  hfc55251_2          conda-forge         \n",
            "  \u001b[32m+ libsqlite              \u001b[0m        3.45.2  h2797004_0          conda-forge         \n",
            "  \u001b[32m+ libssh2                \u001b[0m        1.11.0  h0841786_0          conda-forge         \n",
            "  \u001b[32m+ libxml2                \u001b[0m        2.12.6  h232c23b_1          conda-forge         \n",
            "  \u001b[32m+ readline               \u001b[0m           8.2  h8228510_1          conda-forge         \n",
            "  \u001b[32m+ reproc-cpp             \u001b[0m  14.2.4.post0  h59595ed_1          conda-forge         \n",
            "  \u001b[32m+ tk                     \u001b[0m        8.6.13  noxft_h4845f30_101  conda-forge         \n",
            "  \u001b[32m+ zstd                   \u001b[0m         1.5.5  hfc55251_0          conda-forge         \n",
            "  \u001b[32m+ krb5                   \u001b[0m        1.21.2  h659d440_0          conda-forge         \n",
            "  \u001b[32m+ libarchive             \u001b[0m         3.7.2  h2aa1ff5_1          conda-forge         \n",
            "  \u001b[32m+ python                 \u001b[0m       3.10.14  hd12c33a_0_cpython  conda-forge         \n",
            "  \u001b[32m+ libcurl                \u001b[0m         8.7.1  hca28451_0          conda-forge         \n",
            "  \u001b[32m+ menuinst               \u001b[0m         2.0.2  py310hff52083_0     conda-forge         \n",
            "  \u001b[32m+ archspec               \u001b[0m         0.2.3  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ boltons                \u001b[0m        24.0.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ brotli-python          \u001b[0m         1.1.0  py310hc6cd4ac_1     conda-forge         \n",
            "  \u001b[32m+ certifi                \u001b[0m      2024.2.2  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ charset-normalizer     \u001b[0m         3.3.2  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ colorama               \u001b[0m         0.4.6  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ distro                 \u001b[0m         1.9.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ idna                   \u001b[0m           3.6  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ jsonpointer            \u001b[0m           2.4  py310hff52083_3     conda-forge         \n",
            "  \u001b[32m+ libmamba               \u001b[0m         1.5.8  had39da4_0          conda-forge         \n",
            "  \u001b[32m+ packaging              \u001b[0m          24.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ platformdirs           \u001b[0m         4.2.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ pluggy                 \u001b[0m         1.4.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ pycosat                \u001b[0m         0.6.6  py310h2372a71_0     conda-forge         \n",
            "  \u001b[32m+ pycparser              \u001b[0m          2.22  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ pysocks                \u001b[0m         1.7.1  pyha2e5f31_6        conda-forge         \n",
            "  \u001b[32m+ ruamel.yaml.clib       \u001b[0m         0.2.8  py310h2372a71_0     conda-forge         \n",
            "  \u001b[32m+ setuptools             \u001b[0m        69.5.1  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ truststore             \u001b[0m         0.8.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ wheel                  \u001b[0m        0.43.0  pyhd8ed1ab_1        conda-forge         \n",
            "  \u001b[32m+ cffi                   \u001b[0m        1.16.0  py310h2fee648_0     conda-forge         \n",
            "  \u001b[32m+ jsonpatch              \u001b[0m          1.33  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ libmambapy             \u001b[0m         1.5.8  py310h39ff949_0     conda-forge         \n",
            "  \u001b[32m+ pip                    \u001b[0m          24.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ ruamel.yaml            \u001b[0m        0.18.6  py310h2372a71_0     conda-forge         \n",
            "  \u001b[32m+ tqdm                   \u001b[0m        4.66.2  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ urllib3                \u001b[0m         2.2.1  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ requests               \u001b[0m        2.31.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ zstandard              \u001b[0m        0.22.0  py310h1275a96_0     conda-forge         \n",
            "  \u001b[32m+ conda-package-streaming\u001b[0m         0.9.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ conda-package-handling \u001b[0m         2.2.0  pyh38be061_0        conda-forge         \n",
            "  \u001b[32m+ conda                  \u001b[0m        24.3.0  py310hff52083_0     conda-forge         \n",
            "  \u001b[32m+ conda-libmamba-solver  \u001b[0m        24.1.0  pyhd8ed1ab_0        conda-forge         \n",
            "  \u001b[32m+ mamba                  \u001b[0m         1.5.8  py310h51d5547_0     conda-forge         \n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 78 packages\n",
            "\n",
            "  Total download: 0 B\n",
            "\n",
            "─────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25hLinking _libgcc_mutex-0.1-conda_forge\n",
            "Linking ca-certificates-2024.2.2-hbcca054_0\n",
            "Linking ld_impl_linux-64-2.40-h41732ed_0\n",
            "Linking libstdcxx-ng-13.2.0-h7e041cc_5\n",
            "Linking pybind11-abi-4-hd8ed1ab_3\n",
            "Linking python_abi-3.10-4_cp310\n",
            "Linking tzdata-2024a-h0c530f3_0\n",
            "Linking libgomp-13.2.0-h807b86a_5\n",
            "Linking _openmp_mutex-4.5-2_gnu\n",
            "Linking libgcc-ng-13.2.0-h807b86a_5\n",
            "Linking bzip2-1.0.8-hd590300_5\n",
            "Linking c-ares-1.28.1-hd590300_0\n",
            "Linking fmt-10.2.1-h00ab1b0_0\n",
            "Linking icu-73.2-h59595ed_0\n",
            "Linking keyutils-1.6.1-h166bdaf_0\n",
            "Linking libev-4.33-hd590300_2\n",
            "Linking libffi-3.4.2-h7f98852_5\n",
            "Linking libiconv-1.17-hd590300_2\n",
            "Linking libnsl-2.0.1-hd590300_0\n",
            "Linking libuuid-2.38.1-h0b41bf4_0\n",
            "Linking libxcrypt-4.4.36-hd590300_1\n",
            "Linking libzlib-1.2.13-hd590300_5\n",
            "Linking lz4-c-1.9.4-hcb278e6_0\n",
            "Linking lzo-2.10-h516909a_1000\n",
            "Linking ncurses-6.4.20240210-h59595ed_0\n",
            "Linking openssl-3.2.1-hd590300_1\n",
            "Linking reproc-14.2.4.post0-hd590300_1\n",
            "Linking xz-5.2.6-h166bdaf_0\n",
            "Linking yaml-cpp-0.8.0-h59595ed_0\n",
            "Linking libedit-3.1.20191231-he28a2e2_2\n",
            "Linking libnghttp2-1.58.0-h47da74e_1\n",
            "Linking libsolv-0.7.28-hfc55251_2\n",
            "Linking libsqlite-3.45.2-h2797004_0\n",
            "Linking libssh2-1.11.0-h0841786_0\n",
            "Linking libxml2-2.12.6-h232c23b_1\n",
            "Linking readline-8.2-h8228510_1\n",
            "Linking reproc-cpp-14.2.4.post0-h59595ed_1\n",
            "Linking tk-8.6.13-noxft_h4845f30_101\n",
            "Linking zstd-1.5.5-hfc55251_0\n",
            "Linking krb5-1.21.2-h659d440_0\n",
            "Linking libarchive-3.7.2-h2aa1ff5_1\n",
            "Linking python-3.10.14-hd12c33a_0_cpython\n",
            "Linking libcurl-8.7.1-hca28451_0\n",
            "Linking menuinst-2.0.2-py310hff52083_0\n",
            "Linking archspec-0.2.3-pyhd8ed1ab_0\n",
            "Linking boltons-24.0.0-pyhd8ed1ab_0\n",
            "Linking brotli-python-1.1.0-py310hc6cd4ac_1\n",
            "Linking certifi-2024.2.2-pyhd8ed1ab_0\n",
            "Linking charset-normalizer-3.3.2-pyhd8ed1ab_0\n",
            "Linking colorama-0.4.6-pyhd8ed1ab_0\n",
            "Linking distro-1.9.0-pyhd8ed1ab_0\n",
            "Linking idna-3.6-pyhd8ed1ab_0\n",
            "Linking jsonpointer-2.4-py310hff52083_3\n",
            "Linking libmamba-1.5.8-had39da4_0\n",
            "Linking packaging-24.0-pyhd8ed1ab_0\n",
            "Linking platformdirs-4.2.0-pyhd8ed1ab_0\n",
            "Linking pluggy-1.4.0-pyhd8ed1ab_0\n",
            "Linking pycosat-0.6.6-py310h2372a71_0\n",
            "Linking pycparser-2.22-pyhd8ed1ab_0\n",
            "Linking pysocks-1.7.1-pyha2e5f31_6\n",
            "Linking ruamel.yaml.clib-0.2.8-py310h2372a71_0\n",
            "Linking setuptools-69.5.1-pyhd8ed1ab_0\n",
            "Linking truststore-0.8.0-pyhd8ed1ab_0\n",
            "Linking wheel-0.43.0-pyhd8ed1ab_1\n",
            "Linking cffi-1.16.0-py310h2fee648_0\n",
            "Linking jsonpatch-1.33-pyhd8ed1ab_0\n",
            "Linking libmambapy-1.5.8-py310h39ff949_0\n",
            "Linking pip-24.0-pyhd8ed1ab_0\n",
            "Linking ruamel.yaml-0.18.6-py310h2372a71_0\n",
            "Linking tqdm-4.66.2-pyhd8ed1ab_0\n",
            "Linking urllib3-2.2.1-pyhd8ed1ab_0\n",
            "Linking requests-2.31.0-pyhd8ed1ab_0\n",
            "Linking zstandard-0.22.0-py310h1275a96_0\n",
            "Linking conda-package-streaming-0.9.0-pyhd8ed1ab_0\n",
            "Linking conda-package-handling-2.2.0-pyh38be061_0\n",
            "Linking conda-24.3.0-py310hff52083_0\n",
            "Linking conda-libmamba-solver-24.1.0-pyhd8ed1ab_0\n",
            "Linking mamba-1.5.8-py310h51d5547_0\n",
            "\n",
            "Transaction finished\n",
            "\n",
            "To activate this environment, use:\n",
            "\n",
            "    micromamba activate /root/miniforge3\n",
            "\n",
            "Or to execute a single command in this environment, use:\n",
            "\n",
            "    micromamba run -p /root/miniforge3 mycommand\n",
            "\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniforge3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniforge3: /root/miniforge3\n"
          ]
        }
      ],
      "source": [
        "!curl -L -O \"https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-$(uname)-$(uname -m).sh\"\n",
        "!bash Miniforge3-$(uname)-$(uname -m).sh -b"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8x9r1_oSqMyW",
        "outputId": "3e2b0a9b-c3f2-4a99-814c-8abda4877fc7"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "no change     /root/miniforge3/condabin/conda\n",
            "no change     /root/miniforge3/bin/conda\n",
            "no change     /root/miniforge3/bin/conda-env\n",
            "no change     /root/miniforge3/bin/activate\n",
            "no change     /root/miniforge3/bin/deactivate\n",
            "no change     /root/miniforge3/etc/profile.d/conda.sh\n",
            "no change     /root/miniforge3/etc/fish/conf.d/conda.fish\n",
            "no change     /root/miniforge3/shell/condabin/Conda.psm1\n",
            "no change     /root/miniforge3/shell/condabin/conda-hook.ps1\n",
            "no change     /root/miniforge3/lib/python3.10/site-packages/xontrib/conda.xsh\n",
            "no change     /root/miniforge3/etc/profile.d/conda.csh\n",
            "no change     /root/.bashrc\n",
            "No action taken.\n",
            "Added mamba to /root/.bashrc\n",
            "\n",
            "==> For changes to take effect, close and re-open your current shell. <==\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!~/miniforge3/bin/mamba init"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "7eKYZcKCrUeL",
        "outputId": "c874b79d-c4be-412b-db8f-46da3b1ad565"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "conda-forge/linux-64                                        Using cache\n",
            "conda-forge/noarch                                          Using cache\n",
            "bioconda/linux-64                                           Using cache\n",
            "bioconda/noarch                                             Using cache\n",
            "Transaction\n",
            "\n",
            "  Prefix: /root/miniforge3/envs/snakemake\n",
            "\n",
            "  Updating specs:\n",
            "\n",
            "   - snakemake\n",
            "\n",
            "\n",
            "  Package                                   Version  Build                Channel           Size\n",
            "──────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "  Install:\n",
            "──────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "  + python_abi                                 3.12  4_cp312              conda-forge     Cached\n",
            "  + _libgcc_mutex                               0.1  conda_forge          conda-forge     Cached\n",
            "  + ld_impl_linux-64                           2.40  hf3520f5_7           conda-forge     Cached\n",
            "  + ca-certificates                        2024.7.4  hbcca054_0           conda-forge     Cached\n",
            "  + libgomp                                  14.1.0  h77fa898_0           conda-forge     Cached\n",
            "  + _openmp_mutex                               4.5  2_gnu                conda-forge     Cached\n",
            "  + libgcc-ng                                14.1.0  h77fa898_0           conda-forge     Cached\n",
            "  + libgfortran5                             14.1.0  hc5f4f2c_0           conda-forge     Cached\n",
            "  + libstdcxx-ng                             14.1.0  hc0a3c3a_0           conda-forge     Cached\n",
            "  + openssl                                   3.3.1  h4bc722e_2           conda-forge     Cached\n",
            "  + libzlib                                   1.3.1  h4ab18f5_1           conda-forge     Cached\n",
            "  + libxcrypt                                4.4.36  hd590300_1           conda-forge     Cached\n",
            "  + libffi                                    3.4.2  h7f98852_5           conda-forge     Cached\n",
            "  + bzip2                                     1.0.8  h4bc722e_7           conda-forge     Cached\n",
            "  + yaml                                      0.2.5  h7f98852_2           conda-forge     Cached\n",
            "  + ncurses                                     6.5  h59595ed_0           conda-forge     Cached\n",
            "  + libuuid                                  2.38.1  h0b41bf4_0           conda-forge     Cached\n",
            "  + libnsl                                    2.0.1  hd590300_0           conda-forge     Cached\n",
            "  + libexpat                                  2.6.2  h59595ed_0           conda-forge     Cached\n",
            "  + xz                                        5.2.6  h166bdaf_0           conda-forge     Cached\n",
            "  + libgfortran-ng                           14.1.0  h69a702a_0           conda-forge     Cached\n",
            "  + zstd                                      1.5.6  ha6fb4c9_0           conda-forge     Cached\n",
            "  + tk                                       8.6.13  noxft_h4845f30_101   conda-forge     Cached\n",
            "  + libsqlite                                3.46.0  hde9e2c9_0           conda-forge     Cached\n",
            "  + readline                                    8.2  h8228510_1           conda-forge     Cached\n",
            "  + libopenblas                              0.3.27  pthreads_hac2b453_1  conda-forge     Cached\n",
            "  + libblas                                   3.9.0  23_linux64_openblas  conda-forge     Cached\n",
            "  + libcblas                                  3.9.0  23_linux64_openblas  conda-forge     Cached\n",
            "  + liblapack                                 3.9.0  23_linux64_openblas  conda-forge     Cached\n",
            "  + liblapacke                                3.9.0  23_linux64_openblas  conda-forge     Cached\n",
            "  + coin-or-utils                           2.11.11  h8c65801_1           conda-forge     Cached\n",
            "  + coin-or-osi                            0.108.10  haf5fa05_0           conda-forge     Cached\n",
            "  + coin-or-clp                              1.17.8  h1ee7a9c_0           conda-forge     Cached\n",
            "  + coin-or-cgl                              0.60.7  h516709c_0           conda-forge     Cached\n",
            "  + coin-or-cbc                             2.10.11  h56f689f_0           conda-forge     Cached\n",
            "  + tzdata                                    2024a  h0c530f3_0           conda-forge     Cached\n",
            "  + coincbc                                 2.10.11  0_metapackage        conda-forge     Cached\n",
            "  + python                                   3.12.4  h194c7f8_0_cpython   conda-forge     Cached\n",
            "  + wheel                                    0.43.0  pyhd8ed1ab_1         conda-forge     Cached\n",
            "  + setuptools                               71.0.4  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + pip                                        24.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + pyparsing                                 3.1.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + pycparser                                  2.22  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + mdurl                                     0.1.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + tomli                                     2.0.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + iniconfig                                 2.0.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + exceptiongroup                            1.2.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + colorama                                  0.4.6  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + six                                      1.16.0  pyh6c4a22f_0         conda-forge     Cached\n",
            "  + platformdirs                              4.2.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + pytz                                     2024.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + python-tzdata                            2024.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + hyperframe                                6.0.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + pluggy                                    1.5.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + smmap                                     5.0.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + typing_extensions                        4.12.2  pyha770c72_0         conda-forge     Cached\n",
            "  + zipp                                     3.19.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + attrs                                    23.2.0  pyh71513ae_0         conda-forge     Cached\n",
            "  + pkgutil-resolve-name                     1.3.10  pyhd8ed1ab_1         conda-forge     Cached\n",
            "  + traitlets                                5.14.3  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + python-fastjsonschema                    2.20.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + logmuse                                   0.2.6  pyh8c360ce_0         conda-forge     Cached\n",
            "  + charset-normalizer                        3.3.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + hpack                                     4.0.0  pyh9f0ad1d_0         conda-forge     Cached\n",
            "  + pysocks                                   1.7.1  pyha2e5f31_6         conda-forge     Cached\n",
            "  + idna                                        3.7  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + certifi                                2024.7.4  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + plac                                      1.4.3  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + slack-sdk                                3.31.0  pyha770c72_0         conda-forge     Cached\n",
            "  + pygments                                 2.18.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + argparse-dataclass                        2.0.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + dpath                                     2.2.0  pyha770c72_0         conda-forge     Cached\n",
            "  + throttler                                 1.2.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + stopit                                    1.1.2  py_0                 conda-forge     Cached\n",
            "  + reretry                                  0.11.8  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + tabulate                                  0.9.0  pyhd8ed1ab_1         conda-forge     Cached\n",
            "  + packaging                                  24.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + humanfriendly                              10.0  pyhd8ed1ab_6         conda-forge     Cached\n",
            "  + docutils                                 0.21.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + configargparse                              1.7  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + appdirs                                   1.4.4  pyh9f0ad1d_0         conda-forge     Cached\n",
            "  + toposort                                   1.10  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + connection_pool                           0.0.3  pyhd3deb0d_0         conda-forge     Cached\n",
            "  + markdown-it-py                            3.0.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + python-dateutil                           2.9.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + gitdb                                    4.0.11  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + importlib_resources                       6.4.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + h2                                        4.1.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + slack_sdk                                3.31.0  hd8ed1ab_0           conda-forge     Cached\n",
            "  + pytest                                    8.3.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + amply                                     0.1.6  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + rich                                     13.7.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + gitpython                                3.1.43  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + veracitools                               0.1.3  py_0                 conda-forge     Cached\n",
            "  + ubiquerg                                  0.8.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + attmap                                   0.13.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + psutil                                    6.0.0  py312h9a8786e_0      conda-forge     Cached\n",
            "  + numpy                                     2.0.1  py312h1103770_0      conda-forge     Cached\n",
            "  + markupsafe                                2.1.5  py312h98912ed_0      conda-forge     Cached\n",
            "  + rpds-py                                  0.19.1  py312hf008fa9_0      conda-forge     Cached\n",
            "  + brotli-python                             1.1.0  py312h30efb56_1      conda-forge     Cached\n",
            "  + wrapt                                    1.16.0  py312h98912ed_0      conda-forge     Cached\n",
            "  + pyyaml                                    6.0.1  py312h98912ed_1      conda-forge     Cached\n",
            "  + datrie                                    0.8.2  py312h98912ed_7      conda-forge     Cached\n",
            "  + immutables                                 0.20  py312h98912ed_1      conda-forge     Cached\n",
            "  + cffi                                     1.16.0  py312hf06ca03_0      conda-forge     Cached\n",
            "  + jupyter_core                              5.7.2  py312h7900ff3_0      conda-forge     Cached\n",
            "  + pulp                                      2.8.0  py312h7900ff3_0      conda-forge     Cached\n",
            "  + pandas                                    2.2.2  py312h1d6d2e6_1      conda-forge     Cached\n",
            "  + zstandard                                0.23.0  py312h3483029_0      conda-forge     Cached\n",
            "  + snakemake-interface-common               1.17.2  pyhdfd78af_0         bioconda        Cached\n",
            "  + snakemake-interface-storage-plugins       3.2.3  pyhdfd78af_0         bioconda        Cached\n",
            "  + snakemake-interface-executor-plugins      9.2.0  pyhdfd78af_0         bioconda        Cached\n",
            "  + snakemake-interface-report-plugins        1.0.0  pyhdfd78af_0         bioconda        Cached\n",
            "  + jinja2                                    3.1.4  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + referencing                              0.35.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + smart_open                                7.0.4  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + conda-inject                              1.3.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + yte                                       1.5.4  pyha770c72_0         conda-forge     Cached\n",
            "  + peppy                                    0.40.5  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + urllib3                                   2.2.2  pyhd8ed1ab_1         conda-forge     Cached\n",
            "  + jsonschema-specifications             2023.12.1  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + requests                                 2.32.3  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + jsonschema                               4.23.0  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + eido                                      0.2.2  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + nbformat                                 5.10.4  pyhd8ed1ab_0         conda-forge     Cached\n",
            "  + snakemake-minimal                        8.16.0  pyhdfd78af_0         bioconda        Cached\n",
            "  + snakemake                                8.16.0  hdfd78af_0           bioconda        Cached\n",
            "\n",
            "  Summary:\n",
            "\n",
            "  Install: 128 packages\n",
            "\n",
            "  Total download: 0 B\n",
            "\n",
            "──────────────────────────────────────────────────────────────────────────────────────────────────\n",
            "\n",
            "\n",
            "\n",
            "Looking for: ['snakemake']\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages: ...working... done\n",
            "Preparing transaction: ...working... done\n",
            "Verifying transaction: ...working... done\n",
            "Executing transaction: ...working... done\n",
            "\n",
            "To activate this environment, use\n",
            "\n",
            "     $ mamba activate snakemake\n",
            "\n",
            "To deactivate an active environment, use\n",
            "\n",
            "     $ mamba deactivate\n",
            "\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "__conda_setup=\"$('/root/miniforge3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)\"\n",
        "if [ $? -eq 0 ]; then\n",
        "    eval \"$__conda_setup\"\n",
        "else\n",
        "    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\n",
        "        . \"/root/miniforge3/etc/profile.d/conda.sh\"\n",
        "    else\n",
        "        export PATH=\"/root/miniforge3/bin:$PATH\"\n",
        "    fi\n",
        "fi\n",
        "unset __conda_setup\n",
        "\n",
        "if [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\n",
        "    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\n",
        "fi\n",
        "mamba config --add channels defaults\n",
        "mamba config --add channels bioconda\n",
        "mamba config --add channels conda-forge\n",
        "mamba config --set channel_priority strict\n",
        "mamba create -y -n snakemake snakemake"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pZo4PldUrqT-",
        "outputId": "08ae37b6-b695-4f17-a9d0-9ba42218f0f5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "usage: snakemake [-h] [--dry-run] [--profile PROFILE] [--workflow-profile WORKFLOW_PROFILE]\n",
            "                 [--cache [RULE ...]] [--snakefile FILE] [--cores N] [--jobs N] [--local-cores N]\n",
            "                 [--resources NAME=INT [NAME=INT ...]]\n",
            "                 [--set-threads RULE=THREADS [RULE=THREADS ...]] [--max-threads MAX_THREADS]\n",
            "                 [--set-resources RULE:RESOURCE=VALUE [RULE:RESOURCE=VALUE ...]]\n",
            "                 [--set-scatter NAME=SCATTERITEMS [NAME=SCATTERITEMS ...]]\n",
            "                 [--set-resource-scopes RESOURCE=[global|local] [RESOURCE=[global|local] ...]]\n",
            "                 [--default-resources [NAME=INT ...]]\n",
            "                 [--preemptible-rules [PREEMPTIBLE_RULES ...]]\n",
            "                 [--preemptible-retries PREEMPTIBLE_RETRIES] [--configfile FILE [FILE ...]]\n",
            "                 [--config [KEY=VALUE ...]] [--envvars VARNAME [VARNAME ...]] [--directory DIR]\n",
            "                 [--touch] [--keep-going]\n",
            "                 [--rerun-triggers {code,input,mtime,params,software-env} [{code,input,mtime,params,software-env} ...]]\n",
            "                 [--force] [--executor {local,dryrun,touch}] [--forceall]\n",
            "                 [--forcerun [TARGET ...]] [--prioritize TARGET [TARGET ...]]\n",
            "                 [--batch RULE=BATCH/BATCHES] [--until TARGET [TARGET ...]]\n",
            "                 [--omit-from TARGET [TARGET ...]] [--rerun-incomplete] [--shadow-prefix DIR]\n",
            "                 [--scheduler [{ilp,greedy}]] [--wms-monitor [WMS_MONITOR]]\n",
            "                 [--wms-monitor-arg [NAME=VALUE ...]]\n",
            "                 [--scheduler-ilp-solver {PULP_CBC_CMD,COIN_CMD}]\n",
            "                 [--scheduler-solver-path SCHEDULER_SOLVER_PATH]\n",
            "                 [--conda-base-path CONDA_BASE_PATH] [--no-subworkflows] [--precommand PRECOMMAND]\n",
            "                 [--groups GROUPS [GROUPS ...]]\n",
            "                 [--group-components GROUP_COMPONENTS [GROUP_COMPONENTS ...]] [--report [FILE]]\n",
            "                 [--report-stylesheet CSSFILE] [--reporter PLUGIN] [--draft-notebook TARGET]\n",
            "                 [--edit-notebook TARGET] [--notebook-listen IP:PORT] [--lint [{text,json}]]\n",
            "                 [--generate-unit-tests [TESTPATH]] [--containerize] [--export-cwl FILE]\n",
            "                 [--list-rules] [--list-target-rules] [--dag] [--rulegraph] [--filegraph]\n",
            "                 [--d3dag] [--summary] [--detailed-summary] [--archive FILE]\n",
            "                 [--cleanup-metadata FILE [FILE ...]] [--cleanup-shadow] [--skip-script-cleanup]\n",
            "                 [--unlock] [--list-changes {input,params,code}] [--list-input-changes]\n",
            "                 [--list-params-changes] [--list-untracked]\n",
            "                 [--delete-all-output | --delete-temp-output] [--keep-incomplete]\n",
            "                 [--drop-metadata] [--deploy-sources QUERY CHECKSUM] [--version]\n",
            "                 [--printshellcmds] [--debug-dag] [--nocolor] [--quiet [{all,progress,rules} ...]]\n",
            "                 [--print-compilation] [--verbose] [--force-use-threads] [--allow-ambiguity]\n",
            "                 [--nolock] [--ignore-incomplete] [--max-inventory-time SECONDS]\n",
            "                 [--latency-wait SECONDS] [--wait-for-files [FILE ...]]\n",
            "                 [--wait-for-files-file FILE] [--queue-input-wait-time SECONDS] [--notemp]\n",
            "                 [--all-temp] [--unneeded-temp-files FILE [FILE ...]]\n",
            "                 [--keep-storage-local-copies] [--target-files-omit-workdir-adjustment]\n",
            "                 [--allowed-rules ALLOWED_RULES [ALLOWED_RULES ...]]\n",
            "                 [--target-jobs TARGET_JOBS [TARGET_JOBS ...]] [--local-groupid LOCAL_GROUPID]\n",
            "                 [--max-jobs-per-second MAX_JOBS_PER_SECOND]\n",
            "                 [--max-status-checks-per-second MAX_STATUS_CHECKS_PER_SECOND]\n",
            "                 [--seconds-between-status-checks SECONDS_BETWEEN_STATUS_CHECKS]\n",
            "                 [--retries RETRIES] [--attempt ATTEMPT] [--wrapper-prefix WRAPPER_PREFIX]\n",
            "                 [--default-storage-provider DEFAULT_STORAGE_PROVIDER]\n",
            "                 [--default-storage-prefix DEFAULT_STORAGE_PREFIX]\n",
            "                 [--local-storage-prefix LOCAL_STORAGE_PREFIX]\n",
            "                 [--remote-job-local-storage-prefix REMOTE_JOB_LOCAL_STORAGE_PREFIX]\n",
            "                 [--shared-fs-usage {input-output,persistence,software-deployment,source-cache,sources,storage-local-copies,none} [{input-output,persistence,software-deployment,source-cache,sources,storage-local-copies,none} ...]]\n",
            "                 [--scheduler-greediness SCHEDULER_GREEDINESS] [--no-hooks] [--debug]\n",
            "                 [--runtime-profile FILE] [--mode {subprocess,remote,default}]\n",
            "                 [--show-failed-logs] [--log-handler-script FILE] [--log-service {none,slack,wms}]\n",
            "                 [--job-deploy-sources] [--benchmark-extended] [--container-image IMAGE]\n",
            "                 [--immediate-submit] [--jobscript SCRIPT] [--jobname NAME] [--flux]\n",
            "                 [--software-deployment-method {apptainer,conda,env-modules} [{apptainer,conda,env-modules} ...]]\n",
            "                 [--container-cleanup-images] [--use-conda]\n",
            "                 [--conda-not-block-search-path-envvars] [--list-conda-envs] [--conda-prefix DIR]\n",
            "                 [--conda-cleanup-envs] [--conda-cleanup-pkgs [{tarballs,cache}]]\n",
            "                 [--conda-create-envs-only] [--conda-frontend {conda,mamba}] [--use-apptainer]\n",
            "                 [--apptainer-prefix DIR] [--apptainer-args ARGS] [--use-envmodules]\n",
            "                 [--report-html-path VALUE] [--report-html-stylesheet-path VALUE]\n",
            "                 [targets ...]\n",
            "\n",
            "Snakemake is a Python based language and execution environment for GNU Make-like workflows.\n",
            "\n",
            "options:\n",
            "  -h, --help            show this help message and exit\n",
            "\n",
            "EXECUTION:\n",
            "  targets               Targets to build. May be rules or files.\n",
            "  --dry-run, --dryrun, -n\n",
            "                        Do not execute anything, and display what would be done. If you have a\n",
            "                        very large workflow, use --dry-run --quiet to just print a summary of the\n",
            "                        DAG of jobs. (default: False)\n",
            "  --profile PROFILE     Name of profile to use for configuring Snakemake. Snakemake will search\n",
            "                        for a corresponding folder in /etc/xdg/snakemake and\n",
            "                        /root/.config/snakemake. Alternatively, this can be an absolute or\n",
            "                        relative path. The profile folder has to contain a file 'config.yaml'.\n",
            "                        This file can be used to set default values for command line options in\n",
            "                        YAML format. For example, '--cluster qsub' becomes 'cluster: qsub' in the\n",
            "                        YAML file. Profiles can be obtained from https://github.com/snakemake-\n",
            "                        profiles. The profile can also be set via the environment variable\n",
            "                        $SNAKEMAKE_PROFILE. To override this variable and use no profile at all,\n",
            "                        provide the value 'none' to this argument. [env var: SNAKEMAKE_PROFILE]\n",
            "  --workflow-profile WORKFLOW_PROFILE\n",
            "                        Path (relative to current directory) to workflow specific profile folder\n",
            "                        to use for configuring Snakemake with parameters specific for this\n",
            "                        workflow (like resources). If this flag is not used, Snakemake will by\n",
            "                        default use 'profiles/default' if present (searched both relative to\n",
            "                        current directory and relative to Snakefile, in this order). For skipping\n",
            "                        any workflow specific profile provide the special value 'none'. Settings\n",
            "                        made in the workflow profile will override settings made in the general\n",
            "                        profile (see --profile). The profile folder has to contain a file\n",
            "                        'config.yaml'. This file can be used to set default values for command\n",
            "                        line options in YAML format. For example, '--executor slurm' becomes\n",
            "                        'executor: slurm' in the YAML file. It is advisable to use the workflow\n",
            "                        profile to set or overwrite e.g. workflow specific resources like the\n",
            "                        amount of threads of a particular rule or the amount of memory needed.\n",
            "                        Note that in such cases, the arguments may be given as nested YAML\n",
            "                        mappings in the profile, e.g. 'set-threads: myrule: 4' instead of 'set-\n",
            "                        threads: myrule=4'.\n",
            "  --cache [RULE ...]    Store output files of given rules in a central cache given by the\n",
            "                        environment variable $SNAKEMAKE_OUTPUT_CACHE. Likewise, retrieve output\n",
            "                        files of the given rules from this cache if they have been created before\n",
            "                        (by anybody writing to the same cache), instead of actually executing the\n",
            "                        rules. Output files are identified by hashing all steps, parameters and\n",
            "                        software stack (conda envs or containers) needed to create them.\n",
            "  --snakefile FILE, -s FILE\n",
            "                        The workflow definition in form of a snakefile.Usually, you should not\n",
            "                        need to specify this. By default, Snakemake will search for 'Snakefile',\n",
            "                        'snakefile', 'workflow/Snakefile', 'workflow/snakefile' beneath the\n",
            "                        current working directory, in this order. Only if you definitely want a\n",
            "                        different layout, you need to use this parameter.\n",
            "  --cores N, -c N       Use at most N CPU cores/jobs in parallel. If N is omitted or 'all', the\n",
            "                        limit is set to the number of available CPU cores. In case of\n",
            "                        cluster/cloud execution, this argument sets the maximum number of cores\n",
            "                        requested from the cluster or cloud scheduler. (See https://snakemake.read\n",
            "                        thedocs.io/en/stable/snakefiles/rules.html#resources-remote-execution for\n",
            "                        more info)This number is available to rules via workflow.cores.\n",
            "  --jobs N, -j N        Use at most N CPU cluster/cloud jobs in parallel. For local execution this\n",
            "                        is an alias for --cores (it is though recommended to use --cores in that\n",
            "                        case). Note: Set to 'unlimited' to allow any number of parallel jobs.\n",
            "  --local-cores N       In cluster/cloud mode, use at most N cores of the host machine in parallel\n",
            "                        (default: number of CPU cores of the host). The cores are used to execute\n",
            "                        local rules. This option is ignored when not in cluster/cloud mode.\n",
            "                        (default: <available CPU count>)\n",
            "  --resources NAME=INT [NAME=INT ...], --res NAME=INT [NAME=INT ...]\n",
            "                        Define additional resources that shall constrain the scheduling\n",
            "                        analogously to --cores (see above). A resource is defined as a name and an\n",
            "                        integer value. E.g. --resources mem_mb=1000. Rules can use resources by\n",
            "                        defining the resource keyword, e.g. resources: mem_mb=600. If now two\n",
            "                        rules require 600 of the resource 'mem_mb' they won't be run in parallel\n",
            "                        by the scheduler. In cluster/cloud mode, this argument will also constrain\n",
            "                        the amount of resources requested from the server. (See https://snakemake.\n",
            "                        readthedocs.io/en/stable/snakefiles/rules.html#resources-remote-execution\n",
            "                        for more info) (default: {})\n",
            "  --set-threads RULE=THREADS [RULE=THREADS ...]\n",
            "                        Overwrite thread usage of rules. This allows to fine-tune workflow\n",
            "                        parallelization. In particular, this is helpful to target certain cluster\n",
            "                        nodes by e.g. shifting a rule to use more, or less threads than defined in\n",
            "                        the workflow. Thereby, THREADS has to be a positive integer, and RULE has\n",
            "                        to be the name of the rule. (default: {})\n",
            "  --max-threads MAX_THREADS\n",
            "                        Define a global maximum number of threads available to any rule. Rules\n",
            "                        requesting more threads (via the threads keyword) will have their values\n",
            "                        reduced to the maximum. This can be useful when you want to restrict the\n",
            "                        maximum number of threads without modifying the workflow definition or\n",
            "                        overwriting rules individually with --set-threads.\n",
            "  --set-resources RULE:RESOURCE=VALUE [RULE:RESOURCE=VALUE ...]\n",
            "                        Overwrite resource usage of rules. This allows to fine-tune workflow\n",
            "                        resources. In particular, this is helpful to target certain cluster nodes\n",
            "                        by e.g. defining a certain partition for a rule, or overriding a temporary\n",
            "                        directory. Thereby, VALUE has to be a positive integer or a string, RULE\n",
            "                        has to be the name of the rule, and RESOURCE has to be the name of the\n",
            "                        resource. (default: {})\n",
            "  --set-scatter NAME=SCATTERITEMS [NAME=SCATTERITEMS ...]\n",
            "                        Overwrite number of scatter items of scattergather processes. This allows\n",
            "                        to fine-tune workflow parallelization. Thereby, SCATTERITEMS has to be a\n",
            "                        positive integer, and NAME has to be the name of the scattergather process\n",
            "                        defined via a scattergather directive in the workflow. (default: {})\n",
            "  --set-resource-scopes RESOURCE=[global|local] [RESOURCE=[global|local] ...]\n",
            "                        Overwrite resource scopes. A scope determines how a constraint is reckoned\n",
            "                        in cluster execution. With RESOURCE=local, a constraint applied to\n",
            "                        RESOURCE using --resources will be considered the limit for each group\n",
            "                        submission. With RESOURCE=global, the constraint will apply across all\n",
            "                        groups cumulatively. By default, only `mem_mb` and `disk_mb` are\n",
            "                        considered local, all other resources are global. This may be modified in\n",
            "                        the snakefile using the `resource_scopes:` directive. Note that number of\n",
            "                        threads, specified via --cores, is always considered local. (See https://s\n",
            "                        nakemake.readthedocs.io/en/stable/snakefiles/rules.html#resources-remote-\n",
            "                        execution for more info) (default: {})\n",
            "  --default-resources [NAME=INT ...], --default-res [NAME=INT ...]\n",
            "                        Define default values of resources for rules that do not define their own\n",
            "                        values. In addition to plain integers, python expressions over inputsize\n",
            "                        are allowed (e.g. '2*input.size_mb'). The inputsize is the sum of the\n",
            "                        sizes of all input files of a rule. By default, Snakemake assumes a\n",
            "                        default for mem_mb, disk_mb, and tmpdir (see below). This option allows to\n",
            "                        add further defaults (e.g. account and partition for slurm) or to\n",
            "                        overwrite these default values. The defaults are\n",
            "                        'mem_mb=min(max(2*input.size_mb, 1000), 8000)',\n",
            "                        'disk_mb=max(2*input.size_mb, 1000)' (i.e., default disk and mem usage is\n",
            "                        twice the input file size but at least 1GB), and the system temporary\n",
            "                        directory (as given by $TMPDIR, $TEMP, or $TMP) is used for the tmpdir\n",
            "                        resource. The tmpdir resource is automatically used by shell commands,\n",
            "                        scripts and wrappers to store temporary data (as it is mirrored into\n",
            "                        $TMPDIR, $TEMP, and $TMP for the executed subprocesses). If this argument\n",
            "                        is not specified at all, Snakemake just uses the tmpdir resource as\n",
            "                        outlined above.\n",
            "  --preemptible-rules [PREEMPTIBLE_RULES ...]\n",
            "                        Define which rules shall use a preemptible machine which can be\n",
            "                        prematurely killed by e.g. a cloud provider (also called spot instances).\n",
            "                        This is currently only supported by the Google Life Sciences executor and\n",
            "                        ignored by all other executors. If no rule names are provided, all rules\n",
            "                        are considered to be preemptible.\n",
            "  --preemptible-retries PREEMPTIBLE_RETRIES\n",
            "                        Number of retries that shall be made in order to finish a job from of rule\n",
            "                        that has been marked as preemptible via the --preemptible-rules setting.\n",
            "  --configfile FILE [FILE ...], --configfiles FILE [FILE ...]\n",
            "                        Specify or overwrite the config file of the workflow (see the docs).\n",
            "                        Values specified in JSON or YAML format are available in the global config\n",
            "                        dictionary inside the workflow. Multiple files overwrite each other in the\n",
            "                        given order. Thereby missing keys in previous config files are extended by\n",
            "                        following configfiles. Note that this order also includes a config file\n",
            "                        defined in the workflow definition itself (which will come first).\n",
            "                        (default: [])\n",
            "  --config [KEY=VALUE ...], -C [KEY=VALUE ...]\n",
            "                        Set or overwrite values in the workflow config object. The workflow config\n",
            "                        object is accessible as variable config inside the workflow. Default\n",
            "                        values can be set by providing a YAML JSON file (see --configfile and\n",
            "                        Documentation).\n",
            "  --envvars VARNAME [VARNAME ...]\n",
            "                        Environment variables to pass to cloud jobs.\n",
            "  --directory DIR, -d DIR\n",
            "                        Specify working directory (relative paths in the snakefile will use this\n",
            "                        as their origin).\n",
            "  --touch, -t           Touch output files (mark them up to date without really changing them)\n",
            "                        instead of running their commands. This is used to pretend that the rules\n",
            "                        were executed, in order to fool future invocations of snakemake. Fails if\n",
            "                        a file does not yet exist. Note that this will only touch files that would\n",
            "                        otherwise be recreated by Snakemake (e.g. because their input files are\n",
            "                        newer). For enforcing a touch, combine this with --force, --forceall, or\n",
            "                        --forcerun. Note however that you lose the provenance information when the\n",
            "                        files have been created in reality. Hence, this should be used only as a\n",
            "                        last resort. (default: False)\n",
            "  --keep-going, -k      Go on with independent jobs if a job fails. (default: False)\n",
            "  --rerun-triggers {code,input,mtime,params,software-env} [{code,input,mtime,params,software-env} ...]\n",
            "                        Define what triggers the rerunning of a job. By default, all triggers are\n",
            "                        used, which guarantees that results are consistent with the workflow code\n",
            "                        and configuration. If you rather prefer the traditional way of just\n",
            "                        considering file modification dates, use '--rerun-trigger mtime'.\n",
            "                        (default: frozenset({<RerunTrigger.MTIME: 0>, <RerunTrigger.INPUT: 2>,\n",
            "                        <RerunTrigger.CODE: 4>, <RerunTrigger.SOFTWARE_ENV: 3>,\n",
            "                        <RerunTrigger.PARAMS: 1>}))\n",
            "  --force, -f           Force the execution of the selected target or the first rule regardless of\n",
            "                        already created output. (default: False)\n",
            "  --executor {local,dryrun,touch}, -e {local,dryrun,touch}\n",
            "                        Specify a custom executor, available via an executor plugin:\n",
            "                        snakemake_executor_<name>\n",
            "  --forceall, -F        Force the execution of the selected (or the first) rule and all rules it\n",
            "                        is dependent on regardless of already created output. (default: False)\n",
            "  --forcerun [TARGET ...], -R [TARGET ...]\n",
            "                        Force the re-execution or creation of the given rules or files. Use this\n",
            "                        option if you changed a rule and want to have all its output in your\n",
            "                        workflow updated.\n",
            "  --prioritize TARGET [TARGET ...], -P TARGET [TARGET ...]\n",
            "                        Tell the scheduler to assign creation of given targets (and all their\n",
            "                        dependencies) highest priority.\n",
            "  --batch RULE=BATCH/BATCHES\n",
            "                        Only create the given BATCH of the input files of the given RULE. This can\n",
            "                        be used to iteratively run parts of very large workflows. Only the\n",
            "                        execution plan of the relevant part of the workflow has to be calculated,\n",
            "                        thereby speeding up DAG computation. It is recommended to provide the most\n",
            "                        suitable rule for batching when documenting a workflow. It should be some\n",
            "                        aggregating rule that would be executed only once, and has a large number\n",
            "                        of input files. For example, it can be a rule that aggregates over\n",
            "                        samples.\n",
            "  --until TARGET [TARGET ...], -U TARGET [TARGET ...]\n",
            "                        Runs the pipeline until it reaches the specified rules or files. Only runs\n",
            "                        jobs that are dependencies of the specified rule or files, does not run\n",
            "                        sibling DAGs.\n",
            "  --omit-from TARGET [TARGET ...], -O TARGET [TARGET ...]\n",
            "                        Prevent the execution or creation of the given rules or files as well as\n",
            "                        any rules or files that are downstream of these targets in the DAG. Also\n",
            "                        runs jobs in sibling DAGs that are independent of the rules or files\n",
            "                        specified here.\n",
            "  --rerun-incomplete, --ri\n",
            "                        Re-run all jobs the output of which is recognized as incomplete. (default:\n",
            "                        False)\n",
            "  --shadow-prefix DIR   Specify a directory in which the 'shadow' directory is created. If not\n",
            "                        supplied, the value is set to the '.snakemake' directory relative to the\n",
            "                        working directory.\n",
            "  --scheduler [{ilp,greedy}]\n",
            "                        Specifies if jobs are selected by a greedy algorithm or by solving an ilp.\n",
            "                        The ilp scheduler aims to reduce runtime and hdd usage by best possible\n",
            "                        use of resources. (default: ilp)\n",
            "  --wms-monitor [WMS_MONITOR]\n",
            "                        IP and port of workflow management system to monitor the execution of\n",
            "                        snakemake (e.g. http://127.0.0.1:5000) Note that if your service requires\n",
            "                        an authorization token, you must export WMS_MONITOR_TOKEN in the\n",
            "                        environment.\n",
            "  --wms-monitor-arg [NAME=VALUE ...]\n",
            "                        If the workflow management service accepts extra arguments, provide. them\n",
            "                        in key value pairs with --wms-monitor-arg. For example, to run an existing\n",
            "                        workflow using a wms monitor, you can provide the pair id=12345 and the\n",
            "                        arguments will be provided to the endpoint to first interact with the\n",
            "                        workflow\n",
            "  --scheduler-ilp-solver {PULP_CBC_CMD,COIN_CMD}\n",
            "                        Specifies solver to be utilized when selecting ilp-scheduler. (default:\n",
            "                        COIN_CMD)\n",
            "  --scheduler-solver-path SCHEDULER_SOLVER_PATH\n",
            "                        Set the PATH to search for scheduler solver binaries (internal use only).\n",
            "  --conda-base-path CONDA_BASE_PATH\n",
            "                        Path of conda base installation (home of conda, mamba, activate) (internal\n",
            "                        use only).\n",
            "  --no-subworkflows, --nosw\n",
            "                        Do not evaluate or execute subworkflows. (default: False)\n",
            "  --precommand PRECOMMAND\n",
            "                        Only used in case of remote execution. Command to be executed before\n",
            "                        Snakemake executes each job on the remote compute node.\n",
            "\n",
            "GROUPING:\n",
            "  --groups GROUPS [GROUPS ...]\n",
            "                        Assign rules to groups (this overwrites any group definitions from the\n",
            "                        workflow). (default: {})\n",
            "  --group-components GROUP_COMPONENTS [GROUP_COMPONENTS ...]\n",
            "                        Set the number of connected components a group is allowed to span. By\n",
            "                        default, this is 1, but this flag allows to extend this. This can be used\n",
            "                        to run e.g. 3 jobs of the same rule in the same group, although they are\n",
            "                        not connected. It can be helpful for putting together many small jobs or\n",
            "                        benefitting of shared memory setups. (default: {})\n",
            "\n",
            "REPORTS:\n",
            "  --report [FILE]       Create an HTML report with results and statistics. This can be either a\n",
            "                        .html file or a .zip file. In the former case, all results are embedded\n",
            "                        into the .html (this only works for small data). In the latter case,\n",
            "                        results are stored along with a file report.html in the zip archive. If no\n",
            "                        filename is given, an embedded report.html is the default.\n",
            "  --report-stylesheet CSSFILE\n",
            "                        Custom stylesheet to use for report. In particular, this can be used for\n",
            "                        branding the report with e.g. a custom logo, see docs.\n",
            "  --reporter PLUGIN     Specify a custom report plugin. By default, Snakemake's builtin html\n",
            "                        reporter will be used. For custom reporters, check out their command line\n",
            "                        options starting with --report-.\n",
            "\n",
            "NOTEBOOKS:\n",
            "  --draft-notebook TARGET\n",
            "                        Draft a skeleton notebook for the rule used to generate the given target\n",
            "                        file. This notebook can then be opened in a jupyter server, executed and\n",
            "                        implemented until ready. After saving, it will automatically be reused in\n",
            "                        non-interactive mode by Snakemake for subsequent jobs.\n",
            "  --edit-notebook TARGET\n",
            "                        Interactively edit the notebook associated with the rule used to generate\n",
            "                        the given target file. This will start a local jupyter notebook server.\n",
            "                        Any changes to the notebook should be saved, and the server has to be\n",
            "                        stopped by closing the notebook and hitting the 'Quit' button on the\n",
            "                        jupyter dashboard. Afterwards, the updated notebook will be automatically\n",
            "                        stored in the path defined in the rule. If the notebook is not yet\n",
            "                        present, this will create an empty draft.\n",
            "  --notebook-listen IP:PORT\n",
            "                        The IP address and PORT the notebook server used for editing the notebook\n",
            "                        (--edit-notebook) will listen on. (default: localhost:8888)\n",
            "\n",
            "UTILITIES:\n",
            "  --lint [{text,json}]  Perform linting on the given workflow. This will print snakemake specific\n",
            "                        suggestions to improve code quality (work in progress, more lints to be\n",
            "                        added in the future). If no argument is provided, plain text output is\n",
            "                        used.\n",
            "  --generate-unit-tests [TESTPATH]\n",
            "                        Automatically generate unit tests for each workflow rule. This assumes\n",
            "                        that all input files of each job are already present. Rules without a job\n",
            "                        with present input files will be skipped (a warning will be issued). For\n",
            "                        each rule, one test case will be created in the specified test folder\n",
            "                        (.tests/unit by default). After successful execution, tests can be run\n",
            "                        with 'pytest TESTPATH'.\n",
            "  --containerize        Print a Dockerfile that provides an execution environment for the\n",
            "                        workflow, including all conda environments. (default: False)\n",
            "  --export-cwl FILE     Compile workflow to CWL and store it in given FILE.\n",
            "  --list-rules, --list, -l\n",
            "                        Show available rules in given Snakefile. (default: False)\n",
            "  --list-target-rules, --lt\n",
            "                        Show available target rules in given Snakefile. (default: False)\n",
            "  --dag                 Do not execute anything and print the directed acyclic graph of jobs in\n",
            "                        the dot language. Recommended use on Unix systems: snakemake --dag | dot |\n",
            "                        display. Note print statements in your Snakefile may interfere with\n",
            "                        visualization. (default: False)\n",
            "  --rulegraph           Do not execute anything and print the dependency graph of rules in the dot\n",
            "                        language. This will be less crowded than above DAG of jobs, but also show\n",
            "                        less information. Note that each rule is displayed once, hence the\n",
            "                        displayed graph will be cyclic if a rule appears in several steps of the\n",
            "                        workflow. Use this if above option leads to a DAG that is too large.\n",
            "                        Recommended use on Unix systems: snakemake --rulegraph | dot | display.\n",
            "                        Note print statements in your Snakefile may interfere with visualization.\n",
            "                        (default: False)\n",
            "  --filegraph           Do not execute anything and print the dependency graph of rules with their\n",
            "                        input and output files in the dot language. This is an intermediate\n",
            "                        solution between above DAG of jobs and the rule graph. Note that each rule\n",
            "                        is displayed once, hence the displayed graph will be cyclic if a rule\n",
            "                        appears in several steps of the workflow. Use this if above option leads\n",
            "                        to a DAG that is too large. Recommended use on Unix systems: snakemake\n",
            "                        --filegraph | dot | display. Note print statements in your Snakefile may\n",
            "                        interfere with visualization. (default: False)\n",
            "  --d3dag               Print the DAG in D3.js compatible JSON format. (default: False)\n",
            "  --summary, -S         Print a summary of all files created by the workflow. The has the\n",
            "                        following columns: filename, modification time, rule version, status,\n",
            "                        plan. Thereby rule version contains the versionthe file was created with\n",
            "                        (see the version keyword of rules), and status denotes whether the file is\n",
            "                        missing, its input files are newer or if version or implementation of the\n",
            "                        rule changed since file creation. Finally the last column denotes whether\n",
            "                        the file will be updated or created during the next workflow execution.\n",
            "                        (default: False)\n",
            "  --detailed-summary, -D\n",
            "                        Print a summary of all files created by the workflow. The has the\n",
            "                        following columns: filename, modification time, rule version, input\n",
            "                        file(s), shell command, status, plan. Thereby rule version contains the\n",
            "                        version the file was created with (see the version keyword of rules), and\n",
            "                        status denotes whether the file is missing, its input files are newer or\n",
            "                        if version or implementation of the rule changed since file creation. The\n",
            "                        input file and shell command columns are self explanatory. Finally the\n",
            "                        last column denotes whether the file will be updated or created during the\n",
            "                        next workflow execution. (default: False)\n",
            "  --archive FILE        Archive the workflow into the given tar archive FILE. The archive will be\n",
            "                        created such that the workflow can be re-executed on a vanilla system. The\n",
            "                        function needs conda and git to be installed. It will archive every file\n",
            "                        that is under git version control. Note that it is best practice to have\n",
            "                        the Snakefile, config files, and scripts under version control. Hence,\n",
            "                        they will be included in the archive. Further, it will add input files\n",
            "                        that are not generated by by the workflow itself and conda environments.\n",
            "                        Note that symlinks are dereferenced. Supported formats are .tar, .tar.gz,\n",
            "                        .tar.bz2 and .tar.xz.\n",
            "  --cleanup-metadata FILE [FILE ...], --cm FILE [FILE ...]\n",
            "                        Cleanup the metadata of given files. That means that snakemake removes any\n",
            "                        tracked version info, and any marks that files are incomplete.\n",
            "  --cleanup-shadow      Cleanup old shadow directories which have not been deleted due to failures\n",
            "                        or power loss. (default: False)\n",
            "  --skip-script-cleanup\n",
            "                        Don't delete wrapper scripts used for execution (default: False)\n",
            "  --unlock              Remove a lock on the working directory. (default: False)\n",
            "  --list-changes {input,params,code}, --lc {input,params,code}\n",
            "                        List all output files for which the given items (code, input, params) have\n",
            "                        changed since creation.\n",
            "  --list-input-changes, --li\n",
            "                        List all output files for which the defined input files have changed in\n",
            "                        the Snakefile (e.g. new input files were added in the rule definition or\n",
            "                        files were renamed). For listing input file modification in the\n",
            "                        filesystem, use --summary. (default: False)\n",
            "  --list-params-changes, --lp\n",
            "                        List all output files for which the defined params have changed in the\n",
            "                        Snakefile. (default: False)\n",
            "  --list-untracked, --lu\n",
            "                        List all files in the working directory that are not used in the workflow.\n",
            "                        This can be used e.g. for identifying leftover files. Hidden files and\n",
            "                        directories are ignored. (default: False)\n",
            "  --delete-all-output   Remove all files generated by the workflow. Use together with --dry-run to\n",
            "                        list files without actually deleting anything. Note that this will not\n",
            "                        recurse into subworkflows. Write-protected files are not removed.\n",
            "                        Nevertheless, use with care! (default: False)\n",
            "  --delete-temp-output  Remove all temporary files generated by the workflow. Use together with\n",
            "                        --dry-run to list files without actually deleting anything. Note that this\n",
            "                        will not recurse into subworkflows. (default: False)\n",
            "  --keep-incomplete     Do not remove incomplete output files by failed jobs. (default: False)\n",
            "  --drop-metadata       Drop metadata file tracking information after job finishes. Provenance-\n",
            "                        information based reports (e.g. --report and the --list_x_changes\n",
            "                        functions) will be empty or incomplete. (default: False)\n",
            "  --deploy-sources QUERY CHECKSUM\n",
            "                        Deploy sources archive from given storage provider query to the current\n",
            "                        working sdirectory and control for archive checksum to proceed. Meant for\n",
            "                        internal use only.\n",
            "  --version, -v         show program's version number and exit\n",
            "\n",
            "OUTPUT:\n",
            "  --printshellcmds, -p  Print out the shell commands that will be executed. (default: False)\n",
            "  --debug-dag           Print candidate and selected jobs (including their wildcards) while\n",
            "                        inferring DAG. This can help to debug unexpected DAG topology or errors.\n",
            "                        (default: False)\n",
            "  --nocolor             Do not use a colored output. (default: False)\n",
            "  --quiet [{all,progress,rules} ...], -q [{all,progress,rules} ...]\n",
            "                        Do not output certain information. If used without arguments, do not\n",
            "                        output any progress or rule information. Defining 'all' results in no\n",
            "                        information being printed at all.\n",
            "  --print-compilation   Print the python representation of the workflow. (default: False)\n",
            "  --verbose             Print debugging output. (default: False)\n",
            "\n",
            "BEHAVIOR:\n",
            "  --force-use-threads   Force threads rather than processes. Helpful if shared memory (/dev/shm)\n",
            "                        is full or unavailable. (default: False)\n",
            "  --allow-ambiguity, -a\n",
            "                        Don't check for ambiguous rules and simply use the first if several can\n",
            "                        produce the same file. This allows the user to prioritize rules by their\n",
            "                        order in the snakefile. (default: False)\n",
            "  --nolock              Do not lock the working directory (default: False)\n",
            "  --ignore-incomplete, --ii\n",
            "                        Do not check for incomplete output files. (default: False)\n",
            "  --max-inventory-time SECONDS\n",
            "                        Spend at most SECONDS seconds to create a file inventory for the working\n",
            "                        directory. The inventory vastly speeds up file modification and existence\n",
            "                        checks when computing which jobs need to be executed. However, creating\n",
            "                        the inventory itself can be slow, e.g. on network file systems. Hence, we\n",
            "                        do not spend more than a given amount of time and fall back to individual\n",
            "                        checks for the rest. (default: 20)\n",
            "  --latency-wait SECONDS, --output-wait SECONDS, -w SECONDS\n",
            "                        Wait given seconds if an output file of a job is not present after the job\n",
            "                        finished. This helps if your filesystem suffers from latency (default 5).\n",
            "                        (default: 5)\n",
            "  --wait-for-files [FILE ...]\n",
            "                        Wait --latency-wait seconds for these files to be present before executing\n",
            "                        the workflow. This option is used internally to handle filesystem latency\n",
            "                        in cluster environments.\n",
            "  --wait-for-files-file FILE\n",
            "                        Same behaviour as --wait-for-files, but file list is stored in file\n",
            "                        instead of being passed on the commandline. This is useful when the list\n",
            "                        of files is too long to be passed on the commandline.\n",
            "  --queue-input-wait-time SECONDS\n",
            "                        Set the interval in seconds to check for new input in rules that use\n",
            "                        from_queue to obtain input files. (default: 10)\n",
            "  --notemp, --nt        Ignore temp() declarations. This is useful when running only a part of the\n",
            "                        workflow, since temp() would lead to deletion of probably needed files by\n",
            "                        other parts of the workflow. (default: False)\n",
            "  --all-temp            Mark all output files as temp files. This can be useful for CI testing, in\n",
            "                        order to save space. (default: False)\n",
            "  --unneeded-temp-files FILE [FILE ...]\n",
            "                        Given files will not be uploaded to storage and immediately deleted after\n",
            "                        job or group job completion.\n",
            "  --keep-storage-local-copies\n",
            "                        Keep local copies of remote input files. (default: False)\n",
            "  --target-files-omit-workdir-adjustment\n",
            "                        Do not adjust the paths of given target files relative to the working\n",
            "                        directory. (default: False)\n",
            "  --allowed-rules ALLOWED_RULES [ALLOWED_RULES ...]\n",
            "                        Only consider given rules. If omitted, all rules in Snakefile are used.\n",
            "                        Note that this is intended primarily for internal use and may lead to\n",
            "                        unexpected results otherwise.\n",
            "  --target-jobs TARGET_JOBS [TARGET_JOBS ...]\n",
            "                        Target particular jobs by RULE:WILDCARD1=VALUE,WILDCARD2=VALUE,... This is\n",
            "                        meant for internal use by Snakemake itself only.\n",
            "  --local-groupid LOCAL_GROUPID\n",
            "                        Name for local groupid, meant for internal use only. (default: local)\n",
            "  --max-jobs-per-second MAX_JOBS_PER_SECOND\n",
            "                        Maximal number of cluster/drmaa jobs per second, default is 10, fractions\n",
            "                        allowed. (default: 10)\n",
            "  --max-status-checks-per-second MAX_STATUS_CHECKS_PER_SECOND\n",
            "                        Maximal number of job status checks per second, default is 10, fractions\n",
            "                        allowed. (default: 10)\n",
            "  --seconds-between-status-checks SECONDS_BETWEEN_STATUS_CHECKS\n",
            "                        Number of seconds to wait between two rounds of status checks. (default:\n",
            "                        10)\n",
            "  --retries RETRIES, --restart-times RETRIES, -T RETRIES\n",
            "                        Number of times to restart failing jobs (defaults to 0). (default: 0)\n",
            "  --attempt ATTEMPT     Internal use only: define the initial value of the attempt parameter\n",
            "                        (default: 1). (default: 1)\n",
            "  --wrapper-prefix WRAPPER_PREFIX\n",
            "                        Prefix for URL created from wrapper directive (default:\n",
            "                        https://github.com/snakemake/snakemake-wrappers/raw/). Set this to a\n",
            "                        different URL to use your fork or a local clone of the repository, e.g.,\n",
            "                        use a git URL like 'git+file://path/to/your/local/clone@'. (default:\n",
            "                        https://github.com/snakemake/snakemake-wrappers/raw/)\n",
            "  --default-storage-provider DEFAULT_STORAGE_PROVIDER\n",
            "                        Specify default storage provider to be used for all input and output files\n",
            "                        that don't yet specify one (e.g. 's3'). See\n",
            "                        https://snakemake.github.io/snakemake-plugin-catalog for available storage\n",
            "                        provider plugins. If not set or explicitly 'none', no default storage\n",
            "                        provider will be used.\n",
            "  --default-storage-prefix DEFAULT_STORAGE_PREFIX\n",
            "                        Specify prefix for default storage provider. E.g. a bucket name.\n",
            "  --local-storage-prefix LOCAL_STORAGE_PREFIX\n",
            "                        Specify prefix for storing local copies of storage files and folders. By\n",
            "                        default, this is a hidden subfolder in the workdir. It can however be\n",
            "                        freely chosen, e.g. in order to store those files on a local scratch disk.\n",
            "                        Environment variables will be expanded. (default: .snakemake/storage)\n",
            "  --remote-job-local-storage-prefix REMOTE_JOB_LOCAL_STORAGE_PREFIX\n",
            "                        Specify prefix for storing local copies of storage files and folders in\n",
            "                        case of remote jobs (e.g. cluster or cloud jobs). This may differ from\n",
            "                        --local-storage-prefix. If not set, uses value of --local-storage-prefix.\n",
            "                        By default, this is a hidden subfolder in the workdir. It can however be\n",
            "                        freely chosen, e.g. in order to store those files on a local scratch disk.\n",
            "                        Environment variables will be expanded. In case they shall be expanded\n",
            "                        only within the remote job, mask them with a leading backslash, i.e.\n",
            "                        \\$SLURM_JOB_ID.\n",
            "  --shared-fs-usage {input-output,persistence,software-deployment,source-cache,sources,storage-local-copies,none} [{input-output,persistence,software-deployment,source-cache,sources,storage-local-copies,none} ...]\n",
            "                        Set assumptions on shared filesystem for non-local workflow execution. To\n",
            "                        disable any sharing via the filesystem, specify 'none'. Usually, the\n",
            "                        executor plugin sets this to a correct default. However, sometimes it is\n",
            "                        worth tuning this setting, e.g. for optimizing cluster performance. For\n",
            "                        example, when using '--default-storage-provider fs' together with a\n",
            "                        cluster executor like slurm, you might want to set '--shared-fs-usage\n",
            "                        persistence software-deployment sources source-cache', such that software\n",
            "                        deployment and data provenance will be handled by NFS but input and output\n",
            "                        files will be handled exclusively by the storage provider. (default:\n",
            "                        frozenset({<SharedFSUsage.SOFTWARE_DEPLOYMENT: 2>,\n",
            "                        <SharedFSUsage.STORAGE_LOCAL_COPIES: 4>, <SharedFSUsage.INPUT_OUTPUT: 1>,\n",
            "                        <SharedFSUsage.PERSISTENCE: 0>, <SharedFSUsage.SOURCE_CACHE: 5>,\n",
            "                        <SharedFSUsage.SOURCES: 3>}))\n",
            "  --scheduler-greediness SCHEDULER_GREEDINESS, --greediness SCHEDULER_GREEDINESS\n",
            "                        Set the greediness of scheduling. This value between 0 and 1 determines\n",
            "                        how careful jobs are selected for execution. The default value (1.0)\n",
            "                        provides the best speed and still acceptable scheduling quality.\n",
            "  --no-hooks            Do not invoke onstart, onsuccess or onerror hooks after execution.\n",
            "                        (default: False)\n",
            "  --debug               Allow to debug rules with e.g. PDB. This flag allows to set breakpoints in\n",
            "                        run blocks. (default: False)\n",
            "  --runtime-profile FILE\n",
            "                        Profile Snakemake and write the output to FILE. This requires yappi to be\n",
            "                        installed.\n",
            "  --mode {subprocess,remote,default}\n",
            "                        Set execution mode of Snakemake (internal use only). (default: default)\n",
            "  --show-failed-logs    Automatically display logs of failed jobs. (default: False)\n",
            "  --log-handler-script FILE\n",
            "                        Provide a custom script containing a function 'def log_handler(msg):'.\n",
            "                        Snakemake will call this function for every logging output (given as a\n",
            "                        dictionary msg)allowing to e.g. send notifications in the form of e.g.\n",
            "                        slack messages or emails.\n",
            "  --log-service {none,slack,wms}\n",
            "                        Set a specific messaging service for logging output.Snakemake will notify\n",
            "                        the service on errors and completed execution.Currently slack and workflow\n",
            "                        management system (wms) are supported.\n",
            "  --job-deploy-sources  Whether the workflow sources shall be deployed before a remote job is\n",
            "                        started. Only applies if --no-shared-fs is set or executors are used that\n",
            "                        imply no shared FS (e.g. the kubernetes executor). (default: False)\n",
            "  --benchmark-extended  Write extended benchmarking metrics. (default: False)\n",
            "\n",
            "REMOTE EXECUTION:\n",
            "  --container-image IMAGE\n",
            "                        Docker image to use, e.g., when submitting jobs to kubernetes. Defaults to\n",
            "                        'https://hub.docker.com/r/snakemake/snakemake', tagged with the same\n",
            "                        version as the currently running Snakemake instance. Note that overwriting\n",
            "                        this value is up to your responsibility. Any used image has to contain a\n",
            "                        working snakemake installation that is compatible with (or ideally the\n",
            "                        same as) the currently running version. (default:\n",
            "                        snakemake/snakemake:v8.16.0)\n",
            "  --immediate-submit, --is\n",
            "                        Immediately submit all jobs to the cluster instead of waiting for present\n",
            "                        input files. This will fail, unless you make the cluster aware of job\n",
            "                        dependencies, e.g. via: $ snakemake --cluster 'sbatch --dependency\n",
            "                        {dependencies}. Assuming that your submit script (here sbatch) outputs the\n",
            "                        generated job id to the first stdout line, {dependencies} will be filled\n",
            "                        with space separated job ids this job depends on. Does not work for\n",
            "                        workflows that contain checkpoint rules. (default: False)\n",
            "  --jobscript SCRIPT, --js SCRIPT\n",
            "                        Provide a custom job script for submission to the cluster. The default\n",
            "                        script resides as 'jobscript.sh' in the installation directory.\n",
            "  --jobname NAME, --jn NAME\n",
            "                        Provide a custom name for the jobscript that is submitted to the cluster\n",
            "                        (see --cluster). NAME is \"snakejob.{name}.{jobid}.sh\" per default. The\n",
            "                        wildcard {jobid} has to be present in the name. (default:\n",
            "                        snakejob.{name}.{jobid}.sh)\n",
            "\n",
            "FLUX:\n",
            "  --flux                Execute your workflow on a flux cluster. Flux can work with both a shared\n",
            "                        network filesystem (like NFS) or without. If you don't have a shared\n",
            "                        filesystem, additionally specify --no-shared-fs. (default: False)\n",
            "\n",
            "SOFTWARE DEPLOYMENT:\n",
            "  --software-deployment-method {apptainer,conda,env-modules} [{apptainer,conda,env-modules} ...], --deployment-method {apptainer,conda,env-modules} [{apptainer,conda,env-modules} ...], --deployment {apptainer,conda,env-modules} [{apptainer,conda,env-modules} ...], --sdm {apptainer,conda,env-modules} [{apptainer,conda,env-modules} ...]\n",
            "                        Specify software environment deployment method.\n",
            "  --container-cleanup-images\n",
            "                        Remove unused containers (default: False)\n",
            "\n",
            "CONDA:\n",
            "  --use-conda           If defined in the rule, run job in a conda environment. If this flag is\n",
            "                        not set, the conda directive is ignored. (default: False)\n",
            "  --conda-not-block-search-path-envvars\n",
            "                        Do not block environment variables that modify the search path (R_LIBS,\n",
            "                        PYTHONPATH, PERL5LIB, PERLLIB) when using conda environments. (default:\n",
            "                        False)\n",
            "  --list-conda-envs     List all conda environments and their location on disk. (default: False)\n",
            "  --conda-prefix DIR    Specify a directory in which the 'conda' and 'conda-archive' directories\n",
            "                        are created. These are used to store conda environments and their\n",
            "                        archives, respectively. If not supplied, the value is set to the\n",
            "                        '.snakemake' directory relative to the invocation directory. If supplied,\n",
            "                        the `--use-conda` flag must also be set. The value may be given as a\n",
            "                        relative path, which will be extrapolated to the invocation directory, or\n",
            "                        as an absolute path. The value can also be provided via the environment\n",
            "                        variable $SNAKEMAKE_CONDA_PREFIX. In any case, the prefix may contain\n",
            "                        environment variables which will be properly expanded. Note that if you\n",
            "                        use remote execution e.g. on a cluster and you have node specific values\n",
            "                        for this, you should disable assuming shared fs for software-deployment\n",
            "                        (see --shared-fs-usage).\n",
            "  --conda-cleanup-envs  Cleanup unused conda environments. (default: False)\n",
            "  --conda-cleanup-pkgs [{tarballs,cache}]\n",
            "                        Cleanup conda packages after creating environments. In case of 'tarballs'\n",
            "                        mode, will clean up all downloaded package tarballs. In case of 'cache'\n",
            "                        mode, will additionally clean up unused package caches. If mode is\n",
            "                        omitted, will default to only cleaning up the tarballs.\n",
            "  --conda-create-envs-only\n",
            "                        If specified, only creates the job-specific conda environments then exits.\n",
            "                        The `--use-conda` flag must also be set. (default: False)\n",
            "  --conda-frontend {conda,mamba}\n",
            "                        Choose the conda frontend for installing environments. Mamba is much\n",
            "                        faster and highly recommended. (default: mamba)\n",
            "\n",
            "APPTAINER/SINGULARITY:\n",
            "  --use-apptainer, --use-singularity\n",
            "                        If defined in the rule, run job within a apptainer/singularity container.\n",
            "                        If this flag is not set, the singularity directive is ignored. (default:\n",
            "                        False)\n",
            "  --apptainer-prefix DIR, --singularity-prefix DIR\n",
            "                        Specify a directory in which apptainer/singularity images will be\n",
            "                        stored.If not supplied, the value is set to the '.snakemake' directory\n",
            "                        relative to the invocation directory. If supplied, the `--use-apptainer`\n",
            "                        flag must also be set. The value may be given as a relative path, which\n",
            "                        will be extrapolated to the invocation directory, or as an absolute path.\n",
            "                        If not supplied, APPTAINER_CACHEDIR is used. In any case, the prefix may\n",
            "                        contain environment variables which will be properly expanded. Note that\n",
            "                        if you use remote execution e.g. on a cluster and you have node specific\n",
            "                        values for this, you should disable assuming shared fs for software-\n",
            "                        deployment (see --shared-fs-usage).\n",
            "  --apptainer-args ARGS, --singularity-args ARGS\n",
            "                        Pass additional args to apptainer/singularity.\n",
            "\n",
            "ENVIRONMENT MODULES:\n",
            "  --use-envmodules      If defined in the rule, run job within the given environment modules,\n",
            "                        loaded in the given order. This can be combined with --use-conda and\n",
            "                        --use-singularity, which will then be only used as a fallback for rules\n",
            "                        which don't define environment modules. (default: False)\n",
            "\n",
            "html executor settings:\n",
            "  --report-html-path VALUE\n",
            "                        Path to the report file (either .html or .zip). Use zip if your report\n",
            "                        contains large results or directories with htmlindex as results.\n",
            "  --report-html-stylesheet-path VALUE\n",
            "                        Path to a custom stylesheet for the report.\n",
            "\n",
            " In general, command-line values override environment variables which override defaults.\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "__conda_setup=\"$('/root/miniforge3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)\"\n",
        "if [ $? -eq 0 ]; then\n",
        "    eval \"$__conda_setup\"\n",
        "else\n",
        "    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\n",
        "        . \"/root/miniforge3/etc/profile.d/conda.sh\"\n",
        "    else\n",
        "        export PATH=\"/root/miniforge3/bin:$PATH\"\n",
        "    fi\n",
        "fi\n",
        "unset __conda_setup\n",
        "\n",
        "if [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\n",
        "    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\n",
        "fi\n",
        "mamba activate snakemake\n",
        "snakemake --help"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "pnVDFTkIu81m",
        "outputId": "f8775c34-52f2-40bd-a889-9532fb676f1a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Building DAG of jobs...\n",
            "Singularity image docker://visze/mprasnakeflow:0.1.0 will be pulled.\n",
            "Job stats:\n",
            "job                                    count\n",
            "-----------------------------------  -------\n",
            "all                                        1\n",
            "assignment_attach_idx                     60\n",
            "assignment_bwa_ref                         1\n",
            "assignment_check_design                    1\n",
            "assignment_collect                         1\n",
            "assignment_collectBCs                      1\n",
            "assignment_fastq_split                     3\n",
            "assignment_filter                          1\n",
            "assignment_flagstat                        1\n",
            "assignment_getBCs                         30\n",
            "assignment_idx_bam                         1\n",
            "assignment_mapping_bwa                    30\n",
            "assignment_merge                          30\n",
            "assignment_statistic_assignedCounts        1\n",
            "assignment_statistic_assignment            1\n",
            "assignment_statistic_totalCounts           1\n",
            "total                                    164\n",
            "\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "\n",
        "__conda_setup=\"$('/root/miniforge3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)\"\n",
        "if [ $? -eq 0 ]; then\n",
        "    eval \"$__conda_setup\"\n",
        "else\n",
        "    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\n",
        "        . \"/root/miniforge3/etc/profile.d/conda.sh\"\n",
        "    else\n",
        "        export PATH=\"/root/miniforge3/bin:$PATH\"\n",
        "    fi\n",
        "fi\n",
        "unset __conda_setup\n",
        "\n",
        "if [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\n",
        "    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\n",
        "fi\n",
        "\n",
        "cd assoc_basic\n",
        "mamba activate snakemake\n",
        "snakemake --sdm apptainer --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\n",
        "  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\n",
        "  -c 1 -n --quiet rules\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "collapsed": true,
        "id": "_4baSpGHxNDD",
        "outputId": "1baccdd5-fb97-4ce7-c6f1-ec276d1eff77"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Assuming unrestricted shared filesystem usage.\n",
            "Building DAG of jobs...\n",
            "Pulling singularity image docker://visze/mprasnakeflow:0.1.0.\n",
            "Using shell: /usr/bin/bash\n",
            "Provided cores: 1 (use --cores to define parallelism)\n",
            "Rules claiming more threads will be scaled down.\n",
            "Conda environments: ignored\n",
            "Job stats:\n",
            "job                                    count\n",
            "-----------------------------------  -------\n",
            "all                                        1\n",
            "assignment_attach_idx                     60\n",
            "assignment_bwa_ref                         1\n",
            "assignment_check_design                    1\n",
            "assignment_collect                         1\n",
            "assignment_collectBCs                      1\n",
            "assignment_fastq_split                     3\n",
            "assignment_filter                          1\n",
            "assignment_flagstat                        1\n",
            "assignment_getBCs                         30\n",
            "assignment_idx_bam                         1\n",
            "assignment_mapping_bwa                    30\n",
            "assignment_merge                          30\n",
            "assignment_statistic_assignedCounts        1\n",
            "assignment_statistic_assignment            1\n",
            "assignment_statistic_totalCounts           1\n",
            "total                                    164\n",
            "\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:26:01 2024]\n",
            "localrule assignment_check_design:\n",
            "    input: data/design.fa, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/check_design_file.py (cached)\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fxi, results/assignment/assocBasic/design_check.done, results/assignment/assocBasic/reference/reference.fa\n",
            "    log: results/logs/assignment/check_design.assocBasic.log, results/assignment/assocBasic/design_check.err\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating singularity image /content/assoc_basic/.snakemake/singularity/89c58db8483fea2d66931e10bd087c16.simg\n",
            "FATAL:   container creation failed: mount hook function failure: mount overlay->/var/lib/apptainer/mnt/session/final error: while mounting overlay: can't mount overlay filesystem to /var/lib/apptainer/mnt/session/final: while setting effective capabilities: CAP_DAC_READ_SEARCH is not in the permitted capability set\n",
            "[Tue Jul 30 13:26:01 2024]\n",
            "Error in rule assignment_check_design:\n",
            "    jobid: 8\n",
            "    input: data/design.fa, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/check_design_file.py (cached)\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fxi, results/assignment/assocBasic/design_check.done, results/assignment/assocBasic/reference/reference.fa\n",
            "    log: results/logs/assignment/check_design.assocBasic.log, results/assignment/assocBasic/design_check.err (check log file(s) for error details)\n",
            "    conda-env: /conda-envs/627fecc5254e183621c8e6c6aae87816\n",
            "    shell:\n",
            "        \n",
            "        trap \"cat results/assignment/assocBasic/design_check.err\" ERR\n",
            "        cp data/design.fa results/assignment/assocBasic/reference/reference.fa\n",
            "        python /root/.cache/snakemake/snakemake/source-cache/runtime-cache/tmpcflzbl58/file/content/MPRAsnakeflow/workflow/rules/../scripts/assignment/check_design_file.py --input results/assignment/assocBasic/reference/reference.fa         --start 3 --length 166         --fast-dict --perform-sequence-check > results/logs/assignment/check_design.assocBasic.log 2> results/assignment/assocBasic/design_check.err;\n",
            "        \n",
            "        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)\n",
            "\n",
            "Shutting down, this might take some time.\n",
            "Exiting because a job execution failed. Look above for error message\n",
            "Complete log: .snakemake/log/2024-07-30T131256.126088.snakemake.log\n",
            "WorkflowError:\n",
            "At least one job did not complete successfully.\n"
          ]
        },
        {
          "ename": "CalledProcessError",
          "evalue": "Command 'b'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm apptainer --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'' returned non-zero exit status 1.",
          "output_type": "error",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-51-5e4588e5e704>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm apptainer --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/_shell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m    332\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    333\u001b[0m       \u001b[0mcell\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m' '\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 334\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    335\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2471\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2472\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2473\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2474\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2475\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<decorator-gen-103>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm apptainer --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'' returned non-zero exit status 1."
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "\n",
        "__conda_setup=\"$('/root/miniforge3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)\"\n",
        "if [ $? -eq 0 ]; then\n",
        "    eval \"$__conda_setup\"\n",
        "else\n",
        "    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\n",
        "        . \"/root/miniforge3/etc/profile.d/conda.sh\"\n",
        "    else\n",
        "        export PATH=\"/root/miniforge3/bin:$PATH\"\n",
        "    fi\n",
        "fi\n",
        "unset __conda_setup\n",
        "\n",
        "if [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\n",
        "    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\n",
        "fi\n",
        "\n",
        "cd assoc_basic\n",
        "mamba activate snakemake\n",
        "snakemake --sdm apptainer --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\n",
        "  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\n",
        "  -c 1\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "44AImdbw0mmA",
        "outputId": "3306721c-9e93-4628-fdbf-b1fb5610cd2a"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Assuming unrestricted shared filesystem usage.\n",
            "Building DAG of jobs...\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/python3.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/../envs/python3.yaml created (location: .snakemake/conda/c50457acea2444acd8e75fe0a317f5fb_)\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/bwa_samtools_picard_htslib.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/assignment/../../envs/bwa_samtools_picard_htslib.yaml created (location: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_)\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/default.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/../envs/default.yaml created (location: .snakemake/conda/6e6828c663b278a3ebf966a2b32525c3_)\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/r.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/assignment/../../envs/r.yaml created (location: .snakemake/conda/4befc3cfe6b50ce710a2aef812330db5_)\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/fastqsplitter.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/../envs/fastqsplitter.yaml created (location: .snakemake/conda/b28cd8c63de5fb4f842167f57c76aea9_)\n",
            "Creating conda environment ../MPRAsnakeflow/workflow/envs/NGmerge.yaml...\n",
            "Downloading and installing remote packages.\n",
            "Environment for /content/MPRAsnakeflow/workflow/rules/../envs/NGmerge.yaml created (location: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_)\n",
            "Using shell: /usr/bin/bash\n",
            "Provided cores: 1 (use --cores to define parallelism)\n",
            "Rules claiming more threads will be scaled down.\n",
            "Singularity containers: ignored\n",
            "Job stats:\n",
            "job                                    count\n",
            "-----------------------------------  -------\n",
            "all                                        1\n",
            "assignment_attach_idx                     60\n",
            "assignment_bwa_ref                         1\n",
            "assignment_check_design                    1\n",
            "assignment_collect                         1\n",
            "assignment_collectBCs                      1\n",
            "assignment_fastq_split                     3\n",
            "assignment_filter                          1\n",
            "assignment_flagstat                        1\n",
            "assignment_getBCs                         30\n",
            "assignment_idx_bam                         1\n",
            "assignment_mapping_bwa                    30\n",
            "assignment_merge                          30\n",
            "assignment_statistic_assignedCounts        1\n",
            "assignment_statistic_assignment            1\n",
            "assignment_statistic_totalCounts           1\n",
            "total                                    164\n",
            "\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:14 2024]\n",
            "localrule assignment_check_design:\n",
            "    input: data/design.fa, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/check_design_file.py (cached)\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fxi, results/assignment/assocBasic/design_check.done, results/assignment/assocBasic/reference/reference.fa\n",
            "    log: results/logs/assignment/check_design.assocBasic.log, results/assignment/assocBasic/design_check.err\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/c50457acea2444acd8e75fe0a317f5fb_\n",
            "Touching output file results/assignment/assocBasic/design_check.done.\n",
            "[Tue Jul 30 13:35:15 2024]\n",
            "Finished job 8.\n",
            "1 of 164 steps (1%) done\n",
            "Removing temporary output results/assignment/assocBasic/reference/reference.fa.fxi.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:15 2024]\n",
            "localrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_1.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.FW.log\n",
            "    jobid: 7\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=FW\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/b28cd8c63de5fb4f842167f57c76aea9_\n",
            "\n",
            "gzip: data/SRR10800986_1.fastq.gz: unexpected end of file\n",
            "[Tue Jul 30 13:35:15 2024]\n",
            "Finished job 7.\n",
            "2 of 164 steps (1%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:15 2024]\n",
            "localrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_3.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.REV.log\n",
            "    jobid: 11\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=REV\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/b28cd8c63de5fb4f842167f57c76aea9_\n",
            "\n",
            "gzip: data/SRR10800986_3.fastq.gz: unexpected end of file\n",
            "[Tue Jul 30 13:35:16 2024]\n",
            "Finished job 11.\n",
            "3 of 164 steps (2%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:16 2024]\n",
            "localrule assignment_bwa_ref:\n",
            "    input: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.dict\n",
            "    log: results/logs/assignment/bwa_ref.assocBasic.log\n",
            "    jobid: 12\n",
            "    reason: Missing output files: results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:35:22 2024]\n",
            "Finished job 12.\n",
            "4 of 164 steps (2%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:22 2024]\n",
            "localrule assignment_fastq_split:\n",
            "    input: data/SRR10800986_2.fastq.gz, results/assignment/assocBasic/design_check.done\n",
            "    output: results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz\n",
            "    log: results/logs/assignment/fastq_split.assocBasic.BC.log\n",
            "    jobid: 9\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz; Input files updated by another job: results/assignment/assocBasic/design_check.done\n",
            "    wildcards: assignment=assocBasic, read=BC\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/b28cd8c63de5fb4f842167f57c76aea9_\n",
            "\n",
            "gzip: data/SRR10800986_2.fastq.gz: unexpected end of file\n",
            "[Tue Jul 30 13:35:22 2024]\n",
            "Finished job 9.\n",
            "5 of 164 steps (3%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:22 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.19.FW.log\n",
            "    jobid: 106\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=19\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:23 2024]\n",
            "Finished job 106.\n",
            "6 of 164 steps (4%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split19.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:23 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.19.REV.log\n",
            "    jobid: 107\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=19\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:23 2024]\n",
            "Finished job 107.\n",
            "7 of 164 steps (4%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split19.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split19.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:23 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split19.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.19.log.gz\n",
            "    jobid: 105\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:23 2024]\n",
            "Finished job 105.\n",
            "8 of 164 steps (5%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split19.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split19.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split19.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:24 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.20.REV.log\n",
            "    jobid: 112\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=20\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:24 2024]\n",
            "Finished job 112.\n",
            "9 of 164 steps (5%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split20.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:24 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.20.FW.log\n",
            "    jobid: 111\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=20\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:24 2024]\n",
            "Finished job 111.\n",
            "10 of 164 steps (6%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split20.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split20.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:25 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split20.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.20.log.gz\n",
            "    jobid: 110\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:25 2024]\n",
            "Finished job 110.\n",
            "11 of 164 steps (7%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split20.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split20.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split20.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:25 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.21.REV.log\n",
            "    jobid: 117\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=21\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:25 2024]\n",
            "Finished job 117.\n",
            "12 of 164 steps (7%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split21.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:25 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.21.FW.log\n",
            "    jobid: 116\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=21\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "Finished job 116.\n",
            "13 of 164 steps (8%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split21.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split21.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split21.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.21.log.gz\n",
            "    jobid: 115\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "Finished job 115.\n",
            "14 of 164 steps (9%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split21.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split21.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split21.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.5.FW.log\n",
            "    jobid: 36\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "Finished job 36.\n",
            "15 of 164 steps (9%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split5.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.5.REV.log\n",
            "    jobid: 37\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:26 2024]\n",
            "Finished job 37.\n",
            "16 of 164 steps (10%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split5.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split5.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split5.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.5.log.gz\n",
            "    jobid: 35\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "Finished job 35.\n",
            "17 of 164 steps (10%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split5.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split5.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split5.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.22.REV.log\n",
            "    jobid: 122\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=22\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "Finished job 122.\n",
            "18 of 164 steps (11%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split22.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.22.FW.log\n",
            "    jobid: 121\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=22\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "Finished job 121.\n",
            "19 of 164 steps (12%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split22.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split22.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split22.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.22.log.gz\n",
            "    jobid: 120\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:27 2024]\n",
            "Finished job 120.\n",
            "20 of 164 steps (12%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split22.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split22.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split22.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.4.REV.log\n",
            "    jobid: 32\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=4\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "Finished job 32.\n",
            "21 of 164 steps (13%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split4.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.4.FW.log\n",
            "    jobid: 31\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=4\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "Finished job 31.\n",
            "22 of 164 steps (13%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split4.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split4.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split4.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.4.log.gz\n",
            "    jobid: 30\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "Finished job 30.\n",
            "23 of 164 steps (14%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split4.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split4.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split4.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:28 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.6.FW.log\n",
            "    jobid: 41\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=6\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "Finished job 41.\n",
            "24 of 164 steps (15%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split6.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.6.REV.log\n",
            "    jobid: 42\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=6\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "Finished job 42.\n",
            "25 of 164 steps (15%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split6.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split6.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split6.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.6.log.gz\n",
            "    jobid: 40\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "Finished job 40.\n",
            "26 of 164 steps (16%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split6.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split6.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split6.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.3.FW.log\n",
            "    jobid: 26\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:29 2024]\n",
            "Finished job 26.\n",
            "27 of 164 steps (16%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split3.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.3.REV.log\n",
            "    jobid: 27\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "Finished job 27.\n",
            "28 of 164 steps (17%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split3.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split3.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split3.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.3.log.gz\n",
            "    jobid: 25\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "Finished job 25.\n",
            "29 of 164 steps (18%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split3.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split3.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split3.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.7.FW.log\n",
            "    jobid: 46\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=7\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "Finished job 46.\n",
            "30 of 164 steps (18%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split7.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:30 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.7.REV.log\n",
            "    jobid: 47\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=7\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "Finished job 47.\n",
            "31 of 164 steps (19%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split7.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split7.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split7.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.7.log.gz\n",
            "    jobid: 45\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "Finished job 45.\n",
            "32 of 164 steps (20%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split7.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split7.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split7.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.23.FW.log\n",
            "    jobid: 126\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=23\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "Finished job 126.\n",
            "33 of 164 steps (20%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split23.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.23.REV.log\n",
            "    jobid: 127\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=23\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "Finished job 127.\n",
            "34 of 164 steps (21%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split23.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split23.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:31 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split23.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.23.log.gz\n",
            "    jobid: 125\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "Finished job 125.\n",
            "35 of 164 steps (21%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split23.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split23.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split23.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.18.REV.log\n",
            "    jobid: 102\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=18\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "Finished job 102.\n",
            "36 of 164 steps (22%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split18.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.18.FW.log\n",
            "    jobid: 101\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=18\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "Finished job 101.\n",
            "37 of 164 steps (23%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split18.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split18.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split18.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.18.log.gz\n",
            "    jobid: 100\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "Finished job 100.\n",
            "38 of 164 steps (23%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split18.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split18.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split18.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:32 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.8.FW.log\n",
            "    jobid: 51\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=8\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "Finished job 51.\n",
            "39 of 164 steps (24%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split8.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.8.REV.log\n",
            "    jobid: 52\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=8\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "Finished job 52.\n",
            "40 of 164 steps (24%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split8.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split8.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split8.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.8.log.gz\n",
            "    jobid: 50\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "Finished job 50.\n",
            "41 of 164 steps (25%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split8.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split8.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split8.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.2.FW.log\n",
            "    jobid: 21\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "Finished job 21.\n",
            "42 of 164 steps (26%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split2.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:33 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.2.REV.log\n",
            "    jobid: 22\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "Finished job 22.\n",
            "43 of 164 steps (26%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split2.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split2.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split2.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.2.log.gz\n",
            "    jobid: 20\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "Finished job 20.\n",
            "44 of 164 steps (27%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split2.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split2.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split2.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.9.REV.log\n",
            "    jobid: 57\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=9\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "Finished job 57.\n",
            "45 of 164 steps (27%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split9.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.9.FW.log\n",
            "    jobid: 56\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=9\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:34 2024]\n",
            "Finished job 56.\n",
            "46 of 164 steps (28%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split9.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split9.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split9.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.9.log.gz\n",
            "    jobid: 55\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "Finished job 55.\n",
            "47 of 164 steps (29%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split9.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split9.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split9.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.25.REV.log\n",
            "    jobid: 137\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=25\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "Finished job 137.\n",
            "48 of 164 steps (29%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split25.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.25.FW.log\n",
            "    jobid: 136\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=25\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "Finished job 136.\n",
            "49 of 164 steps (30%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split25.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split25.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:35 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split25.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.25.log.gz\n",
            "    jobid: 135\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:36 2024]\n",
            "Finished job 135.\n",
            "50 of 164 steps (30%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split25.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split25.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split25.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:36 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.24.FW.log\n",
            "    jobid: 131\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=24\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:36 2024]\n",
            "Finished job 131.\n",
            "51 of 164 steps (31%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split24.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:36 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.24.REV.log\n",
            "    jobid: 132\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=24\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:36 2024]\n",
            "Finished job 132.\n",
            "52 of 164 steps (32%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split24.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split24.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:37 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split24.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.24.log.gz\n",
            "    jobid: 130\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:37 2024]\n",
            "Finished job 130.\n",
            "53 of 164 steps (32%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split24.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split24.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split24.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:37 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.26.REV.log\n",
            "    jobid: 142\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=26\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:37 2024]\n",
            "Finished job 142.\n",
            "54 of 164 steps (33%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split26.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:37 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.26.FW.log\n",
            "    jobid: 141\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=26\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "Finished job 141.\n",
            "55 of 164 steps (34%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split26.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split26.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split26.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.26.log.gz\n",
            "    jobid: 140\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "Finished job 140.\n",
            "56 of 164 steps (34%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split26.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split26.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split26.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.17.REV.log\n",
            "    jobid: 97\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=17\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "Finished job 97.\n",
            "57 of 164 steps (35%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split17.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:38 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.17.FW.log\n",
            "    jobid: 96\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=17\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "Finished job 96.\n",
            "58 of 164 steps (35%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split17.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split17.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split17.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.17.log.gz\n",
            "    jobid: 95\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "Finished job 95.\n",
            "59 of 164 steps (36%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split17.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split17.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split17.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.10.REV.log\n",
            "    jobid: 62\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "Finished job 62.\n",
            "60 of 164 steps (37%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split10.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:39 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.10.FW.log\n",
            "    jobid: 61\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "Finished job 61.\n",
            "61 of 164 steps (37%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split10.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split10.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split10.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.10.log.gz\n",
            "    jobid: 60\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "Finished job 60.\n",
            "62 of 164 steps (38%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split10.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split10.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split10.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.16.FW.log\n",
            "    jobid: 91\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=16\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "Finished job 91.\n",
            "63 of 164 steps (38%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split16.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.16.REV.log\n",
            "    jobid: 92\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=16\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "Finished job 92.\n",
            "64 of 164 steps (39%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split16.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split16.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:40 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split16.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.16.log.gz\n",
            "    jobid: 90\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "Finished job 90.\n",
            "65 of 164 steps (40%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split16.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split16.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split16.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.11.REV.log\n",
            "    jobid: 67\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=11\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "Finished job 67.\n",
            "66 of 164 steps (40%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split11.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.11.FW.log\n",
            "    jobid: 66\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=11\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "Finished job 66.\n",
            "67 of 164 steps (41%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split11.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split11.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split11.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.11.log.gz\n",
            "    jobid: 65\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "Finished job 65.\n",
            "68 of 164 steps (41%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split11.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split11.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split11.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:41 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.1.REV.log\n",
            "    jobid: 17\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "Finished job 17.\n",
            "69 of 164 steps (42%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split1.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.1.FW.log\n",
            "    jobid: 16\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "Finished job 16.\n",
            "70 of 164 steps (43%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split1.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split1.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split1.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.1.log.gz\n",
            "    jobid: 15\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "Finished job 15.\n",
            "71 of 164 steps (43%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split1.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split1.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split1.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.15.FW.log\n",
            "    jobid: 86\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=15\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "Finished job 86.\n",
            "72 of 164 steps (44%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split15.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.15.REV.log\n",
            "    jobid: 87\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=15\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:42 2024]\n",
            "Finished job 87.\n",
            "73 of 164 steps (45%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split15.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split15.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split15.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.15.log.gz\n",
            "    jobid: 85\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "Finished job 85.\n",
            "74 of 164 steps (45%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split15.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split15.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split15.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.29.REV.log\n",
            "    jobid: 157\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=29\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "Finished job 157.\n",
            "75 of 164 steps (46%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split29.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.29.FW.log\n",
            "    jobid: 156\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=29\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "Finished job 156.\n",
            "76 of 164 steps (46%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split29.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split29.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split29.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.29.log.gz\n",
            "    jobid: 155\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "Finished job 155.\n",
            "77 of 164 steps (47%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split29.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split29.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split29.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:43 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.14.REV.log\n",
            "    jobid: 82\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=14\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "Finished job 82.\n",
            "78 of 164 steps (48%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split14.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.14.FW.log\n",
            "    jobid: 81\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=14\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "Finished job 81.\n",
            "79 of 164 steps (48%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split14.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split14.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split14.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.14.log.gz\n",
            "    jobid: 80\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "Finished job 80.\n",
            "80 of 164 steps (49%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split14.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split14.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split14.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.13.FW.log\n",
            "    jobid: 76\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=13\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "Finished job 76.\n",
            "81 of 164 steps (49%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split13.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:44 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.13.REV.log\n",
            "    jobid: 77\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=13\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "Finished job 77.\n",
            "82 of 164 steps (50%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split13.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split13.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split13.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.13.log.gz\n",
            "    jobid: 75\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "Finished job 75.\n",
            "83 of 164 steps (51%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split13.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split13.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split13.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.28.REV.log\n",
            "    jobid: 152\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=28\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "Finished job 152.\n",
            "84 of 164 steps (51%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split28.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.28.FW.log\n",
            "    jobid: 151\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=28\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "Finished job 151.\n",
            "85 of 164 steps (52%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split28.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split28.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split28.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.28.log.gz\n",
            "    jobid: 150\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "Finished job 150.\n",
            "86 of 164 steps (52%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split28.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split28.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split28.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:45 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.27.FW.log\n",
            "    jobid: 146\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=27\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "Finished job 146.\n",
            "87 of 164 steps (53%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split27.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.27.REV.log\n",
            "    jobid: 147\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=27\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "Finished job 147.\n",
            "88 of 164 steps (54%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split27.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split27.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split27.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.27.log.gz\n",
            "    jobid: 145\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "Finished job 145.\n",
            "89 of 164 steps (54%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split27.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split27.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split27.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.0.REV.log\n",
            "    jobid: 10\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=0\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "Finished job 10.\n",
            "90 of 164 steps (55%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split0.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:46 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.12.REV.log\n",
            "    jobid: 72\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=REV, split=12\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "Finished job 72.\n",
            "91 of 164 steps (55%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split12.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.12.FW.log\n",
            "    jobid: 71\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=12\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "Finished job 71.\n",
            "92 of 164 steps (56%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split12.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split12.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split12.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.12.log.gz\n",
            "    jobid: 70\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "Finished job 70.\n",
            "93 of 164 steps (57%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split12.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split12.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split12.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "localrule assignment_attach_idx:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/attachBCToFastQ.py (cached), /content/MPRAsnakeflow/workflow/rules/../scripts/common.py (cached)\n",
            "    output: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz\n",
            "    log: results/logs/assignment/attach_idx.assocBasic.0.FW.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz, results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz\n",
            "    wildcards: assignment=assocBasic, read=FW, split=0\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "Finished job 6.\n",
            "94 of 164 steps (57%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split0.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/BC.split0.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "localrule assignment_merge:\n",
            "    input: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz\n",
            "    output: results/assignment/assocBasic/fastq/merge_split0.un.fastq.gz, results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz\n",
            "    log: results/logs/assignment/merge.assocBasic.0.log.gz\n",
            "    jobid: 5\n",
            "    reason: Missing output files: results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz; Input files updated by another job: results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz, results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6b3957ac5de78f88a5d5f08764c8c120_\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "Finished job 5.\n",
            "95 of 164 steps (58%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/FW.split0.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/splits/REV.split0.BCattached.fastq.gz.\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split0.un.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:47 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.6.log\n",
            "    jobid: 39\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split6.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.008 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:35:50 2024]\n",
            "Finished job 39.\n",
            "96 of 164 steps (59%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split6.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:50 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.6.log\n",
            "    jobid: 38\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split6.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=6\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:35:54 2024]\n",
            "Finished job 38.\n",
            "97 of 164 steps (59%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:54 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.23.log\n",
            "    jobid: 124\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split23.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:35:56 2024]\n",
            "Finished job 124.\n",
            "98 of 164 steps (60%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split23.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:56 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.23.log\n",
            "    jobid: 123\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split23.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=23\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:35:59 2024]\n",
            "Finished job 123.\n",
            "99 of 164 steps (60%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:35:59 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.8.log\n",
            "    jobid: 49\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split8.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:01 2024]\n",
            "Finished job 49.\n",
            "100 of 164 steps (61%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split8.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:01 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.8.log\n",
            "    jobid: 48\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split8.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=8\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:04 2024]\n",
            "Finished job 48.\n",
            "101 of 164 steps (62%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:04 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.25.log\n",
            "    jobid: 134\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split25.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.009 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:36:08 2024]\n",
            "Finished job 134.\n",
            "102 of 164 steps (62%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split25.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:08 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.25.log\n",
            "    jobid: 133\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split25.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=25\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:10 2024]\n",
            "Finished job 133.\n",
            "103 of 164 steps (63%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:10 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.10.log\n",
            "    jobid: 59\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split10.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:12 2024]\n",
            "Finished job 59.\n",
            "104 of 164 steps (63%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split10.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:12 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.10.log\n",
            "    jobid: 58\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split10.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:15 2024]\n",
            "Finished job 58.\n",
            "105 of 164 steps (64%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:15 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.12.log\n",
            "    jobid: 69\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split12.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:17 2024]\n",
            "Finished job 69.\n",
            "106 of 164 steps (65%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split12.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:17 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.12.log\n",
            "    jobid: 68\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split12.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=12\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:21 2024]\n",
            "Finished job 68.\n",
            "107 of 164 steps (65%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:21 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.27.log\n",
            "    jobid: 144\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split27.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:36:24 2024]\n",
            "Finished job 144.\n",
            "108 of 164 steps (66%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split27.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:24 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.27.log\n",
            "    jobid: 143\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split27.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=27\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:26 2024]\n",
            "Finished job 143.\n",
            "109 of 164 steps (66%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:26 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.29.log\n",
            "    jobid: 154\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split29.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:28 2024]\n",
            "Finished job 154.\n",
            "110 of 164 steps (67%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split29.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:28 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.29.log\n",
            "    jobid: 153\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=29\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:31 2024]\n",
            "Finished job 153.\n",
            "111 of 164 steps (68%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:31 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.14.log\n",
            "    jobid: 79\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split14.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.010 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:36:34 2024]\n",
            "Finished job 79.\n",
            "112 of 164 steps (68%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split14.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:34 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.14.log\n",
            "    jobid: 78\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=14\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:37 2024]\n",
            "Finished job 78.\n",
            "113 of 164 steps (69%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:37 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.1.log\n",
            "    jobid: 14\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split1.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:40 2024]\n",
            "Finished job 14.\n",
            "114 of 164 steps (70%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split1.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:40 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.1.log\n",
            "    jobid: 13\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split1.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:42 2024]\n",
            "Finished job 13.\n",
            "115 of 164 steps (70%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:42 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.16.log\n",
            "    jobid: 89\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split16.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:45 2024]\n",
            "Finished job 89.\n",
            "116 of 164 steps (71%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split16.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:45 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.16.log\n",
            "    jobid: 88\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split16.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=16\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:48 2024]\n",
            "Finished job 88.\n",
            "117 of 164 steps (71%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:48 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.18.log\n",
            "    jobid: 99\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split18.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:51 2024]\n",
            "Finished job 99.\n",
            "118 of 164 steps (72%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split18.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:51 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.18.log\n",
            "    jobid: 98\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split18.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=18\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:54 2024]\n",
            "Finished job 98.\n",
            "119 of 164 steps (73%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:54 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.13.log\n",
            "    jobid: 74\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split13.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:36:56 2024]\n",
            "Finished job 74.\n",
            "120 of 164 steps (73%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split13.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:56 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.13.log\n",
            "    jobid: 73\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split13.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=13\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:36:58 2024]\n",
            "Finished job 73.\n",
            "121 of 164 steps (74%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:36:58 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.21.log\n",
            "    jobid: 114\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split21.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.009 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:01 2024]\n",
            "Finished job 114.\n",
            "122 of 164 steps (74%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split21.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:01 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.21.log\n",
            "    jobid: 113\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split21.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=21\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:05 2024]\n",
            "Finished job 113.\n",
            "123 of 164 steps (75%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:05 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.3.log\n",
            "    jobid: 24\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split3.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:07 2024]\n",
            "Finished job 24.\n",
            "124 of 164 steps (76%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split3.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:07 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.3.log\n",
            "    jobid: 23\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split3.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:10 2024]\n",
            "Finished job 23.\n",
            "125 of 164 steps (76%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:10 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.19.log\n",
            "    jobid: 104\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split19.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:37:12 2024]\n",
            "Finished job 104.\n",
            "126 of 164 steps (77%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split19.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:12 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.19.log\n",
            "    jobid: 103\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split19.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=19\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:14 2024]\n",
            "Finished job 103.\n",
            "127 of 164 steps (77%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:14 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.26.log\n",
            "    jobid: 139\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split26.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.009 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:18 2024]\n",
            "Finished job 139.\n",
            "128 of 164 steps (78%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split26.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:18 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.26.log\n",
            "    jobid: 138\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split26.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=26\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:21 2024]\n",
            "Finished job 138.\n",
            "129 of 164 steps (79%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:21 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.11.log\n",
            "    jobid: 64\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split11.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:37:23 2024]\n",
            "Finished job 64.\n",
            "130 of 164 steps (79%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split11.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:23 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.11.log\n",
            "    jobid: 63\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split11.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=11\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:26 2024]\n",
            "Finished job 63.\n",
            "131 of 164 steps (80%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:26 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.4.log\n",
            "    jobid: 29\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split4.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:28 2024]\n",
            "Finished job 29.\n",
            "132 of 164 steps (80%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split4.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:28 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.4.log\n",
            "    jobid: 28\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split4.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=4\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:31 2024]\n",
            "Finished job 28.\n",
            "133 of 164 steps (81%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:32 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.0.log\n",
            "    jobid: 4\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split0.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:34 2024]\n",
            "Finished job 4.\n",
            "134 of 164 steps (82%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split0.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:34 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.0.log\n",
            "    jobid: 3\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split0.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=0\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:37 2024]\n",
            "Finished job 3.\n",
            "135 of 164 steps (82%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:37 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.20.log\n",
            "    jobid: 109\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split20.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:37:39 2024]\n",
            "Finished job 109.\n",
            "136 of 164 steps (83%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split20.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:39 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.20.log\n",
            "    jobid: 108\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split20.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=20\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:41 2024]\n",
            "Finished job 108.\n",
            "137 of 164 steps (84%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:41 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.28.log\n",
            "    jobid: 149\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split28.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.009 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:37:44 2024]\n",
            "Finished job 149.\n",
            "138 of 164 steps (84%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split28.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:44 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.28.log\n",
            "    jobid: 148\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split28.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=28\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:37:57 2024]\n",
            "Finished job 148.\n",
            "139 of 164 steps (85%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:57 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.2.log\n",
            "    jobid: 19\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split2.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.004 sec; CPU: 0.007 sec\n",
            "[Tue Jul 30 13:37:59 2024]\n",
            "Finished job 19.\n",
            "140 of 164 steps (85%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split2.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:37:59 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.2.log\n",
            "    jobid: 18\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split2.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:01 2024]\n",
            "Finished job 18.\n",
            "141 of 164 steps (86%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:01 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.17.log\n",
            "    jobid: 94\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split17.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:38:04 2024]\n",
            "Finished job 94.\n",
            "142 of 164 steps (87%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split17.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:04 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.17.log\n",
            "    jobid: 93\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split17.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=17\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:07 2024]\n",
            "Finished job 93.\n",
            "143 of 164 steps (87%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:07 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.9.log\n",
            "    jobid: 54\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split9.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:38:10 2024]\n",
            "Finished job 54.\n",
            "144 of 164 steps (88%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split9.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:10 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.9.log\n",
            "    jobid: 53\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split9.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=9\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:13 2024]\n",
            "Finished job 53.\n",
            "145 of 164 steps (88%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:13 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.24.log\n",
            "    jobid: 129\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split24.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:38:15 2024]\n",
            "Finished job 129.\n",
            "146 of 164 steps (89%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split24.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:15 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.24.log\n",
            "    jobid: 128\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split24.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=24\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:18 2024]\n",
            "Finished job 128.\n",
            "147 of 164 steps (90%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:18 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.7.log\n",
            "    jobid: 44\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split7.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:38:20 2024]\n",
            "Finished job 44.\n",
            "148 of 164 steps (90%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split7.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:20 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.7.log\n",
            "    jobid: 43\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split7.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=7\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:24 2024]\n",
            "Finished job 43.\n",
            "149 of 164 steps (91%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:24 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.15.log\n",
            "    jobid: 84\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split15.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.009 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:38:26 2024]\n",
            "Finished job 84.\n",
            "150 of 164 steps (91%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split15.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:26 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.15.log\n",
            "    jobid: 83\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split15.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=15\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:29 2024]\n",
            "Finished job 83.\n",
            "151 of 164 steps (92%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:29 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.5.log\n",
            "    jobid: 34\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split5.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.006 sec; CPU: 0.008 sec\n",
            "[Tue Jul 30 13:38:31 2024]\n",
            "Finished job 34.\n",
            "152 of 164 steps (93%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split5.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:31 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.5.log\n",
            "    jobid: 33\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split5.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:33 2024]\n",
            "Finished job 33.\n",
            "153 of 164 steps (93%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:33 2024]\n",
            "localrule assignment_mapping_bwa:\n",
            "    input: results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/reference/reference.fa.fai, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.amb\n",
            "    output: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    log: results/logs/assignment/mapping.assocBasic.22.log\n",
            "    jobid: 119\n",
            "    reason: Missing output files: results/assignment/assocBasic/bam/merge_split22.mapped.bam; Input files updated by another job: results/assignment/assocBasic/reference/reference.fa, results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz, results/assignment/assocBasic/reference/reference.fa.sa, results/assignment/assocBasic/reference/reference.fa.amb, results/assignment/assocBasic/reference/reference.fa.bwt, results/assignment/assocBasic/reference/reference.fa.dict, results/assignment/assocBasic/reference/reference.fa.ann, results/assignment/assocBasic/reference/reference.fa.pac, results/assignment/assocBasic/reference/reference.fa.fai\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -t 1 -L 80 -M -C results/assignment/assocBasic/reference/reference.fa /dev/fd/63\n",
            "[main] Real time: 0.011 sec; CPU: 0.009 sec\n",
            "[Tue Jul 30 13:38:37 2024]\n",
            "Finished job 119.\n",
            "154 of 164 steps (94%) done\n",
            "Removing temporary output results/assignment/assocBasic/fastq/merge_split22.join.fastq.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:37 2024]\n",
            "localrule assignment_collect:\n",
            "    input: results/assignment/assocBasic/bam/merge_split0.mapped.bam, results/assignment/assocBasic/bam/merge_split1.mapped.bam, results/assignment/assocBasic/bam/merge_split2.mapped.bam, results/assignment/assocBasic/bam/merge_split3.mapped.bam, results/assignment/assocBasic/bam/merge_split4.mapped.bam, results/assignment/assocBasic/bam/merge_split5.mapped.bam, results/assignment/assocBasic/bam/merge_split6.mapped.bam, results/assignment/assocBasic/bam/merge_split7.mapped.bam, results/assignment/assocBasic/bam/merge_split8.mapped.bam, results/assignment/assocBasic/bam/merge_split9.mapped.bam, results/assignment/assocBasic/bam/merge_split10.mapped.bam, results/assignment/assocBasic/bam/merge_split11.mapped.bam, results/assignment/assocBasic/bam/merge_split12.mapped.bam, results/assignment/assocBasic/bam/merge_split13.mapped.bam, results/assignment/assocBasic/bam/merge_split14.mapped.bam, results/assignment/assocBasic/bam/merge_split15.mapped.bam, results/assignment/assocBasic/bam/merge_split16.mapped.bam, results/assignment/assocBasic/bam/merge_split17.mapped.bam, results/assignment/assocBasic/bam/merge_split18.mapped.bam, results/assignment/assocBasic/bam/merge_split19.mapped.bam, results/assignment/assocBasic/bam/merge_split20.mapped.bam, results/assignment/assocBasic/bam/merge_split21.mapped.bam, results/assignment/assocBasic/bam/merge_split22.mapped.bam, results/assignment/assocBasic/bam/merge_split23.mapped.bam, results/assignment/assocBasic/bam/merge_split24.mapped.bam, results/assignment/assocBasic/bam/merge_split25.mapped.bam, results/assignment/assocBasic/bam/merge_split26.mapped.bam, results/assignment/assocBasic/bam/merge_split27.mapped.bam, results/assignment/assocBasic/bam/merge_split28.mapped.bam, results/assignment/assocBasic/bam/merge_split29.mapped.bam\n",
            "    output: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    log: results/logs/assignment/collect.assocBasic.log\n",
            "    jobid: 162\n",
            "    reason: Missing output files: results/assignment/assocBasic/aligned_merged_reads.bam; Input files updated by another job: results/assignment/assocBasic/bam/merge_split7.mapped.bam, results/assignment/assocBasic/bam/merge_split18.mapped.bam, results/assignment/assocBasic/bam/merge_split6.mapped.bam, results/assignment/assocBasic/bam/merge_split15.mapped.bam, results/assignment/assocBasic/bam/merge_split20.mapped.bam, results/assignment/assocBasic/bam/merge_split29.mapped.bam, results/assignment/assocBasic/bam/merge_split13.mapped.bam, results/assignment/assocBasic/bam/merge_split3.mapped.bam, results/assignment/assocBasic/bam/merge_split9.mapped.bam, results/assignment/assocBasic/bam/merge_split11.mapped.bam, results/assignment/assocBasic/bam/merge_split16.mapped.bam, results/assignment/assocBasic/bam/merge_split25.mapped.bam, results/assignment/assocBasic/bam/merge_split27.mapped.bam, results/assignment/assocBasic/bam/merge_split26.mapped.bam, results/assignment/assocBasic/bam/merge_split12.mapped.bam, results/assignment/assocBasic/bam/merge_split21.mapped.bam, results/assignment/assocBasic/bam/merge_split8.mapped.bam, results/assignment/assocBasic/bam/merge_split10.mapped.bam, results/assignment/assocBasic/bam/merge_split17.mapped.bam, results/assignment/assocBasic/bam/merge_split19.mapped.bam, results/assignment/assocBasic/bam/merge_split0.mapped.bam, results/assignment/assocBasic/bam/merge_split2.mapped.bam, results/assignment/assocBasic/bam/merge_split5.mapped.bam, results/assignment/assocBasic/bam/merge_split28.mapped.bam, results/assignment/assocBasic/bam/merge_split4.mapped.bam, results/assignment/assocBasic/bam/merge_split22.mapped.bam, results/assignment/assocBasic/bam/merge_split1.mapped.bam, results/assignment/assocBasic/bam/merge_split23.mapped.bam, results/assignment/assocBasic/bam/merge_split24.mapped.bam, results/assignment/assocBasic/bam/merge_split14.mapped.bam\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:40 2024]\n",
            "Finished job 162.\n",
            "155 of 164 steps (95%) done\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split0.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split1.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split2.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split3.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split4.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split5.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split6.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split7.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split8.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split9.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split10.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split11.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split12.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split13.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split14.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split15.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split16.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split17.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split18.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split19.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split20.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split21.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split23.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split24.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split25.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split26.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split27.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split28.mapped.bam.\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split29.mapped.bam.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:40 2024]\n",
            "localrule assignment_getBCs:\n",
            "    input: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    output: results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv\n",
            "    log: results/logs/assignment/getBCs.assocBasic.22.log\n",
            "    jobid: 118\n",
            "    reason: Missing output files: results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv; Input files updated by another job: results/assignment/assocBasic/bam/merge_split22.mapped.bam\n",
            "    wildcards: assignment=assocBasic, split=22\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:42 2024]\n",
            "Finished job 118.\n",
            "156 of 164 steps (95%) done\n",
            "Removing temporary output results/assignment/assocBasic/bam/merge_split22.mapped.bam.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:42 2024]\n",
            "localrule assignment_collectBCs:\n",
            "    input: results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv\n",
            "    output: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    log: results/logs/assignment/collectBCs.assocBasic.log\n",
            "    jobid: 2\n",
            "    reason: Missing output files: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz; Input files updated by another job: results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv, results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/6e6828c663b278a3ebf966a2b32525c3_\n",
            "[Tue Jul 30 13:38:43 2024]\n",
            "Finished job 2.\n",
            "157 of 164 steps (96%) done\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.0.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.1.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.2.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.3.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.4.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.5.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.6.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.7.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.8.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.9.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.10.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.11.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.12.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.13.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.14.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.15.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.16.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.17.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.18.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.19.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.20.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.21.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.22.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.23.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.24.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.25.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.26.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.27.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.28.tsv.\n",
            "Removing temporary output results/assignment/assocBasic/BCs/barcodes_incl_other.29.tsv.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:43 2024]\n",
            "localrule assignment_statistic_totalCounts:\n",
            "    input: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_total_counts.py (cached)\n",
            "    output: results/assignment/assocBasic/statistic/total_counts.tsv.gz\n",
            "    log: results/logs/assignment/statistic_totalCounts.assocBasic.log\n",
            "    jobid: 160\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/total_counts.tsv.gz; Input files updated by another job: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/c50457acea2444acd8e75fe0a317f5fb_\n",
            "[Tue Jul 30 13:38:45 2024]\n",
            "Finished job 160.\n",
            "158 of 164 steps (96%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:45 2024]\n",
            "localrule assignment_filter:\n",
            "    input: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/filterAssignmentTsv.py (cached)\n",
            "    output: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    log: results/logs/assignment/filter.assocBasic.default.log\n",
            "    jobid: 1\n",
            "    reason: Missing output files: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz; Input files updated by another job: results/assignment/assocBasic/barcodes_incl_other.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/c50457acea2444acd8e75fe0a317f5fb_\n",
            "[Tue Jul 30 13:38:45 2024]\n",
            "Finished job 1.\n",
            "159 of 164 steps (97%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:45 2024]\n",
            "localrule assignment_idx_bam:\n",
            "    input: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    output: results/assignment/assocBasic/aligned_merged_reads.bam.bai\n",
            "    log: results/logs/assignment/assocBasic/assignment_idx_bam.log\n",
            "    jobid: 163\n",
            "    reason: Missing output files: results/assignment/assocBasic/aligned_merged_reads.bam.bai; Input files updated by another job: results/assignment/assocBasic/aligned_merged_reads.bam\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:48 2024]\n",
            "Finished job 163.\n",
            "160 of 164 steps (98%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:48 2024]\n",
            "localrule assignment_flagstat:\n",
            "    input: results/assignment/assocBasic/aligned_merged_reads.bam, results/assignment/assocBasic/aligned_merged_reads.bam.bai\n",
            "    output: results/assignment/assocBasic/statistic/assignment/bam_stats.txt\n",
            "    log: results/logs/assignment/flagstat.assocBasic.log\n",
            "    jobid: 161\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assignment/bam_stats.txt; Input files updated by another job: results/assignment/assocBasic/aligned_merged_reads.bam, results/assignment/assocBasic/aligned_merged_reads.bam.bai\n",
            "    wildcards: assignment=assocBasic\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/7b8791bda87729024a6bda910c75d51e_\n",
            "[Tue Jul 30 13:38:51 2024]\n",
            "Finished job 161.\n",
            "161 of 164 steps (98%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:51 2024]\n",
            "localrule assignment_statistic_assignedCounts:\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_total_counts.py (cached)\n",
            "    output: results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz\n",
            "    log: results/logs/assignment/statistic_assignedCounts.assocBasic.default.log\n",
            "    jobid: 158\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assigned_counts.default.tsv.gz; Input files updated by another job: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/c50457acea2444acd8e75fe0a317f5fb_\n",
            "[Tue Jul 30 13:38:52 2024]\n",
            "Finished job 158.\n",
            "162 of 164 steps (99%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue Jul 30 13:38:52 2024]\n",
            "localrule assignment_statistic_assignment:\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_assignment.R (cached)\n",
            "    output: results/assignment/assocBasic/statistic/assignment.default.tsv.gz, results/assignment/assocBasic/statistic/assignment.default.png\n",
            "    log: results/logs/assignment/statistic_assignment.assocBasic.default.log\n",
            "    jobid: 159\n",
            "    reason: Missing output files: results/assignment/assocBasic/statistic/assignment.default.png; Input files updated by another job: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz\n",
            "    wildcards: assignment=assocBasic, assignment_config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: .snakemake/conda/4befc3cfe6b50ce710a2aef812330db5_\n",
            "[Tue Jul 30 13:38:57 2024]\n",
            "Error in rule assignment_statistic_assignment:\n",
            "    jobid: 159\n",
            "    input: results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz, /content/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_assignment.R (cached)\n",
            "    output: results/assignment/assocBasic/statistic/assignment.default.tsv.gz, results/assignment/assocBasic/statistic/assignment.default.png\n",
            "    log: results/logs/assignment/statistic_assignment.assocBasic.default.log (check log file(s) for error details)\n",
            "    conda-env: /content/assoc_basic/.snakemake/conda/4befc3cfe6b50ce710a2aef812330db5_\n",
            "    shell:\n",
            "        \n",
            "        Rscript /root/.cache/snakemake/snakemake/source-cache/runtime-cache/tmp32cox1ez/file/content/MPRAsnakeflow/workflow/rules/../scripts/assignment/statistic_assignment.R --input results/assignment/assocBasic/assignment_barcodes.default.sorted.tsv.gz --statistic results/assignment/assocBasic/statistic/assignment.default.tsv.gz --plot results/assignment/assocBasic/statistic/assignment.default.png &> results/logs/assignment/statistic_assignment.assocBasic.default.log\n",
            "        \n",
            "        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)\n",
            "\n",
            "Shutting down, this might take some time.\n",
            "Exiting because a job execution failed. Look above for error message\n",
            "Complete log: .snakemake/log/2024-07-30T132651.351231.snakemake.log\n",
            "WorkflowError:\n",
            "At least one job did not complete successfully.\n"
          ]
        },
        {
          "ename": "CalledProcessError",
          "evalue": "Command 'b'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm conda --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'' returned non-zero exit status 1.",
          "output_type": "error",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-52-319dffa7d295>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm conda --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/google/colab/_shell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m    332\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    333\u001b[0m       \u001b[0mcell\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m' '\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 334\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    335\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2471\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2472\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2473\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2474\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2475\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m<decorator-gen-103>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\n__conda_setup=\"$(\\'/root/miniforge3/bin/conda\\' \\'shell.bash\\' \\'hook\\' 2> /dev/null)\"\\nif [ $? -eq 0 ]; then\\n    eval \"$__conda_setup\"\\nelse\\n    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\\n        . \"/root/miniforge3/etc/profile.d/conda.sh\"\\n    else\\n        export PATH=\"/root/miniforge3/bin:$PATH\"\\n    fi\\nfi\\nunset __conda_setup\\n\\nif [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\\n    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\\nfi\\n\\ncd assoc_basic\\nmamba activate snakemake\\nsnakemake --sdm conda --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\\\\n  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\\\\n  -c 1\\n'' returned non-zero exit status 1."
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "\n",
        "__conda_setup=\"$('/root/miniforge3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)\"\n",
        "if [ $? -eq 0 ]; then\n",
        "    eval \"$__conda_setup\"\n",
        "else\n",
        "    if [ -f \"/root/miniforge3/etc/profile.d/conda.sh\" ]; then\n",
        "        . \"/root/miniforge3/etc/profile.d/conda.sh\"\n",
        "    else\n",
        "        export PATH=\"/root/miniforge3/bin:$PATH\"\n",
        "    fi\n",
        "fi\n",
        "unset __conda_setup\n",
        "\n",
        "if [ -f \"/root/miniforge3/etc/profile.d/mamba.sh\" ]; then\n",
        "    . \"/root/miniforge3/etc/profile.d/mamba.sh\"\n",
        "fi\n",
        "\n",
        "cd assoc_basic\n",
        "mamba activate snakemake\n",
        "snakemake --sdm conda --snakefile ../MPRAsnakeflow/workflow/Snakefile \\\n",
        "  --configfile ../MPRAsnakeflow/resources/assoc_basic/config.yml \\\n",
        "  -c 1\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QiSvx3Oy6qzD",
        "outputId": "43384c74-2cb4-417e-a9bc-d95f0eaea390"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "total 9.3G\n",
            "-rw-r--r-- 1 root root 9.3G Jul 30 14:26 SRR10800986.sra\n"
          ]
        }
      ],
      "source": [
        "!ls -lh /content/assoc_basic/data/SRR10800986/"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "authorship_tag": "ABX9TyO9Rt29ByR/S1kLW5tnVhMm",
      "collapsed_sections": [
        "32PHp-vU99x0",
        "6GilAsqZ4t9r"
      ],
      "include_colab_link": true,
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
