{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "colab_type": "text",
        "id": "view-in-github"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/kircherlab/MPRAsnakeflow_tutorial/blob/main/tutorial_experiment.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NsBG7lOCPnnE",
        "outputId": "6da1caeb-905a-42f8-843a-54409a379c0f"
      },
      "outputs": [],
      "source": [
        "%%capture\n",
        "import matplotlib.pyplot as plt\n",
        "import os\n",
        "import cv2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "BOSJecyQPnnC"
      },
      "source": [
        "# MPRAsnakeflow Experiment Tutorial\n",
        "\n",
        "This notebook is used to generate activity measurements from DNA and RNA barcode counts. The input file of the assignment workflow is needed to associate the barcodes to designed oligos.\n",
        "\n",
        "After going through the prerequisites, we will cover the config file, run MPRAsnakeflow, and look at interesting results like the correlation of the DNA and RNA counts among the replicates. This is useful in practice to check if the data preparation went well, because if the resulting data is not correlating between replicates, problems during the experiment will introduce a bias in the results of our analysis that we would like to know about."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uAAYJQqv9zIk"
      },
      "source": [
        "## Prerequisites"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "dOsEkOFu-MbM"
      },
      "source": [
        "### Docker/uDocker/Apptainer\n",
        "\n",
        "When running locally, you need to install `docker` or `apptainer`. For running remotely on Google Colab, `udocker` from github.com/drengskapur/docker-in-colab works nicely.\n",
        "\n",
        "We try to automatically find out what you need:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Docker is installed! We will use Docker to run MPRAsnakeflow\n"
          ]
        }
      ],
      "source": [
        "tutorial_container = \"visze/mprasnakeflow_tutorial:0.5.3\"\n",
        "docker= False\n",
        "apptainer = False\n",
        "\n",
        "docker = !docker --help &>/dev/null; if [ $? -eq 0 ]; then echo 1; else echo 0; fi\n",
        "docker = bool(int(docker[0]))\n",
        "apptainer = !apptainer --help &>/dev/null; if [ $? -eq 0 ]; then echo 1; else echo 0; fi\n",
        "apptainer = bool(int(apptainer[0]))\n",
        "\n",
        "if docker:\n",
        "    print(\"Docker is installed! We will use Docker to run MPRAsnakeflow\")\n",
        "elif apptainer:\n",
        "    print(\"Apptainer is installed! We will use Apptainer to run MPRAsnakeflow\")\n",
        "else:\n",
        "    print(\"Neither Docker nor Apptainer is installed. We assume you run the tutorial on Colab and we will install uDocker for Colab\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "When neither Docker nor Apptainer is installed, we will install uDocker for Colab now."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NSTDSHTN4JJk",
        "outputId": "5c2ad4b0-cc4b-4ced-9bfe-a75960cc36bd"
      },
      "outputs": [],
      "source": [
        "def udocker_init():\n",
        "    import os\n",
        "    if not os.path.exists(\"/home/user\"):\n",
        "        !pip install udocker > /dev/null\n",
        "        !udocker --allow-root install > /dev/null\n",
        "        !useradd -m user > /dev/null\n",
        "    print(f'Docker-in-Colab 1.1.0\\n')\n",
        "    print(f'Usage:     udocker(\"--help\")')\n",
        "    print(f'Examples:  https://github.com/indigo-dc/udocker?tab=readme-ov-file#examples')\n",
        "\n",
        "    def execute(command: str):\n",
        "        user_prompt = \"\\033[1;32muser@pc\\033[0m\"\n",
        "        print(f\"{user_prompt}$ udocker {command}\")\n",
        "        !su - user -c \"udocker $command\"\n",
        "\n",
        "    return execute\n",
        "\n",
        "if not docker and not apptainer:\n",
        "    udocker = udocker_init()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "We define a general function to run our MPRAsnakeflow tutorial container that will check if Docker, Apptainer, or uDocker has to be run. It will also bind your data directory `${PWD}/MPRAsnakeflow_tutorial` (will be downloaded later) to `/data/run` within the container and defines this as the working directory."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "def runContainer(command):\n",
        "    user_prompt = \"\\033[1;32muser@pc\\033[0m\"\n",
        "    if docker:\n",
        "        print(f\"{user_prompt}$ docker run -v=${{PWD}}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run {tutorial_container} {command}\")\n",
        "        !docker run -v=${{PWD}}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run {tutorial_container} {command}\n",
        "    elif apptainer:\n",
        "        print(f\"apptainer run -B=${{PWD}}/MPRAsnakeflow_tutorial:/data/run --cwd /data/run mprasnakeflow_tutorial.sif {command}\")\n",
        "        !apptainer run -B=${{PWD}}/MPRAsnakeflow_tutorial:/data/run --cwd /data/run mprasnakeflow_tutorial.sif {command}\n",
        "    else:\n",
        "        udocker(f\"run -v=${{PWD}}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run mprasnakeflow_tutorial {command}\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Mb0rhjfo8dbA"
      },
      "source": [
        "### Test data\n",
        "\n",
        "The data used in this tutorial is present in the GitHub repository. We will download it using git so that it is also present within a Colab environment."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "BGLsBxpyst9V"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Cloning into 'MPRAsnakeflow_tutorial'...\n",
            "Updating files: 100% (37/37), done.\n",
            "Updating files: 100% (37/37), done.\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "if [ ! -d \"MPRAsnakeflow_tutorial\" ]; then\n",
        "    git clone https://github.com/kircherlab/MPRAsnakeflow_tutorial.git\n",
        "else\n",
        "    echo \"Folder 'MPRAsnakeflow_tutorial' already exists. Skipping clone.\"\n",
        "fi"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4WYuqwGr-fEp"
      },
      "source": [
        "### snakemake\n",
        "\n",
        "MPRAsnakeflow and its software dependencies are packed in the official snakemake v9.3.4 container, which you can find on DockerHub with the tag [`visze/mprasnakeflow_tutorial:0.5.3`](https://hub.docker.com/r/visze/mprasnakeflow_tutorial)."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZZv9f_eu4bB9",
        "outputId": "67fa0e9c-386e-454a-c88d-11ce125ee5c3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "docker pull visze/mprasnakeflow_tutorial:0.5.3\n",
            "0.5.3: Pulling from visze/mprasnakeflow_tutorial\n",
            "Digest: sha256:abbb0dd917d690b7cbf17dd4c3d37da847edfa73d2f39c91182bca19d74b8374\n",
            "Status: Image is up to date for visze/mprasnakeflow_tutorial:0.5.3\n",
            "docker.io/visze/mprasnakeflow_tutorial:0.5.3\n",
            "0.5.3: Pulling from visze/mprasnakeflow_tutorial\n",
            "Digest: sha256:abbb0dd917d690b7cbf17dd4c3d37da847edfa73d2f39c91182bca19d74b8374\n",
            "Status: Image is up to date for visze/mprasnakeflow_tutorial:0.5.3\n",
            "docker.io/visze/mprasnakeflow_tutorial:0.5.3\n",
            "docker create --name=mprasnakeflow_tutorial visze/mprasnakeflow_tutorial:0.5.3\n",
            "docker create --name=mprasnakeflow_tutorial visze/mprasnakeflow_tutorial:0.5.3\n",
            "Error response from daemon: Conflict. The container name \"/mprasnakeflow_tutorial\" is already in use by container \"26d8834b6ccb212d6aabb1097dc745c36c43a3040d2d3f76a3f87b2a0ebbb730\". You have to remove (or rename) that container to be able to reuse that name.\n",
            "Error response from daemon: Conflict. The container name \"/mprasnakeflow_tutorial\" is already in use by container \"26d8834b6ccb212d6aabb1097dc745c36c43a3040d2d3f76a3f87b2a0ebbb730\". You have to remove (or rename) that container to be able to reuse that name.\n"
          ]
        }
      ],
      "source": [
        "if docker:\n",
        "    print(f\"docker pull {tutorial_container}\")\n",
        "    !docker pull \"{tutorial_container}\"\n",
        "    print(f\"docker create --name=mprasnakeflow_tutorial {tutorial_container}\")\n",
        "    !docker create --name=mprasnakeflow_tutorial \"{tutorial_container}\" > /dev/null\n",
        "elif apptainer:\n",
        "    print(f\"apptainer pull mprasnakeflow_tutorial.sif docker://{tutorial_container}\")\n",
        "    !apptainer pull mprasnakeflow_tutorial.sif docker://{tutorial_container} > /dev/null\n",
        "else:\n",
        "    udocker(f\"pull {tutorial_container}\")\n",
        "    udocker(f\"create --name=mprasnakeflow_tutorial {tutorial_container}\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "zYhKnwlV-orz"
      },
      "source": [
        "Try to run `snakemake --version` in the container:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "jQi5e3634sDP",
        "outputId": "844eb31f-20d4-463a-ef20-6c6c3b013e7a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ docker run -v=${PWD}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run visze/mprasnakeflow_tutorial:0.5.3 snakemake --version\n",
            "9.3.5.dev0\n",
            "9.3.5.dev0\n"
          ]
        }
      ],
      "source": [
        "runContainer(\"snakemake --version\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Experiment Workflow\n",
        "\n",
        "For the experiment step, where the counts per oligo are calculated, we need all the files in the `example_data/counts` folder, as well as the config file and the barcode assignment file, which is the output of the assignment step (but also stored under `example_data/counts`).\n",
        "\n",
        "We investigated the assignment file in the association tutorial. This table is now used to count the number of observed barcodes for each target sequence."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "We can take a look at the input files to understand what we later have to give the MPRAsnakeflow pipeline via a config file. Each count experiment, in our case, is described using three files (the paths of which are in the `experiment.csv` file): forward reads, reverse reads, and UMIs. Note that the UMIs are optional, and serve to deduplicate barcodes with overdispersed counts due to preferential amplification related to GC content or library preparation steps. From one of the UMI read files, we can see that indeed our UMI length is 16:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Average length of a UMI:\n",
            "16\n",
            "16\n"
          ]
        }
      ],
      "source": [
        "print(\"\\nAverage length of a UMI:\")\n",
        "!zcat \"MPRAsnakeflow_tutorial/example_data/counts/dna-1_S1_R2_001.fastq.gz\" | awk 'NR % 4 == 2 {{sum += length($$0)}} END {{print sum*4/NR}}'"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "We can also check the length of the barcode, which should be 15:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Average length of a barcode:\n",
            "15\n",
            "15\n"
          ]
        }
      ],
      "source": [
        "print(\"\\nAverage length of a barcode:\")\n",
        "!zcat \"MPRAsnakeflow_tutorial/example_data/counts/dna-1_S1_R1_001.fastq.gz\" | awk 'NR % 4 == 2 {{sum += length($$0)}} END {{print sum*4/NR}}'"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0KqdtmOb_gHg"
      },
      "source": [
        "## Run MPRAsnakeflow"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Snakemake\n",
        "\n",
        "Before we come to more MPRAsnakeflow pipeline specific things, we quickly want to introduce the snakemake workflow management system. A workflow management system, also like Nextflow or CWL, is a tool to create reproducible and scalable data analyses. Snakemake workflows are described via a human-readable, Python-based language. They can be seamlessly scaled to server, cluster, grid, and cloud environments, without the need to modify the workflow definition. Finally, Snakemake workflows can entail a description of required software, which will be automatically deployed to any execution environment.\n",
        "\n",
        "Each step in snakemake is defined as a `rule`, which has an `input`, `output`, and mostly a `shell` command. The `input` and `output` are files that are used as input and output for the rule. The `shell` command is the command that is executed to create the output file from the input file. There are other commands like `conda`, `params`, `log`, `threads`, `resources` that can be used to define the rule.\n",
        "\n",
        "E.g. here is an example rule `assigned_counts_combine_replicates` of MPRAsnakeflow, which combines counts of replicates into one final count. In short: merging counts of replicates. The rule takes as input a tab separated file with all counts in it per condition, and the python script that will merge it `combine_replicates.py`. Path names in curly brackets like `{project}` or `{assignment}` are wildcards and the name will be the project, assignment name etc. that is configured in the config files. Finally, the `shell` command takes the python script, the inputs and runs it creating the output and logs.\n",
        "\n",
        "```python\n",
        "rule assigned_counts_combine_replicates:\n",
        "    \"\"\"\n",
        "    Combine replicates of master table by summing counts up and using also the average.\n",
        "    \"\"\"\n",
        "    conda:\n",
        "        \"../envs/python3.yaml\"\n",
        "    input:\n",
        "        master_table=\"results/experiments/{project}/assigned_counts/{assignment}/{config}/{condition}_{allreps_or_threshold}_merged.tsv.gz\",\n",
        "        script=getScript(\"count/combine_replicates.py\"),\n",
        "    output:\n",
        "        \"results/experiments/{project}/assigned_counts/{assignment}/{config}/{condition}_{allreps_or_threshold}_merged.combined.tsv.gz\",\n",
        "    params:\n",
        "        label_file=lambda wc: (\n",
        "            \"--labels %s\" % config[\"experiments\"][wc.project][\"label_file\"]\n",
        "            if \"label_file\" in config[\"experiments\"][wc.project]\n",
        "            else \"\"\n",
        "        ),\n",
        "    log:\n",
        "        temp(\n",
        "            \"results/logs/assigned_counts/combine_replicates.{project}.{condition}.{config}.{assignment}.{allreps_or_threshold}.log\"\n",
        "        ),\n",
        "    shell:\n",
        "        \"\"\"\n",
        "        python {input.script} \\\n",
        "        --input {input.master_table} \\\n",
        "        {params.label_file} \\\n",
        "        --output {output}  &> {log}\n",
        "        \"\"\"\n",
        "```"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "When running snakemake, you need to specify where your workflow is, basically the main `Snakefile` of it. By default, Snakemake will search for ‘Snakefile’, ‘snakefile’, ‘workflow/Snakefile’, ‘workflow/snakefile’ beneath the current working directory, in this order. Then you do not have to use this command. But practically your workflow, here MPRAsnakeflow, might live somewhere else and you want to run the workflow in a different directory where you want to produce results. Then you need to specify the `--snakefile` parameter to the MPRAsnakeflow file in `workflow/Snakefile`. Otherwise, you have to check/copy the workflow in every \"run\" directory which is not practical. Usually, you start snakemake in the directory where you want to produce results."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### The config file\n",
        "\n",
        "The heart of your workflow run is the config file. You can specify it via `--configfile`. It is in `yaml` or `json` format. It contains all the parameters and input files that are used in the workflow.\n",
        "\n",
        "The config file has many parameters, but most of them have a default value, based on best practice, and you don't have to specify them. But you can overwrite them if you want. Config files are first validated using JSON schema. If your config file is not correct, MPRAsnakeflow will tell you what is missing or wrong and stop. Naming, parameters, etc. might change from MPRAsnakeflow version and you might have to update your config file if you run a new version of MPRAsnakeflow. An extensive documentation of the config file can be found in the [MPRAsnakeflow documentation](https://mprasnakeflow.readthedocs.io/en/development/config.html).\n",
        "\n",
        "Here is the config file that we will use for running the experiment tutorial.\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "```yaml\n",
        "---\n",
        "experiments:\n",
        "  experimentMPRAworkshop:\n",
        "    bc_length: 15\n",
        "    umi_length: 16\n",
        "    data_folder: example_data/counts\n",
        "    experiment_file: example_data/experiment.csv\n",
        "    assignments:\n",
        "      MPRAworkshop:\n",
        "        type: file\n",
        "        assignment_file: example_data/assignment/assignment_barcodes.default.tsv.gz\n",
        "    label_file: example_data/design/workshop_labels.tsv\n",
        "    configs:\n",
        "      default: {}\n",
        "      tutorialConfig:\n",
        "        filter:\n",
        "          bc_threshold: 5\n",
        "          DNA:\n",
        "            min_counts: 1\n",
        "          RNA:\n",
        "            min_counts: 1\n",
        "```"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "The previous block shows the part of the config file that concerns the experiment part. Since assignment and experiment parts can be run independently, the `bc_length` needs to be set again.\n",
        "\n",
        "The `experiment_file` holds the path to the file of the experiment table (i.e. condition, replicate and name of read files given)."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Because we run the experiment workflow, we start with `experiments`. Then a list with possible names of experiments will follow. Here we have just one: `experimentMPRAworkshop`. The whole config file has the minimal information, which is needed by MPRAsnakeflow (except for `configs`, see later). This is the barcode length `bc_length` and the UMI length `umi_length` which we calculated previously. Then a link to folder `data_folder` where all fastqs of the experiment are. What conditions, replicates and belonging fastq files we have is stored in the `experiment_file`."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "As a sanity check and to understand where these values come from, we will take a look at whether the information matches our input files. We used 15 bp adapters for this MPRA experiment, which were added at both ends of each designed oligo. The `alignment_start` configuration refers to these adapters as our actual target sequences only start after the 15 bp of adapter sequence."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Average length of a barcode:\n",
            "15\n",
            "15\n",
            "\n",
            "Average length of a target sequence (oligo length - 2 x adapter):\n",
            "\n",
            "Average length of a target sequence (oligo length - 2 x adapter):\n",
            "200\n",
            "200\n"
          ]
        }
      ],
      "source": [
        "adapter_length=15\n",
        "\n",
        "print(\"\\nAverage length of a barcode:\")\n",
        "!zcat \"MPRAsnakeflow_tutorial/example_data/assignment/assoc_bc.fastq.gz\" | awk 'NR % 4 == 2 {{sum += length($$0)}} END {{print sum*4/NR}}'\n",
        "\n",
        "print(\"\\nAverage length of a target sequence (oligo length - 2 x adapter):\")\n",
        "seq_length_with_bc = !cat \"MPRAsnakeflow_tutorial/example_data/design/workshop_design.fa\" | awk 'NR % 4 == 2 {{sum +=length($$0)}} END {{print sum*4/NR}}'\n",
        "seq_length_with_bc = int(seq_length_with_bc[0])\n",
        "seq_length = seq_length_with_bc - 2*adapter_length\n",
        "print(seq_length)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### MPRAsnakeflow tutorial profile\n",
        "\n",
        "Snakemake can make use of profiles to pre-configure command-line interfaces as well as resources and number of threads as default or per rule. A profile has to be stored (on Linux) in the folder `$HOME/.config/snakemake` or `/etc/xdg/snakemake`. It can be activated by using the `--profile` option in the snakemake command. In addition, you can set an environment variable `SNAKEMAKE_PROFILE`, e.g. by specifying `export SNAKEMAKE_PROFILE=myprofile` in your `~/.bashrc` to use the profile by default. In addition, a specific workflow profile can be set using `--workflow-profile path_to_your_profile`. E.g. we pre-defined a [workflow profile for MPRAsnakeflow](https://github.com/kircherlab/MPRAsnakeflow/blob/development/profiles/default/config.yaml) adding resources to each rule with some additional commands for the [snakemake executor plugin slurm](https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html).\n",
        "\n",
        "For more information on snakemake using profiles, please have a look at the [snakemake profile documentation](https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles).\n",
        "\n",
        "For the MPRAsnakeflow tutorial, we will use the profile `mprasnakeflow` which is stored in the folder `/etc/xdg/snakemake` within the container and the `SNAKEMAKE_PROFILE=mprasnakeflow` is already set. Have a look at the variable and the profile:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ docker run -v=${PWD}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run visze/mprasnakeflow_tutorial:0.5.3 cat /etc/xdg/snakemake/mprasnakeflow/config.yaml\n",
            "---\n",
            "snakefile: /data/MPRAsnakeflow/workflow/Snakefile\n",
            "software-deployment-method: conda\n",
            "conda-prefix: /data/conda_envs\n",
            "cores: 1\n",
            "set-threads:\n",
            "  assignment_mapping_bbmap: 10\n",
            "---\n",
            "snakefile: /data/MPRAsnakeflow/workflow/Snakefile\n",
            "software-deployment-method: conda\n",
            "conda-prefix: /data/conda_envs\n",
            "cores: 1\n",
            "set-threads:\n",
            "  assignment_mapping_bbmap: 10\n"
          ]
        }
      ],
      "source": [
        "runContainer(\"cat /etc/xdg/snakemake/mprasnakeflow/config.yaml\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "You see that we already defined where the MPRAsnakeflow `Snakefile` is and `--snakefile` is not necessary. Also, all software packages are installed already in the container using conda and we enable this software deployment method and tell snakemake where the conda environments live (the default will be your execution folder under `.snakemake/conda`). Also, the `--cores/-c` option is not necessary anymore and snakemake just needs your configuration file of the workflow using the command `--configfile`"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Run the workflow\n",
        "\n",
        "First, we run a dry-run to see what will be executed using `-n` and `--quiet rules` to see only the overview of executed rules and not each rule command individually. The most important part is setting the config file via `--configfile config_experiment.yaml`"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TXlOomWL5Am2",
        "outputId": "5c32a998-eaef-426e-ccc8-8102d2487950"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ docker run -v=${PWD}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run visze/mprasnakeflow_tutorial:0.5.3 snakemake --configfile config_experiment.yaml -n --quiet rules\n",
            "Using profile mprasnakeflow for setting default command line arguments.\n",
            "Using profile mprasnakeflow for setting default command line arguments.\n",
            "\u001b[95mRunning MPRAsnakeflow version 0.5.3\u001b[0m\n",
            "host: af8541701f5d\n",
            "Building DAG of jobs...\n",
            "\u001b[95mRunning MPRAsnakeflow version 0.5.3\u001b[0m\n",
            "host: af8541701f5d\n",
            "Building DAG of jobs...\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd\n",
            "\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "Reasons:\n",
            "    (check individual jobs above for details)\n",
            "    input files updated by another job:\n",
            "        all, experiment_assigned_counts_assignBarcodes, experiment_assigned_counts_combine_replicates, experiment_assigned_counts_combine_replicates_barcode_output, experiment_assigned_counts_copy_final_all_files, experiment_assigned_counts_copy_final_thresh_files, experiment_assigned_counts_dna_rna_merge, experiment_assigned_counts_make_master_tables, experiment_counts_dna_rna_merge_counts, experiment_counts_filter_counts, experiment_counts_final_counts, experiment_counts_umi_raw_counts, experiment_statistic_assigned_counts_combine_BC_assignment_stats, experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all, experiment_statistic_bc_overlap_combine_assigned_counts, experiment_statistic_bc_overlap_combine_counts, experiment_statistic_bc_overlap_run, experiment_statistic_correlation_bc_counts, experiment_statistic_correlation_bc_counts_hist, experiment_statistic_correlation_calculate, experiment_statistic_correlation_combine_bc_assigned, experiment_statistic_correlation_combine_bc_raw, experiment_statistic_correlation_combine_oligo, experiment_statistic_correlation_hist_box_plots, experiment_statistic_counts_BC_in_RNA_DNA, experiment_statistic_counts_BC_in_RNA_DNA_merge, experiment_statistic_counts_barcode_base_composition, experiment_statistic_counts_final, experiment_statistic_counts_frequent_umis, experiment_statistic_counts_stats_merge, experiment_statistic_counts_table, experiment_statistic_quality_metric, qc_report_count\n",
            "    output files have to be generated:\n",
            "        experiment_assigned_counts_assignBarcodes, experiment_assigned_counts_combine_replicates, experiment_assigned_counts_combine_replicates_barcode_output, experiment_assigned_counts_copy_final_all_files, experiment_assigned_counts_copy_final_thresh_files, experiment_assigned_counts_dna_rna_merge, experiment_assigned_counts_filterAssignment, experiment_assigned_counts_make_master_tables, experiment_counts_dna_rna_merge_counts, experiment_counts_filter_counts, experiment_counts_final_counts, experiment_counts_umi_create_BAM, experiment_counts_umi_raw_counts, experiment_statistic_assigned_counts_combine_BC_assignment_stats, experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all, experiment_statistic_bc_overlap_combine_assigned_counts, experiment_statistic_bc_overlap_combine_counts, experiment_statistic_bc_overlap_run, experiment_statistic_correlation_bc_counts, experiment_statistic_correlation_bc_counts_hist, experiment_statistic_correlation_calculate, experiment_statistic_correlation_combine_bc_assigned, experiment_statistic_correlation_combine_bc_raw, experiment_statistic_correlation_combine_oligo, experiment_statistic_correlation_hist_box_plots, experiment_statistic_counts_BC_in_RNA_DNA, experiment_statistic_counts_BC_in_RNA_DNA_merge, experiment_statistic_counts_barcode_base_composition, experiment_statistic_counts_final, experiment_statistic_counts_frequent_umis, experiment_statistic_counts_stats_merge, experiment_statistic_counts_table, experiment_statistic_quality_metric, qc_report_count\n",
            "This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "Would remove temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd\n",
            "\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "Reasons:\n",
            "    (check individual jobs above for details)\n",
            "    input files updated by another job:\n",
            "        all, experiment_assigned_counts_assignBarcodes, experiment_assigned_counts_combine_replicates, experiment_assigned_counts_combine_replicates_barcode_output, experiment_assigned_counts_copy_final_all_files, experiment_assigned_counts_copy_final_thresh_files, experiment_assigned_counts_dna_rna_merge, experiment_assigned_counts_make_master_tables, experiment_counts_dna_rna_merge_counts, experiment_counts_filter_counts, experiment_counts_final_counts, experiment_counts_umi_raw_counts, experiment_statistic_assigned_counts_combine_BC_assignment_stats, experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all, experiment_statistic_bc_overlap_combine_assigned_counts, experiment_statistic_bc_overlap_combine_counts, experiment_statistic_bc_overlap_run, experiment_statistic_correlation_bc_counts, experiment_statistic_correlation_bc_counts_hist, experiment_statistic_correlation_calculate, experiment_statistic_correlation_combine_bc_assigned, experiment_statistic_correlation_combine_bc_raw, experiment_statistic_correlation_combine_oligo, experiment_statistic_correlation_hist_box_plots, experiment_statistic_counts_BC_in_RNA_DNA, experiment_statistic_counts_BC_in_RNA_DNA_merge, experiment_statistic_counts_barcode_base_composition, experiment_statistic_counts_final, experiment_statistic_counts_frequent_umis, experiment_statistic_counts_stats_merge, experiment_statistic_counts_table, experiment_statistic_quality_metric, qc_report_count\n",
            "    output files have to be generated:\n",
            "        experiment_assigned_counts_assignBarcodes, experiment_assigned_counts_combine_replicates, experiment_assigned_counts_combine_replicates_barcode_output, experiment_assigned_counts_copy_final_all_files, experiment_assigned_counts_copy_final_thresh_files, experiment_assigned_counts_dna_rna_merge, experiment_assigned_counts_filterAssignment, experiment_assigned_counts_make_master_tables, experiment_counts_dna_rna_merge_counts, experiment_counts_filter_counts, experiment_counts_final_counts, experiment_counts_umi_create_BAM, experiment_counts_umi_raw_counts, experiment_statistic_assigned_counts_combine_BC_assignment_stats, experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge, experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all, experiment_statistic_bc_overlap_combine_assigned_counts, experiment_statistic_bc_overlap_combine_counts, experiment_statistic_bc_overlap_run, experiment_statistic_correlation_bc_counts, experiment_statistic_correlation_bc_counts_hist, experiment_statistic_correlation_calculate, experiment_statistic_correlation_combine_bc_assigned, experiment_statistic_correlation_combine_bc_raw, experiment_statistic_correlation_combine_oligo, experiment_statistic_correlation_hist_box_plots, experiment_statistic_counts_BC_in_RNA_DNA, experiment_statistic_counts_BC_in_RNA_DNA_merge, experiment_statistic_counts_barcode_base_composition, experiment_statistic_counts_final, experiment_statistic_counts_frequent_umis, experiment_statistic_counts_stats_merge, experiment_statistic_counts_table, experiment_statistic_quality_metric, qc_report_count\n",
            "This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.\n"
          ]
        }
      ],
      "source": [
        "# MPRAsnakeflow_tutorial must be accessible and writable for all (and docker)\n",
        "!chmod ugo+rwx MPRAsnakeflow_tutorial\n",
        "# Run MPRAsnakeflow\n",
        "runContainer(\"snakemake --configfile config_experiment.yaml -n --quiet rules\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "You see a list of rules and how often this rule is executed depending on the input file. A short summary of each rule is in the [MPRAsnakeflow documentation](https://mprasnakeflow.readthedocs.io/en/latest/experiment.html#rules).\n",
        "\n",
        "Now let's run snakemake using 10 cores. When you add the `-p` flag you will see the exact code that is executed in each rule. But we omit this here because it is a lot of output."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "gNESKLR_UkLV"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[1;32muser@pc\u001b[0m$ docker run -v=${PWD}/MPRAsnakeflow_tutorial:/data/run --workdir /data/run visze/mprasnakeflow_tutorial:0.5.3 snakemake --configfile config_experiment.yaml -c 10\n",
            "Using profile mprasnakeflow for setting default command line arguments.\n",
            "Using profile mprasnakeflow for setting default command line arguments.\n",
            "Assuming unrestricted shared filesystem usage.\n",
            "host: 56f159cc3a1a\n",
            "Building DAG of jobs...\n",
            "Assuming unrestricted shared filesystem usage.\n",
            "host: 56f159cc3a1a\n",
            "Building DAG of jobs...\n",
            "Using shell: /usr/bin/bash\n",
            "Provided cores: 10\n",
            "Rules claiming more threads will be scaled down.\n",
            "Singularity containers: ignored\n",
            "Using shell: /usr/bin/bash\n",
            "Provided cores: 10\n",
            "Rules claiming more threads will be scaled down.\n",
            "Singularity containers: ignored\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "Select jobs to execute...\n",
            "Job stats:\n",
            "job                                                                        count\n",
            "-----------------------------------------------------------------------  -------\n",
            "all                                                                            1\n",
            "experiment_assigned_counts_assignBarcodes                                     12\n",
            "experiment_assigned_counts_combine_replicates                                  4\n",
            "experiment_assigned_counts_combine_replicates_barcode_output                   2\n",
            "experiment_assigned_counts_copy_final_all_files                                2\n",
            "experiment_assigned_counts_copy_final_thresh_files                             2\n",
            "experiment_assigned_counts_dna_rna_merge                                       6\n",
            "experiment_assigned_counts_filterAssignment                                    1\n",
            "experiment_assigned_counts_make_master_tables                                  2\n",
            "experiment_counts_dna_rna_merge_counts                                        12\n",
            "experiment_counts_filter_counts                                                6\n",
            "experiment_counts_final_counts                                                 6\n",
            "experiment_counts_umi_create_BAM                                               6\n",
            "experiment_counts_umi_raw_counts                                               6\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats               2\n",
            "experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper        2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge               2\n",
            "experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all           2\n",
            "experiment_statistic_bc_overlap_combine_assigned_counts                        2\n",
            "experiment_statistic_bc_overlap_combine_counts                                 2\n",
            "experiment_statistic_bc_overlap_run                                            8\n",
            "experiment_statistic_correlation_bc_counts                                     4\n",
            "experiment_statistic_correlation_bc_counts_hist                                4\n",
            "experiment_statistic_correlation_calculate                                     2\n",
            "experiment_statistic_correlation_combine_bc_assigned                           2\n",
            "experiment_statistic_correlation_combine_bc_raw                                2\n",
            "experiment_statistic_correlation_combine_oligo                                 2\n",
            "experiment_statistic_correlation_hist_box_plots                                2\n",
            "experiment_statistic_counts_BC_in_RNA_DNA                                      6\n",
            "experiment_statistic_counts_BC_in_RNA_DNA_merge                                2\n",
            "experiment_statistic_counts_barcode_base_composition                           6\n",
            "experiment_statistic_counts_final                                              2\n",
            "experiment_statistic_counts_frequent_umis                                      6\n",
            "experiment_statistic_counts_stats_merge                                        2\n",
            "experiment_statistic_counts_table                                             12\n",
            "experiment_statistic_quality_metric                                            2\n",
            "qc_report_count                                                                2\n",
            "total                                                                        146\n",
            "\n",
            "Select jobs to execute...\n",
            "Execute 7 jobs...\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-3_S3_R1_001.fastq.gz, example_data/counts/dna-3_S3_R3_001.fastq.gz, example_data/counts/dna-3_S3_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 29\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 7 jobs...\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-3_S3_R1_001.fastq.gz, example_data/counts/dna-3_S3_R3_001.fastq.gz, example_data/counts/dna-3_S3_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 29\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-1_S1_R1_001.fastq.gz, example_data/counts/dna-1_S1_R3_001.fastq.gz, example_data/counts/dna-1_S1_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-3_S6_R1_001.fastq.gz, example_data/counts/rna-3_S6_R3_001.fastq.gz, example_data/counts/rna-3_S6_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 33\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-1_S1_R1_001.fastq.gz, example_data/counts/dna-1_S1_R3_001.fastq.gz, example_data/counts/dna-1_S1_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 8\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-3_S6_R1_001.fastq.gz, example_data/counts/rna-3_S6_R3_001.fastq.gz, example_data/counts/rna-3_S6_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 33\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-1_S4_R1_001.fastq.gz, example_data/counts/rna-1_S4_R3_001.fastq.gz, example_data/counts/rna-1_S4_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 12\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-2_S5_R1_001.fastq.gz, example_data/counts/rna-2_S5_R3_001.fastq.gz, example_data/counts/rna-2_S5_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 23\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-2_S2_R1_001.fastq.gz, example_data/counts/dna-2_S2_R3_001.fastq.gz, example_data/counts/dna-2_S2_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 19\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-1_S4_R1_001.fastq.gz, example_data/counts/rna-1_S4_R3_001.fastq.gz, example_data/counts/rna-1_S4_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 12\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/rna-2_S5_R1_001.fastq.gz, example_data/counts/rna-2_S5_R3_001.fastq.gz, example_data/counts/rna-2_S5_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 23\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_counts_umi_create_BAM:\n",
            "    input: example_data/counts/dna-2_S2_R1_001.fastq.gz, example_data/counts/dna-2_S2_R3_001.fastq.gz, example_data/counts/dna-2_S2_R2_001.fastq.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/FastQ2doubleIndexBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/library_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReadsBAM_python3.py (cached), /data/MPRAsnakeflow/workflow/rules/../scripts/count/MergeTrimReads_python3.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    log: results/logs/experiment/counts/umi/create_BAM.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 19\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_assigned_counts_filterAssignment:\n",
            "    input: example_data/assignment/assignment_barcodes.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/samplerer_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/filterAssignment.experimentMPRAworkshop.MPRAworkshop.log\n",
            "    jobid: 13\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:24:40 2025]\n",
            "localrule experiment_assigned_counts_filterAssignment:\n",
            "    input: example_data/assignment/assignment_barcodes.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/samplerer_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/filterAssignment.experimentMPRAworkshop.MPRAworkshop.log\n",
            "    jobid: 13\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:24:53 2025]\n",
            "Finished jobid: 13 (Rule: experiment_assigned_counts_filterAssignment)\n",
            "1 of 146 steps (1%) done\n",
            "[Tue May 13 10:24:53 2025]\n",
            "Finished jobid: 13 (Rule: experiment_assigned_counts_filterAssignment)\n",
            "1 of 146 steps (1%) done\n",
            "[Tue May 13 10:29:27 2025]\n",
            "Finished jobid: 29 (Rule: experiment_counts_umi_create_BAM)\n",
            "2 of 146 steps (1%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:27 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 28\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:27 2025]\n",
            "Finished jobid: 29 (Rule: experiment_counts_umi_create_BAM)\n",
            "2 of 146 steps (1%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:27 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 28\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:34 2025]\n",
            "Finished jobid: 19 (Rule: experiment_counts_umi_create_BAM)\n",
            "3 of 146 steps (2%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:34 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 18\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:34 2025]\n",
            "Finished jobid: 19 (Rule: experiment_counts_umi_create_BAM)\n",
            "3 of 146 steps (2%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:34 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 18\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:37 2025]\n",
            "Finished jobid: 28 (Rule: experiment_counts_umi_raw_counts)\n",
            "4 of 146 steps (3%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:37 2025]\n",
            "Finished jobid: 28 (Rule: experiment_counts_umi_raw_counts)\n",
            "4 of 146 steps (3%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:37 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_DNA_raw.log\n",
            "    jobid: 92\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:37 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 27\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:37 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_DNA_raw.log\n",
            "    jobid: 92\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:37 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 27\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "15\n",
            "[Tue May 13 10:29:39 2025]\n",
            "Finished jobid: 92 (Rule: experiment_statistic_counts_table)\n",
            "5 of 146 steps (3%) done\n",
            "[Tue May 13 10:29:39 2025]\n",
            "Finished jobid: 92 (Rule: experiment_statistic_counts_table)\n",
            "5 of 146 steps (3%) done\n",
            "[Tue May 13 10:29:40 2025]\n",
            "Finished jobid: 8 (Rule: experiment_counts_umi_create_BAM)\n",
            "6 of 146 steps (4%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:40 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 7\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:40 2025]\n",
            "Finished jobid: 8 (Rule: experiment_counts_umi_create_BAM)\n",
            "6 of 146 steps (4%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:40 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 7\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:29:43 2025]\n",
            "Finished jobid: 27 (Rule: experiment_counts_filter_counts)\n",
            "7 of 146 steps (5%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:43 2025]\n",
            "Finished jobid: 27 (Rule: experiment_counts_filter_counts)\n",
            "7 of 146 steps (5%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_DNA_filtered.log\n",
            "    jobid: 104\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_DNA_filtered.log\n",
            "    jobid: 104\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 112\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 26\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 112\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:43 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.3.DNA.log\n",
            "    jobid: 26\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:29:44 2025]\n",
            "Finished jobid: 18 (Rule: experiment_counts_umi_raw_counts)\n",
            "8 of 146 steps (5%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:44 2025]\n",
            "Finished jobid: 18 (Rule: experiment_counts_umi_raw_counts)\n",
            "8 of 146 steps (5%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:44 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 17\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:44 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_DNA_raw.log\n",
            "    jobid: 90\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:44 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 17\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:44 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_DNA_raw.log\n",
            "    jobid: 90\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "15\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 104 (Rule: experiment_statistic_counts_table)\n",
            "9 of 146 steps (6%) done\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 104 (Rule: experiment_statistic_counts_table)\n",
            "9 of 146 steps (6%) done\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 26 (Rule: experiment_counts_final_counts)\n",
            "10 of 146 steps (7%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 26 (Rule: experiment_counts_final_counts)\n",
            "10 of 146 steps (7%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_3_DNA.log\n",
            "    jobid: 82\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 67\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_3_DNA.log\n",
            "    jobid: 82\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 67\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.DNA.default.MPRAworkshop.log\n",
            "    jobid: 40\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 112 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "11 of 146 steps (8%) done\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:46 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.DNA.default.MPRAworkshop.log\n",
            "    jobid: 40\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 112 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "11 of 146 steps (8%) done\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 90 (Rule: experiment_statistic_counts_table)\n",
            "12 of 146 steps (8%) done\n",
            "[Tue May 13 10:29:46 2025]\n",
            "Finished jobid: 90 (Rule: experiment_statistic_counts_table)\n",
            "12 of 146 steps (8%) done\n",
            "[Tue May 13 10:29:51 2025]\n",
            "Finished jobid: 17 (Rule: experiment_counts_filter_counts)\n",
            "[Tue May 13 10:29:51 2025]\n",
            "Finished jobid: 17 (Rule: experiment_counts_filter_counts)\n",
            "13 of 146 steps (9%) done\n",
            "Select jobs to execute...\n",
            "13 of 146 steps (9%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_DNA_filtered.log\n",
            "    jobid: 102\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 16\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 111\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_DNA_filtered.log\n",
            "    jobid: 102\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 16\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:29:51 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.2.DNA.log\n",
            "    jobid: 111\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:29:52 2025]\n",
            "Finished jobid: 7 (Rule: experiment_counts_umi_raw_counts)\n",
            "14 of 146 steps (10%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:52 2025]\n",
            "Finished jobid: 7 (Rule: experiment_counts_umi_raw_counts)\n",
            "14 of 146 steps (10%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:52 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:52 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 6\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 102 (Rule: experiment_statistic_counts_table)\n",
            "15 of 146 steps (10%) done\n",
            "Execute 1 jobs...\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 102 (Rule: experiment_statistic_counts_table)\n",
            "15 of 146 steps (10%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_DNA_raw.log\n",
            "    jobid: 88\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_DNA_raw.log\n",
            "    jobid: 88\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 111 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "16 of 146 steps (11%) done\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 16 (Rule: experiment_counts_final_counts)\n",
            "17 of 146 steps (12%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 66\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_2_DNA.log\n",
            "    jobid: 81\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 111 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "16 of 146 steps (11%) done\n",
            "[Tue May 13 10:29:54 2025]\n",
            "Finished jobid: 16 (Rule: experiment_counts_final_counts)\n",
            "17 of 146 steps (12%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 66\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:29:54 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_2_DNA.log\n",
            "    jobid: 81\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:56 2025]\n",
            "Finished jobid: 88 (Rule: experiment_statistic_counts_table)\n",
            "18 of 146 steps (12%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:56 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.DNA.default.MPRAworkshop.log\n",
            "    jobid: 39\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:29:56 2025]\n",
            "Finished jobid: 88 (Rule: experiment_statistic_counts_table)\n",
            "18 of 146 steps (12%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:56 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.DNA.default.MPRAworkshop.log\n",
            "    jobid: 39\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:29:59 2025]\n",
            "Finished jobid: 6 (Rule: experiment_counts_filter_counts)\n",
            "19 of 146 steps (13%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:29:59 2025]\n",
            "Finished jobid: 6 (Rule: experiment_counts_filter_counts)\n",
            "19 of 146 steps (13%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:59 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 110\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:29:59 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 110\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:04 2025]\n",
            "Finished jobid: 110 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "20 of 146 steps (14%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:04 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 5\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:30:04 2025]\n",
            "Finished jobid: 110 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "20 of 146 steps (14%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:04 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.1.DNA.log\n",
            "    jobid: 5\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:08 2025]\n",
            "Finished jobid: 5 (Rule: experiment_counts_final_counts)\n",
            "21 of 146 steps (14%) done\n",
            "[Tue May 13 10:30:08 2025]\n",
            "Finished jobid: 5 (Rule: experiment_counts_final_counts)\n",
            "21 of 146 steps (14%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:08 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.tutorialConfig.counts.log\n",
            "    jobid: 126\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:08 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.tutorialConfig.counts.log\n",
            "    jobid: 126\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:16 2025]\n",
            "Finished jobid: 67 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "22 of 146 steps (15%) done\n",
            "[Tue May 13 10:30:16 2025]\n",
            "Finished jobid: 67 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "22 of 146 steps (15%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:16 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.DNA.default.MPRAworkshop.log\n",
            "    jobid: 38\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:16 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.DNA.default.MPRAworkshop.log\n",
            "    jobid: 38\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:16 2025]\n",
            "Finished jobid: 40 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "23 of 146 steps (16%) done\n",
            "[Tue May 13 10:30:16 2025]\n",
            "Finished jobid: 40 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "23 of 146 steps (16%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:17 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.default.counts.log\n",
            "    jobid: 123\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:17 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.default.counts.log\n",
            "    jobid: 123\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:21 2025]\n",
            "Finished jobid: 126 (Rule: experiment_statistic_bc_overlap_run)\n",
            "24 of 146 steps (16%) done\n",
            "[Tue May 13 10:30:21 2025]\n",
            "Finished jobid: 126 (Rule: experiment_statistic_bc_overlap_run)\n",
            "24 of 146 steps (16%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:21 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_DNA_filtered.log\n",
            "    jobid: 100\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:21 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_DNA_filtered.log\n",
            "    jobid: 100\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:23 2025]\n",
            "Finished jobid: 100 (Rule: experiment_statistic_counts_table)\n",
            "25 of 146 steps (17%) done\n",
            "[Tue May 13 10:30:23 2025]\n",
            "Finished jobid: 100 (Rule: experiment_statistic_counts_table)\n",
            "25 of 146 steps (17%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:23 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 65\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:23 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.DNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 65\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:30:26 2025]\n",
            "Finished jobid: 123 (Rule: experiment_statistic_bc_overlap_run)\n",
            "26 of 146 steps (18%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:26 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_1_DNA.log\n",
            "    jobid: 80\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:30:26 2025]\n",
            "Finished jobid: 123 (Rule: experiment_statistic_bc_overlap_run)\n",
            "26 of 146 steps (18%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:26 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_1_DNA.log\n",
            "    jobid: 80\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=DNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:30:27 2025]\n",
            "Finished jobid: 66 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "[Tue May 13 10:30:27 2025]\n",
            "Finished jobid: 66 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "27 of 146 steps (18%) done\n",
            "27 of 146 steps (18%) done\n",
            "[Tue May 13 10:30:28 2025]\n",
            "Finished jobid: 39 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "28 of 146 steps (19%) done\n",
            "[Tue May 13 10:30:28 2025]\n",
            "Finished jobid: 39 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "28 of 146 steps (19%) done\n",
            "[Tue May 13 10:30:39 2025]\n",
            "Finished jobid: 82 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "29 of 146 steps (20%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:30:39 2025]\n",
            "Finished jobid: 82 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "29 of 146 steps (20%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:30:43 2025]\n",
            "Finished jobid: 38 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "30 of 146 steps (21%) done\n",
            "[Tue May 13 10:30:43 2025]\n",
            "Finished jobid: 38 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "30 of 146 steps (21%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:43 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 133\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:43 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 133\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=DNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:30:45 2025]\n",
            "Finished jobid: 65 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "31 of 146 steps (21%) done\n",
            "[Tue May 13 10:30:45 2025]\n",
            "Finished jobid: 65 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "31 of 146 steps (21%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:45 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 136\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:30:45 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.DNA.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 136\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=DNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:30:47 2025]\n",
            "Finished jobid: 81 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "32 of 146 steps (22%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:30:47 2025]\n",
            "Finished jobid: 81 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "32 of 146 steps (22%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:30:48 2025]\n",
            "Finished jobid: 133 (Rule: experiment_statistic_bc_overlap_run)\n",
            "33 of 146 steps (23%) done\n",
            "[Tue May 13 10:30:48 2025]\n",
            "Finished jobid: 133 (Rule: experiment_statistic_bc_overlap_run)\n",
            "33 of 146 steps (23%) done\n",
            "[Tue May 13 10:30:50 2025]\n",
            "Finished jobid: 136 (Rule: experiment_statistic_bc_overlap_run)\n",
            "34 of 146 steps (23%) done\n",
            "[Tue May 13 10:30:50 2025]\n",
            "Finished jobid: 136 (Rule: experiment_statistic_bc_overlap_run)\n",
            "34 of 146 steps (23%) done\n",
            "[Tue May 13 10:30:59 2025]\n",
            "Finished jobid: 80 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "35 of 146 steps (24%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:30:59 2025]\n",
            "Finished jobid: 80 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "35 of 146 steps (24%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:35:56 2025]\n",
            "Finished jobid: 12 (Rule: experiment_counts_umi_create_BAM)\n",
            "36 of 146 steps (25%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:35:56 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 11\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:35:56 2025]\n",
            "Finished jobid: 12 (Rule: experiment_counts_umi_create_BAM)\n",
            "36 of 146 steps (25%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:35:56 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 11\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:36:10 2025]\n",
            "Finished jobid: 11 (Rule: experiment_counts_umi_raw_counts)\n",
            "37 of 146 steps (25%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:10 2025]\n",
            "Finished jobid: 11 (Rule: experiment_counts_umi_raw_counts)\n",
            "37 of 146 steps (25%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 10\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_1_raw.log\n",
            "    jobid: 95\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_RNA_raw.log\n",
            "    jobid: 89\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 10\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_1_raw.log\n",
            "    jobid: 95\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_RNA_raw.log\n",
            "    jobid: 89\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:10 2025]\n",
            "Finished jobid: 23 (Rule: experiment_counts_umi_create_BAM)\n",
            "38 of 146 steps (26%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 22\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:36:10 2025]\n",
            "Finished jobid: 23 (Rule: experiment_counts_umi_create_BAM)\n",
            "38 of 146 steps (26%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:10 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 22\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "15\n",
            "15\n",
            "[Tue May 13 10:36:12 2025]\n",
            "Finished jobid: 95 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "39 of 146 steps (27%) done\n",
            "[Tue May 13 10:36:12 2025]\n",
            "Finished jobid: 95 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "39 of 146 steps (27%) done\n",
            "[Tue May 13 10:36:13 2025]\n",
            "Finished jobid: 89 (Rule: experiment_statistic_counts_table)\n",
            "40 of 146 steps (27%) done\n",
            "[Tue May 13 10:36:13 2025]\n",
            "Finished jobid: 89 (Rule: experiment_statistic_counts_table)\n",
            "40 of 146 steps (27%) done\n",
            "[Tue May 13 10:36:19 2025]\n",
            "Finished jobid: 10 (Rule: experiment_counts_filter_counts)\n",
            "41 of 146 steps (28%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:19 2025]\n",
            "Finished jobid: 10 (Rule: experiment_counts_filter_counts)\n",
            "41 of 146 steps (28%) done\n",
            "Select jobs to execute...\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_1_filtered.log\n",
            "    jobid: 107\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 9\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_RNA_filtered.log\n",
            "    jobid: 101\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 113\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_1_filtered.log\n",
            "    jobid: 107\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 9\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_1_RNA_filtered.log\n",
            "    jobid: 101\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:19 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.1.RNA.log\n",
            "    jobid: 113\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:21 2025]\n",
            "Finished jobid: 9 (Rule: experiment_counts_final_counts)\n",
            "42 of 146 steps (29%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:21 2025]\n",
            "Finished jobid: 9 (Rule: experiment_counts_final_counts)\n",
            "42 of 146 steps (29%) done\n",
            "Select jobs to execute...\n",
            "Execute 5 jobs...\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.1.tutorialConfig.log\n",
            "    jobid: 56\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=1, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 68\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_1_RNA.log\n",
            "    jobid: 83\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 5 jobs...\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.1.tutorialConfig.log\n",
            "    jobid: 56\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=1, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 68\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_1_RNA.log\n",
            "    jobid: 83\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=1, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.1.default.log\n",
            "    jobid: 4\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=1, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.RNA.default.MPRAworkshop.log\n",
            "    jobid: 41\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:21 2025]\n",
            "Finished jobid: 107 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "43 of 146 steps (29%) done\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.1.default.log\n",
            "    jobid: 4\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=1, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:21 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.1.RNA.default.MPRAworkshop.log\n",
            "    jobid: 41\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=1, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:21 2025]\n",
            "Finished jobid: 107 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "43 of 146 steps (29%) done\n",
            "[Tue May 13 10:36:23 2025]\n",
            "Finished jobid: 101 (Rule: experiment_statistic_counts_table)\n",
            "44 of 146 steps (30%) done\n",
            "[Tue May 13 10:36:23 2025]\n",
            "Finished jobid: 101 (Rule: experiment_statistic_counts_table)\n",
            "44 of 146 steps (30%) done\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 4 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "45 of 146 steps (31%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:24 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.1.default.log\n",
            "    jobid: 3\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 4 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "45 of 146 steps (31%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:24 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.1.default.log\n",
            "    jobid: 3\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 56 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "46 of 146 steps (32%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 56 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "46 of 146 steps (32%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:24 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.1.tutorialConfig.log\n",
            "    jobid: 55\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:24 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.1.tutorialConfig.log\n",
            "    jobid: 55\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=1\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 113 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "47 of 146 steps (32%) done\n",
            "[Tue May 13 10:36:24 2025]\n",
            "Finished jobid: 113 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "47 of 146 steps (32%) done\n",
            "[Tue May 13 10:36:26 2025]\n",
            "Finished jobid: 22 (Rule: experiment_counts_umi_raw_counts)\n",
            "48 of 146 steps (33%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:26 2025]\n",
            "Finished jobid: 22 (Rule: experiment_counts_umi_raw_counts)\n",
            "48 of 146 steps (33%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 21\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_2_raw.log\n",
            "    jobid: 96\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_RNA_raw.log\n",
            "    jobid: 91\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 21\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_2_raw.log\n",
            "    jobid: 96\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:26 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_RNA_raw.log\n",
            "    jobid: 91\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "15\n",
            "[Tue May 13 10:36:28 2025]\n",
            "Finished jobid: 3 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "49 of 146 steps (34%) done\n",
            "[Tue May 13 10:36:28 2025]\n",
            "Finished jobid: 3 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "49 of 146 steps (34%) done\n",
            "[Tue May 13 10:36:29 2025]\n",
            "Finished jobid: 55 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "50 of 146 steps (34%) done\n",
            "[Tue May 13 10:36:29 2025]\n",
            "Finished jobid: 96 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "51 of 146 steps (35%) done\n",
            "[Tue May 13 10:36:29 2025]\n",
            "Finished jobid: 55 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "50 of 146 steps (34%) done\n",
            "[Tue May 13 10:36:29 2025]\n",
            "Finished jobid: 96 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "51 of 146 steps (35%) done\n",
            "[Tue May 13 10:36:30 2025]\n",
            "Finished jobid: 91 (Rule: experiment_statistic_counts_table)\n",
            "52 of 146 steps (36%) done\n",
            "[Tue May 13 10:36:30 2025]\n",
            "Finished jobid: 91 (Rule: experiment_statistic_counts_table)\n",
            "52 of 146 steps (36%) done\n",
            "[Tue May 13 10:36:38 2025]\n",
            "Finished jobid: 21 (Rule: experiment_counts_filter_counts)\n",
            "53 of 146 steps (36%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:38 2025]\n",
            "Finished jobid: 21 (Rule: experiment_counts_filter_counts)\n",
            "53 of 146 steps (36%) done\n",
            "Select jobs to execute...\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 114\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_2_filtered.log\n",
            "    jobid: 108\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 20\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_RNA_filtered.log\n",
            "    jobid: 103\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 114\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_2_filtered.log\n",
            "    jobid: 108\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.2.RNA.log\n",
            "    jobid: 20\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:36:38 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_2_RNA_filtered.log\n",
            "    jobid: 103\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:41 2025]\n",
            "Finished jobid: 20 (Rule: experiment_counts_final_counts)\n",
            "54 of 146 steps (37%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:41 2025]\n",
            "Finished jobid: 20 (Rule: experiment_counts_final_counts)\n",
            "54 of 146 steps (37%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.2.tutorialConfig.log\n",
            "    jobid: 58\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.RNA.default.MPRAworkshop.log\n",
            "    jobid: 42\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.2.tutorialConfig.log\n",
            "    jobid: 58\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.RNA.default.MPRAworkshop.log\n",
            "    jobid: 42\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 69\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:41 2025]\n",
            "Finished jobid: 108 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "55 of 146 steps (38%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_2_RNA.log\n",
            "    jobid: 84\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.2.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 69\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:41 2025]\n",
            "Finished jobid: 108 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "55 of 146 steps (38%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:41 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_2_RNA.log\n",
            "    jobid: 84\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=2, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:43 2025]\n",
            "Finished jobid: 103 (Rule: experiment_statistic_counts_table)\n",
            "56 of 146 steps (38%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:43 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.2.default.log\n",
            "    jobid: 15\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:43 2025]\n",
            "Finished jobid: 103 (Rule: experiment_statistic_counts_table)\n",
            "56 of 146 steps (38%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:43 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.2.default.log\n",
            "    jobid: 15\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:44 2025]\n",
            "Finished jobid: 114 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "57 of 146 steps (39%) done\n",
            "[Tue May 13 10:36:44 2025]\n",
            "Finished jobid: 114 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "57 of 146 steps (39%) done\n",
            "[Tue May 13 10:36:44 2025]\n",
            "Finished jobid: 58 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "58 of 146 steps (40%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:44 2025]\n",
            "Finished jobid: 58 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "58 of 146 steps (40%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:44 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.2.tutorialConfig.log\n",
            "    jobid: 57\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:44 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.2.tutorialConfig.log\n",
            "    jobid: 57\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:45 2025]\n",
            "Finished jobid: 15 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "59 of 146 steps (40%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:36:45 2025]\n",
            "Finished jobid: 15 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "59 of 146 steps (40%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:45 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.2.default.log\n",
            "    jobid: 14\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:45 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.2.default.log\n",
            "    jobid: 14\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:36:46 2025]\n",
            "Finished jobid: 41 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "60 of 146 steps (41%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:46 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.1.default.log\n",
            "    jobid: 46\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=1, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:46 2025]\n",
            "Finished jobid: 41 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "60 of 146 steps (41%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:46 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.1.default.log\n",
            "    jobid: 46\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=1, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:47 2025]\n",
            "Finished jobid: 68 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "61 of 146 steps (42%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:47 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.1.tutorialConfig.log\n",
            "    jobid: 73\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=1, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:47 2025]\n",
            "Finished jobid: 68 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "61 of 146 steps (42%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:36:47 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.1.tutorialConfig.log\n",
            "    jobid: 73\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=1, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:36:48 2025]\n",
            "Finished jobid: 46 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "62 of 146 steps (42%) done\n",
            "[Tue May 13 10:36:48 2025]\n",
            "Finished jobid: 46 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "62 of 146 steps (42%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 57 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "63 of 146 steps (43%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 73 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "64 of 146 steps (44%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 57 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "63 of 146 steps (43%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 73 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "64 of 146 steps (44%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 14 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "65 of 146 steps (45%) done\n",
            "[Tue May 13 10:36:49 2025]\n",
            "Finished jobid: 14 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "65 of 146 steps (45%) done\n",
            "[Tue May 13 10:37:01 2025]\n",
            "Finished jobid: 83 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "66 of 146 steps (45%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:37:01 2025]\n",
            "Finished jobid: 83 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "66 of 146 steps (45%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:37:02 2025]\n",
            "Finished jobid: 42 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "67 of 146 steps (46%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:02 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.2.default.log\n",
            "    jobid: 47\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:02 2025]\n",
            "Finished jobid: 42 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "67 of 146 steps (46%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:02 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.2.default.log\n",
            "    jobid: 47\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:02 2025]\n",
            "Finished jobid: 69 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "68 of 146 steps (47%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:02 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.2.tutorialConfig.log\n",
            "    jobid: 74\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:02 2025]\n",
            "Finished jobid: 69 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "68 of 146 steps (47%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:02 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.2.tutorialConfig.log\n",
            "    jobid: 74\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:03 2025]\n",
            "Finished jobid: 47 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "69 of 146 steps (47%) done\n",
            "[Tue May 13 10:37:03 2025]\n",
            "Finished jobid: 47 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "69 of 146 steps (47%) done\n",
            "[Tue May 13 10:37:03 2025]\n",
            "Finished jobid: 74 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "70 of 146 steps (48%) done\n",
            "[Tue May 13 10:37:03 2025]\n",
            "Finished jobid: 74 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "70 of 146 steps (48%) done\n",
            "[Tue May 13 10:37:07 2025]\n",
            "Finished jobid: 33 (Rule: experiment_counts_umi_create_BAM)\n",
            "71 of 146 steps (49%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:07 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 32\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:37:07 2025]\n",
            "Finished jobid: 33 (Rule: experiment_counts_umi_create_BAM)\n",
            "71 of 146 steps (49%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:07 2025]\n",
            "localrule experiment_counts_umi_raw_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/umi/raw_counts.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 32\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA.bam\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/22750e19c251ab6076f7519e48544945_\n",
            "[Tue May 13 10:37:10 2025]\n",
            "Finished jobid: 84 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "72 of 146 steps (49%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:37:10 2025]\n",
            "Finished jobid: 84 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "72 of 146 steps (49%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:37:16 2025]\n",
            "Finished jobid: 32 (Rule: experiment_counts_umi_raw_counts)\n",
            "73 of 146 steps (50%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:16 2025]\n",
            "Finished jobid: 32 (Rule: experiment_counts_umi_raw_counts)\n",
            "73 of 146 steps (50%) done\n",
            "Select jobs to execute...\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_3_raw.log\n",
            "    jobid: 97\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_RNA_raw.log\n",
            "    jobid: 93\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 31\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 3 jobs...\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_3_raw.log\n",
            "    jobid: 97\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_RNA_raw.log\n",
            "    jobid: 93\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:16 2025]\n",
            "localrule experiment_counts_filter_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/filter_counts.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 31\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "15\n",
            "15\n",
            "[Tue May 13 10:37:17 2025]\n",
            "Finished jobid: 97 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "74 of 146 steps (51%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:17 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA_merge.experimentMPRAworkshop.raw.log\n",
            "    jobid: 94\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:17 2025]\n",
            "Finished jobid: 97 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "74 of 146 steps (51%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:17 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA_merge.experimentMPRAworkshop.raw.log\n",
            "    jobid: 94\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:18 2025]\n",
            "Finished jobid: 94 (Rule: experiment_statistic_counts_BC_in_RNA_DNA_merge)\n",
            "75 of 146 steps (51%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "[Tue May 13 10:37:18 2025]\n",
            "Finished jobid: 94 (Rule: experiment_statistic_counts_BC_in_RNA_DNA_merge)\n",
            "75 of 146 steps (51%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_raw_BC_in_RNA_DNA.tsv.gz.\n",
            "[Tue May 13 10:37:18 2025]\n",
            "Finished jobid: 93 (Rule: experiment_statistic_counts_table)\n",
            "76 of 146 steps (52%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:18 2025]\n",
            "localrule experiment_statistic_counts_stats_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/stats_merge.experimentMPRAworkshop.raw.log\n",
            "    jobid: 87\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:18 2025]\n",
            "Finished jobid: 93 (Rule: experiment_statistic_counts_table)\n",
            "76 of 146 steps (52%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:18 2025]\n",
            "localrule experiment_statistic_counts_stats_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/stats_merge.experimentMPRAworkshop.raw.log\n",
            "    jobid: 87\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:19 2025]\n",
            "Finished jobid: 87 (Rule: experiment_statistic_counts_stats_merge)\n",
            "77 of 146 steps (53%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz.\n",
            "[Tue May 13 10:37:19 2025]\n",
            "Finished jobid: 87 (Rule: experiment_statistic_counts_stats_merge)\n",
            "77 of 146 steps (53%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:19 2025]\n",
            "localrule experiment_statistic_counts_final:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_count_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/final.experimentMPRAworkshop.raw.log\n",
            "    jobid: 86\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_raw_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_raw_counts.tsv.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:19 2025]\n",
            "localrule experiment_statistic_counts_final:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_count_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv\n",
            "    log: results/logs/experiment/statistic/counts/final.experimentMPRAworkshop.raw.log\n",
            "    jobid: 86\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, countType=raw\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "Attaching package: ‘dplyr’\n",
            "\n",
            "The following objects are masked from ‘package:stats’:\n",
            "\n",
            "    filter, lag\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    intersect, setdiff, setequal, union\n",
            "\n",
            "\n",
            "Attaching package: ‘dplyr’\n",
            "\n",
            "The following objects are masked from ‘package:stats’:\n",
            "\n",
            "    filter, lag\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    intersect, setdiff, setequal, union\n",
            "\n",
            "Rows: 6 Columns: 8\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (2): X1, X3\n",
            "dbl (6): X2, X4, X5, X6, X7, X8\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Rows: 6 Columns: 8\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (2): X1, X3\n",
            "dbl (6): X2, X4, X5, X6, X7, X8\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Rows: 3 Columns: 3\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (1): X1\n",
            "dbl (2): X2, X3\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Rows: 3 Columns: 3\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (1): X1\n",
            "dbl (2): X2, X3\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Joining with `by = join_by(Condition, Replicate)`\n",
            "Joining with `by = join_by(Condition, Replicate)`\n",
            "[Tue May 13 10:37:20 2025]\n",
            "Finished jobid: 86 (Rule: experiment_statistic_counts_final)\n",
            "78 of 146 steps (53%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv.\n",
            "[Tue May 13 10:37:20 2025]\n",
            "Finished jobid: 86 (Rule: experiment_statistic_counts_final)\n",
            "78 of 146 steps (53%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_raw.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_raw.tsv.\n",
            "[Tue May 13 10:37:23 2025]\n",
            "Finished jobid: 31 (Rule: experiment_counts_filter_counts)\n",
            "79 of 146 steps (54%) done\n",
            "Select jobs to execute...\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_RNA_filtered.log\n",
            "    jobid: 105\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_3_filtered.log\n",
            "    jobid: 109\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 115\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:23 2025]\n",
            "Finished jobid: 31 (Rule: experiment_counts_filter_counts)\n",
            "79 of 146 steps (54%) done\n",
            "Select jobs to execute...\n",
            "Execute 4 jobs...\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_table:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/table.experimentMPRAworkshop.HepG2_3_RNA_filtered.log\n",
            "    jobid: 105\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA.experimentMPRAworkshop.HepG2_3_filtered.log\n",
            "    jobid: 109\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_statistic_counts_frequent_umis:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt\n",
            "    log: results/logs/experiment/statistic/counts/frequent_umis.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 115\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 30\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:23 2025]\n",
            "localrule experiment_counts_final_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    log: results/logs/experiment/counts/final_counts_umi.experimentMPRAworkshop.HepG2.3.RNA.log\n",
            "    jobid: 30\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:25 2025]\n",
            "Finished jobid: 109 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "80 of 146 steps (55%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "Finished jobid: 109 (Rule: experiment_statistic_counts_BC_in_RNA_DNA)\n",
            "80 of 146 steps (55%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA_merge.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 106\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_counts_BC_in_RNA_DNA_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/BC_in_RNA_DNA_merge.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 106\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:25 2025]\n",
            "Finished jobid: 30 (Rule: experiment_counts_final_counts)\n",
            "81 of 146 steps (55%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:25 2025]\n",
            "Finished jobid: 30 (Rule: experiment_counts_final_counts)\n",
            "81 of 146 steps (55%) done\n",
            "Select jobs to execute...\n",
            "Execute 7 jobs...\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.3.tutorialConfig.log\n",
            "    jobid: 60\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=3, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_3_RNA.log\n",
            "    jobid: 85\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 7 jobs...\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.3.tutorialConfig.log\n",
            "    jobid: 60\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=3, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_counts_barcode_base_composition:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/nucleotideCountPerPosition.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz\n",
            "    log: results/logs/experiment/statistic/counts/barcode_base_composition.experimentMPRAworkshop.HepG2_3_RNA.log\n",
            "    jobid: 85\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, replicate=3, type=RNA\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.3.default.log\n",
            "    jobid: 25\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=3, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.RNA.default.MPRAworkshop.log\n",
            "    jobid: 43\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.counts.HepG2.3.default.log\n",
            "    jobid: 25\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_DNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, replicate=3, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.RNA.default.MPRAworkshop.log\n",
            "    jobid: 43\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.tutorialConfig.counts.log\n",
            "    jobid: 127\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.default.counts.log\n",
            "    jobid: 124\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 70\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.tutorialConfig.counts.log\n",
            "    jobid: 127\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.default.counts.log\n",
            "    jobid: 124\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:25 2025]\n",
            "localrule experiment_assigned_counts_assignBarcodes:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_BC_and_assignment.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/assignBarcodes.experimentMPRAworkshop.HepG2.3.RNA.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 70\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, replicate=3, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:37:26 2025]\n",
            "Finished jobid: 106 (Rule: experiment_statistic_counts_BC_in_RNA_DNA_merge)\n",
            "82 of 146 steps (56%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "[Tue May 13 10:37:26 2025]\n",
            "Finished jobid: 106 (Rule: experiment_statistic_counts_BC_in_RNA_DNA_merge)\n",
            "82 of 146 steps (56%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_filtered_BC_in_RNA_DNA.tsv.gz.\n",
            "[Tue May 13 10:37:27 2025]\n",
            "Finished jobid: 105 (Rule: experiment_statistic_counts_table)\n",
            "83 of 146 steps (57%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_counts_stats_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/stats_merge.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 99\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:27 2025]\n",
            "Finished jobid: 105 (Rule: experiment_statistic_counts_table)\n",
            "83 of 146 steps (57%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_counts_stats_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/stats_merge.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 99\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:37:27 2025]\n",
            "Finished jobid: 25 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "84 of 146 steps (58%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:27 2025]\n",
            "Finished jobid: 25 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "84 of 146 steps (58%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.default.counts.log\n",
            "    jobid: 117\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.default.counts.log\n",
            "    jobid: 117\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.default.counts.log\n",
            "    jobid: 120\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "\n",
            "[Tue May 13 10:37:27 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.default.counts.log\n",
            "    jobid: 120\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 60 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "85 of 146 steps (58%) done\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 60 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "85 of 146 steps (58%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.tutorialConfig.counts.log\n",
            "    jobid: 119\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.tutorialConfig.counts.log\n",
            "    jobid: 119\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 99 (Rule: experiment_statistic_counts_stats_merge)\n",
            "86 of 146 steps (59%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz.\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 99 (Rule: experiment_statistic_counts_stats_merge)\n",
            "86 of 146 steps (59%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_1_RNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_2_RNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_DNA_filtered_counts.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/HepG2_3_RNA_filtered_counts.tsv.gz.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_statistic_counts_final:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_count_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/final.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 98\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_statistic_counts_final:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_count_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv\n",
            "    log: results/logs/experiment/statistic/counts/final.experimentMPRAworkshop.filtered.log\n",
            "    jobid: 98\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, countType=filtered\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 115 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "87 of 146 steps (60%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:28 2025]\n",
            "Finished jobid: 115 (Rule: experiment_statistic_counts_frequent_umis)\n",
            "87 of 146 steps (60%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.3.default.log\n",
            "    jobid: 24\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:28 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.3.default.log\n",
            "    jobid: 24\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, replicate=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "\n",
            "Attaching package: ‘dplyr’\n",
            "\n",
            "The following objects are masked from ‘package:stats’:\n",
            "\n",
            "    filter, lag\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    intersect, setdiff, setequal, union\n",
            "\n",
            "\n",
            "Attaching package: ‘dplyr’\n",
            "\n",
            "The following objects are masked from ‘package:stats’:\n",
            "\n",
            "    filter, lag\n",
            "\n",
            "The following objects are masked from ‘package:base’:\n",
            "\n",
            "    intersect, setdiff, setequal, union\n",
            "\n",
            "[Tue May 13 10:37:33 2025]\n",
            "Finished jobid: 24 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "88 of 146 steps (60%) done\n",
            "[Tue May 13 10:37:33 2025]\n",
            "Finished jobid: 24 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "88 of 146 steps (60%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:33 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_statistic_tables.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 35\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:33 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_statistic_tables.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 35\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Rows: 6 Columns: 8\n",
            "Rows: 6 Columns: 8\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (2): X1, X3\n",
            "dbl (6): X2, X4, X5, X6, X7, X8\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (2): X1, X3\n",
            "dbl (6): X2, X4, X5, X6, X7, X8\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Rows: 3 Columns: 3\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Rows: 3 Columns: 3\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (1): X1\n",
            "dbl (2): X2, X3\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "── Column specification ────────────────────────────────────────────────────────\n",
            "Delimiter: \"\\t\"\n",
            "chr (1): X1\n",
            "dbl (2): X2, X3\n",
            "\n",
            "ℹ Use `spec()` to retrieve the full column specification for this data.\n",
            "ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.\n",
            "Joining with `by = join_by(Condition, Replicate)`\n",
            "Joining with `by = join_by(Condition, Replicate)`\n",
            "[Tue May 13 10:37:34 2025]\n",
            "Finished jobid: 98 (Rule: experiment_statistic_counts_final)\n",
            "89 of 146 steps (61%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv.\n",
            "[Tue May 13 10:37:34 2025]\n",
            "Finished jobid: 98 (Rule: experiment_statistic_counts_final)\n",
            "89 of 146 steps (61%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/count_filtered.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/counts/BC_in_RNA_DNA_filtered.tsv.\n",
            "Execute 1 jobs...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:34 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates_barcode_output:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_replicates_barcode_counts.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates_barcode_output.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 52\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "\n",
            "[Tue May 13 10:37:34 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates_barcode_output:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_replicates_barcode_counts.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates_barcode_output.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 52\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:35 2025]\n",
            "Finished jobid: 124 (Rule: experiment_statistic_bc_overlap_run)\n",
            "90 of 146 steps (62%) done\n",
            "[Tue May 13 10:37:35 2025]\n",
            "Finished jobid: 124 (Rule: experiment_statistic_bc_overlap_run)\n",
            "90 of 146 steps (62%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:35 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_counts.experimentMPRAworkshop.default.log\n",
            "    jobid: 122\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:35 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_counts.experimentMPRAworkshop.default.log\n",
            "    jobid: 122\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:36 2025]\n",
            "Finished jobid: 122 (Rule: experiment_statistic_bc_overlap_combine_counts)\n",
            "91 of 146 steps (62%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv.\n",
            "[Tue May 13 10:37:36 2025]\n",
            "Finished jobid: 122 (Rule: experiment_statistic_bc_overlap_combine_counts)\n",
            "91 of 146 steps (62%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.default.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.default.tsv.\n",
            "Execute 1 jobs...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:36 2025]\n",
            "localrule experiment_assigned_counts_make_master_tables:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/make_master_tables.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_average_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/make_master_tables.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 51\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "\n",
            "[Tue May 13 10:37:36 2025]\n",
            "localrule experiment_assigned_counts_make_master_tables:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/make_master_tables.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_average_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/make_master_tables.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 51\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:37 2025]\n",
            "Finished jobid: 127 (Rule: experiment_statistic_bc_overlap_run)\n",
            "92 of 146 steps (63%) done\n",
            "[Tue May 13 10:37:37 2025]\n",
            "Finished jobid: 127 (Rule: experiment_statistic_bc_overlap_run)\n",
            "92 of 146 steps (63%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:37 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_counts.experimentMPRAworkshop.tutorialConfig.log\n",
            "    jobid: 125\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:37 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_counts.experimentMPRAworkshop.tutorialConfig.log\n",
            "    jobid: 125\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:37 2025]\n",
            "Finished jobid: 35 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge)\n",
            "93 of 146 steps (64%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz.\n",
            "[Tue May 13 10:37:37 2025]\n",
            "Finished jobid: 35 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge)\n",
            "93 of 146 steps (64%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:38 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge_all.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 34\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:38 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge_all.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 34\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:39 2025]\n",
            "Finished jobid: 125 (Rule: experiment_statistic_bc_overlap_combine_counts)\n",
            "94 of 146 steps (64%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv.\n",
            "[Tue May 13 10:37:39 2025]\n",
            "Finished jobid: 125 (Rule: experiment_statistic_bc_overlap_combine_counts)\n",
            "94 of 146 steps (64%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/counts/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:39 2025]\n",
            "localrule experiment_statistic_correlation_calculate:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/calculate.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 2\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:39 2025]\n",
            "localrule experiment_statistic_correlation_calculate:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/calculate.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 2\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:39 2025]\n",
            "Finished jobid: 34 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all)\n",
            "95 of 146 steps (65%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz.\n",
            "[Tue May 13 10:37:39 2025]\n",
            "Finished jobid: 34 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all)\n",
            "95 of 146 steps (65%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/combined/HepG2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:39 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.3.tutorialConfig.log\n",
            "    jobid: 59\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:39 2025]\n",
            "localrule experiment_assigned_counts_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_label.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcodesRemoved_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/MPRAworkshop/dna_rna_merge.experimentMPRAworkshop.HepG2.3.tutorialConfig.log\n",
            "    jobid: 59\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, replicate=3\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:41 2025]\n",
            "Finished jobid: 51 (Rule: experiment_assigned_counts_make_master_tables)\n",
            "96 of 146 steps (66%) done\n",
            "[Tue May 13 10:37:41 2025]\n",
            "Finished jobid: 51 (Rule: experiment_assigned_counts_make_master_tables)\n",
            "96 of 146 steps (66%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:41 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.default.MPRAworkshop.allreps.log\n",
            "    jobid: 138\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, allreps_or_threshold=allreps\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:41 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.default.MPRAworkshop.allreps.log\n",
            "    jobid: 138\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, allreps_or_threshold=allreps\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:45 2025]\n",
            "Finished jobid: 120 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "97 of 146 steps (66%) done\n",
            "[Tue May 13 10:37:45 2025]\n",
            "Finished jobid: 120 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "97 of 146 steps (66%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:46 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.default.MPRAworkshop.allreps_minThreshold.log\n",
            "    jobid: 140\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, allreps_or_threshold=allreps_minThreshold\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:37:46 2025]\n",
            "Finished jobid: 52 (Rule: experiment_assigned_counts_combine_replicates_barcode_output)\n",
            "98 of 146 steps (67%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:46 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.default.MPRAworkshop.allreps_minThreshold.log\n",
            "    jobid: 140\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2, allreps_or_threshold=allreps_minThreshold\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:37:46 2025]\n",
            "Finished jobid: 52 (Rule: experiment_assigned_counts_combine_replicates_barcode_output)\n",
            "98 of 146 steps (67%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:46 2025]\n",
            "localrule experiment_statistic_correlation_hist_box_plots:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png\n",
            "    log: results/logs/experiment/statistic/correlation/hist_box_plots.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 44\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:46 2025]\n",
            "localrule experiment_statistic_correlation_hist_box_plots:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png\n",
            "    log: results/logs/experiment/statistic/correlation/hist_box_plots.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 44\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:47 2025]\n",
            "Finished jobid: 138 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "99 of 146 steps (68%) done\n",
            "[Tue May 13 10:37:47 2025]\n",
            "Finished jobid: 138 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "99 of 146 steps (68%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:47 2025]\n",
            "localrule experiment_assigned_counts_copy_final_all_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_all_files.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 50\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:47 2025]\n",
            "localrule experiment_assigned_counts_copy_final_all_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_all_files.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 50\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:47 2025]\n",
            "Finished jobid: 59 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "[Tue May 13 10:37:47 2025]\n",
            "Finished jobid: 59 (Rule: experiment_assigned_counts_dna_rna_merge)\n",
            "100 of 146 steps (68%) done\n",
            "100 of 146 steps (68%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:47 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_statistic_tables.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 62\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:47 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_statistic_tables.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 62\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:49 2025]\n",
            "Finished jobid: 50 (Rule: experiment_assigned_counts_copy_final_all_files)\n",
            "101 of 146 steps (69%) done\n",
            "[Tue May 13 10:37:49 2025]\n",
            "Finished jobid: 50 (Rule: experiment_assigned_counts_copy_final_all_files)\n",
            "101 of 146 steps (69%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:49 2025]\n",
            "localrule experiment_statistic_quality_metric:\n",
            "    input: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/quality_metrics.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json\n",
            "    log: results/logs/experiment/statistic_quality_metric.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 49\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/edbb616bb1f2f0b411141153b7cbcd9a_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:49 2025]\n",
            "localrule experiment_statistic_quality_metric:\n",
            "    input: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/quality_metrics.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json\n",
            "    log: results/logs/experiment/statistic_quality_metric.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 49\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json; Input files updated by another job: results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/edbb616bb1f2f0b411141153b7cbcd9a_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:50 2025]\n",
            "Finished jobid: 140 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "102 of 146 steps (70%) done\n",
            "[Tue May 13 10:37:50 2025]\n",
            "Finished jobid: 140 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "102 of 146 steps (70%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:50 2025]\n",
            "localrule experiment_statistic_correlation_hist_box_plots:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png\n",
            "    log: results/logs/experiment/statistic/correlation/hist_box_plots.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 71\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:50 2025]\n",
            "localrule experiment_statistic_correlation_hist_box_plots:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png\n",
            "    log: results/logs/experiment/statistic/correlation/hist_box_plots.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 71\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:51 2025]\n",
            "Finished jobid: 62 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge)\n",
            "103 of 146 steps (71%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz.\n",
            "[Tue May 13 10:37:51 2025]\n",
            "Finished jobid: 62 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge)\n",
            "103 of 146 steps (71%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:51 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge_all.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 61\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:51 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_stats_dna_rna_merge_all.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 61\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:37:53 2025]\n",
            "Finished jobid: 61 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all)\n",
            "104 of 146 steps (71%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "[Tue May 13 10:37:53 2025]\n",
            "Finished jobid: 61 (Rule: experiment_statistic_assigned_counts_combine_stats_dna_rna_merge_all)\n",
            "104 of 146 steps (71%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/combined/HepG2_merged_assigned_counts.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:37:54 2025]\n",
            "localrule experiment_assigned_counts_make_master_tables:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/make_master_tables.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_average_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/make_master_tables.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 78\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "\n",
            "[Tue May 13 10:37:54 2025]\n",
            "localrule experiment_assigned_counts_make_master_tables:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/make_master_tables.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_average_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/make_master_tables.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 78\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:00 2025]\n",
            "Finished jobid: 49 (Rule: experiment_statistic_quality_metric)\n",
            "105 of 146 steps (72%) done\n",
            "[Tue May 13 10:38:00 2025]\n",
            "Finished jobid: 49 (Rule: experiment_statistic_quality_metric)\n",
            "105 of 146 steps (72%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:00 2025]\n",
            "localrule experiment_statistic_correlation_calculate:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/calculate.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 54\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:00 2025]\n",
            "localrule experiment_statistic_correlation_calculate:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, example_data/design/workshop_labels.tsv, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perInsertCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/calculate.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 54\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_merged_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_merged_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:01 2025]\n",
            "Finished jobid: 78 (Rule: experiment_assigned_counts_make_master_tables)[Tue May 13 10:38:01 2025]\n",
            "Finished jobid: 78 (Rule: experiment_assigned_counts_make_master_tables)\n",
            "106 of 146 steps (73%) done\n",
            "\n",
            "106 of 146 steps (73%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:02 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.allreps_minThreshold.log\n",
            "    jobid: 141\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, allreps_or_threshold=allreps_minThreshold\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:02 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.allreps_minThreshold.log\n",
            "    jobid: 141\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, allreps_or_threshold=allreps_minThreshold\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:06 2025]\n",
            "Finished jobid: 141 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "107 of 146 steps (73%) done\n",
            "[Tue May 13 10:38:06 2025]\n",
            "Finished jobid: 141 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "107 of 146 steps (73%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:06 2025]\n",
            "localrule experiment_assigned_counts_copy_final_thresh_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_thresh_files.experimentMPRAworkshop.HepG2.MPRAworkshop.default.10.log\n",
            "    jobid: 142\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default, threshold=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:06 2025]\n",
            "localrule experiment_assigned_counts_copy_final_thresh_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_thresh_files.experimentMPRAworkshop.HepG2.MPRAworkshop.default.10.log\n",
            "    jobid: 142\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default, threshold=10\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:08 2025]\n",
            "Finished jobid: 142 (Rule: experiment_assigned_counts_copy_final_thresh_files)\n",
            "108 of 146 steps (74%) done\n",
            "[Tue May 13 10:38:08 2025]\n",
            "Finished jobid: 142 (Rule: experiment_assigned_counts_copy_final_thresh_files)\n",
            "108 of 146 steps (74%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:08 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.allreps.log\n",
            "    jobid: 139\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, allreps_or_threshold=allreps\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:08 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/combine_replicates.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.allreps.log\n",
            "    jobid: 139\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2, allreps_or_threshold=allreps\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:08 2025]\n",
            "Finished jobid: 70 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "109 of 146 steps (75%) done\n",
            "[Tue May 13 10:38:08 2025]\n",
            "Finished jobid: 70 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "109 of 146 steps (75%) done\n",
            "Execute 1 jobs...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:08 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats_helper.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 64\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "\n",
            "[Tue May 13 10:38:08 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats_helper.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 64\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:09 2025]\n",
            "Finished jobid: 64 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper)\n",
            "110 of 146 steps (75%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "[Tue May 13 10:38:09 2025]\n",
            "Finished jobid: 64 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper)\n",
            "110 of 146 steps (75%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_tutorialConfig.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:09 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats.experimentMPRAworkshop.MPRAworkshop_tutorialConfig.log\n",
            "    jobid: 63\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:09 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats.experimentMPRAworkshop.MPRAworkshop_tutorialConfig.log\n",
            "    jobid: 63\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:11 2025]\n",
            "Finished jobid: 63 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats)\n",
            "111 of 146 steps (76%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz.\n",
            "[Tue May 13 10:38:11 2025]\n",
            "Finished jobid: 63 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats)\n",
            "111 of 146 steps (76%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.tutorialConfig.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:11 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.tutorialConfig.counts.log\n",
            "    jobid: 121\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:38:11 2025]\n",
            "Finished jobid: 43 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "112 of 146 steps (77%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:11 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.tutorialConfig.counts.log\n",
            "    jobid: 121\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/counts/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/counts/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=counts, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:38:11 2025]\n",
            "Finished jobid: 43 (Rule: experiment_assigned_counts_assignBarcodes)\n",
            "112 of 146 steps (77%) done\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:11 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats_helper.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 37\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:11 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats_helper.experimentMPRAworkshop.HepG2.default.MPRAworkshop.log\n",
            "    jobid: 37\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:12 2025]\n",
            "Finished jobid: 139 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "113 of 146 steps (77%) done\n",
            "[Tue May 13 10:38:12 2025]\n",
            "Finished jobid: 139 (Rule: experiment_assigned_counts_combine_replicates)\n",
            "113 of 146 steps (77%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:12 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.3.default.log\n",
            "    jobid: 48\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=3, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:38:12 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.3.default.log\n",
            "    jobid: 48\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=3, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:13 2025]\n",
            "Finished jobid: 44 (Rule: experiment_statistic_correlation_hist_box_plots)\n",
            "114 of 146 steps (78%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:13 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 134\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:38:13 2025]\n",
            "Finished jobid: 44 (Rule: experiment_statistic_correlation_hist_box_plots)\n",
            "114 of 146 steps (78%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:13 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 134\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=RNA, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:13 2025]\n",
            "Finished jobid: 37 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper)\n",
            "115 of 146 steps (79%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz.\n",
            "[Tue May 13 10:38:13 2025]\n",
            "Finished jobid: 37 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats_helper)\n",
            "115 of 146 steps (79%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_DNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_1_RNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_2_RNA_default.statistic.tsv.gz.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/HepG2_3_RNA_default.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:13 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats.experimentMPRAworkshop.MPRAworkshop_default.log\n",
            "    jobid: 36\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmpExecute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:13 2025]\n",
            "localrule experiment_statistic_assigned_counts_combine_BC_assignment_stats:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/assigned_counts/combine_BC_assignment_stats.experimentMPRAworkshop.MPRAworkshop_default.log\n",
            "    jobid: 36\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:14 2025]\n",
            "Finished jobid: 48 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "116 of 146 steps (79%) done\n",
            "[Tue May 13 10:38:14 2025]\n",
            "Finished jobid: 48 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "116 of 146 steps (79%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:14 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 129\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:14 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 129\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "[Tue May 13 10:38:14 2025]\n",
            "Finished jobid: 71 (Rule: experiment_statistic_correlation_hist_box_plots)\n",
            "117 of 146 steps (80%) done\n",
            "[Tue May 13 10:38:14 2025]\n",
            "Finished jobid: 71 (Rule: experiment_statistic_correlation_hist_box_plots)\n",
            "117 of 146 steps (80%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:15 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 45\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:15 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.default.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 45\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.default.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.default.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:15 2025]\n",
            "Finished jobid: 36 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats)\n",
            "118 of 146 steps (81%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz.\n",
            "[Tue May 13 10:38:15 2025]\n",
            "Finished jobid: 36 (Rule: experiment_statistic_assigned_counts_combine_BC_assignment_stats)\n",
            "118 of 146 steps (81%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/helper.HepG2.default.statistic.tsv.gz.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:15 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.3.tutorialConfig.log\n",
            "    jobid: 75\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=3, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:15 2025]\n",
            "localrule experiment_counts_dna_rna_merge_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    log: results/logs/experiment/counts/dna_rna_merge_counts.experimentMPRAworkshop.assigned_counts/MPRAworkshop.HepG2.3.tutorialConfig.log\n",
            "    jobid: 75\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, replicate=3, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:17 2025]\n",
            "Finished jobid: 75 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "119 of 146 steps (82%) done\n",
            "[Tue May 13 10:38:17 2025]\n",
            "Finished jobid: 75 (Rule: experiment_counts_dna_rna_merge_counts)\n",
            "119 of 146 steps (82%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:17 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 72\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:17 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts_hist:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_stats.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts_hist.experimentMPRAworkshop.HepG2.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 72\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:21 2025]\n",
            "Finished jobid: 134 (Rule: experiment_statistic_bc_overlap_run)\n",
            "[Tue May 13 10:38:21 2025]\n",
            "Finished jobid: 134 (Rule: experiment_statistic_bc_overlap_run)\n",
            "120 of 146 steps (82%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:21 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_assigned_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_assigned_counts.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 132\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "120 of 146 steps (82%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:21 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_assigned_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_assigned_counts.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 132\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:22 2025]\n",
            "Finished jobid: 132 (Rule: experiment_statistic_bc_overlap_combine_assigned_counts)\n",
            "121 of 146 steps (83%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv.\n",
            "[Tue May 13 10:38:22 2025]\n",
            "Finished jobid: 132 (Rule: experiment_statistic_bc_overlap_combine_assigned_counts)\n",
            "121 of 146 steps (83%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.default.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.default.tsv.\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:22 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 131\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:22 2025]\n",
            "localrule experiment_statistic_correlation_bc_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/plot_perBCCounts_correlation.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/correlate_bc_counts.experimentMPRAworkshop.HepG2.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 131\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:29 2025]\n",
            "Finished jobid: 45 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "122 of 146 steps (84%) done\n",
            "[Tue May 13 10:38:29 2025]\n",
            "Finished jobid: 45 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "122 of 146 steps (84%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:29 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 137\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:29 2025]\n",
            "localrule experiment_statistic_bc_overlap_run:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/BCCounts_betweenReplicates.R (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/run.experimentMPRAworkshop.HepG2.RNA.tutorialConfig.assigned_counts/MPRAworkshop.log\n",
            "    jobid: 137\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, raw_or_assigned=assigned_counts/MPRAworkshop, condition=HepG2, type=RNA, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "Activating conda environment: ../conda_envs/da1f95a3d9f5dc8bae0b5a92efc25676_\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:30 2025]\n",
            "Finished jobid: 72 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "123 of 146 steps (84%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:30 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates_barcode_output:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_replicates_barcode_counts.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates_barcode_output.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 79\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:38:30 2025]\n",
            "Finished jobid: 121 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "124 of 146 steps (85%) done\n",
            "[Tue May 13 10:38:30 2025]\n",
            "Finished jobid: 72 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "123 of 146 steps (84%) done\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:30 2025]\n",
            "localrule experiment_assigned_counts_combine_replicates_barcode_output:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/count/merge_replicates_barcode_counts.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/combine_replicates_barcode_output.experimentMPRAworkshop.HepG2.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 79\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_3_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_1_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_2_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig, condition=HepG2\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/184c66443732f7e76af6696323c485e3_\n",
            "[Tue May 13 10:38:30 2025]\n",
            "Finished jobid: 121 (Rule: experiment_statistic_correlation_bc_counts_hist)\n",
            "124 of 146 steps (85%) done\n",
            "[Tue May 13 10:38:35 2025]\n",
            "Finished jobid: 137 (Rule: experiment_statistic_bc_overlap_run)\n",
            "125 of 146 steps (86%) done\n",
            "[Tue May 13 10:38:35 2025]\n",
            "Finished jobid: 137 (Rule: experiment_statistic_bc_overlap_run)\n",
            "125 of 146 steps (86%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:35 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_assigned_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_assigned_counts.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 135\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:35 2025]\n",
            "localrule experiment_statistic_bc_overlap_combine_assigned_counts:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv\n",
            "    log: results/logs/experiment/statistic/bc_overlap/combine_assigned_counts.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 135\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig, assignment=MPRAworkshop\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:38:37 2025]\n",
            "Finished jobid: 85 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "126 of 146 steps (86%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:38:37 2025]\n",
            "Finished jobid: 85 (Rule: experiment_statistic_counts_barcode_base_composition)\n",
            "126 of 146 steps (86%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/counts/HepG2_3_RNA_final.BC.tsv.gz.\n",
            "[Tue May 13 10:38:37 2025]\n",
            "Finished jobid: 135 (Rule: experiment_statistic_bc_overlap_combine_assigned_counts)\n",
            "127 of 146 steps (87%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv.\n",
            "[Tue May 13 10:38:37 2025]\n",
            "Finished jobid: 135 (Rule: experiment_statistic_bc_overlap_combine_assigned_counts)\n",
            "127 of 146 steps (87%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_DNA.tutorialConfig.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/bc_overlap/assigned_counts/MPRAworkshop/overlapBCandCounts.HepG2_RNA.tutorialConfig.tsv.\n",
            "[Tue May 13 10:38:39 2025]\n",
            "Finished jobid: 79 (Rule: experiment_assigned_counts_combine_replicates_barcode_output)\n",
            "128 of 146 steps (88%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:39 2025]\n",
            "Finished jobid: 79 (Rule: experiment_assigned_counts_combine_replicates_barcode_output)\n",
            "128 of 146 steps (88%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:38:39 2025]\n",
            "localrule experiment_assigned_counts_copy_final_thresh_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_thresh_files.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.5.log\n",
            "    jobid: 143\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig, threshold=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:38:39 2025]\n",
            "localrule experiment_assigned_counts_copy_final_thresh_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_thresh_files.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.5.log\n",
            "    jobid: 143\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig, threshold=5\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:38:39 2025]\n",
            "localrule experiment_assigned_counts_copy_final_all_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_all_files.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 77\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:38:39 2025]\n",
            "localrule experiment_assigned_counts_copy_final_all_files:\n",
            "    input: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    output: results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz\n",
            "    log: results/logs/experiment/assigned_counts/copy_final_all_files.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 77\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz; Input files updated by another job: results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:38:40 2025]\n",
            "Finished jobid: 143 (Rule: experiment_assigned_counts_copy_final_thresh_files)\n",
            "129 of 146 steps (88%) done\n",
            "[Tue May 13 10:38:40 2025]\n",
            "Finished jobid: 77 (Rule: experiment_assigned_counts_copy_final_all_files)\n",
            "130 of 146 steps (89%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:40 2025]\n",
            "Finished jobid: 143 (Rule: experiment_assigned_counts_copy_final_thresh_files)\n",
            "129 of 146 steps (88%) done\n",
            "[Tue May 13 10:38:40 2025]\n",
            "Finished jobid: 77 (Rule: experiment_assigned_counts_copy_final_all_files)\n",
            "130 of 146 steps (89%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:40 2025]\n",
            "localrule experiment_statistic_quality_metric:\n",
            "    input: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/quality_metrics.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json\n",
            "    log: results/logs/experiment/statistic_quality_metric.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 76\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json; Input files updated by another job: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/edbb616bb1f2f0b411141153b7cbcd9a_\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:38:40 2025]\n",
            "localrule experiment_statistic_quality_metric:\n",
            "    input: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz, /data/MPRAsnakeflow/workflow/rules/../scripts/quality_metrics.py (cached)\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json\n",
            "    log: results/logs/experiment/statistic_quality_metric.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 76\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json; Input files updated by another job: results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/assignment/MPRAworkshop.tsv.gz\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/edbb616bb1f2f0b411141153b7cbcd9a_\n",
            "[Tue May 13 10:38:45 2025]\n",
            "Finished jobid: 76 (Rule: experiment_statistic_quality_metric)\n",
            "131 of 146 steps (90%) done\n",
            "[Tue May 13 10:38:45 2025]\n",
            "Finished jobid: 76 (Rule: experiment_statistic_quality_metric)\n",
            "131 of 146 steps (90%) done\n",
            "[Tue May 13 10:38:56 2025]\n",
            "Finished jobid: 2 (Rule: experiment_statistic_correlation_calculate)\n",
            "132 of 146 steps (90%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:38:56 2025]\n",
            "Finished jobid: 2 (Rule: experiment_statistic_correlation_calculate)\n",
            "132 of 146 steps (90%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:38:56 2025]\n",
            "localrule experiment_statistic_correlation_combine_oligo:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_oligo.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 144\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:38:56 2025]\n",
            "localrule qc_report_count:\n",
            "    input: /data/MPRAsnakeflow/workflow/rules/../scripts/report/qc_report_count.qmd (cached), results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd\n",
            "    log: results/logs/qc_report/count.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 1\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/47964f8f064a674e28931fac7dfc1fb3_\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:38:56 2025]\n",
            "localrule experiment_statistic_correlation_combine_oligo:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_oligo.experimentMPRAworkshop.default.MPRAworkshop.log\n",
            "    jobid: 144\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "\n",
            "[Tue May 13 10:38:56 2025]\n",
            "localrule qc_report_count:\n",
            "    input: /data/MPRAsnakeflow/workflow/rules/../scripts/report/qc_report_count.qmd (cached), results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd\n",
            "    log: results/logs/qc_report/count.experimentMPRAworkshop.HepG2.MPRAworkshop.default.log\n",
            "    jobid: 1\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/47964f8f064a674e28931fac7dfc1fb3_\n",
            "[Tue May 13 10:38:57 2025]\n",
            "Finished jobid: 144 (Rule: experiment_statistic_correlation_combine_oligo)\n",
            "133 of 146 steps (91%) done\n",
            "[Tue May 13 10:38:57 2025]\n",
            "Finished jobid: 144 (Rule: experiment_statistic_correlation_combine_oligo)\n",
            "133 of 146 steps (91%) done\n",
            "[Tue May 13 10:39:12 2025]\n",
            "Finished jobid: 54 (Rule: experiment_statistic_correlation_calculate)\n",
            "134 of 146 steps (92%) done\n",
            "Select jobs to execute...\n",
            "[Tue May 13 10:39:12 2025]\n",
            "Finished jobid: 54 (Rule: experiment_statistic_correlation_calculate)\n",
            "134 of 146 steps (92%) done\n",
            "Select jobs to execute...\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:39:12 2025]\n",
            "localrule qc_report_count:\n",
            "    input: /data/MPRAsnakeflow/workflow/rules/../scripts/report/qc_report_count.qmd (cached), results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd\n",
            "    log: results/logs/qc_report/count.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 53\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/47964f8f064a674e28931fac7dfc1fb3_\n",
            "\n",
            "[Tue May 13 10:39:12 2025]\n",
            "localrule experiment_statistic_correlation_combine_oligo:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_oligo.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 145\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "Execute 2 jobs...\n",
            "\n",
            "[Tue May 13 10:39:12 2025]\n",
            "localrule qc_report_count:\n",
            "    input: /data/MPRAsnakeflow/workflow/rules/../scripts/report/qc_report_count.qmd (cached), results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json\n",
            "    output: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd\n",
            "    log: results/logs/qc_report/count.experimentMPRAworkshop.HepG2.MPRAworkshop.tutorialConfig.log\n",
            "    jobid: 53\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_barcodesPerInsert.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_dna_vs_rna.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_Ratio_pairwise_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, condition=HepG2, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/47964f8f064a674e28931fac7dfc1fb3_\n",
            "\n",
            "[Tue May 13 10:39:12 2025]\n",
            "localrule experiment_statistic_correlation_combine_oligo:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_oligo.experimentMPRAworkshop.tutorialConfig.MPRAworkshop.log\n",
            "    jobid: 145\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:13 2025]\n",
            "Finished jobid: 145 (Rule: experiment_statistic_correlation_combine_oligo)\n",
            "135 of 146 steps (92%) done\n",
            "[Tue May 13 10:39:13 2025]\n",
            "Finished jobid: 145 (Rule: experiment_statistic_correlation_combine_oligo)\n",
            "135 of 146 steps (92%) done\n",
            "[Tue May 13 10:39:15 2025]\n",
            "Finished jobid: 129 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "136 of 146 steps (93%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:15 2025]\n",
            "Finished jobid: 129 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "136 of 146 steps (93%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:15 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_assigned:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_assigned.experimentMPRAworkshop.MPRAworkshop_default.log\n",
            "    jobid: 128\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:15 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_assigned:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_assigned.experimentMPRAworkshop.MPRAworkshop_default.log\n",
            "    jobid: 128\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:16 2025]\n",
            "Finished jobid: 128 (Rule: experiment_statistic_correlation_combine_bc_assigned)\n",
            "137 of 146 steps (94%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:16 2025]\n",
            "Finished jobid: 128 (Rule: experiment_statistic_correlation_combine_bc_assigned)\n",
            "137 of 146 steps (94%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:21 2025]\n",
            "Finished jobid: 131 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "138 of 146 steps (95%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:21 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_assigned:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_assigned.experimentMPRAworkshop.MPRAworkshop_tutorialConfig.log\n",
            "    jobid: 130\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:21 2025]\n",
            "Finished jobid: 131 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "138 of 146 steps (95%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:21 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_assigned:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_assigned.experimentMPRAworkshop.MPRAworkshop_tutorialConfig.log\n",
            "    jobid: 130\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, assignment=MPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:22 2025]\n",
            "Finished jobid: 130 (Rule: experiment_statistic_correlation_combine_bc_assigned)\n",
            "139 of 146 steps (95%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:22 2025]\n",
            "Finished jobid: 130 (Rule: experiment_statistic_correlation_combine_bc_assigned)\n",
            "139 of 146 steps (95%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:29 2025]\n",
            "Finished jobid: 1 (Rule: qc_report_count)\n",
            "140 of 146 steps (96%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd.\n",
            "[Tue May 13 10:39:29 2025]\n",
            "Finished jobid: 1 (Rule: qc_report_count)\n",
            "140 of 146 steps (96%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_correlation_minThreshold.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.qmd.\n",
            "[Tue May 13 10:39:36 2025]\n",
            "Finished jobid: 53 (Rule: qc_report_count)\n",
            "141 of 146 steps (97%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd.\n",
            "[Tue May 13 10:39:36 2025]\n",
            "Finished jobid: 53 (Rule: qc_report_count)\n",
            "141 of 146 steps (97%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_correlation_minThreshold.tsv.\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.qmd.\n",
            "[Tue May 13 10:39:50 2025]\n",
            "Finished jobid: 117 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "142 of 146 steps (97%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:50 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_raw:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_raw.experimentMPRAworkshop.default.log\n",
            "    jobid: 116\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:50 2025]\n",
            "Finished jobid: 117 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "142 of 146 steps (97%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:50 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_raw:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_raw.experimentMPRAworkshop.default.log\n",
            "    jobid: 116\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=default\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:50 2025]\n",
            "Finished jobid: 116 (Rule: experiment_statistic_correlation_combine_bc_raw)\n",
            "143 of 146 steps (98%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:50 2025]\n",
            "Finished jobid: 116 (Rule: experiment_statistic_correlation_combine_bc_raw)\n",
            "143 of 146 steps (98%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_barcode_correlation.tsv.\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 119 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "144 of 146 steps (99%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_raw:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_raw.experimentMPRAworkshop.tutorialConfig.log\n",
            "    jobid: 118\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 119 (Rule: experiment_statistic_correlation_bc_counts)\n",
            "144 of 146 steps (99%) done\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "localrule experiment_statistic_correlation_combine_bc_raw:\n",
            "    input: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    output: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv\n",
            "    log: results/logs/experiment/statistic/correlation/combine_bc_raw.experimentMPRAworkshop.tutorialConfig.log\n",
            "    jobid: 118\n",
            "    reason: Missing output files: results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv; Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv\n",
            "    wildcards: project=experimentMPRAworkshop, config=tutorialConfig\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "Activating conda environment: ../conda_envs/32041765d78c16e5382d1a3cd6df4b7b_\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 118 (Rule: experiment_statistic_correlation_combine_bc_raw)\n",
            "145 of 146 steps (99%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "localrule all:\n",
            "    input: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv\n",
            "    jobid: 0\n",
            "    reason: Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 0 (Rule: all)\n",
            "146 of 146 steps (100%) done\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 118 (Rule: experiment_statistic_correlation_combine_bc_raw)\n",
            "145 of 146 steps (99%) done\n",
            "Removing temporary output results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_barcode_correlation.tsv.\n",
            "Select jobs to execute...\n",
            "Execute 1 jobs...\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "localrule all:\n",
            "    input: results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv\n",
            "    jobid: 0\n",
            "    reason: Input files updated by another job: results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.default.json, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.default.tsv, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_2_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.counts.tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.raw.tsv, results/experiments/experimentMPRAworkshop/statistic/counts.filtered.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_3_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_DNA.txt, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box_minThreshold.png, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_1_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_2_RNA.txt, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged.combined.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_RNA.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_merged.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/statistic_bc_correlation_merged_default.tsv, results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_group_barcodesPerInsert_box.png, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_merged_MPRAworkshop_default.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_RNA_perBarcode.png, results/experiments/experimentMPRAworkshop/statistic/bc_overlap.assigned_counts.default.MPRAworkshop.tsv, results/experiments/experimentMPRAworkshop/qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json, results/experiments/experimentMPRAworkshop/statistic/counts.freqUMIs.HepG2_3_RNA.txt, results/experiments/experimentMPRAworkshop/statistic/counts/BCNucleotideComposition.HepG2_1_DNA.tsv.gz, results/experiments/experimentMPRAworkshop/reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_default_DNA_perBarcode.png, results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default/HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz, results/experiments/experimentMPRAworkshop/statistic/barcode/counts/HepG2_tutorialConfig_DNA_perBarcode.png\n",
            "    resources: tmpdir=/tmp\n",
            "\n",
            "[Tue May 13 10:39:52 2025]\n",
            "Finished jobid: 0 (Rule: all)\n",
            "146 of 146 steps (100%) done\n",
            "Complete log(s): /data/run/.snakemake/log/2025-05-13T102436.012200.snakemake.log\n",
            "\u001b[95mRunning MPRAsnakeflow version 0.5.3\u001b[0m\n",
            "Complete log(s): /data/run/.snakemake/log/2025-05-13T102436.012200.snakemake.log\n",
            "\u001b[95mRunning MPRAsnakeflow version 0.5.3\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "runContainer(\"snakemake --configfile config_experiment.yaml -c 10\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Experiment results\n",
        "\n",
        "Let's take a look at the files in the experiment results folder!"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "MPRAsnakeflow_tutorial/results/experiments:\n",
            "experimentMPRAworkshop\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop:\n",
            "assigned_counts\n",
            "assignment\n",
            "counts\n",
            "qc_metrics.HepG2.MPRAworkshop.default.json\n",
            "qc_metrics.HepG2.MPRAworkshop.tutorialConfig.json\n",
            "qc_report.HepG2.MPRAworkshop.default.html\n",
            "qc_report.HepG2.MPRAworkshop.tutorialConfig.html\n",
            "reporter_experiment.barcode.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "reporter_experiment.barcode.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz\n",
            "reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz\n",
            "reporter_experiment.barcode.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz\n",
            "reporter_experiment.oligo.HepG2.MPRAworkshop.default.all.tsv.gz\n",
            "reporter_experiment.oligo.HepG2.MPRAworkshop.default.min_oligo_threshold_10.tsv.gz\n",
            "reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.all.tsv.gz\n",
            "reporter_experiment.oligo.HepG2.MPRAworkshop.tutorialConfig.min_oligo_threshold_5.tsv.gz\n",
            "statistic\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/assigned_counts:\n",
            "MPRAworkshop\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop:\n",
            "HepG2_1.merged.config.default.tsv.gz\n",
            "HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_1_DNA_final_counts.config.default.tsv.gz\n",
            "HepG2_1_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "HepG2_1_RNA_final_counts.config.default.tsv.gz\n",
            "HepG2_1_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "HepG2_2.merged.config.default.tsv.gz\n",
            "HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_2_DNA_final_counts.config.default.tsv.gz\n",
            "HepG2_2_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "HepG2_2_RNA_final_counts.config.default.tsv.gz\n",
            "HepG2_2_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "HepG2_3.merged.config.default.tsv.gz\n",
            "HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_3_DNA_final_counts.config.default.tsv.gz\n",
            "HepG2_3_DNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "HepG2_3_RNA_final_counts.config.default.tsv.gz\n",
            "HepG2_3_RNA_final_counts.config.tutorialConfig.tsv.gz\n",
            "default\n",
            "tutorialConfig\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/default:\n",
            "HepG2_1_barcode_assigned_counts.tsv.gz\n",
            "HepG2_1_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_1_merged_assigned_counts.tsv.gz\n",
            "HepG2_2_barcode_assigned_counts.tsv.gz\n",
            "HepG2_2_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_2_merged_assigned_counts.tsv.gz\n",
            "HepG2_3_barcode_assigned_counts.tsv.gz\n",
            "HepG2_3_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_3_merged_assigned_counts.tsv.gz\n",
            "HepG2_allreps_merged.combined.tsv.gz\n",
            "HepG2_allreps_merged.tsv.gz\n",
            "HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig:\n",
            "HepG2_1_barcode_assigned_counts.tsv.gz\n",
            "HepG2_1_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_1_merged_assigned_counts.tsv.gz\n",
            "HepG2_2_barcode_assigned_counts.tsv.gz\n",
            "HepG2_2_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_2_merged_assigned_counts.tsv.gz\n",
            "HepG2_3_barcode_assigned_counts.tsv.gz\n",
            "HepG2_3_barcodesRemoved_assigned_counts.tsv.gz\n",
            "HepG2_3_merged_assigned_counts.tsv.gz\n",
            "HepG2_allreps_merged.combined.tsv.gz\n",
            "HepG2_allreps_merged.tsv.gz\n",
            "HepG2_allreps_merged_barcode_assigned_counts.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged.combined.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged.tsv.gz\n",
            "HepG2_allreps_minThreshold_merged_barcode_assigned_counts.tsv.gz\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/assignment:\n",
            "MPRAworkshop.tsv.gz\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/counts:\n",
            "HepG2_1.merged.config.default.tsv.gz\n",
            "HepG2_1.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_1_DNA_filtered_counts.tsv.gz\n",
            "HepG2_1_DNA_final_counts.tsv.gz\n",
            "HepG2_1_RNA_filtered_counts.tsv.gz\n",
            "HepG2_1_RNA_final_counts.tsv.gz\n",
            "HepG2_2.merged.config.default.tsv.gz\n",
            "HepG2_2.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_2_DNA_filtered_counts.tsv.gz\n",
            "HepG2_2_DNA_final_counts.tsv.gz\n",
            "HepG2_2_RNA_filtered_counts.tsv.gz\n",
            "HepG2_2_RNA_final_counts.tsv.gz\n",
            "HepG2_3.merged.config.default.tsv.gz\n",
            "HepG2_3.merged.config.tutorialConfig.tsv.gz\n",
            "HepG2_3_DNA_filtered_counts.tsv.gz\n",
            "HepG2_3_DNA_final_counts.tsv.gz\n",
            "HepG2_3_RNA_filtered_counts.tsv.gz\n",
            "HepG2_3_RNA_final_counts.tsv.gz\n",
            "useUMI.HepG2_1_DNA.bam\n",
            "useUMI.HepG2_1_DNA_raw_counts.tsv.gz\n",
            "useUMI.HepG2_1_RNA.bam\n",
            "useUMI.HepG2_1_RNA_raw_counts.tsv.gz\n",
            "useUMI.HepG2_2_DNA.bam\n",
            "useUMI.HepG2_2_DNA_raw_counts.tsv.gz\n",
            "useUMI.HepG2_2_RNA.bam\n",
            "useUMI.HepG2_2_RNA_raw_counts.tsv.gz\n",
            "useUMI.HepG2_3_DNA.bam\n",
            "useUMI.HepG2_3_DNA_raw_counts.tsv.gz\n",
            "useUMI.HepG2_3_RNA.bam\n",
            "useUMI.HepG2_3_RNA_raw_counts.tsv.gz\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic:\n",
            "assigned_counts\n",
            "barcode\n",
            "bc_overlap.assigned_counts.default.MPRAworkshop.tsv\n",
            "bc_overlap.assigned_counts.tutorialConfig.MPRAworkshop.tsv\n",
            "bc_overlap.counts.default.tsv\n",
            "bc_overlap.counts.tutorialConfig.tsv\n",
            "counts\n",
            "counts.filtered.tsv\n",
            "counts.freqUMIs.HepG2_1_DNA.txt\n",
            "counts.freqUMIs.HepG2_1_RNA.txt\n",
            "counts.freqUMIs.HepG2_2_DNA.txt\n",
            "counts.freqUMIs.HepG2_2_RNA.txt\n",
            "counts.freqUMIs.HepG2_3_DNA.txt\n",
            "counts.freqUMIs.HepG2_3_RNA.txt\n",
            "counts.raw.tsv\n",
            "statistic_assigned_bc_correlation_merged_MPRAworkshop_default.tsv\n",
            "statistic_assigned_bc_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "statistic_assigned_counts_merged_MPRAworkshop_default.tsv\n",
            "statistic_assigned_counts_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "statistic_assigned_counts_single_MPRAworkshop_default.tsv\n",
            "statistic_assigned_counts_single_MPRAworkshop_tutorialConfig.tsv\n",
            "statistic_bc_correlation_merged_default.tsv\n",
            "statistic_bc_correlation_merged_tutorialConfig.tsv\n",
            "statistic_oligo_correlation_merged_MPRAworkshop_default.tsv\n",
            "statistic_oligo_correlation_merged_MPRAworkshop_tutorialConfig.tsv\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/assigned_counts:\n",
            "MPRAworkshop\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop:\n",
            "default  tutorialConfig\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default:\n",
            "HepG2_DNA_pairwise.png\n",
            "HepG2_DNA_pairwise_minThreshold.png\n",
            "HepG2_RNA_pairwise.png\n",
            "HepG2_RNA_pairwise_minThreshold.png\n",
            "HepG2_Ratio_pairwise.png\n",
            "HepG2_Ratio_pairwise_minThreshold.png\n",
            "HepG2_average_allreps_merged.tsv.gz\n",
            "HepG2_barcodesPerInsert.png\n",
            "HepG2_dna_vs_rna.png\n",
            "HepG2_dna_vs_rna_minThreshold.png\n",
            "HepG2_group_barcodesPerInsert_box.png\n",
            "HepG2_group_barcodesPerInsert_box_minThreshold.png\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig:\n",
            "HepG2_DNA_pairwise.png\n",
            "HepG2_DNA_pairwise_minThreshold.png\n",
            "HepG2_RNA_pairwise.png\n",
            "HepG2_RNA_pairwise_minThreshold.png\n",
            "HepG2_Ratio_pairwise.png\n",
            "HepG2_Ratio_pairwise_minThreshold.png\n",
            "HepG2_average_allreps_merged.tsv.gz\n",
            "HepG2_barcodesPerInsert.png\n",
            "HepG2_dna_vs_rna.png\n",
            "HepG2_dna_vs_rna_minThreshold.png\n",
            "HepG2_group_barcodesPerInsert_box.png\n",
            "HepG2_group_barcodesPerInsert_box_minThreshold.png\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/barcode:\n",
            "assigned_counts  counts\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts:\n",
            "MPRAworkshop\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/barcode/assigned_counts/MPRAworkshop:\n",
            "HepG2_default_DNA_perBarcode.png\n",
            "HepG2_default_RNA_perBarcode.png\n",
            "HepG2_default_barcode_DNA_pairwise.png\n",
            "HepG2_default_barcode_RNA_pairwise.png\n",
            "HepG2_default_barcode_Ratio_pairwise.png\n",
            "HepG2_tutorialConfig_DNA_perBarcode.png\n",
            "HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "HepG2_tutorialConfig_barcode_DNA_pairwise.png\n",
            "HepG2_tutorialConfig_barcode_RNA_pairwise.png\n",
            "HepG2_tutorialConfig_barcode_Ratio_pairwise.png\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/barcode/counts:\n",
            "HepG2_default_DNA_perBarcode.png\n",
            "HepG2_default_RNA_perBarcode.png\n",
            "HepG2_default_barcode_DNA_pairwise.png\n",
            "HepG2_default_barcode_RNA_pairwise.png\n",
            "HepG2_default_barcode_Ratio_pairwise.png\n",
            "HepG2_tutorialConfig_DNA_perBarcode.png\n",
            "HepG2_tutorialConfig_RNA_perBarcode.png\n",
            "HepG2_tutorialConfig_barcode_DNA_pairwise.png\n",
            "HepG2_tutorialConfig_barcode_RNA_pairwise.png\n",
            "HepG2_tutorialConfig_barcode_Ratio_pairwise.png\n",
            "\n",
            "MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/statistic/counts:\n",
            "BCNucleotideComposition.HepG2_1_DNA.tsv.gz\n",
            "BCNucleotideComposition.HepG2_1_RNA.tsv.gz\n",
            "BCNucleotideComposition.HepG2_2_DNA.tsv.gz\n",
            "BCNucleotideComposition.HepG2_2_RNA.tsv.gz\n",
            "BCNucleotideComposition.HepG2_3_DNA.tsv.gz\n",
            "BCNucleotideComposition.HepG2_3_RNA.tsv.gz\n"
          ]
        }
      ],
      "source": [
        "results_dir = \"MPRAsnakeflow_tutorial/results/experiments\"\n",
        "!ls -R \"{results_dir}\""
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "We can take a look at the amount of barcodes that have been removed due to UMI collision, by checking how many unique BC x UMI combinations we have in our data. (We are now checking the RNA counts in replicate 1, but we should look at both DNA and RNA counts in all three replicates.)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "First few lines of a count file, where column 1 contains the barcode, 2 the UMI and 3 the count.\n",
            "AAAAAAAAAAAAAAA\tCGAAGTACACTTTCCG\t1\n",
            "AAAAAAAAAAAAAAA\tTATAGTAAACACGCCC\t1\n",
            "AAAAAAAAAAAAAGG\tCGTGAGCAGGATTAAG\t1\n",
            "AAAAAAAAAACAAGT\tCTATCCGCACTGTGTC\t1\n",
            "AAAAAAAAAACCCAA\tTTCCTAGGGACAGTCC\t1\n",
            "AAAAAAAAAAGATGC\tGGAGCATGTAATGGGT\t1\n",
            "AAAAAAAAAAGCTGC\tAAGATATACGGACGAT\t1\n",
            "AAAAAAAAAAGCTGC\tAGAGGTTAGAAGGGCA\t1\n",
            "AAAAAAAAAAGCTGC\tATATGGACCGCCCTGG\t1\n",
            "AAAAAAAAAAGCTGC\tCATGCACCCTGCATAA\t1\n",
            "AAAAAAAAAAAAAAA\tCGAAGTACACTTTCCG\t1\n",
            "AAAAAAAAAAAAAAA\tTATAGTAAACACGCCC\t1\n",
            "AAAAAAAAAAAAAGG\tCGTGAGCAGGATTAAG\t1\n",
            "AAAAAAAAAACAAGT\tCTATCCGCACTGTGTC\t1\n",
            "AAAAAAAAAACCCAA\tTTCCTAGGGACAGTCC\t1\n",
            "AAAAAAAAAAGATGC\tGGAGCATGTAATGGGT\t1\n",
            "AAAAAAAAAAGCTGC\tAAGATATACGGACGAT\t1\n",
            "AAAAAAAAAAGCTGC\tAGAGGTTAGAAGGGCA\t1\n",
            "AAAAAAAAAAGCTGC\tATATGGACCGCCCTGG\t1\n",
            "AAAAAAAAAAGCTGC\tCATGCACCCTGCATAA\t1\n",
            "\n",
            "Amount of total BC x UMI combinations:\n",
            "\n",
            "Amount of total BC x UMI combinations:\n",
            "871810\n",
            "871810\n",
            "\n",
            "Amount of duplicate BC x UMI combinations (the file is sorted):\n",
            "\n",
            "Amount of duplicate BC x UMI combinations (the file is sorted):\n",
            "0\n",
            "0\n"
          ]
        }
      ],
      "source": [
        "print(\"First few lines of a count file, where column 1 contains the barcode, 2 the UMI and 3 the count.\")\n",
        "!zcat \"{results_dir}\"/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz | awk 'NR <= 10 {{print}}'\n",
        "\n",
        "print(\"\\nAmount of total BC x UMI combinations:\")\n",
        "!zcat \"{results_dir}\"/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz | wc -l\n",
        "\n",
        "print(\"\\nAmount of duplicate BC x UMI combinations (the file is sorted):\")\n",
        "!zcat \"{results_dir}\"/experimentMPRAworkshop/counts/HepG2_1_RNA_filtered_counts.tsv.gz | uniq -d | wc -l"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "There are no collisions, indicating that this subset of the data is of high quality. Great!"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "The count files are then joined with the assignment file (the output of the assignment step) to assign counts to oligos. The barcode counts are then aggregated over the oligos. This is done within MPRAsnakeflow with additional filter steps, but to give you a feeling of what's happening we give you some simplified code below:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Joining the assignment and the counts of replicate 1:\n",
            "\n",
            "Barcode\toligo\tDNA\tRNA\n",
            "AAAAAAAAATCGATC\toligo_006917\t2\t3\n",
            "AAAAAAAACACATGA\toligo_005412\t10\t4\n",
            "AAAAAAAACAGTAAG\toligo_006290\t4\t3\n",
            "AAAAAAAACCAAATA\toligo_189277\t4\t2\n",
            "AAAAAAAACGAACAG\toligo_005299\t4\t1\n",
            "AAAAAAAAGAATTAC\toligo_144932\t3\t4\n",
            "AAAAAAAAGAGTAGT\toligo_006091\t7\t11\n",
            "AAAAAAACCGCACAT\toligo_005747\t1\t4\n",
            "AAAAAAAGACCTAAG\toligo_006429\t1\t4\n",
            "AAAAAAAGAGGTAAA\toligo_006350\t4\t6\n",
            "AAAAAAAAATCGATC\toligo_006917\t2\t3\n",
            "AAAAAAAACACATGA\toligo_005412\t10\t4\n",
            "AAAAAAAACAGTAAG\toligo_006290\t4\t3\n",
            "AAAAAAAACCAAATA\toligo_189277\t4\t2\n",
            "AAAAAAAACGAACAG\toligo_005299\t4\t1\n",
            "AAAAAAAAGAATTAC\toligo_144932\t3\t4\n",
            "AAAAAAAAGAGTAGT\toligo_006091\t7\t11\n",
            "AAAAAAACCGCACAT\toligo_005747\t1\t4\n",
            "AAAAAAAGACCTAAG\toligo_006429\t1\t4\n",
            "AAAAAAAGAGGTAAA\toligo_006350\t4\t6\n",
            "\n",
            "Aggregating the barcodes over oligos and the counts of replicate 1:\n",
            "\n",
            "oligo\tDNA\tRNA\n",
            "\n",
            "Aggregating the barcodes over oligos and the counts of replicate 1:\n",
            "\n",
            "oligo\tDNA\tRNA\n",
            "oligo_002500\t23\t41\n",
            "oligo_002501:oligo_002502\t12\t26\n",
            "oligo_002503\t14\t23\n",
            "oligo_002504\t1\t2\n",
            "oligo_002505\t24\t41\n",
            "oligo_002506\t7\t15\n",
            "oligo_002507\t11\t21\n",
            "oligo_002508\t11\t29\n",
            "oligo_002510\t15\t31\n",
            "oligo_002511\t23\t29\n",
            "oligo_002500\t23\t41\n",
            "oligo_002501:oligo_002502\t12\t26\n",
            "oligo_002503\t14\t23\n",
            "oligo_002504\t1\t2\n",
            "oligo_002505\t24\t41\n",
            "oligo_002506\t7\t15\n",
            "oligo_002507\t11\t21\n",
            "oligo_002508\t11\t29\n",
            "oligo_002510\t15\t31\n",
            "oligo_002511\t23\t29\n"
          ]
        }
      ],
      "source": [
        "counts_rep1 = os.path.join(results_dir, \"experimentMPRAworkshop/assigned_counts/MPRAworkshop/HepG2_1.merged.config.tutorialConfig.tsv.gz\")\n",
        "assignment = \"MPRAsnakeflow_tutorial/example_data/assignment/assignment_barcodes.default.tsv.gz\"\n",
        "\n",
        "print(\"Joining the assignment and the counts of replicate 1:\\n\\nBarcode\\toligo\\tDNA\\tRNA\")\n",
        "!join <(zcat {assignment}) <(zcat {counts_rep1}) -t $'\\t' | cut -f 1,2,5-6 | awk 'NR <= 10 {{print}}'\n",
        "\n",
        "print(\"\\nAggregating the barcodes over oligos and the counts of replicate 1:\\n\\noligo\\tDNA\\tRNA\")\n",
        "!join <(zcat {assignment}) <(zcat {counts_rep1}) -t $'\\t' | cut -f 1,2,5-6 | sort -k2,2 | \\\n",
        " awk -v OFS=\"\\t\" '{{if ($$2 == prev) {{dna += $$3; rna += $$4}} else {{ if (NR > 1) print prev,dna,rna; prev=$$2; dna=$$3;rna=$$4}}}}' | awk 'NR <= 10 {{print}}'"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "In the next steps, barcodes are already aggregated to their oligos/targets.  \n",
        "  \n",
        "We will have a look at the correlation plots between the replicates for DNA and RNA now. Typically, we will sequence RNA counts deeper than DNA -- because of the larger dynamic range of RNA. We therefore expect that the correlation of RNA counts is higher than that of the DNA. Our example data shows this trend as well.\n",
        "\n",
        "We have a correlation of DNA counts per insert between the replicates of 0.27 and 0.31. This correlation is for a subset of the complete MPRA experiment. The shown output uses the user set minThreshold for removing barcodes with too few counts. In order to see the influence of the parameter another plot is generated without this threshold."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "With the threshold of 5 barcodes per oligo and at least one DNA and RNA count\n",
        "- DNA correlation: 0.28 - 0.31\n",
        "- RNA correlation: 0.68 - 0.70"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAARYAAAGiCAYAAAAiFuksAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/TGe4hAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB3tUlEQVR4nO3dd3hUx734//f2ot5Xq15QF72bZoNxxx23JE7s5LonXCdx4pubxPE3F2z8C4lriMsFOy644RqbYpveO0hICPXepV2V7Xt+f5A9F1FswBKrMq/n4UE6O7s75+jsZ+fMzPmMQpIkCUEQhH6k9HcFBEEYfkRgEQSh34nAIghCvxOBRRCEficCiyAI/U4EFkEQ+p0ILIIg9DsRWARB6HcisAiC0O9EYBEEod/5NbC89NJLpKSkoNfrmTBhAlu2bPFndQRB6Cd+CyzvvvsuixYt4ne/+x0HDhxg5syZXHXVVVRXV/urSoIg9BOFv25CnDJlCuPHj+fvf/+7vC07O5sbbriBJUuW+KNKgiD0E7U/3tTpdLJv3z5++9vf9tk+f/58tm/fflp5h8OBw+GQf/d6vbS3txMREYFCoRjw+gqCAJIk0dXVhdlsRqn89osdvwSW1tZWPB4PMTExfbbHxMTQ2Nh4WvklS5bwpz/96WJVTxCEb1FTU0N8fPy3lvFLYPE5tbUhSdIZWyCPP/44jz76qPy7xWIhMTGRmpoagoOD+71eFRUV2Gw2cnJyzvk5x48f59ChQ2i1WgIDA5kzZ44c1e12O5999hlGoxG3282kSZOIjY2lqKiIiooKrrzySnp6ejh8+DAzZsxAkiQOHjyIJEmMGzduQFplVVVV6HQ6TCZTv7/2mRQUFLB3715uvPFGvv76a2688UbR2hxirFYrCQkJBAUFfWdZvwSWyMhIVCrVaa2T5ubm01oxADqdDp1Od9r24ODgAQkspaWlNDY2MnXqVODEpduuXbtwu92kpaWRkJBAaWkptbW1BAQEEBwcTGRkJFOnTiU4OJj77ruPOXPmyHVzu93861//4tVXX6WxsZHly5fzk5/8hH379vHJJ58QGBhIXl4en3zyCVdffTVWq5Xly5cTGBjIpEmTCAgIkOtmtVrZu3cvSqWS5ORkIiMj2bNnD5IkkZycTHx8PIcPH2bChAk4nU46OjowGo0UFxfjcDhQqVSMHj2aL774AoPBwHXXXQdAZWUlKpWKcePGySdOaWkpXV1dWCwWYmJiyMrKoqmpiaKiIgBGjRpFQEAApaWl2Gw2Ro0aRWxsLAB1dXUcO3YMtVrN5MmTkSSJlpYWoqKiGDduHIGBgezevRu73Y7T6WTq1KlotVp2796Nx+MhKyuL2NhYEXwGoXP5m/hlVEir1TJhwgTWr1/fZ/v69euZPn26P6r0rV599VXq6+sxm80sWbKEY8eO8fLLL5OYmMiBAwd46623SEpKIjs7m46ODrKystDr9X1eQ61WExoaSk5ODpdddhlr1qwB4LbbbmPt2rV0dHQAJ1ptn3/+OXPnzmX8+PGsW7cOX/+61+vlpZdeoqqqipSUFEJDQ3nrrbeoqKggLCyMxx9/nJaWFp5++mkkScJisbBz506qqqp4+umniYuL4+2336a0tJSoqCjMZjOhoaH88Y9/RJIkTCZTn5Nm9erVbNiwgaSkJF555RWOHj3K008/jVarJSAggBdeeIHi4mKeeeaZPt9kXV1dLFmyBJPJRHt7Oy+++KL8ml1dXaxevZqCggI+//xzEhMTeemllygrK+Pdd9+lsbGR6OhonnzySXp6egb07yoMHL8NNz/66KO8+uqr/O///i9FRUX853/+J9XV1dx///3+qtJZFRQUMHbsWDIyMnC5XJSWluLxeEhJSSEvLw84ERCOHDnCe++9x+9///sztrB85crLywkJCQEgLCyMm266ib///e9IkoTD4eCTTz4BTgSj999/H7vdDoDH4+Hw4cMsWLCAxMREgoODKSwsZP78+YwePZqoqChcLpf8Xj09PXg8HgDi4+NJTU1Fr9fj9XqJiIggKiqK0NBQfv7zn1NSUsLf//53Ojs7+9Q3MzOT5ORkQkJCKCoq4vDhwxw/fpySkhImTpyIJEmMGjWK5ORkAgMDAeju7sZisZCdnc24ceMoLCw87ThUV1dz6aWXkpqaSlZWFgA7duxg0qRJ5Obm0tXVJe+3MPT4rY/ltttuo62tjSeffJKGhgby8vL44osvSEpK8leVZFqtloqKCtavX49Go+Gqq67is88+Y9SoUcTHxzNr1iy++eYbXnvtNSorK1EqlezcuZM///nP/OQnP+HgwYNMnTpVbrUoFApcLhfr169ny5YtBAQEcM899/Dee++h1WqZNWsWX331Fa2trRQXF5Obm8vChQtRKBR0dHRw5MgRJk+ejFqt5pZbbuHZZ59lxowZJCQkcMUVV/CXv/yFKVOmEB0dTVRUFDk5OXzwwQfs3r2biRMnolKpMBqNABiNRlQqFeHh4WzdupW4uDjKy8tJTU3l6NGjp7USNm/ejEqlwmazMXPmTBobG+ns7JQ//BqNBoPB0KelExERQUZGBqtXr6a2tpbrr78etVotlzMYDEyYMIFnnnkGm83G8ePHgRPnxGuvvca4cePIzc0dkMtc4eLw2zyW78NqtRISEoLFYhmQk89ms8nf3AqFgsjISNrb2/F4PISGhqLX6ykqKuL48ePs2bOHGTNmMGvWLCwWCwBKpVLuR4ITlzAtLS14vV6USiVRUVEolUp6enpQqVTo9Xq6u7txuVwYjUacTqd8WdHb2yt/GOFEi6etrQ2Xy0VgYCABAQHyKFtERARarZbe3l4sFgt6vR6DwYBKpaK3t5eQkBCsVqu8raWlRX4/p9OJWq0mMjJSDhJ/+ctfSE9PZ/LkyQQGBhIUFITX65XfT6/XExAQgN1uP+3vYLfb6ejokPfX7XbjcDgwGo309vYSGBjIvn37qK2tZdWqVfzpT39i1KhR8nEKDw8/a6tP8I/z+dyJwHIBJEmioaEBi8WCVqslOTlZDiLDSWtrK3q9Xr7E6U+SJFFVVYXNZkOn05GSkiI6age58/nc+XW4eahSKBSYzWbMZrO/qzKgIiMjB+y1FQoFycnJA/b6gn+Ju5vPwO1209PT06fz82y8Xi8HDx7EZrNRV1fHtzUAvV4vbrf7rI9LkkRvby82m+1bX+dsbDYb7e3t8mvZ7XZKSkro6uq6oNcThAslAssZbNu2jXfeeYe//OUv7Nq1C4/HQ0lJCWVlZXg8HlpbWyktLaWqqgqXy8WTTz6J1WqltLQUgLKyMo4ePYrT6aS0tJSioiKcTieHDx/mzTffpLa2lt7eXgoLC2loaJA/9E6nk7Vr1/Kzn/0Mp9Mp18fr9VJeXk5RURF2u53Ozk6qq6spLi7G6XTi8XgoLi7mwIEDHD58WH7erl272LRpE3/84x+prKy8qMdQGNnEpdAZ9PT00NzcjNvtxu12895771FWVkZgYCCjR49m3759TJkyhY0bN3LttdfKw8SVlZU4nU4KCwuZNWsWHo+HyspKGhoa+Oqrr5g0aRKdnZ3YbDb+/Oc/k56eTkVFBQ8//DAxMTHodDquuuoq3njjjT4tjJ6eHg4dOkR7ezs7d+7EYDBQUFBAXl4eBw8exOVyye81ZcoU4MSlxuzZs5k4cSJPPvkkGo3GX4dTGIFEi+UMlEol0dHRqNVqOjs7KSoqIj09nZycHFJTUwkKCmLKlClkZmbKLQRJkvB6vZSUlHD11Vczbtw46uvr2bBhAwaDgfXr1xMaGkpSUhJxcXE0NzdjNpuZMWPGd3aONjQ00NXVhcFgoL29HZVKxcyZM8nPz2fbtm0cOnSIyy67jFmzZqFWq+X6NDU18eabb3LTTTcRFxc34MdNEHxEYDmDiIgIFixYwG9+8xsqKiq48cYbKS8vp7q6mqamJrq6uuRWzNVXX82oUaPQ6XRERkZy3XXX8d577/Huu+/K9z5ZrVbS09OJioqioKCA/fv3c91113H8+HF5Ojyc6NvZsGEDer2ebdu24fV6ATAYDHR2dlJXV4fJZCI8PJzQ0FAMBgNJSUn86Ec/4p133mHnzp197uP4xz/+gd1up7S0lNbWVr8cS2FkEsPNZ3DyIfEFB9+HHOD999/npptuQq1Wy48plcrTyiqVSvlnhUIhP3byz75yCoVCbvX4nGm777lnq993vZYgXCgx3Pw9nfwB9P3sm6ciSRI333yzHFROfuzU30/9+bseUygUZ5wPc7btZ3vP73qOIAw0EVjOk0KhEB2hgvAdRB/LKXbu3ElxcbG/qyEIfuX1eqmvr+8zfeF8iMByit27d3Ps2DF/V0MQ/MLj8dDQ0MDRo0dRKBRkZ2df0OuMmEsh31wThUKBVquVOzw7OjoICgpCq9X6u4qC4DcOh0NO7JWcnEx2dvb36qMbMYGls7OTL774gqysLMaPHw/A1q1bCQwMpLKykuuvv14eTRGEoco3Gugbpfyu89nlclFbW4vL5cJkMpGTk9Mvo4cjJrDo9XomT57MgQMHyMjIIDAwELvdzrRp09iyZQs2m43a2lrKyspISUnxd3UF4ZycHDgUCgXHjh1j2bJl/O53v+O5557D6XSyd+9eFAqFnPbDaDRSV1eHTqfjxz/+MQsXLiQwMLBfpyOMmMDS0dFBT08PQUFBdHd3s2vXLsxmM/v372fMmDEYjUaysrJIT08XrRZhyCgqKmLHjh2kpKQwY8YM+f6xP/zhD3Ku4qlTp/L2228TGBjI3r17UavVOJ1OwsLCmDx58jklxz5fIyawhIWFodVqyczMRKvVEhQUREBAAJ2dnQQGBn7nOimCf0iSRGNjI3V1dZjNZjnBdldXFyUlJSiVSnJzc7FYLFRWVpKWlkZ4eLi/q33RZGZmYjabeeqpp4iKikKr1dLc3Mwnn3xCVVUVjz32GNdddx1lZWUcOHCA0NBQJk+eTHt7OzU1NdTW1pKfn9/v9RoxnyaDwUBkZKScPc3X9AsLCxPzUi4yl8vFO++8w9dffy3fxe3rDzj5n2/7kiVLsNvtPP/881itVuDETGKdTsehQ4fYuHEjKpWKjz76iK1bt/ptv/xBpVIRGhrK4sWLycvLY86cOVx66aUolUr+8Y9/cODAAV577TWuuuoqli5disvlYu3atSxcuBCz2TxgqWBHTItFGDxcLhcrV65k5cqVclDft28fmzdvlsvceuutJCQkIEkSaWlpTJ06lY8++ojKykrGjBlDQEAAbrebwsJC4uLiCAsLw2QyjdjLWF+LW6/Xc8cdd7Bjxw65X9HtdhMVFcW6dev4zW9+Q3d3NzU1NaSkpAxIdkAYQS0WYXAxGo19lsgNDw8nMzNT/udL/q1QKKipqaG8vJzg4GAiIiIoLi7GYrGQlJTEbbfdRlFREZIk4XK5cLlcfe6RGil8ScKOHz9Oc3MzkiQxd+5c/vGPf7BixQqSk5MJCgqiubmZW2+9FYPBwLJlywYsC6JosQgDzuPxUF9fT3x8vHxLxB133NFnnkRqaiqpqamnPVehUPDoo4/ywQcfkJ2djVKppLS0lMDAQD788EPcbjfz58+ns7OTtrY2WltbGT9+/IjJoetbCK61tRW1Wk18fDy9vb28//77/Md//IecRmPs2LHExMTIrZfRo0cTEhIiP97fRItFGDAtLS18+eWXbN68mUWLFvHll19SWlqKUqkkLi6Ozz77TO4zORuFQkFwcDCdnZ0YDAZWrlzJlVdeSXx8PLfccgszZ87k448/prOzk+nTp3PNNdfw5ptv9llfaTiSJInW1lYOHjxIU1MTycnJ8sqUkZGR3HfffX2ChlqtJiEhQb78ueOOOwY0Eb0ILMKAqa6u5q233iI1NZXu7m7S09N56qmn2Lp1K7t27SIsLIw333xT7qz9/PPPefHFF3nxxRf53//9XzlPTWdnJ62trVx55ZXs27cPh8MBgNlspqOjg+rqanQ6Hddccw1msxmVSjVs7+x2uVzU1NRw9OhRHA4H+fn55OfnYzQa5RbaYLizXVwKCQMqNjZWXmM7LS2Nnp4eWltbiYqKIjw8vM99WWFhYXLQ0Ol0coekRqOhp6eH9vZ2QkND6enpwW63o1AouPTSS6mtrWXTpk1ER0ezb98+fvKTnwy76QNut5uamhoaGxsxmUxkZmYO2GVMfxi8NROGvOjoaGbOnCm3JhQKBVdddRXTpk1j586dbN++nd/+9rfyN+0ll1xy1tf58Y9/zIcffsiiRYvo6Oigu7ub7u5uqqurUalU3HDDDaxcuZKYmBg2btzIVVddRVhY2MXc3X7n65BtamrCbrcTGxtLUlLSkAiaIoPcKZ577jmSkpK4/vrr+/V1BeFcSZJET08P5eXl9Pb2kpGRQVhYmN87o0UGOUEYgrxeL1arlcbGRjQaDaNGjUKv1/s9oFwIEVgEwc98KyqUlZUREhJCeno6Op1uSAYUHxFYBMFP3G43LS0ttLW1ERkZyZQpUwZ1h+z5GB57IQhDiNPppKKigubmZlJTU793UqXBSAQWQbgIJEnC6XRSW1uL0+kkNjaWjIyMIX25821EYBGEAdbZ2cmxY8fQ6XTySprDNaD4iMAiCANAkiQsFgv19fXo9Xry8/MxGAzDPqD4jJjAUldXx/Hjx1EoFMycOROlUkl9fT2FhYUkJiaSmZnp7yoKw4DX66WmpkZeDjc9PR2NRjNiAorPiAksZrOZsLAwNmzYIG/r6urC6XSi1+uRJIkDBw5w5MgRkpOT/VdRYcjxJbBuaGigo6NDTvmoUqlGXEDxGTGBpbe3lw8++IApU6YgSRJ2u52EhATCwsI4ePAgiYmJhIWFERIS4u+qCkOIzWbj+PHjeL1ezGYzubm5Q2LK/UAbMYHFarWSmJiI0+nEbrfT3NwMnLi1Pzc3F4VCQUpKComJiSM2C5lwbnxfTDU1NTidTsxmc5+kVcIICiyxsbHExsbKv/uW+BBLfQjnypdUqb6+HoPBQHx8fJ90BcL/GTGBRRAulCRJtLe3U19fj1qtJiMjY0SN8FwIEVgE4Sx8qwR2d3cTGhpKVlaWWNHhHInAIggnkSQJj8dDXV0dra2thIWFyVPuRQvl3A35wCJJEkePHuXQoUNcddVVGAwGGhsbxZCxcN66urpoamrC6XRiMplISEgQIzwXaMgftfb2dt555x2CgoJYtmwZhYWFHD582N/VEoYIX1Klo0ePUlxcTGhoKNnZ2YSHh4ug8j0M+RaL1+slLS2Na6+9lpkzZ/Lee++JuSjCd/J6vXR0dNDS0oJarSYlJWXIJlUajIZ8YImMjOSHP/whAKGhodxzzz3Y7XY/10oYrHxDxpWVleh0OjIyMkRAGQBDPrAoFIrT1k8ZqGUjhaHL6XTS1tZGW1sbERERjB8/ftgkVRqMhsVFpMfj4auvvuLll1+mra2NzZs3i9mzAnBiyLi0tJRdu3bhcrnIzs4mNjZWBJUBNiyOrs1mY//+/SQnJ9Pd3c3q1auZMWOGaN6OUL4p9/X19fKU+7S0NHE+XETDIrDo9XpiY2N5//332bFjB7feeqs4iUYoq9VKSUkJXq+XjIwMQkJCxLngB8MisPT29mKz2Xj22WdxOBy0t7eLk2kE8S2b4buHJzc3V3TI+tmwCCwrVqzgH//4Bzt27ECSJB5++GF/V0m4CHw5UMrLy4mJiSEtLQ2tVisCyiAwLALLAw88QH5+PuXl5bhcLkJDQ/1dJWEAeTweGhsb6ejoICIigmnTpokp94PMsBkV2rJlCw6Hg9jYWHbt2iVGhYYhh8PBkSNH2Lt3L2q1us8Ijwgqg8uwaLGoVCry8vLQaDRs27aN+Pj408pIkkRDQwMGg4HQ0FAUCoWcnzQyMpKAgAA/1Fz4LpIk4XA4qKmpwe12ExcXNyjWMRa+3bAILEqlkujoaKZPn05WVhbd3d2nlSkrK6OhoYHe3l6mTJlCaGgoe/bsITAwkIKCAq644opht2jUUObLgVJaWkpQUBCJiYkEBASIgDJEDItLIbvdzqZNm5AkieDgYFasWHHapVB1dTUhISGYTCY6OzsBsFgsZGRkUFZWht1up7i4WM7kL/iHL6AUFhbS3t5Ofn4+2dnZBAYGir/LEDIsWixGo5HJkyezcuVKrFYrd99992knYW5uLvv27cPtdjNt2jSKi4uJiIhg7969ZGVlYTQayc7OZtSoUaJ/xg88Hg+VlZVYLBYiIyPJzMwUSZWGsCEfWKxWK//6179oaGiQWyIKheK0wBIdHc1ll12GWq1GpVIRGRkJnJi1q9frxS3yfuBbNqOuro6Ojg5CQ0MZPXq0GOEZBoZ8YNFoNJhMJoxGIw6HA4CwsLDTyikUCvR6/WnbjUbjgNdROF13dzfV1dV4vV5iY2OJj48XwX0YGfKBxW63s2rVKnp6euRtUVFRpKam+rFWwplIkoTNZqO6uhqbzUZCQoJIqDRMDfnAEhoayksvvcQnn3xCb28vbrebtLQ0JEkSzelBwuv10tLSQltbGxqNhsTERJHlfpgb8l8VCoUCu91OWVkZCxcu5NJLL2X58uUAohPWjyRJkpMqHT58mKamJpKTk0lPTxdr8YwAQ77FAif6WbRaLW+//TZ2u52AgAA+/vhjrrvuOpF3ww+cTifNzc10dnYSHh5OXl6e+DuMMOfVYlmyZAmTJk0iKCiI6OhobrjhBo4dO9anjCRJPPHEE5jNZgwGA3PmzKGwsLBPGYfDwSOPPCLPeF2wYAG1tbUXvBMajYZrr70WhULBD37wA+677z5uuOEGcTJfZG63m4qKCvbv34/b7SYrKwuz2Sz+DiPQeQWWTZs28dBDD7Fz507Wr1+P2+1m/vz5fTpOly5dyrJly3jhhRfYs2cPJpOJyy+/nK6uLrnMokWL+Oijj1i1ahVbt26lu7uba6+9Fo/Hc0E70dvby4oVK+jq6qK9vZ3/+Z//EZdBF4kvy315eTklJSWEhYUxZcoUkpOTRUAZyaTvobm5WQKkTZs2SZIkSV6vVzKZTNJTTz0ll7Hb7VJISIi0fPlySZIkqbOzU9JoNNKqVavkMnV1dZJSqZTWrFlzTu9rsVgkQLJYLJIkSZLT6ZReeOEF6YknnpCef/556YMPPpC8Xu8F7dOzzz4rffzxxxf03JHGarVKhw4dkrZv3y61tbVd8DEXhoZTP3ff5nt9pVgsFgDCw8MBqKiooLGxkfnz58tldDods2fPZvv27dx3333s27cPl8vVp4zZbCYvL4/t27dzxRVXnPY+DodDnqMCJybFnVqP/Px8kpOTKSgoIDIyUnQODhCv14vFYqGxsVHOcq/T6cTxFvq44FEhSZJ49NFHmTFjBnl5eQA0NjYCEBMT06dsTEyM/FhjYyNarfa0SWwnlznVkiVLCAkJkf8lJCTIj3V1dfGnP/2JPXv2sHTpUrq7u8nMzLzQ3RLOQvr33eHbt2+nvr6elJQUsRaPcFYXHFgefvhhDh8+zDvvvHPaY6eeaNI5zCn5tjKPP/44FotF/ldTUyM/ZrVaqaioYOLEiYSGhmI2m7HZbBewR8KZuN1u6uvrKSwsRKFQMHXqVHJyckRAEb7VBV0KPfLII3z66ads3ry5T+4Tk8kEnGiVxMbGytubm5vlVozJZMLpdNLR0dGn1dLc3Mz06dPP+H46nQ6dTnfGx8LDw7n//vvp7u5m0qRJWCwWnE7nheyWcBKn00lZWRmtra2kpaXJC6MLwrk4r8AiSRKPPPIIH330ERs3biQlJaXP4ykpKZhMJtavX8+4ceOAEyfopk2bePrppwGYMGECGo2G9evXs3DhQgAaGhooKChg6dKl570DBoOBa6+99ryfJ5xO+ndSpdraWlwuF7GxsWRlZYmWiXDeziuwPPTQQ7z99tt88sknBAUFyX0iISEh8hTtRYsWsXjxYkaNGsWoUaNYvHgxRqORO++8Uy5777338stf/pKIiAjCw8P51a9+RX5+PvPmzev/PRS+kyRJdHZ2cuzYMQwGA6mpqSL/ifC9nFdg+fvf/w7AnDlz+mxfsWIFP/7xjwF47LHHsNlsPPjgg3R0dDBlyhTWrVtHUFCQXP6vf/0rarWahQsXYrPZmDt3LitXrhRN7YvMF1B8y2aMGTNG9J0I/UIhSUNvJpnVaiUkJASLxUJwcPA5P8/r9Z4xV8vJnnvuOZKSkrj++uv7o6qDksfjobq6Wu4LM5vNaLVaf1dLGOTO53M3YqZG1tfX09LSgs1mY8qUKSgUCpqamjh69Cjx8fGMGjXK31UcUNK/kyrV19fL9/BMmjRJJFUSBsSwDywul4uenh4iIyMJDQ3lyy+/lIe27XY7arUat9uNJElUVlZSU1NDcnKyv6vdr3p7ezl+/DgAcXFxxMXFiRwowoAa9oHFZrNRXl5OUlISR44cIT8/X777NjY2lrCwMDZv3kxWVhYmk4moqKhhcZ+R9O+F0aurq3G5XMTHxxMWFiYCinBRDPvAEhwczPjx46mtrcXr9dLV1YXdbqelpQWHw0FjY6M8NG4wGM6YvnIokSSJ5uZmGhoaMBqNxMfHi/wnwkU37AOLT3x8fJ/JfBMmTAAYNn0rkiTR1tZGfX09arWajIwMkaVN8JsRE1iGK6fTSV1dHd3d3YSHh5OTkyPSFQh+J87AIcrtdlNbW0traysRERFkZ2eLgCIMGuJMHEIkSaK7u5vGxkZ5yn1SUpK43BEGHRFYhgjfOjzd3d2kpaURHh4uAoowaInAMoh5vV46Ojpobm5Gq9WSmpoqkioJQ4IILIOQb8i4qqoKvV7PqFGjxD08wpAiAssg4nQ6aW1tpb29ncjISCZMmIBSqRQBRRhyRGAZBJxOJ9XV1XLKR5FUSRjqRGDxk1OTKsXFxZGWliZaJ8KwIALLRSZJElarlZKSEgAyMjIIDg4WAUUYVkRguUi8Xi9Wq5W6ujqMRiP5+flihEcYtkRgGWBer5e6ujoqKiowm82kp6efNTG4IAwXIrAMAF9SpYaGBjo6OoiKiuKSSy4RIzzCiCECSz+z2+0cO3YMp9NJUlISubm5IgeKMOKIwNIPfCM81dXVeDweEhMTCQkJEQFFGLFGTGBxu900NTWhVquJjo6WU1O2tLQQEhJyXkm5fSRJorW1lfLycnnpV5FUSRC+xxKrQ43FYmHnzp20tLTI27Zt20ZzczNr1qyRc+P29vZ+52v5kioVFBRgsVgYPXo0mZmZBAQEiKAiCIyAwNLd3U1BQQFKpZLU1FQOHjxId3c3cGIZjPz8fGpra3E4HGzdupXdu3ef9RLG5XJRWlrKwYMH6e3tJSsri/T0dJGpTRBOMewvhTQaDdHR0fKC8lqtFrfbzbFjx4iIiOCrr75i3LhxGI1G5s+fz7Fjx/B6vfLzfSM8tbW1dHR0EB4eTn5+vlg2QxC+xbAPLDqdjujoaLxeLwkJCQAolUpCQkJQKBTyImantlJ8SZWqq6sBiI2NJTExUQQTQTgHwz6w+JwaOHwB4kw3+1mtVvbs2YPdbichIUFOqtTV1XVR6ioIg4VCoSAgIOC8RzhHTGA5V5MnT2bZsmWsXbv2rEuBWCwWjEYjGo3mItfuwvX29mI0Gv1djXPm60QfKnX2rYMdFhbm76qcM5fLhSRJ37q8rtFo5MknnyQ0NPS8XntErd18LiRJkv+dzT//+U+mT59OWlpav773QJEkiS1btjBr1ix/V+WcbdiwAbVazcyZM/1dlXNis9l44403uO+++/xdlXNWVlaGy+UiKyvrrGV8a50rFIrz+tyJwHKOJEni8OHDeDwe0tLSCAoKQqFQcOzYMRwOB9nZ2YNuYXWbzcaxY8cIDw/HbDajVqtpaWmhoqKCyMhIUlNT/V3FPnyzllNSUggICACgpaWF+vp6cnJyBt1ich6Ph8LCQoxGI8nJyUiShNPp5OjRo4SGhpKenj7o+uR8864iIiJISUkBTqxrXl9fT2ZmJiEhIWd97vl87ob9cHN/kSSJ7du3Y7PZUKvVKJVKent7KSkpweVyUV5e7u8qnsbhcJCYmEhZWZm8rbKykqampkH3IZUkiXXr1hEWFsaaNWuQJAmPx8ORI0fQ6/UcOXJk0C19q1QqSU5OprS0lNraWjQaDZ2dndTX12MwGPxdvTOqrq6moaEBvV4v37u2d+9ewsPD2b9/f7+9jwgs5+DYsWPU1NRw6623EhQUxIEDB5AkCbfbTXR0NLGxsYPqpLdYLGzevBmPx4PFYiEuLg6VSoUkSeTl5ZGfn9+vJ1F/6erqIjQ0FJVKhdfrxev1yqN6HR0d/q7eGXV1dWGz2TCbzUiSRHR0NJMnT2bHjh3nNNnyYsvOzmbs2LHs379fvuQPCgoiISGhX9elEp235yA9PR2AiooKlEol0dHRtLa2EhgYiNvtpr29neTkZP9W8iTBwcFMmzaN5uZmenp6yM3Nxel0yrceuFwuIiMj/V3N04wdO5a9e/eSkJBAa2srCoWC8PBwKioqyMjIGHSXFR6PB5vNxnXXXYdSqaS6uhqtVovVaiU8PHzQXRoDdHR04HQ6iYyMpLu7G4VCQUREBKWlpf16aSz6WM6Dx+PB6/WiVqvlk9z3zTpUViGUJAmXy4VGoxl0H1RfK/Dk4+vbNlRG4Abz8YUz1+9Mx/1MzudzNzQ+DYOESqU6bd6LUqkcUncxKxSKi/JNWlZWhtVqJSIigoSEhG89Yb1eL+3t7fIlUERERJ/6nhxUfLOiz3bMJUmiuroatVpNXFxcP+3NubtYx/dCnal+px7j/jB0PhHCkLJ48WIaGxt5/vnnKSoqoru7m8rKSpqbm/F6vbS0tMgdyU6nky+//JLe3l66urpwuVxUVlZSX1+Pw+GgsrKSqqoqXC4Xy5cv55NPPqGrq0se4bJarXIfl9Pp5PDhw/zrX//y8xEY2USLRRgQHo+H1tZWwsPDcblcvPjii2RkZFBVVcVVV13FM888w5133slnn33Go48+iiRJ7Ny5k87OTrq7u4mPjyc1NRWDwcChQ4eoqqoiOTmZ8vJyenp6SEpK4ssvvyQnJ4eqqioeeOABdDodOp2OOXPm8OGHH/r7EIxoosUiDAilUklGRgZarZbOzk527dolj6IBaLVapkyZgtlsxm63Aydmgtrtdo4cOcKll15Kbm4uW7dupaysDK/XS2trKyaTiaysLFwuF8ePH8fj8RAbG4vH4wGQ55K43W5cLpff9n+kEy0WYUAsWLCAjIwMMjIyOHDgAI888gjbt28nNjaW6OhodDodL7zwAmazmfj4eEaPHo1KpcLhcDB79mxeffVVkpKSGDduHAcPHiQzM5OsrCwCAwPZtGkTo0aN4uqrr2b//v1cdtll8rwRl8vF5s2bcbvd7N27l2nTpvn5SIxMYlRIuCh8E96USiVut5uvvvqKyy+//IwjEb5UFb6p5F6v94yJyM91NEPoH2JUSBh0FAqFPCSv1Wq5+uqrv7XsyaNvZ1tudiBGM4T+IfpYBEHodyKwCILQ70RgEQSh34nAcgbt7e00NTWd13OcTicHDhxg//79dHZ29rkp0e12c+DAAXbv3k1paSlutxtJkmhqauLYsWPyukQVFRUA8mMtLS0DdnOj71aEi6Wrq4va2lpcLhe1tbWD6qZNof+JwHIGGzdu7DPBSpIk6urq5NmfkiTR09NDRUUFzc3NtLe3U1VVxbZt2+jq6uL+++/vk8bSarXywAMPsHr1at5++21++tOf0trayocffsj1119PRUUF9fX1LF26FDgxZLpo0SJ+9atf4Xa7+9RNkiQaGhooLy+no6NDDkIVFRU4HA48Hg9NTU3yKExPTw9Op5OmpiYqKytpbGzE5XLx4Ycf8uWXX9LR0UF3dzcVFRVUVlbicDjk9+rs7KS5ubnPe3m9Xmpqaqiursbr9WK322lubqaqqqrPPns8Hqqrq6mpqcHr9XLkyBFeeeUVbDYbO3bskI9LZWUlLS0tOBwOJEmipaWF8vJybDbbgPxthYtDjAqdwakZ5D777DMOHTqEyWSiurqaxx57jMcff5wpU6ZQXl6OQqHgD3/4A6NGjaKtrY3u7u7TWgOJiYksXrwYhULBihUr+OCDD5AkiXvvvZdly5bx85//XH7O4cOHGTduHDabrc9cDEmS2Lx5M9u2bWP69OnU1taiUqlYt24d48aN46233uK+++7jV7/6Fa+//jrt7e3s3LmT1NRUfvWrX3HnnXfy8ccf86c//YnKykoMBgO5ubk8++yzzJ49G6fTyfTp04mPjwdgxYoVbN26ldtuu42tW7fyhz/8gQ0bNtDS0oJKpSIoKIj09HQefvhhfv7zn5Obm8u4cePweDw89dRThIeHY7VaSU5OJj4+Hq/Xi8PhoKSkhLq6Op555hmuvfZaXn/9dX71q1/R2trKunXrmDFjBitXruR3v/sdOp3uIv3Vhf4kWiznYM2aNSxcuJB7772Xuro6Nm3ahE6n4wc/+AHz5s2TZ3g2NTXxhz/8gbvvvvus4/wKhUK+ZR0gNTWVcePGyfe2eDwe3n//fWJjY0lLS+P999+XZ5W63W5WrlzJAw88wJw5c5gxYwarV6/mZz/7Gddddx21tbX09PTIM1k9Hg9utxuv10tGRgY//OEPSUlJwel0kpmZyZgxY+T1pSsqKk6bn+ByufjhD3/IrbfeSkREBNu2beP1119Hr9ej1WppbW3F7XYzc+ZM7rrrLsaNGwecyFJWXFzMf/zHf3D77bezdu3aPsfA6XSyf/9+FixYwOWXXy4Hnffee48HH3yQ6667jtLSUpG8fAgTgeUMFAoFTU1NFBcXc/z4cbKzs9m6dSuHDh3CaDSSnZ2NUqmksLCQffv2AdDc3Mxvf/tbrrjiCsaOHSsHAx9f2sUNGzawfv16br75ZuDE1PfbbruNb775BpvNRkVFBR0dHUyYMIHx48fT3d1NbW0tcGI+R15eHhs3bqSoqIj29nZGjx7Nxo0bKSwsxGKxEBAQgNFoZN++fXz11VdyAPNNIPP973K5KCgooLW1lfnz5zN79my2bt16Wia8Q4cOcfToUVpaWuTZrxkZGUyZMoWbbroJpVKJXq/vM0HNaDQSFhbGwYMH2bJlC9nZ2afVIS0tjbVr13LkyBEOHToEwPjx49m1axf79++nt7d3yKSiEE4nZt6eQUtLCyUlJahUKjQaDWlpaezdu5e6ujquuOIKYmJiKCwspKWlhcLCQiRJ4rbbbuPYsWNoNBo0Gg15eXlyM97lcskByOv1kpycTGxsLPX19Wi1WiIjI6mqqsJmsxEbG0tnZ6ecOKq8vJzIyEh5P7u7u9mzZw8lJSVcc801ckuioaGBqVOnkp6eTklJCVu2bGHKlCmYTCaMRiMNDQ2kp6dTVlZGdHQ0NpuN48ePk5iYiMVi4ZtvvmHSpElMmjRJ/kAvXbqUgIAAgoKCmDRpEllZWbS2trJ//35qa2uZOXMmJpOJtrY2OX/qycdw7dq1REZGMn36dCRJor29nbi4OOrr60lISGDLli14vV6WL1/Of//3f5OZmcm2bduoqalh8uTJZGVliRm1g4hIpj3AJEniww8/5Pjx4wQFBXHHHXf0ySEyXLz//vukpKQwceLEfn9tSZJ4+eWX6ezsJDMzk2uuuUbMoh3kRGC5CE4+bMP1W9W3jwOxf6eedsP1GA4n4l6h7+lcgsbJ2yVJQqFQyP9/12t/W5nv82E+9bnfN/gN5IddBJLhTXTensG+ffv4xz/+wbJly6irqwNOdL76RlvgxNIaLpcLl8vF008/jcVioaysTJ7s1tPTI+cX9f1cX18v97X41oY+eZ5Kb28vK1as4L/+679wOp196uR0OuXXgRMjRF1dXfLQeG9vLw0NDRQWFsqvX1xczBtvvMHTTz9NS0vLgB4zQTiZaLGcQWdnJ1dccQWVlZUcPnyYmpoaduzYgdvt5q677mL79u00NzdjtVr5yU9+wq5du7jnnntoaWnBZrPx6aefEhISwt13380XX3xBY2MjaWlpaLVa1q5di81mo76+noaGBgwGA3fffTd6vR69Xs/ChQv54Q9/2GceTE9PD59++inHjx/nhhtuwOVysWvXLpqbm7nmmmvo7e3lwIEDuN1uxowZQ15eHgqFgvT0dJxOJ2VlZTQ2NhIVFeXHoyqMJKLFcgYul4t//etfrF+/nujoaD766CMCAgIIDAzk6NGjNDY28qMf/YhRo0axZs0a4MSqg8XFxXzzzTfcfvvtPPjgg6hUKjo7O9HpdLzyyiskJCQwe/Zsxo0bxyeffCKPqPjWzFEqlWccYu3u7paXbFi/fj2lpaXEx8dzyy238Pbbb/Phhx9yyy23cMkll/S5xDg5VYGYaCZcTCKwnEFAQAA33XQTDz/8MJ999hmJiYkoFApiYmLIyMjAYrGwdu1aysvLmTRpEsHBwahUKvR6PSaTifXr1/P1119TWVlJQUEBYWFhBAUFERAQQEFBAcXFxSQnJ8vzUnzLWno8Hvbu3YvNZuPQoUNyq6WoqIiGhgbCw8MxGAwYDAaCg4PRaDTo9XrGjBnDxo0b2bdvn5y7RJIkDhw4QE1NzVnzmQjCQBGjQmfgdDpRqVQolUqsVit6vR6LxSKvGvfOO+9w7bXXotPp5HoEBQXhdDrRaDS0t7cDEB4ejsVikdfFCQ0NpaOjA7VajUajoaurS14wyrdSYUdHBy6XS35thUKBx+Ohra0NtVqNwWCQM6upVCpsNpucV1an02E0GtFqtUiSRGdnJ06nE7VaTXh4uOgwFb4XMdw8gCRJwmKxyB96QRgpxHDzAFIoFISGhvq7GoIwqIk+llNUVFScdy4WQRiO3G43DQ0NF5Q7RwSWU3z22Wfs3LnT39UQBL+QJAmbzUZZWRnFxcXo9foLep0Rcyl0plmoI2FaviCcC98kS1/6jKysLFJTUy/4czFiAktXVxcbN27EbDYzYcIEFAoFRUVFVFZWEhERweTJk0VwEUYcr9dLV1cXdXV1aLVa0tPT0el03/uz8L0uhZYsWYJCoWDRokXyNkmSeOKJJzCbzRgMBubMmSNPM/dxOBw88sgjREZGEhAQwIIFC+ScIwPFl1axo6NDzpVSW1vL3Llz2bVrFz09PWzatIkdO3agVIorRGHo8Hg8FBUVUVJSwqZNm2hra+P999/nhRde4Pnnn2fXrl20traybNkytm3bxptvvsk333zD3r172bp1KzU1NaSmppKenn5abp0LdcGfoD179vDyyy8zevToPtuXLl3KsmXLeOGFF9izZw8mk4nLL7+8TzawRYsW8dFHH7Fq1Sq2bt1Kd3c311577WnJkfqTWq1m0qRJNDU10dnZSWdnJwBVVVWEhYWh0WjIzc0lMzNTJHoWhhSn08m+ffsICwtj165d/OIXvyA4OJg333yTZ555hoceeojHH3+c559/nl//+tf87Gc/Y968eVx66aV4vV5yc3MvuC/lbC4osHR3d3PXXXfxyiuvEBYWJm+XJIm//e1v/O53v+Omm24iLy+P119/nd7eXt5++20ALBYLr732Gn/5y1+YN28e48aN48033+TIkSN89dVX/bNXZ+B0OrFarUybNo2goCBsNhvTpk3D7XZzxRVXoNPpiIyMJDw8XASWQcbj8dDa2tonwbYvcVRbWxter1cuY7FY/FjTi89ut+NwOLjhhhvkqRBVVVXs2bOHq666ittvv12+/SQtLY3U1FTuu+8+eTa2y+UakC6AC+pjeeihh7jmmmuYN28ef/7zn+XtFRUVNDY2Mn/+fHmbTqdj9uzZbN++nfvuu499+/bhcrn6lDGbzeTl5bF9+3auuOKK097P4XD0yR5vtVrPu85hYWF9gmBsbCwAOTk55/1awvcjSZKc3d9sNn9rgidJknjnnXfo7Oykq6uLX//616jVahobG1m7di11dXXMmjWLsrIynE4n9fX1PPjgg0RHR1/EPfKf2tpaDh06hFKpZMOGDfzyl7/k+PHjxMXFsWvXLq644grefPNNtmzZwsKFC1m9erUcoB9++GHmzJkzIPU678CyatUq9u/fz549e057rLGxEYCYmJg+22NiYqiqqpLLaLXaPh9yXxnf80+1ZMkS/vSnP51vVYVByuFw8NOf/pQrr7ySW2+9FY1GQ3FxMQcOHJDLXHbZZcTExOD1eqmsrOTXv/41TzzxhJyD2GQycffdd/PKK69gtVpRKBSUlZWNuMmLaWlppKWlAXDDDTcA8Pjjj7Nt2zZmzpxJYWEhDz30EDNnzmTDhg089thjrF+/nvHjx6PVagdswOK8AktNTQ2/+MUvWLdu3bdek51a2e+bAOnxxx/n0UcflX+3Wq0kJCScR82FwcTr9dLb28ttt92GVqsF+t7Z7bsXyqetrQ2XyyXfdNnR0YFer+fzzz9HrVZz+eWXs3z5cq688krKysrYsmWLnKx8uDs14ZjVaqWkpIS2tjYWLlyIy+Vi1apVXHbZZUybNo0JEyZw7733otPpaGtrG7AbVM8rsOzbt4/m5mYmTJggb/N4PGzevJkXXniBY8eOASdaJb5LDTiRwd7XijGZTDidTjo6Ovq0Wpqbm5k+ffoZ31en04nb/ocw36WPb8RBpVIxadKkPh+KxMTEPvliAgMDgRMBZ9KkSbz++uvExsai1+tZvXo106dPZ9euXSQlJVFRUcG4ceM4cOAAFouFSy+99KLvoz+53W5qa2tpaGggNjaW8ePHM3XqVPnx++67r0/5oKAgent7ee+993j44YcHJLicV+ft3LlzOXLkCAcPHpT/TZw4kbvuuouDBw+Smpoqpw3wcTqdbNq0SQ4aEyZMQKPR9CnT0NBAQUHBWQOLMDRJkoTb7aajo4NHH30Ul8uFx+NBq9Xy2GOPAcipIfR6vdwP5hulgxPfyLfccgs2mw23282OHTu45557yMrK4sYbb0Sn07Fhwways7OJi4sjICCAN954Y9h34vqmT9TX11NUVIRarWbixIkkJSXJrcBvYzAY+PnPfz44WixBQUHk5eX12RYQEEBERIS8fdGiRSxevJhRo0YxatQoFi9ejNFo5M477wQgJCSEe++9l1/+8pdEREQQHh7Or371K/Lz85k3b14/7ZYwGBQXF/Pss89y9dVXU1RUxLvvvktzczPXXXcd7777LtHR0SQmJnLllVcCsH37drmfTafTcfnll6PT6eRlXv/2t79x//33c9VVV2EwGJg2bRpTpkzh5Zdfpre3V14Xu7i4mJ6eHjnPzXDim3JfWlqK1+slLi5Ozhh4PgZ6Mmi/z7x97LHHsNlsPPjgg3R0dDBlyhTWrVtHUFCQXOavf/0rarWahQsXYrPZmDt3LitXrhQJiYaZ3t5egoKCmDlzJv/85z+58847+dGPfkRqairz5s3DbDazePFiObCc3Ldy8iRFpVKJRqPB4/HI2fSUSiVKpZJ//vOfREZGYjabKSsrY/78+Rw/frxPfuLhwm63U1NTg8PhIDY2loiIiEE7mfN7B5aNGzf2+V2hUPDEE0/wxBNPnPU5er2e559/nueff/77vr0wiAUGBpKSkoJarSY7OxuFQiHPp/j8888xGo089NBDcvmsrCx54TNfMIETI4ajRo3itdde4+6776a8vJyuri750mj27Nl0dHSQn5+PxWIhMTHRb8vC9Dev10trayt1dXUYjUYSEhIwGo3+rtZ3EomeTvHcc8+RlJTE9ddf36+vOxKdfGqdukTKyUuVnEuz3NcXc6YRx5O3n+/rDka+fejo6KC+vh6FQkFiYiKBgYF+3SeR6EkYFE5N7H2m/8/V2Zr8p77OUA0mPi6Xi7q6OiwWC+Hh4WRnZw/JLgIRWARhEPAlVWptbSU4OJicnJwhveSsCCyC4Ce+HCj19fU4HA7MZjPx8fHA0G95icAiCH7Q09NDTU0NVquV5ORkIiMjB+0Iz4UQgUUQLhKv14vFYqGhoQGNRkNycnK/JFUajERgEYQB5vV6aWtro6qqCrVaTUZGxpAYMv4+RGARhAHgm3Lf3NxMa2sr4eHhjB8/fkgPg5+PERNYent7KSkpITo6mtjYWBQKBR0dHZSXlxMZGUlSUpK/qygME06nk5qaGurq6khOTiYnJ2dIDhl/HyMmsNhsNhISEjh06BAxMTGoVCrq6+tpbm6We+LdbveApscUhi9JknA6ndTV1WGz2YiLi/teWe6HuuHTDf0dwsLC6OrqIj4+HqVSiSRJpKWlkZWVxYEDB5AkifLycqqqqkbsySCcP18OlL1793Lo0CEiIiLIyckhNDR0RJ9HI6bFUl9fj91uJzMzE7vdTlVVFUFBQbhcLjnrWEZGBqmpqSLnrfCdfAHFt2xGfn7+sB3huRAjJrD4LnfgRC6KrKwsufk6atQocUII58Tj8dDQ0EB5eTlms5n09PRzyn8y0oyYwHImCoVCZKYTvpMkSUiSRFNTE62trURERDB9+nRUKpX4QjqLER1YBOG72Gw2SkpKcDqdJCUlkZubO6xmyA4UEVgE4RS+S+SamhqcTicJCQmEhoaKgHIehkVg6ejowGaz4fV6aWxslDvSBOF8SJJEW1sb5eXlBAUFkZiYSEBAgL+rNSQN+cDS1dXF0qVLCQ0NpaSkhKioKEpKSrjjjjvE9a/wnXwjgBaLhdraWnQ6HTk5OQQEBIjz53sY8oGlu7tbzqF69OhRpk2bxpIlS7j99tvFiSF8K5fLRXV1NRaLhcjISLKysuS1jYTvZ8gfxaioKC699FKSk5NJSUnBbrfz4IMPiqAinJVvyLi9vZ3g4GDy8vLEkHE/G/KBRa1Wk56eLv+u1+vlhMyC4CNJEj09PVRXV+PxeDCbzeTn5wNDP6nSYDTkA4tPdXU1+/fv57LLLqO9vZ2kpCRxwgjAiRtQa2tr6e3tJTY2lqioKDHCM8CGxdG12Wy89tpr1NXV0dnZyauvvurvKgl+5ls2o6CggNraWhISEhg7diwxMTEiqFwEw6LFolKpUKvVNDQ0sHnzZjIzM/1dJcFPfEPGdXV1AKSnp2M0GkXr9SIbFoEFYMyYMcTFxeHxeAgODhYn0gjiWyO6qamJ9vZ2wsPDycvLQ6lUivPAT4ZFm/DAgQO8+uqrtLW1UV9fT0lJib+rJFwkviHjvXv34nQ6ycnJIT4+XtzH42fDosWSkpLCf/zHf5CamorX6yUmJsbfVRIGkCRJ2O126uvrsdlsxMfHk5iYKALJIDIsAktISAhHjhxh586dcurJW265RZxow4wkSXR3d1NRUUFPTw9ZWVkjPqHSYDUsAosvWZNvfZbNmzdzyy23nFbO7XajVCrlUQFJknC5XGg0GnFyDmJer5euri45qVJmZiZarVb8zQaxYRFYjEYjCxcuBGD79u389Kc/Pa1MR0cHe/bsQZIkZs2ahcFgoLi4mPb2diRJ4pJLLhEn6iDj8XhoaWmhtLSU8PBw0tLSxM2lQ8Sw6Lzt7e1l1apVhIWFMXPmTD7++OPTyhw8eJCYmBiio6Npbm4GTkyqmzhxIgcPHqSnp4djx45RVlYmAowf+ZbNaGho4OjRo3i9XqZNm0Z2drYIKkPIsGixaDQaWltbefXVV1EqlURHR59WJjo6GqvVSnd3NxERERQUFGAwGKivr8dsNqPT6UhPTyc5OVnkvPUTu91OeXk57e3tpKeni6RKQ9iQDyxerxev18sDDzzA5s2biYmJkReGOll2djbl5eUkJycTFhaG0+lk+vTplJSUcNlll6HRaABG3Pov/uZLqlRbWysvjJ6VlSUCyhA35AOLw+Hg0KFDdHZ2olQqaWlpobm5+bQbEZVKZZ+bFX0/5+TkXNT6CidIkkRHRwfFxcUEBgaSmpoqcqAMI0M+sAA0Nzfjcrn8XQ3hHEiShMVioa6uDr1ez9ixYzEYDCKgDDNDPrDodDouu+wyGhsbOXjwIAqFguDgYH9XSziF2+2murqa5uZmTCYTGRkZ8uWnMPwM+QtZpVKJWq3mzTffJDc3l+joaF566SXRATsISJKEx+Ohrq6OoqIitFot48ePJzk5WQSVYW7It1jgRAduZ2cn69ato6uri56eHr755hvmzJkjOmP9QJIkent7KS0tBSAuLo68vDxxuTOCDIvAotVqufHGG2ltbSU0NBRJkggJCREnsh/YbDZqa2txOp2YzWYiIiLECM8INCz+4h6PhyNHjlBeXk5mZiaHDh1iwoQJ4oS+SLxeL83NzRw8eJDq6mri4+PJzc0VmdpGsGHRYnG73bS0tMhLgIgRoovDN2RcV1eHSqUiLS2NwMBA0VIUhkdgMRqN3H333bz11lvs2bOH2267TZzcA8jlclFXV4fVaiUsLIycnByRVEnoY8gHFrfbzYoVK+SlMNvb2/F6vf6u1rDkcrloaGigtbWV4OBgsrOzxeiOcEZDPrC0tLSwY8cO/va3v/Hyyy/z0EMPiZvV+pEkSX06ZOPi4khISBCtE+FbDfnAYjQayc3N5bPPPkOv17N69WqmTJnSZ/q+cP586/DU1NRgtVpJTU0lMjJSBBThnAz5wBISEsIvf/lLf1dj2PB6vVgsFhoaGtBoNKSmpoqkSsJ5G/KBRegfvnV4Kisr0ev1ZGRkoNfr/V0tYYgSgWUE8yVVampqoq2tjYiICCZOnIhCoRAtFOF7EYFlhHI4HNTU1FBfX09qaio5OTni9geh34yYwNLe3k5DQwM6nY60tDQUCgVWq5Xq6mrCwsKIi4vzdxUHnCRJOByOPstm+I6FIPSnETPfWq/XEx4eTmFhoXznc3V1NTU1NXKZlpYW2traht0HzZcDZc+ePRQUFBAZGUlOTo64n0oYMMO+xdLZ2UlxcTHZ2dkUFxcTHx+PQqFAkiRSU1PR6/UUFRVhNpspKiqitLSUiRMn+rva/UKSJKxWK7W1tej1esaMGSNGeISLYtgHlqCgIEaPHk1DQwOlpaXk5ubS1dVFfX09kiTR0NAgJ9+eNWsWBw8eHPIzd91uN/X19VRVVWE2m0VSJeGiG/aBRaVSYTQaSUtLIy0tTd7uyzKXnZ3tr6r1K98IT3NzM62trURERDBt2jTU6mH/JxYGIXHWDXG+Kfe+u7qTk5NFUiXB70RgGaJ8Izy1tbW4XC4SExMJDQ0V+U+EQUEEliHG6/XS1tZGRUUFQUFBJCYmEhAQ4O9qCUIfIrAMEZIk0dnZSV1dHTqdjpycHLEOjzBoicAyyLlcLqqrq7FYLERGRpKdnS1myAqDnggsg5Bv2YyGhgY6OzsJCgoiPz9fDBkLQ4YILIOILwdKdXU1Ho+HuLg44uPjAcQljzCkiMAyCPjW4amtrcVms2E2m4mKihLBRBiyRGDxI6/XS3t7O42NjWi1WpKSktDpdCKgCEPeeU96qKur4wc/+AEREREYjUbGjh3Lvn375MclSeKJJ57AbDZjMBiYM2cOhYWFfV7D4XDwyCOPEBkZSUBAAAsWLKC2tvb7780Q4fV6aWlp4fDhw9TV1ZGamionVhJBRRgOziuwdHR0cMkll6DRaPjyyy85evQof/nLXwgNDZXLLF26lGXLlvHCCy+wZ88eTCYTl19+OV1dXXKZRYsW8dFHH7Fq1Sq2bt1Kd3c31157LR6Pp992bLCRJAm3201tbS0FBQU4nU5Gjx7N6NGjMRqN/q6eIPQv6Tz85je/kWbMmHHWx71er2QymaSnnnpK3ma326WQkBBp+fLlkiRJUmdnp6TRaKRVq1bJZerq6iSlUimtWbPmnOphsVgkQLJYLOdT/XPy7LPPSh9//HG/vqbD4ZDKy8ulbdu2SRUVFZLb7e7X1xeEi+F8Pnfn1WL59NNPmThxIrfeeivR0dGMGzeOV155RX68oqKCxsZG5s+fL2/T6XTMnj2b7du3A7Bv3z5cLlefMmazmby8PLnMqRwOB1artc+/wU769z08ZWVlHD9+nPDwcKZNm0ZycrKYhyIMe+cVWMrLy/n73//OqFGjWLt2Lffffz8///nPeeONNwBobGwEICYmps/zYmJi5Md8HZVhYWFnLXOqJUuWEBISIv9LSEg4n2pfVNK/c6AcOnSIgwcPEhERIZIqCSPOeY0Keb1eJk6cyOLFiwEYN24chYWF/P3vf+dHP/qRXO7UD5AkSd/5ofq2Mo8//jiPPvqo/LvVah10wUX6d5a2+vp6dDod2dnZIqmSMGKdV4slNjaWnJycPtuys7Oprq4GwGQyAZzW8mhubpZbMSaTCafTSUdHx1nLnEqn0xEcHNzn3/nyXZrY7XY5NaUkSdjt9u/VaezxeKivr2fr1q00NTWRnp5OWlqaGDYWRrTzCiyXXHIJx44d67OtpKSEpKQkAFJSUjCZTKxfv15+3Ol0smnTJqZPnw7AhAkT0Gg0fco0NDRQUFAglxkInZ2dfPnll5SUlMjbysvL2b17N3v37j2vrHHSSVPujx49ikKhYPr06WRkZKDVagei+oIwpJxXYPnP//xPdu7cyeLFiyktLeXtt9+W10uGE5dAixYtYvHixXz00UcUFBTw4x//GKPRyJ133gmcWLnw3nvv5Ze//CVff/01Bw4c4Ac/+AH5+fnMmzev33fQ4XDQ1NSEQqEgKCgIq9WK2+0GoLS0lKlTp7J79256e3spLS2loqLirC0NXwunsLCQnTt3olQqyc3NJTY2FpVKJVoogvBv59XHMmnSJD766CMef/xxnnzySVJSUvjb3/7GXXfdJZd57LHHsNlsPPjgg3R0dDBlyhTWrVtHUFCQXOavf/0rarWahQsXYrPZmDt3LitXrhyQ0RKXy0VLSwvh4eFMnjyZTz75hLS0NFwuFyqVirKyMiIjI9HpdCQmJhIfHy9fKvlI/06qVFdXh9PpJCEhgZycHJFUSRDOQiGd+ikaAqxWKyEhIVgslnPub3E6nTQ2NhIcHExQUBB2ux2tVktjYyNhYWEEBgYC8Nxzz5GUlMT111+PJEm0t7dTUlJCYGAgKSkpcjlBGGnO53M3Yu4V0mq1JCYmyr/7sq6dbXTJl1TJt2yGwWAQlzqCcI5GTGA5V3FxcXz66afU1dURHx+PSqWiuLi4Txm32z3k+lS8Xu+QunTzeDwoFIohVWe32z2kVkXwXax823ms0WiYNWsWer3+vF576ByFi+T6669nzpw5p/WznGzlypXMmDGD9PT0i1izCydJEps2bWLOnDn+rso5+/rrr1Gr1cyePdvfVTknvb29rFy5kgceeGDIfOGUlpbicrm+dQkchUJxQSOdI6aP5fuSJEkONlarFaPRiEajweVy4fF4BuWdyZIk0d3djV6vx+l0EhAQ0Gc/BltrQPp3oiuj0YjNZgNOLI1rs9kGZX5fX6e+QqFAo9FgsVgIDQ3t0xIYjHX21c/tdiNJEmq1+pyOsehjGSDvvPMOERERTJ8+Ha1Wi91uZ/v27URFRWEymYiKivJ3Fftoamqis7OT8vJy+d6so0ePUlBQQF5eHrm5uX6u4f+RJIktW7YAYLfbmTdvHl6vl71796LRaDAYDINucTmPx0NzczOHDx9m4sSJmEwmmpub2bhxIxkZGYwZM8bfVTxNcXExhw8fJicnh/z8fCRJYvv27cTExOD1esnIyOiX9xlcX1mDVE1NDU1NTSQnJxMXFycvJO9wODAajURERNDa2urnWv6fnp4eDh8+TEBAANHR0TidTvmbKCwsjNGjR9PW1ubnWp6upqaG8ePH09XVhdvtxu1243Q6SUxMpKKi4lsvT/1BrVYTEhKCzWaT8xFrtVqysrKoqanB5XL5uYanCw0NZfTo0bS3t8vb7HY7iYmJNDU19dv7iMByDkJDQwkKCiI5ORmbzYbVaqWqqgqAqqoqCgsLB9WMW61WS2xsLD09PWzevJmcnBxsNhubN2+W69nd3e3nWp7ObDZTVlaGTqejvLycyspKent7KS4uJjExcdBdVjidTt544w0SEhLQarUcOnRIvqTweDyDMr+Q7/617u5uWltb6ezspLOzk7179/Zr4BZ9LOdIkiQaGxtxuVzExsbidDoxGAy0tbXhcrkwmUyDrs+iu7ublpYWtFot0dHRdHR0oFarsVqtREVFDbqFzny3SURHR+NwOIAT/RQdHR2YzeZBl27C6/VSW1uL1+uVW4a+m1F92REHWzBsb2+X//5qtRqFQoHD4aC9vZ3Y2Nhv/YI8n8+dCCyC30mShNPplH8+36FN4eIQnbeC3/kSgNlsNhYuXCh/o6tUKtRqNW63G4/Hg1qtxuPx8MknnxAbG4vNZmPevHk4nU6USiVqtVoOOlqtli+//JLY2FjGjBkjp/vUaDRya8bhcPDxxx8TFBTE1Vdf7bf9H+lEYBEGxPbt27nrrrvYtGkTR44coaamBqfTSUtLCw888ABPPvkkkyZNoqOjgx/+8IfY7Xa6urro6uri888/p7q6mvT0dMaOHcs333xDY2MjEydO5IsvviA4OBiVSsWnn36K2WzG5XJx3333yXMurrrqKlavXu3vQzCiDa5OAWHYkCSJqqoqiouLiYqK4p///Cd6vZ7AwEAaGhpwuVzcfvvtaLVaWlpagBNN7dbWVj7++GPuuecerrjiChwOB729vSiVSioqKkhKSmL27NlYrVYOHDiAWq2Wy8DgnDsyEokWizAgjEYj8+bNY9KkSXzzzTdcc801mM1mQkND5TSkn376KfX19URGRqJWq1Gr1RgMBi6//HK++eYbzGYzDoeDnp4eeTLXqFGjaGxsZOzYsUycOJHk5GS6u7vllQ7cbjf79u2jvLycY8eOkZmZ6ecjMTKJzlthQLS0tBAaGoparaarq4uAgADq6+uBE3NpXnzxRe644w7Cw8MJCAigu7sbhUKB1+slKCiIhoYGFAoFJpOJ+vp6AgIC0Gg0GI1GGhsbCQwMRKPR0NDQQEBAgLxypMfjoa6uDrfbTWBgINHR0X4+EsOHGBUSBjWv10tTUxMmk0lctgwhYlRIGNSUSiWxsbH+roYwgETnrSAI/U4EFkEQ+p0ILP3It6zJEOy2GnAul4uenh68Xi89PT3iGA1zIrCcwd69e/n666/7bPPdaXvqmkS+3/ft28eyZcv49a9/zauvvtrnztaenh6eeOIJ3njjDV577TXee+897HY7X3/9NX/4wx/o6uqipaWF119/XX7tlStX8vzzz5/xA+jxeLDb7fKSJR6PB4fD8Z0fVqfTKa9QcODAAXkpFF9eEZfLdcZE4ie/F5wIEifvn29Kvu+1z7TtwIEDvPDCC3R3d/PFF18AJzpxTz6m57MvwuAmOm/PoKqqisbGRubOnQucyLS1fPlyent7uf7667n88st57733+Pzzz8nLy2PixIlMnTqV0aNHo1AouOOOO7jlllvkZWQdDgdHjx7l97//PV6vl6effpp169ZRU1NDTU0Nr7/+OldffTXbt2/n7rvvprm5mYKCAvmmPLPZDPzfpLPXX3+doqIiHn30UeLj41m5ciWlpaUsWLCAyy+/nM8//5yFCxfS29tLdXU14eHh/O///i91dXWEh4fzyCOP8N5776HT6bjppps4cuQIa9asITU1lUWLFsn13rJlC1u2bKGwsJBJkybx0EMPcfjwYd5++20kSeL2228nNjaWTz75hGPHjnHDDTcwd+5cOcfHu+++iyRJ3HPPPbhcLrq7u1GpVERERGC321m5ciXFxcVkZGRw4403otFoWLFiBQUFBVxzzTXcfPPNQyrVo/B/RIvlHLz22mtcf/31LF26lHfffZfNmzezbds2Xn75ZTIyMti0aROBgYEolUrWrl3LjBkz+ix34uPLNDZp0iSOHj0KwJVXXklVVRUFBQXAieDxz3/+U07Es3r1avnb2+v18uKLL3LnnXfy+uuvM378eN566y2uuuoqXnjhBT766CNaW1v54IMP5Gxsx44do7m5maKiIpYuXUpPTw81NTVMnDiRWbNmkZ2dzdatW5k8eTJ33HFHn3rv2LGD2NhYXn31Verq6tixYwfPPvss48aNY9y4cXz++efU1tZy5MgRli5dKqe+bGtr45VXXuHPf/4zP/rRj+T7huBECsdNmzaxe/duDAYDS5cupaysjMbGRv7yl78wf/58li9fzocffojFYhmoP6kwwMTXwVk4HA66urpQKBTYbDY5S7/H40GlUslNfd/t/b29vSxduhSNRsODDz54WgoFj8dDd3c3brebL7/8kuuvv56jR4+i0Wh45JFHeOihh4iMjKSpqYmvv/6akJAQ4ETqgzvuuIOIiAgAVCqVfCOfx+PB7XajVCpRKpVy8mm1Wo3dbuf48eNyisewsDCMRqM8b0SpVGK1WvF6vfy///f/KCgo4P/7//4/Fi1aRF5enlxv342DSqUSr9dLYGAg06ZNk1MwHj9+nMjISAwGg/wcX1lAzk1yKpVKhdPpxOl0yo+ffHxFS2VoE3+9MwgLC2P58uVs27aNoKAgHnzwQdasWcNXX33FtddeyyWXXEJbWxs///nPiYiIICgoiB07drB27Vqio6MpLS3lmWeeITIyEjjxIXK5XPzkJz9h+vTpXHrppcyYMUOenZqQkMAPf/hDjhw5wr59+/jZz37GjTfeCMDHH3/Mnj17uPLKK1EqlTzyyCO8+OKLJCYmMm7cOH7yk5/wyiuvoNfrufbaa4mNjeW6667jv/7rv8jPzyc5ORmDwSAvcxIfH4/RaCQ7O5u//vWv8s1/cCLRUnh4eJ9jsWPHDrq7u0lISGD69OlERESwYsUKEhMTiY6OJjU1Vb5UO/n43Xzzzbzwwgt4PB4eeeQR3G43ZrMZtVpNbGwskyZN4vjx4/zhD3+gtrYWhULB/fffz8qVK1m3bh033XQToaGhA/yXFgaKmHl7Bl6vt8+3qEqlwuv1IkmS3FpZv349VquVwsJCbr31VrKzs097jq914Fvr2fd8X2vG962uVCqRJEn+/eRlL3zbfWkBfL/7Xsv3LX/y7750Ar7n+KbKq1QqPB7Pae/v+/nkugEsX76czMxMZs6c2ee1T32/k+vn4yt38r74liDx/f/FF19gtVo5duwYjz76KMHBwae9tjB4iCn9A8w3UuJbpvXkS4zh5OScKP1NkiR6e3vlFQ4GU2pP4czElP4BplAoMBgMffoVhqOB/LArFIpBlxpT6D9iVOgMSkpKeOutt/jXv/4l5/k4G4/HwwcffEB3d/d3ZpJ3Op309PSc9XG3280333zD6tWr5TuBz0dnZ6ec5FuSJIqLi3nnnXdYv369nIVNEC4G0WI5g+rqaiZNmsTBgwfZunUreXl5HDhwAIBZs2ZRUVEhJ6meMGECb731FpdddhkOh4PW1lb27NmDWq1mxowZHDhwgM7OTlJTU2lpaWHTpk0sWLAAgLq6OoxGo9yHIUkS2dnZOBwO3nrrLX79618DJ9I7bt26FbvdzsSJE3E4HBQVFeHxeJg4cSIul4tDhw7hdrsJCQkhKSkJgKioKKZMmcIzzzxDdHT0oFznRhieRIvlDFwuF1988QUbNmzAbDbz8ssvU19fT1FREVu3bmXdunUkJSVRVFTEhg0bAOjq6mLHjh288847BAYGMn78eNRqNb29vXR0dLB48WJCQkJITU0lNjaWZcuWYbPZ+Ne//iWv56LRaIiMjGTt2rV9FufyeDzYbDZaWlr44IMP2L59O7W1tYSGhrJixQpWrFhBQkICkiTJM2J9c2aWLl1KdXU1MTExF/9ACiOWCCxnoNVqufnmm7n33nvZunUrKpWKgIAApk2bRl5eHk6nU75E0mg08qiHQqFAp9PhdDpxuVwUFRWxatUqAgMD6erqQqfTYbFY6OrqwmAwEBYWxjXXXCPPdHW73bz00ku0t7czduxYuT579+6ltLSU4OBgnE4nWq2W5ORkwsPDaWhokOfH+ObUAPIyFIsXL2bOnDk0NjZevAMojHhiVOgMOjs7MRgMaLVaqqqqCAwMZM2aNajVaubOnctbb71FcnIyGo2GuXPnUllZSUpKChaLBYPBwLp16+jp6eGGG25gzZo1KBQKsrOzSU1NpbCwUJ7fsnbtWpKSkpg7dy5arRa3282hQ4fwer1ERUWRnJwMnLgUeuutt0hKSiI/P19O46jVamlsbCQ4OJh169aRkpJCVlYWYWFhSJJEYWEhO3bsIDc3l8mTJ4tJZ8L3IoabB5AkSVRUVJCcnDzoFigThIEkhpsHkEKhIDU11d/VEIRBTXzlnsJqtX7nELMgjASSJGGz2fqkwzhXIrCc4q233pJHegRhJJIkiba2Ng4ePEhlZeUFLW4vLoVO4XK5LihCC8JQ5/V66ezslJdeSU9PP2P6j3MxYgJLb28vu3fvJjQ0lDFjxqBQKLDb7Rw6dIjExESxFIUwIvluaq2rq6Ojo4PQ0FAyMzO/902gIyaw2O122tvb5VmpcCJD2rhx41i7di233HKLPBFNEIaCnp4eNBoNWq0Wl8uF3W4nMDBQDgh2ux2Hw0F3dzdBQUHodDo6OjqA/8t9Y7VasdlsBAUFkZub229TEkZMYAkODmbu3LmsXbuW4OBggoODcbvd6PV6urq68Hg8tLS00NbWJlouwqAnSRJr1qwhOzubnJwcWltbWbt2LT/4wQ9Qq9U0NDTw+9//noKCAjo7Oxk9ejTh4eF88MEH6PV63G43LpeLRx55hD/+8Y/9fs6PmMDS2dlJS0sL8fHx8sS3/Px8jhw5wvTp0zEYDCQlJREfHy8SOQ9CXq/3tAXfT85nMxLddNNN8s8xMTFs2LCB7du388wzz6BWq7nrrrv46KOPKC8vZ//+/bS2tmKxWOS8Onq9npiYmAH5Ih0xgSUiIoLw8HD55AwKCkKSJMxm82knrDB4SJLEvn37eP/995k0aRI33XQTSqWS1tZWXn75Zbq7u7nnnntQqVRs376d6dOnk5KS4u9qDziFQoHVakWr1aLX61EoFPzud79j165dfPPNN0ycOJHS0lJaW1vJyclBr9cTFRXFl19+idfrpba2FmDAplaMmMBypuAhAop/OBwOfv/73zN27FhuvvlmdDod1dXVlJaWAif+LmPHjiUsLAyv18sHH3zAf/7nf/L8888zZcoUEhISCA0N5ac//Snr16/n+PHjXHbZZdTW1nL48OEREVgkSeLrr78mKyuL7OxsioqKMJlM3H777WzcuBGtVsuCBQtITk6ms7OTtrY2HnzwQaZOnUpVVRU7duygt7d3wM7/kdmGFPzK4/FQUlLCTTfdhE6nA07kqunq6qKrqwur1dpn7kRcXByRkZE4HA46OzsBUKvVtLW10djYSG1tLVqtdtgn3jqZQqHghhtuICsrCzhxqe9wOFAoFBiNRt544w0cDgfJyck0NjZyzz338MILL9DS0sK9997L1KlTeeONN7j//vsHpH4jpsUi+M/JfVa+VqLJZOrTNxISEiK3NBQKBXq9Xv65vb2df/7znwQGBhIbG8u+fftITEyksLCQ8PBwNBoNFouFyspKurq6mD179ohIxH1yTuWJEyfKrb6kpCQeffRR+fj6Vo2Ii4vjyy+/xGQyER0dzZYtW/jBD34wMHUTNyH29dxzz5GUlMT111/fr687knV3d/Pss8/y+OOPy4nDHQ4HOp3unJriDoeDTZs2YbfbSUpKIisrC7VaTUFBAWVlZWi1WmbMmEFVVRVlZWWEhYUxe/bsYd+p68u93NTURG9vL7GxsYSEhJzTfvvy9mg0mnN+P3ETojAo+FZtHDNmDJ999hlRUVGEhIRw6aWXsmrVKpRKJQsWLCAxMRFJkjh27Jh8qaPRaBg9ejQajYaOjg4++ugjfvOb3/DMM8+wbNkyVCoVaWlphIaG8umnnxIfH8+bb77J/fffzyuvvEJycvKw7WuRJAmr1SovkZuRkUFSUtJ59ZecT0C5EMM7pAt+1draitVqZebMmcTExPDDH/6QdevWsW3bNhITE5k3bx5PPfWUfKnU2dlJU1MTTU1NtLS0yNvdbjdGo5G4uDgaGhrkb9uAgABKS0upqanB6XRSWVlJdHS03F8z3PiSdxUXF9Pa2iov7xsSEjLoBiFEi0UYMBqNhuDgYFQqFdHR0Wi1WjQaDREREWzcuJHGxkYmTJgglz+5X0Sj0cgfloCAAGw2G++88w7Tpk2ju7ubzs5Oent7aWlpISQkhLq6Oi655BLef/99nE6nnJVvOPCt4V1RUUFMTAxpaWmDfrkU0cdyCtHH0n986TINBgM9PT0EBgbS3d2N0WikoaGB6upqJkyYII8MfZuOjg4OHDjAlClT0Gg08oJoW7ZsISEhgczMTJxOJ7t37yYrK4vo6OhB9y1+vtxuN62trbS1tREWFkZkZKRfA4roYxEGBV8KTUC+S9b3f3x8PPHx8ef8WmFhYVx22WWnbb/iiivknw0GA7Nnz/4+VfY7X8f28ePH6e3tJSkpiZycnCEXJEVgEYRBwDfCU19fj8vlIjY2lvDw8CE7siUCiyD4kSRJdHR0UFZWhsFgIDk5mYCAgCHXQjmVCCyC4AeSJMlJlbRaLVlZWRecVGkwEoFFEC4ij8dDTU0Nra2tREZGkpGR8b2TKg1GIrAIwkXgdrtpbGyks7OTgIAAxowZM+CT1PxJBBZBGCC+DtnKykq5QzY3N3fYtU7OZMQEFkmS6OnpQa/Xy01PSZKQJEmkTxD6lSRJ9Pb2UldXJ9/DMxzm1ZyPERNY6urq6Onpoby8nCuuuAKFQkF5eTkHDhwgNze3zyLsgnAhfMtmNDU1oVarSUhIwGAwjKiA4jNiAkt8fDwtLS0cOXJE3hYaGkp2djZtbW1IkkRxcTHHjx+X10wWhHPh9Xppa2ujrq4OhULBqFGjMBqN/q6WX42YwNLS0sKWLVvIz8+nq6uLwsJCRo0ahVqtpqenB4CsrCxGjRolct4K58TXIetbNiM/Px+lUjkiWyinGjGBRavVMmbMGLRaLTqdjrS0NODEzW4zZ84UJ4NwTiRJwuVyUVdXR2Njo9whO1RnyA6UERNYQkJCCAkJkX/3ZSiLioryV5WEIcS3jrEvqZLZbCYpKUkElLM4r6Pidrv57//+b1JSUjAYDKSmpvLkk0/KyzDAiT/AE088gdlsxmAwMGfOHAoLC/u8jsPh4JFHHiEyMpKAgAAWLFggZw0XhMHEl1Tp8OHDHD58mNDQUHJycggLCxNB5Vuc15F5+umnWb58OS+88AJFRUUsXbqUZ555hueff14us3TpUpYtW8YLL7zAnj17MJlMXH755X0S7yxatIiPPvqIVatWsXXrVrq7u7n22msvaPFpQRgIvin3x44do7m5mczMTKZMmUJYWJi4bD4X0nm45pprpHvuuafPtptuukn6wQ9+IEmSJHm9XslkMklPPfWU/LjdbpdCQkKk5cuXS5IkSZ2dnZJGo5FWrVoll6mrq5OUSqW0Zs2ac6qHxWKRAMlisfTZ7vV6JY/Hcz67dJpnn31W+vjjj7/XawhDl9vtlurr66UtW7ZIR48elex2u7+rNGic7XN3JufVxzJjxgyWL19OSUkJGRkZHDp0iK1bt/K3v/0NgIqKChobG5k/f778HJ1Ox+zZs9m+fTv33Xcf+/btw+Vy9SljNpvJy8tj+/btffJr+DgcDhwOh/y71WqVf7bb7bz++usYjUZaWlowGo3k5uYyY8YM8c0inDO3201bWxutra2EhYXJCaWEC3Nel0K/+c1vuOOOO8jKykKj0TBu3DgWLVrEHXfcAUBjYyNwYrnHk8XExMiPNTY2otVqT0sdeHKZUy1ZskTufA0JCSEhIUF+rK2tjYCAAMaOHUtgYCC33nor77//vhgyFr6T9O+kSkVFRezcuROv10t2djZms1kEle/pvFos7777Lm+++SZvv/02ubm5HDx4kEWLFmE2m7n77rvlcqe2FKR/T5v/Nt9W5vHHH+fRRx+Vf7darXJwCQwMJDg4mNzcXHJycrDb7dxyyy2itSKclfTve3gaGhpwOByYTCYyMzNFZ2w/Oq/A8utf/5rf/va33H777QDk5+dTVVXFkiVLuPvuuzGZTADy+L5Pc3Oz3IoxmUw4nU46Ojr6tFqam5uZPn36Gd9Xp9OdNS9qSEgICxYskH8PCAhg1qxZ57Nbwggh/btDtqSkRJ7LFBgYKL6EBsB5heje3t7TorpKpZKHm1NSUjCZTKxfv15+3Ol0smnTJjloTJgwAY1G06dMQ0MDBQUFZw0s58Lj8VBbW4vH4+kz/C0IvoBSXFxMR0cHubm5jBkzhqCgIBFUBsh5tViuu+46/ud//ofExERyc3M5cOAAy5Yt45577gFOXAItWrSIxYsXM2rUKEaNGsXixYsxGo3ceeedwIkWxr333ssvf/lLIiIiCA8P51e/+hX5+fnMmzfvgnbC6XTyt7/9jaamJn75y1+yZs0afvKTn4iTZoTzJVWqr6/HZDKRnp6OWq0W58VFcF6B5fnnn+f3v/89Dz74IM3NzZjNZu677z7+8Ic/yGUee+wxbDYbDz74IB0dHUyZMoV169b1Sbv317/+FbVazcKFC7HZbMydO5eVK1eiUqkuaCfcbjeSJJGXl0d5eTnl5eUX9DrC8OB2u2lubqajo4Pg4GAmTpw46NfhGW6GxbpCkiRx+PBhSkpKUKvVzJo1i4iIiAt6bbGu0NDk65AtKyvD6XQSHx9PVFSUaJ30o/NZV2hYdIOXl5fzxhtvkJ+fT1JSEgcOHPB3lYSLRPp3UqXS0lIqKiqIjo5m3LhxIy6x0mAzLG5CtNvtxMbG0t7eDkBubq6fayQMNOnfSZVqa2vR6/UkJCRgNBpFMBkkhkVgycrKYv/+/WzdulWe9GQymcRJNgx5vV46OjpobGyUkyoFBAT4u1rCKYZFYOnt7aW+vp7U1FRyc3N5//33mTNnjr+rJfQjt9tNXV0dnZ2dhIWFkZmZOSyXzRguhkVgMRqNTJo0CbPZzM6dO5k2bdppZRwOB4cOHUKpVDJ+/HiUSiXNzc2UlpYSGxtLSkqKH2oufBeXy0V9fT1tbW2EhoaSl5d3waOHwsUzLDpvnU4nDQ0NZGRkcPPNN/e5YdHn4MGDAFgsFpqbmwHYv38/6enpfP3119hsNsrKyqiqqhLfgn4m/XtFhdLSUkpKSggKCmLcuHGkpqaKoDJEDIvA4vV6KSkpwWq1UllZecZRIZfLhUajQa/X43a75dm5BoMBl8uFJEl0dXXR3d19sasv/JskSXR3d1NUVERBQQFhYWHk5OQQHh4ugv0QM+QvhSRJwmg0cvPNN/Phhx8C8LOf/ey0cuPHj2fXrl0olUrCwsI4fvw42dnZ7Nq1i9mzZ2MwGBg7diy5ubnizuiLTJIk2tvbaW5uRq1Wk5KSgl6vF8FkCBvygcVms7Fr1y7a29tRq0/sTldXV5+bIOFEP8yll14q/56ZmQlAUlLSxaus0IfX66WlpYWKigoCAwNJT0+XcxELQ9uQDyxqtZqgoCAMBoO8LTQ01H8VEr6Ty+WitbWV9vZ2wsLCmDhxovylIAwPQ/6vqVAoCAkJwe12U1VVhUajEfeFDEKSJOF0OqmurqahoYHU1FSys7NFDpRhasgHFo1GQ1JSEi+99BJ1dXUolUqampq44447xDX6ICCdtGyGb4Z0WlqaCCjD3JAPLPB/K9I9+uijtLW18dlnn2Gz2UbsurmDgfTvZTNKSkpQKpWkp6cTHBws/h4jxLAILBqNhtzcXL766iu8Xi8mk4ktW7Ywb948Me/hIvN6vXR1dVFfX49OpyMvL0+M8IxAw6I9qlKpCAwMpKSkhMsvvxyTycT8+fNFULmIfEmVtm7dSktLC2lpafLCdiKojDzDIrD09vayb98+srOzsdvtbNiwQcxFuUjcbjcNDQ0UFxejVquZOnUq6enpaLVaEVBGsGFxKWQ0GpkwYQJbtmyhsrKSW2+9VZzUA8h3B/nx48fp7e0lOTlZjPAIfQz5wCJJEocOHaKtrY3rrruO/fv3i7teB4gvS1tdXR1ut5vY2FgiIiLEsRZOM+S/Ytra2nj++eeZMWMGb7zxBrfccgtjx471d7WGFV9SpT179lBeXi6vwxMZGSmCinBGQz6w+K7x33rrLVwuF6+99hr79+/3d7WGBa/XS3t7O0VFRbS3t5OdnU1ubq5Yi0f4TkP+UigqKooVK1b06awNCQnxY42GPo/HQ3V1NW1tbURGRpKRkSEuL4XzMuQDi0qlkldgFL4fl8tFY2MjnZ2dBAUFMXbsWHEPj3BBxFkzwvmy3FdXV8sdsnFxcWKER/heRGAZoXxZ2urq6rDZbMTFxYnOWKHfjJjA4vF46OjowGg0YjQagRMpLTs7OzEYDH1WahzOJEmitbWV5uZmNBoNCQkJYnas0O9GTGBpamqitbWVtrY2Zs+ejVKppLS0lKqqKnJzc4d9YPF6vbS2tlJXV4darSY9Pb1PDhtB6E/DPrD09PRQX19PfHw8AQEBOJ1O+bGYmBhaWlqorKwkISGB4uJijh8/TnJysv8q3M9cLhdNTU1YLBZCQkLIz88XHbLCgBv2Z5hSqUSn09HY2EhzczMTJkzA6XRSU1NDUFAQmZmZcgb/rKws0tPTh/x9RpIk4XK5qK2tpbGxkbi4ODHlXriohn1gMRgMJCYm4vF4SExMRKVSoVKpSElJQZIkOjo6mDt3rtzHMJT7GnwjPM3NzdhsNmJjY0lOThYBRbjohn1g8Tk5hYJCoZAvB6Kjo/1VpX7jW7qkrKwMl8tFRkYGycnJQzpICkPbiAksw5EkSXR2dtLU1IRGoyE7OxudTicCiuB3IrAMQR6Ph8bGRsrKyoiOjiY1NRWNRiMCijBoiMAyhJy8bEZERATTpk1Do9H4u1qCcBoRWAY537IZZWVldHR0iGUzhCFBBJZBypdUqb6+HpfLhdlsJisrSwQUYUgQgWWQ8XXIHjt2DKPRSEpKish/Igw5IrAMEl6vF4vFQkNDAwaDgby8PAIDA/1dLUG4ICKw+Jlv2YyGhgZMJhOjRo1CrVaLFoowpInA4ie+1Rt99/BMnDhRjPAIw4YILBeRbx3j0tJSPB4P8fHxYoRHGJZEYLkIfAGltrYWl8uFyWQiKipKXO4Iw5YILAPIl1SptrYWo9EokioJI4YILAPA6/XS1tZGY2MjKpWKzMxMOWudIIwEIrD0I7fbTW1tLVarlbCwMLKzs0VSJWFEGtFn/ckJnS708kSSJNxuN3V1dbS1tREeHk5ubm6fNA2CMNKMmMDicDgoLCwkICCAjIwMFAoFbW1tVFVVERISQlpa2nkFF1+W+8bGRhwOB7GxsSQlJYn+E0FgGCyx+l28Xi9Op5Pu7m5qa2ux2WxyS2X37t0kJyezYcMG7Ha7nL7y24KDL6lSYWEhRUVFhIeHk5OTQ3h4uAgqgvBvw77FYrFYKCgoICcnh/nz5/P+++9jMpnQ6XSoVCo0Go3cD6LT6dDpdGfMeetbGL21tVXOci+SKgnCmQ37wBIWFsbMmTOpqqri0KFDjB07Fr1ez6FDh5g2bRoHDx5kzpw56PV6DAbDaes+e71empubqaysJCgoiPT0dLRarQgogvAthn1g8UlKSiIpKUn+ffbs2QDMmjXrjOWdTidtbW20t7cTHh7OxIkTxQiPIJwj8Uk5hUKhYPv27dTW1pKamkpsbCzNzc00Nzf7u2qCcNH55mGd731sIrCc4qabbmL37t3AiVZLVVXVaWXWrl1LXl4ecXFxF7t6F0SSJPkycKjYv38/KpWKMWPG+Lsq58Rut7N27VoWLFgwZC6T6+rqcLvdfVryp9JoNKSlpZ13YFFIQ3B1LqvVSkhICBaLheDg4Iv+/r4k1kNlWVZJkmhqasJkMvm7KuesoaEBpVJJTEyMv6tyTtxuN+Xl5YwaNWrIBJauri68Xu9p/Ypncz6fO9FiOUeSJLF+/Xq0Wi3jx48nKCgIj8fDwYMHaWtrY9q0aYMu0DQ2NlJUVIRKpeKSSy4BoLKykmPHjhEfH09ubq6fa/h/JEmioqJCDtqjR49GkiS2bt2Kx+Nh7Nix5/wBuFjcbje7d++ms7OTGTNmkJGRQUdHB7t378ZkMjF69OhBF2SqqqooKioiPj6evLw8JEni2LFj1NTUkJOTg9ls7pf3GfbzWPqDJElIkkR9fT1wIjkTnFgX2ul0kp6eTk1NjT+r2IevvjExMUyZMoWWlhZ5CN1iseBwOAZdEATYt28fkyZN4ujRo7jdbpxOJ729vWRnZ7N///5Bt/StSqViypQpKJVKLBYLcGIipt1ux263+7l2Z2a1Wk/7+9fU1DBt2jSOHz/eb+8jAss5KC4upqqqihtuuIGUlBRKSkpwuVwoFAo8Hg9Wq3VQ5VSxWCxs3LiR+vp63n33Xbl5brPZSE9PZ9q0aZSVlfm7mqfRaDR0dXUBJ5Y6cbvd9Pb20tvbS0BAgJ9rdzqv18uBAwdISEjAZDLhcDgICQlh+vTpHDt2jJ6eHn9X8TRpaWlMnz6dsrIyPB4PXq8XoN/PYXEpdA6ys7PlJmNPTw+5ubl0d3cTFBSEyWTCYrGQmprq72rKQkNDufTSS2lpaSE1NRWlUonL5aKhoQG1Wk1zczPZ2dn+rmYfCoWC6dOnU1lZybRp0+jt7QVOHPuOjo5BeVkhSRIej4fe3l66u7ux2+14vV4aGhoYN27coAyGbW1tNDU1kZ2djd1uR6lUkpOTQ21tLaNHj+639xGdt4LfnXoKDrYAIpwgOm8Fv/v666/p6upCpVJx2WWXfeu3t8vlYsOGDQQFBWGz2Zg7d+5Zyx46dIjw8HASEhLO+HhdXR379u3D6/UyefLkfuuMFM6PCCzCgHjrrbf4xS9+weeff45er8dkMnHw4EGio6OZPn067733Hmq1mtDQUObPn09TUxN6vR6n00lpaSlr1qwhNDSUBQsW8Mknn2CxWLj11ltZvnw5gYGB3HvvvTQ0NFBTU0N2djYTJ05EoVDIfRylpaWsXr2ahx9+2N+HYkQaPD2OwrDidrv55ptvaGxsJCsri6eeeopRo0axd+9eDh06xJYtW7jhhhsoLi6moaEBODF3pbKykhdffJEbb7yRG264AbfbTVdXF01NTaxdu5aUlBTmzp2Lw+Hg/fffJzs7m7ffflvu9A0ICECj0fDFF18wf/58fx6CEU20WIQBodFouO666ygoKGDNmjVoNBqOHz/O6NGjSU5Oxu1209raisViOW00Qq/XU1tbi0qlYsuWLRgMBiZNmkRraytGo5GWlhaio6OxWCzU1tZyzTXXoNfrgRNzd1555RWmTp1KeHi4P3ZdQHTeCgPk4MGDZGRkoFQqKSsrIyoqioKCAnQ6HXl5eTzzzDPMnTuXoKAgxo4dK8+0dbvdhISEcODAAUJCQkhKSuLw4cMEBQVhNpsJCAhg//79ZGZmYrFYaGpqIjw8nPz8fBQKBU1NTRw6dAitVktMTMygG/0ays7ncycCi3DRud1ujhw5wtixY8UI0BAiRoWEQU2tVjNu3Dh/V0MYQEMysPgaWVar1c81EYSRw/d5O5eLnCEZWHwjAGebyyAIwsDp6ur6zhtCh2Qfi9fr5dixY+Tk5FBTUyP6Wc7CarWSkJAgjtF3EMfp3PgSyZvN5u+8r2hItliUSqWcZCk4OFicDN9BHKNzI47TdzvX1BVigpwgCP1OBBZBEPrdkA0sOp2OP/7xj+h0On9XZdASx+jciOPU/4Zk560gCIPbkG2xCIIweInAIghCvxOBRRCEficCiyAI/U4EFkEQ+t2QDCwvvfQSKSkp6PV6JkyYwJYtW/xdpYtmyZIlTJo0iaCgIKKjo7nhhhs4duxYnzKSJPHEE09gNpsxGAzMmTOHwsLCPmUcDgePPPIIkZGRBAQEsGDBAmpray/mrlxUS5YsQaFQsGjRInmbOE4DSBpiVq1aJWk0GumVV16Rjh49Kv3iF7+QAgICpKqqKn9X7aK44oorpBUrVkgFBQXSwYMHpWuuuUZKTEyUuru75TJPPfWUFBQUJH344YfSkSNHpNtuu02KjY2VrFarXOb++++X4uLipPXr10v79++XLr30UmnMmDGS2+32x24NqN27d0vJycnS6NGjpV/84hfydnGcBs6QCyyTJ0+W7r///j7bsrKypN/+9rd+qpF/NTc3S4C0adMmSZIkyev1SiaTSXrqqafkMna7XQoJCZGWL18uSZIkdXZ2ShqNRlq1apVcpq6uTlIqldKaNWsu7g4MsK6uLmnUqFHS+vXrpdmzZ8uBRRyngTWkLoWcTif79u07LUny/Pnz2b59u59q5V++pT19+V0rKipobGzsc4x0Oh2zZ8+Wj9G+fftwuVx9ypjNZvLy8obdcXzooYe45pprmDdvXp/t4jgNrCF1d3Nraysej4eYmJg+22NiYmhsbPRTrfxHkiQeffRRZsyYQV5eHoB8HM50jKqqquQyWq2WsLCw08oMp+O4atUq9u/fz549e057TByngTWkAovPqXlSJUkakblTH374YQ4fPszWrVtPe+xCjtFwOo41NTX84he/YN26dXIG/zMZ6cdpoAypS6HIyEhUKtVp3xbNzc2nffMMd4888giffvopGzZsID4+Xt5uMpkAvvUYmUwmnE4nHR0dZy0z1O3bt4/m5mYmTJiAWq1GrVazadMmnnvuOdRqtbyfI/04DZQhFVi0Wi0TJkxg/fr1fbavX7+e6dOn+6lWF5ckSTz88MOsXr2ab775hpSUlD6Pp6SkYDKZ+hwjp9PJpk2b5GM0YcIENBpNnzINDQ0UFBQMm+M4d+5cjhw5wsGDB+V/EydO5K677uLgwYOkpqaK4zSQ/NhxfEF8w82vvfaadPToUWnRokVSQECAVFlZ6e+qXRQPPPCAFBISIm3cuFFqaGiQ//X29splnnrqKSkkJERavXq1dOTIEemOO+444zBqfHy89NVXX0n79++XLrvssmE/jHryqJAkieM0kIZcYJEkSXrxxRelpKQkSavVSuPHj5eHWkcC4Iz/VqxYIZfxer3SH//4R8lkMkk6nU6aNWuWdOTIkT6vY7PZpIcfflgKDw+XDAaDdO2110rV1dUXeW8urlMDizhOA0fkYxEEod8NqT4WQRCGBhFYBEHodyKwCILQ70RgEQSh34nAIghCvxOBRRCEficCiyAI/U4EFkEQ+p0ILIIg9DsRWARB6HcisAiC0O/+f9kTBZl0MzdyAAAAAElFTkSuQmCC",
            "text/plain": [
              "<Figure size 640x480 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "dna_pairwise = cv2.imread(f'{results_dir}/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_DNA_pairwise_minThreshold.png')\n",
        "\n",
        "dna_pairwise_resize = cv2.resize(dna_pairwise, (600, 1000))\n",
        "\n",
        "plt.imshow(dna_pairwise_resize)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAARYAAAGiCAYAAAAiFuksAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/TGe4hAAAACXBIWXMAAA9hAAAPYQGoP6dpAACES0lEQVR4nOzdd3gc1dn4/e+utmnVe++9uRdcwAYbgymmF0MogeSFUJ44kMaThIc4iU0gIRBqCAkOwdj04oAbYNy7JdmSZcnqXau2RdL2nfcPZ+dn2QYsI1ntfK5Ll63d2d0zo5l7z5xyH4UkSRKCIAiDSDncBRAEYewRgUUQhEEnAosgCINOBBZBEAadCCyCIAw6EVgEQRh0IrAIgjDoRGARBGHQicAiCMKgE4FFEIRBN6yB5aWXXiIlJQWdTsfUqVPZvn37cBZHEIRBMmyB5e2332bZsmX86le/orCwkAsvvJDFixdTX18/XEUSBGGQKIZrEuLMmTOZMmUKL7/8svxYTk4O1157LStXrhyOIgmCMEhUw/GhDoeDgwcP8stf/rLf44sWLWLXrl2nbW+327Hb7fLvHo+Hrq4uwsLCUCgUQ15eQRBAkiQsFguxsbEold98szMsgaWjowO3201UVFS/x6OiomhtbT1t+5UrV/Lb3/72fBVPEIRv0NDQQHx8/DduMyyBxevU2oYkSWesgTz22GM88sgj8u8mk4nExEQaGhoIDAwc9HLV1NRgtVrJzc0969ccP36c4uJiNBoN/v7+zJ8/X47qNpuNdevWodfrUSgUXHTRRfj5+bFu3Try8/NJTU2luLiY0NBQEhIS6O7uZtOmTSxZsgRfX99B3z+z2UxRUREXXXTRoL/31/nggw+49NJL+fe//83tt99OUFDQeftsYXCYzWYSEhIICAj41m2HJbCEh4fj4+NzWu3EYDCcVosB0Gq1aLXa0x4PDAwcksBSWVlJa2srF1xwAXDi1m3v3r24XC7S0tJISEigsrKSxsZG/Pz8CAwMJDw8nAsuuIDAwEDuu+8+5s+fL5fN5XLx6aef8ve//53nn38epVLJDTfcwD/+8Q/y8vJ4+umn2b17NxMmTCA3N5e3336bTZs2kZiYyKJFi+RgK0kSFRUVNDc34+/vz+TJk6mrq6O+vh6NRsPMmTPp6urC398fX19fDh06RG5uLmVlZbhcLnp7e5k4cSItLS28/PLLBAYGkp6eTnl5OQqFgtTUVBITE4ETt5t79uzBarUCMGXKFIKCgigqKsJsNhMWFkZ+fj4lJSW0tbURFxdHTk6OfLwOHjyIzWYjPDyc/Px8TCYTer2eOXPmEBERQV9fH0VFRQQGBuLxeJg2bRoVFRW0tLTg4+PDBRdcgEajGfS/rfDdnU3zw7D0Cmk0GqZOncrmzZv7Pb5582Zmz549HEX6Rq+99hrNzc3ExsaycuVKysvLefXVV0lMTKSwsJDVq1eTlJRETk4O3d3dZGdno9Pp+r2H3W7nyJEjNDY2kpmZCUBYWBhJSUl88skneNvQjUYju3fv5ve//z3vv/8+DodDfo+amhpeeuklkpKSiImJobW1lWeeeYaEhAQqKip48803OXToEAaDAYA//elPdHR08Nvf/haPx4OPjw///Oc/8fX1JSAggOTkZDZv3szBgweJjY3tV4twu9389re/RafT0d7ezuuvv862bdv49NNPiY+P56233qKkpIS///3v1NfXExERIb9248aNbN68mYSEBJ599tl+XyAvvfQSnZ2d/Pa3vyUwMBCn08njjz9Oa2srK1asIDY2ll27drFp06bB/0MK582wdTc/8sgjvPbaa/zzn/+krKyMn/zkJ9TX13P//fcPV5G+VklJCZMmTSIzMxOn00llZSVut5uUlBTy8/OBE7WJI0eO8M477/Cb3/zmjDWsoKAg4uLi5FoAwB133MG2bdtobGwEYNOmTSgUCo4cOUJbWxtfffWVHHTa2trw9/cnJSWFuLg4TCYTSqWStLQ0CgoKOHr0KJIkYbPZ+n1uWFgYkydPZtKkSVRUVODr64ufnx8JCQncdNNN6HQ61q1bx86dOzm5kzA8PJzJkyezcOFCenp6qKiooK2tjb179xIVFYVCocDHx4dp06YRHh4uv669vZ1p06aRlpZGZmYmZWVl/crjcrlobGxkwoQJZGRkoNPp6OvrIzIykszMTC699FJ27tz5Hf9qwnAatjaWW265hc7OTpYvX05LSwv5+fl89tlnJCUlDVeRZBqNhpqaGjZv3oxarWbx4sWsW7eOjIwM4uPjueiii/jyyy/5xz/+QW1tLUqlkj179vD73/+e73//+xQVFXHBBRfItRaFQkFUVBQTJ04kOjqalStXkp+fL9+y3Hfffdx+++1cccUV7Nmzh//93/8lMjKS3Nxc/vnPf7Jw4UJ8fHzIzMzE7XazYcMG9Ho9ubm5BAQE8Omnn1JcXMx1112Hx+Ph3XffZfLkyVgsFpRKJX5+figUChQKBXq9Hj8/P1QqFV988QV9fX2kp6fT3NxMS0tLv3autrY23n77bSoqKsjLy2PGjBnU19cTGRlJQEAA6enp+Pr64uPj0+/4zZgxgzVr1tDb20tZWRkPPvggFRUVKBQK/Pz88PX1Zf78+fzrX/+ivr4ei8VCREQEGo2Gjz/+mE8++YSHH374vP/dhcEzbONYvguz2UxQUBAmk2lI2lisVitGoxE4ERTCw8Pp6urC7XYTHByMTqejrKyM48ePs3//fubOnctFF12EyWQCQKlUyu1IcKK9wmQyERwcjEKhwGg04u/vj8ViISgoCIVCQUdHB/7+/thsNnk7j8eD0WgkJCREvth7enqwWCyoVCrCwsKw2WyYTCZUKhXh4eFIkkR7ezsKhQKVSkVQUJD8OZIkyceuu7sbu92OTqfDZrPJ+6lSnfiucTqdPPTQQzz22GPodDr5ue7ubmw2G0qlkoiICCwWC3q9HrVa3e8Yerfz9/cnICCA3t5efH195b9ZfX09hw8fpq2tDZvNxkMPPYTdbsdoNKJWq8VQghFoINedCCznQJIkWlpaMJlMaDQakpOTT/vWHu28+xgdHf2tYxbOhcFgoLOzE6VSSUpKimioHQVEYBEEYdAN5LoTs5vPwNs129vbi9vt/sZtPR4PRUVFWK1Wmpqa+KY47fF4cLlcX/u8JEn09fVhtVq/8X2+jtVqpaurS34vm81GRUUFFovlnN5PEM6VCCxnsHPnTtasWcOf//xn9u7di9vtpqKigqqqKtxuNx0dHVRWVlJXV4fT6WT58uWYzWYqKysBqKqq4ujRozgcDiorKykrK8PhcHD48GHefPNNGhsb6evro7S0VG4whRPjPzZu3MgPf/jDft3MHo+H6upqysrKsNlsGI1G6uvrOXbsGA6HA7fbzbFjxygsLOTw4cPy6/bu3cvWrVv5v//7P2pra8/rMRTGt2EdeTtS9fb2YjAYcLlcuFwu3nnnHaqqqvD392fChAkcPHiQmTNn8tVXX3HVVVchSRJ2u53a2locDgelpaVcdNFFuN1uamtraWlp4fPPP2f69OkYjUasViu///3vSU9Pp6amhoceeoioqCi0Wi2LFy/mjTfe6FfD6O3tpbi4mK6uLvbs2YOvry8lJSXk5+dTVFSE0+mUP2vmzJnAiUbnefPmMW3aNJYvX35a46ogDCVRYzkDpVJJZGQkKpUKo9FIWVkZ6enp5ObmkpqaSkBAADNnziQrK0uuIUiShMfjoaKigiuuuILJkyfT3NzMli1b8PX1ZfPmzQQHB5OUlERcXBwGg4HY2Fjmzp2Lv7//N5anpaUFi8WCr68vXV1d+Pj4cOGFF1JQUMDOnTspLi7mkksu4aKLLpJ7dSRJoq2tjTfffJPrr7+euLi4IT9uguAlAssZhIWFsWTJEn7xi19QU1PDddddR3V1NfX19bS1tWGxWORazBVXXEFGRgZarZbw8HCuvvpq3nnnHd5++215TIjZbCY9PZ2IiAhKSko4dOgQV199NcePH6eyslIeMOdyudiyZQs6nY6dO3fi8XgA8PX1xWg00tTURHR0NKGhoQQHB+Pr60tSUhJ33nkna9asYc+ePf3mcfztb3/DZrNRWVlJR0fHsBxLYXwSvUJncPIh8QYH70UO8O6773L99dejUqnk55RK5WnbKpVK+f/eAWoej6ff/73bKRQKudbjdabHva/9uvJ923sJwrkayHUn2ljO4OQL0Pt/7zgVSZK44YYb5KBy8nOn/n7q/7/tOe8Q+TOV5+vGyZzpM7/tNYIw1ERgGSCFQiEaQgXhW4g2llPs2bOHY8eODXcxBGFYeTwempub+w1fGAgRWE6xb98+ysvLh7sYgjAs3G43LS0tHD16FIVCIefYGahxcyvkHWuiUCjQaDRyg2d3dzcBAQFiroowrtntdiorK7FYLCQnJ5OTk/Od2ujGTWAxGo189tlnZGdnM2XKFAB27NiBv78/tbW1XHPNNXJviiCMVJIk9UttcXIP4Jl6Jr29iN7HT369QqHAbrfT2NiI0+kkOjqa3NzcQek9HDeBRafTMWPGDAoLC8nMzJRTFMyaNYvt27djtVppbGykqqqKlJSU4S6uIPQLBp2dnbhcLoxGI4cOHcLf35+IiAguuOACnE4nDz/8MLGxsTQ1NREfH8+WLVvo7e3lggsuICIigoqKCqZPn47FYmHLli3MnDmTadOmkZGRQVJSEv7+/oM6HGHcBJbu7m56e3sJCAigp6eHvXv3Ehsby6FDh5g4cSJ6vZ7s7GzS09NFrUU4LwwGA263m5iYGPkxq9WKRqPB5XLxpz/9icsvv5xjx46h1Wr59NNPue666wgICODIkSMcPXqUF198kaVLl1JUVMSbb75JYGBgv1Sg+/fvl///zjvvEBoaisFgIDAwkC+++IJ169adVXLsgRo3gSUkJASNRkNWVhYajYaAgAD8/PzkpEtDkXNE+O4kSaK1tZWmpiZiY2OJiYlBoVBgsVioqKjA4/EQHx9PYGAgx44dIyAggIyMjFExGLCrqwuFQiEHlra2Nh555BFWrlxJdHQ0DQ0NNDU1MXfuXPbt20dcXBydnZ0UFRVRXl4ur1JRVVXFpEmTOHz4ME6nEwCVSoUkSf1m5zscDlpbW/Hx8WHnzp1oNBpWrVrFz372s0Hft3ETWHx9ffstpeGdnxMSEjJcRRq3nE4n7733HpGRkVx44YVoNJoz1hK9twErV67k5ptv5vnnn+eXv/wlQUFBKJVKtFotq1evZtq0afT09KDX66moqODyyy9n6tSpw7BnA2O1WvuNiQoLC+OBBx7g8OHD6HQ6rr76an7yk5+waNEiJElCp9ORlJSE3W5Ho9HQ0dFBd3c3H374Ibt378btdtPZ2QmcWNli4sSJ7Nu3D7VajdVqJSAgAEmSiImJISgoqN9KFINNfE0L553T6WTVqlXk5ubKF9bBgwf5y1/+Iv94k4tLkkRaWhoXXHABDodDTv/g5+dHUlISHR0dLFy4kO7ubmbNmkV2djZ79+4drl0bkO7ubnp6eoAT3bxms5mcnBx6e3tZvnw5l1xyCRs3buSnP/0pLpeLf/3rX/z5z3+W8wR5PB7S0tJob2/H5XL1m8LR29vLwYMH8fPzQ6vVolAo5CTqdXV1dHV1YbVa+6XnGEwisAjDQq/X98trGxoaSlZWlvyj1+uBE7WWhoYGqqurCQwMJCwsTB7AuHfvXubPn09QUBA+Pj4UFxdjsViYPHnysO3XQFxyySVyjaGjo4PLL7+cgwcPMmHCBODEIm/33HMP//M//8Pdd9/ND37wAwwGA/v27aOurk6ufXR3dxMcHIxSqcTHx0dO0u7xePj5z39Odna2nAvZZrORlZVFXFwcS5Ys6bdixGAaN7dCwvBxu900NzcTHx8vT4lYunRpv3ESqamppKamnvZahULBI488wnvvvUdOTg5KpZLKykqys7MJCQkhMjKS8vJy7rrrLj788EPMZjOpqalfu6rmSHBqly+cuHVJT08nLi6OtLQ0fv3rX6PRaPjzn//M4cOH+elPf0piYiKLFy/m8ssv54UXXiA+Pp7S0lKSkpLo6+sjISFBrr3k5OTQ0NDAX//6VzmIG41G5s+fT2ZmJq+99pq8tMtQEDUWYci0t7ezfv16tm3bxrJly1i/fj2VlZUolUri4uJYt24dZrP5G99DoVAQGBiI0WjE19eXVatWcfnllwMnltp99913qaurQ61W09HRQUREBM8++6zc1jBSSJLE/v37aW1tZdWqVfzf//0fhYWFbNq0idLSUvz8/Ljzzjv56quvaGtr47333uPNN9/kD3/4A0uXLmX+/PkUFhayf/9+/vSnP3HxxRfT0dFBYGAgtbW1XH/99RiNRjweDx0dHRQVFREZGYmfnx+1tbVERERwxRVX4Ha7Wbp0KbGxsSxcuPCM618NBlFjEYZMfX09q1ev5g9/+AM9PT2kp6fz5JNPcscdd3Dw4EGmTp3Km2++yY9+9CMAPv30U+rq6oATje1Lly6Vc9F4bxVWrVqF3W5HpVKxbds2fHx82LVrF+Hh4ezatYvXX3+dffv20dzc3G8RteEmSRIOhwOFQsHNN9+M0+mUlxk+ePAgqampTJ06lYCAAKKiorj11ltRqVS4XC50Oh2JiYnk5uaSm5vLunXrcDgczJgxg+bmZv74xz/yv//7vzgcDjweDzqdjrS0NJRKJRqNhquuugqj0Uh3dzfXXnstzz33HCEhIaet1jmYRGARhlRMTIy8xnZaWhq9vb1yzSI0NLTfvKyQkBDsdjtw4tbAOwRArVbT29tLV1cXwcHB9Pb2YrPZCA0NJSEhAZvNxp49ewgKCqKrqwun0zmkF825UCqVzJkzB4/HI3ed+/j4sHjxYnkb7/pNtbW1/POf/2T58uV4PB7mz5/PO++8w6uvvsp1113HnDlz0Ol0fPzxxyQlJXHw4EH6+vp44okncLlcHDhwgLvvvpsdO3awfft2Dh8+TFhYGLfeeiuff/45eXl5TJkyhR07drBgwYIhuWUUgUUYMt7uZK1Wy5VXXolCoWDx4sXMmjWLPXv2sGvXLn75y1/KJ/acOXO+9n3uvvtu3n//fZYtWyb3ptx77728/fbbKJVK7rjjDi6//HLWr1/PlVdeSXp6+vnc1bNmt9tZt24dt912m7ww3aliYmK49957qaqq4ne/+x0PPvgg+fn5vP/++4SGhrJ582bq6urIycnB4XCwePFilEolx48fJyAggClTplBVVYXVaiU0NBSlUsmiRYu47777mDBhAs8//zyLFy9m/vz5Q9YOJTLIneKvf/0rSUlJXHPNNYP6voJwMu/yLCfXzLzcbjcWi4WysjLMZjNTpkwhODgYj8dDVVUVf/3rX7nxxhspKiri+uuvJyoqCp1OJ6/Y8NVXX3H8+HGWLFnCk08+SXp6Oo888ggVFRWo1WoiIyPR6/UDnngrMsgJwijw6quvyjmT4UQOFLPZTGFhIUePHsXtdnPTTTcRERFBcXEx+/bt49Zbb+Xxxx8nKChI7tF5+eWXmTdvHrm5uQAsXLiQ1tZWzGazvDSNSqVCq9XidDp5++23ueSSS0hPTx+yGosILIIwTB5++GF5dHFbWxtVVVUEBQUxa9Ys5s2bB5zoFXO73dTV1XHzzTfLa2HD/+u2jo+PJzo6Wn7f+vp6NBoNkyZNora2ll27dpGeni7Pg8vKymLTpk34+fkRGxs7JPsmAosgDANvwGhvb6ezs5Pw8HBmzpwpL9/i9c4775CcnExaWhp6vf60Gsbzzz9PXFxcv27j9PR0AgICWLNmDTExMXz/+9/v97l9fX10d3cP6XQWMY5FEM4zh8NBeXk5u3fvxuPxkJOTQ3R09GlBBeCKK65g69at9PT0nDHX8ve//32uv/56/Pz85MeUSiU9PT0kJiZ+7VwgtVrNl19+OWQz+UWNRRDOA+84lsbGRhwOBzExMWRmZn5rG4efnx8//elPz/icd/DgmWRkZMhtN2d6z0WLFtHc3DywnRgAEVgEYYgZjUbKy8vRarXySppn22g6kMZVl8uFj4/PWb3GO7ZoqIjAIghDQJIkTCYTzc3N6HQ6CgoK8PX1HbJeGEmSePPNN5kzZ87X1lTOp3ETWJqamjh+/DgKhYILL7wQpVJJc3MzpaWlJCYmkpWVNdxFFMYAj8cjJ2iKjo4mPT0dtVo95BMivVMFhmruz0CNm8ASGxtLSEgIW7ZskR+zWCw4HA50Oh2SJFFYWMiRI0dITk4evoIKo443mXVLS4vc2zJjxoyzvi0ZLN5UEyPBuAksfX19vPfee8ycOVMe9ZiQkEBISAhFRUUkJiYSEhJCUFDQcBdVGEWsVivHjx/H4/EQGxtLXl6eSHPKOAosZrOZxMREHA4HNpsNg8EAnJjan5eXh0KhICUlhcTERJFMW/hG3i+mhoYGHA4HsbGx/ZJWCeMosMTExPTLhu5d4kMs9SGcLW8WtubmZnx9fYmPjz/joDVhHAUWQThXkiTR1dVFc3MzKpWKzMzMIe3hGQtEYBGEr+F0OmlsbKSnp4fg4GCys7PPOPpVOJ0ILIJwEu9aPE1NTXR0dBASEiKvYyxqKGdv1AcWSZI4evQoxcXFLF68GF9fX1pbW0WXsTBgFouFtrY2HA4H0dHRJCQkiB6eczTqj1pXVxdr1qwhICCAZ555htLSUg4fPjzcxRJGCUmS6O3t5ejRoxw7dozg4GBycnLkzGvCuRn1NRbvok1XXXUVF154Ie+8844YiyJ8K4/HQ3d3N+3t7ahUKlJSUtDpdOJ2Z5CM+sASHh7OHXfcAUBwcDD33HMPNpttmEsljFTeLuPa2lq0Wi2ZmZkioAyBUR9YFApFvzwWKpVKXpdZELwcDgednZ10dnYSFhbGlClTzpj/RBgcY+Im0u128/nnn/Pqq6/S2dnJtm3bxOhZATjRZVxZWcnevXtxOp3k5OQQExMjgsoQGxNH12q1cujQIZKTk+np6eGDDz5g7ty5ono7TnmH3Dc3N8tD7tPS0sT5cB6NicCi0+mIiYnh3XffZffu3dx0003iJBqnzGYzFRUVeDweMjMzCQoKEufCMBgTgaWvrw+r1cpzzz2H3W6nq6tLnEzjiHfZDO8cnry8PNEgO8zGRGB5/fXX+dvf/sbu3buRJImHHnpouIsknAfeHCjV1dVERUWRlpaGRqMRAWUEGBOB5Uc/+hEFBQVUV1fjdDoJDg4e7iIJQ8jtdtPa2kp3dzdhYWHMmjVLDLkfYcZMr9D27dux2+3ExMSwd+9e0Ss0Btntdo4cOcKBAwdQqVT9enhEUBlZxkSNxcfHh/z8fNRqNTt37iQ+Pv60bSRJoqWlBV9fX3kxbm9+0vDw8H7rsggjhyRJ2O12GhoacLlcxMXFERISIgLJCDcmAotSqSQyMpLZs2eTnZ1NT0/PadtUVVXR0tJCX18fM2fOJDg4mP379+Pv709JSQmXXXYZPj4+w1B64Uy8OVAqKysJCAggMTERPz8/EVBGiTFxK2Sz2di6dSuSJBEYGMjrr79+2q1QfX09QUFBREdHYzQaATCZTGRmZlJVVYXNZuPYsWNyJn9heHgDSmlpKV1dXRQUFJCTk4O/v7/4u4wiY6LGotfrmTFjBqtWrcJsNnPXXXeddhLm5eVx8OBBXC4Xs2bN4tixY4SFhXHgwAGys7PR6/Xk5OSQkZEh2meGgdvtpra2FpPJRHh4OFlZWSKp0ig26gOL2Wzm008/paWlRa6JKBSK0wJLZGQkl1xyCSqVCh8fH8LDw4ETo3Z1Op2YIj8MvMtmNDU10d3dTXBwMBMmTBA9PGPAqA8sarWa6Oho9Ho9drsdgJCQkNO2UygU6HS60x4fSWuxjCc9PT3U19fj8XiIiYkhPj5eBPcxZNQHFpvNxtq1a+nt7ZUfi4iIIDU1dRhLJZyJJElYrVbq6+uxWq0kJCSIhEpj1KgPLMHBwbz00kt8/PHH9PX14XK5SEtLQ5IkUZ0eITweD+3t7XR2dqJWq0lMTBRZ7se4Uf9VoVAosNlsVFVVcfPNN3PxxRfzyiuvAIhG2GEkSZKcVOnw4cO0tbWRnJxMenq6WItnHBj1NRY40c6i0Wh46623sNls+Pn58dFHH3H11VeLvBvDwOFwYDAYMBqNhIaGkp+fL/4O48yAaiwrV65k+vTpBAQEEBkZybXXXkt5eXm/bSRJ4oknniA2NhZfX1/mz59PaWlpv23sdjsPP/ywPOJ1yZIlNDY2nvNOqNVqrrrqKhQKBd/73ve47777uPbaa8XJfJ65XC5qamo4dOgQLpeL7OxsYmNjxd9hHBpQYNm6dSsPPvgge/bsYfPmzbhcLhYtWtSv4fSpp57imWee4YUXXmD//v1ER0dz6aWXYrFY5G2WLVvGhx9+yNq1a9mxYwc9PT1cddVVuN3uc9qJvr4+Xn/9dSwWC11dXfzhD38Qt0HniTfLfXV1NRUVFYSEhDBz5kySk5NFQBnPpO/AYDBIgLR161ZJkiTJ4/FI0dHR0pNPPilvY7PZpKCgIOmVV16RJEmSjEajpFarpbVr18rbNDU1SUqlUtqwYcNZfa7JZJIAyWQySZIkSQ6HQ3rhhRekJ554Qnr++eel9957T/J4POe0T88995z00UcfndNrxxuz2SwVFxdLu3btkjo7O8/5mAujw6nX3Tf5Tl8pJpMJgNDQUABqampobW1l0aJF8jZarZZ58+axa9cu7rvvPg4ePIjT6ey3TWxsLPn5+ezatYvLLrvstM+x2+3yGBU4MSju1HIUFBSQnJxMSUkJ4eHhonFwiHg8HkwmE62trXKWe61WK4630M859wpJksQjjzzC3Llzyc/PB6C1tRWAqKiofttGRUXJz7W2tqLRaE4bxHbyNqdauXIlQUFB8k9CQoL8nMVi4be//S379+/nqaeeoqenh6ysrHPdLeFrSP+dHb5r1y6am5tJSUkRa/EIX+ucA8tDDz3E4cOHWbNmzWnPnXqiSWcxpuSbtnnssccwmUzyT0NDg/yc2WympqaGadOmERwcTGxsLFar9Rz2SDgTl8tFc3MzpaWlKBQKLrjgAnJzc0VAEb7ROd0KPfzww3zyySds27atX+6T6Oho4EStJCYmRn7cYDDItZjo6GgcDgfd3d39ai0Gg4HZs2ef8fO0Wi1arfaMz4WGhnL//ffT09PD9OnTMZlMOByOc9kt4SQOh4Oqqio6OjpIS0uTF0YXhLMxoMAiSRIPP/wwH374IV999RUpKSn9nk9JSSE6OprNmzczefJk4MQJunXrVv74xz8CMHXqVNRqNZs3b+bmm28GoKWlhZKSEp566qkB74Cvry9XXXXVgF8nnE76b1KlxsZGnE4nMTExZGdni5qJMGADCiwPPvggb731Fh9//DEBAQFym0hQUJA8RHvZsmWsWLGCjIwMMjIyWLFiBXq9nttuu03e9t577+XRRx8lLCyM0NBQfvrTn1JQUMDChQsHfw+FbyVJEkajkfLycnx9fUlNTRX5T4TvZECB5eWXXwZg/vz5/R5//fXXufvuuwH4+c9/jtVq5YEHHqC7u5uZM2eyadMmAgIC5O3/8pe/oFKpuPnmm7FarSxYsIBVq1aJqvZ55g0o3mUzJk6cKNpOhEGhkKTRN5LMbDYTFBSEyWQiMDDwrF/n8XjOmKvlZH/9619JSkrimmuuGYyijkhut5v6+nq5LSw2NhaNRjPcxRJGuIFcd+NmaGRzczPt7e1YrVZmzpyJQqGgra2No0ePEh8fT0ZGxnAXcUhJ/02q1NzcLM/hmT59ukiqJAyJMR9YnE4nvb29hIeHExwczPr16+WubZvNhkqlwuVyIUkStbW1NDQ0kJycPNzFHlR9fX0cP34cgLi4OOLi4kQOFGFIjfnAYrVaqa6uJikpiSNHjlBQUCDPvo2JiSEkJIRt27aRnZ1NdHQ0ERERY2KekfTfhdHr6+txOp3Ex8cTEhIiAopwXoz5wBIYGMiUKVNobGzE4/FgsViw2Wy0t7djt9tpbW2Vu8Z9fX3PmL5yNJEkCYPBQEtLC3q9nvj4eJH/RDjvxnxg8YqPj+83mG/q1KkAY6ZtRZIkOjs7aW5uRqVSkZmZKbK0CcNm3ASWscrhcNDU1ERPTw+hoaHk5uaKdAXCsBNn4CjlcrlobGyko6ODsLAwcnJyREARRgxxJo4ikiTR09NDa2urPOQ+KSlJ3O4II44ILKOEdx2enp4e0tLSCA0NFQFFGLFEYBnBPB4P3d3dGAwGNBoNqampIqmSMCqIwDICebuM6+rq0Ol0ZGRkiDk8wqgiAssI4nA46OjooKuri/DwcKZOnYpSqRQBRRh1RGAZARwOB/X19XLKR5FUSRjtRGAZJqcmVYqLiyMtLU3UToQxQQSW80ySJMxmMxUVFQBkZmYSGBgoAoowpojAcp54PB7MZjNNTU3o9XoKCgpED48wZonAMsQ8Hg9NTU3U1NQQGxtLenr61yYGF4SxQgSWIeBNqtTS0kJ3dzcRERHMmTNH9PAI44YILIPMZrNRXl6Ow+EgKSmJvLw8kQNFGHdEYBkE3h6e+vp63G43iYmJBAUFiYAijFvjJrC4XC7a2tpQqVRERkbKqSnb29sJCgoaUFJuL0mS6OjooLq6Wl76VSRVEoTvsMTqaGMymdizZw/t7e3yYzt37sRgMLBhwwY5N25fX9+3vpc3qVJJSQkmk4kJEyaQlZWFn5+fCCqCwDgILD09PZSUlKBUKklNTaWoqIienh7gxDIYBQUFNDY2Yrfb2bFjB/v27fvaWxin00llZSVFRUX09fWRnZ1Nenq6yNQmCKcY87dCarWayMhIeUF5jUaDy+WivLycsLAwPv/8cyZPnoxer2fRokWUl5fj8Xjk13t7eBobG+nu7iY0NJSCggKxbIYgfIMxH1i0Wi2RkZF4PB4SEhIAUCqVBAUFoVAo5EXMTq2leJMq1dfXAxATE0NiYqIIJoJwFsZ8YPE6NXB4A8SZJvuZzWb279+PzWYjISFBTqpksVjOS1kFYaRQKBT4+fkNuIdz3ASWszVjxgyeeeYZNm7c+LVLgZhMJvR6PWq1+jyX7tz19fWh1+uHuxhnzduIPlrK7F0HOyQkZLiLctacTieSJH3j8rp6vZ7ly5cTHBw8oPceV2s3nw1JkuSfr/Pvf/+b2bNnk5aWNqifPVQkSWL79u1cdNFFw12Us7ZlyxZUKhUXXnjhcBflrFitVt544w3uu+++4S7KWauqqsLpdJKdnf2123jXOlcoFAO67kRgOUuSJHH48GHcbjdpaWkEBASgUCgoLy/HbreTk5Mz4hZWt1qtlJeXExoaSmxsLCqVivb2dmpqaggPDyc1NXW4i9iPd9RySkoKfn5+ALS3t9Pc3Exubu6IW0zO7XZTWlqKXq8nOTkZSZJwOBwcPXqU4OBg0tPTR1ybnHfcVVhYGCkpKcCJdc2bm5vJysoiKCjoa187kOtuzHc3DxZJkti1axdWqxWVSoVSqaSvr4+KigqcTifV1dXDXcTT2O12EhMTqaqqkh+rra2lra1txF2kkiSxadMmQkJC2LBhA5Ik4Xa7OXLkCDqdjiNHjoy4pW+VSiXJyclUVlbS2NiIWq3GaDTS3NyMr6/vcBfvjOrr62lpaUGn08lz1w4cOEBoaCiHDh0atM8RgeUslJeX09DQwE033URAQACFhYVIkoTL5SIyMpKYmJgRddKbTCa2bduG2+3GZDIRFxeHj48PkiSRn59PQUHBoJ5Eg8VisRAcHIyPjw8ejwePxyP36nV3dw938c7IYrFgtVqJjY1FkiQiIyOZMWMGu3fvPqvBludbTk4OkyZN4tChQ/Itf0BAAAkJCYO6LpVovD0L6enpANTU1KBUKomMjKSjowN/f39cLhddXV0kJycPbyFPEhgYyKxZszAYDPT29pKXl4fD4ZCnHjidTsLDw4e7mKeZNGkSBw4cICEhgY6ODhQKBaGhodTU1JCZmTnibivcbjdWq5Wrr74apVJJfX09Go0Gs9lMaGjoiLs1Buju7sbhcBAeHk5PTw8KhYKwsDAqKysH9dZYtLEMgNvtxuPxoFKp5JPc+806WlYhlCQJp9OJWq0ecReqtxZ48vH1PjZaeuBG8vGFM5fvTMf9TAZy3Y2Oq2GE8PHxOW3ci1KpHFWzmBUKxXn5Jq2qqsJsNhMWFkZCQsI3nrAej4euri75FigsLKxfeU8OKt5R0V93zCVJor6+HpVKRVxc3CDtzdk7X8f3XJ2pfKce48Eweq4IYVRZsWIFra2tPP/885SVldHT00NtbS0GgwGPx0N7e7vckOxwOFi/fj19fX1YLBacTie1tbU0Nzdjt9upra2lrq4Op9PJK6+8wscff4zFYpF7uMxms9zG5XA4OHz4MJ9++ukwH4HxTdRYhCHhdrvp6OggNDQUp9PJiy++SGZmJnV1dSxevJinn36a2267jXXr1vHII48gSRJ79uzBaDTS09NDfHw8qamp+Pr6UlxcTF1dHcnJyVRXV9Pb20tSUhLr168nNzeXuro6fvSjH6HVatFqtcyfP5/3339/uA/BuCZqLMKQUCqVZGZmotFoMBqN7N27V+5FA9BoNMycOZPY2FhsNhtwYiSozWbjyJEjXHzxxeTl5bFjxw6qqqrweDx0dHQQHR1NdnY2TqeT48eP43a7iYmJwe12A8hjSVwuF06nc9j2f7wTNRZhSCxZsoTMzEwyMzMpLCzk4YcfZteuXcTExBAZGYlWq+WFF14gNjaW+Ph4JkyYgI+PD3a7nXnz5vHaa6+RlJTE5MmTKSoqIisri+zsbPz9/dm6dSsZGRlcccUVHDp0iEsuuUQeN+J0Otm2bRsul4sDBw4wa9asYT4S45PoFRLOC++AN6VSicvl4vPPP+fSSy89Y0+EN1WFdyi5x+M5YyLys+3NEAaH6BUSRhyFQiF3yWs0Gq644opv3Pbk3revW252KHozhMEh2lgEQRh0IrAIgjDoRGARBGHQicByBl1dXbS1tQ3oNQ6Hg8LCQg4dOoTRaOw3KdHlclFYWMi+fftoaGiQJ38dPXoUo9EIQEtLi/x/q9VKUVERLpdrsHapH29D6vnU2tqKy+WisrJSdAOPAyKwnMFXX33Vb4CVJEk0NTXJoz8lSaK3t5eamhoMBgNdXV3U1dWxc+dOLBYL999/f780lmazmR/96Ed88MEHrFy5kkOHDuHxeLjvvvt4+eWX8Xg8vPnmm+zZswdJkvj888/58Y9/THFxcb9yeRdGq6mpoampCY/HQ09PDzU1NXR1dSFJEn19fXIZDQYDLpeL9vZ2jEYjtbW18koDv/jFL6ipqcHhcNDS0kJ1dbUc2Lyf1d7eTktLC83NzXIw7OrqoqamBovFIi+DYjAYMBgM/V7b1tZGTU0NdrsdSZJ45513sFgsfPrpp1itVhwOBzU1NTQ3N9PW1ibnGK6urpb3RRi9RGA5g1MzyK1bt45//vOfbNq0ieXLl9PT08MvfvELduzYwcsvv8wLL7xARkYGDz30EPn5+fT09PTL9A8QGxvLo48+SmRkpJy7JSwsDIPBIE9hB+Su2N/85jesXr26X82ivb2d//u//+PAgQMcPHiQ9vZ2nnjiCQ4cOMBvfvMbDh06xM6dO2lpaQHg5z//Oa2trdx777289957vP3223z44YcYjUaOHTtGU1MT27dv5+WXX6auro6SkhL5s1wuFw888AAHDx7kX//6F1u3bqW0tJS//OUvlJeX8+c//5mOjg6efPJJXn31VTo6OuTXlpSU8MILL1BcXMxjjz1GX1+fvH+FhYVYrVaee+45tm/fzt///nfuuusuzGYzv//97ykvL+d3v/tdv7IIo48ILGdhw4YN3Hzzzdx77700NTWxdetWtFot3/ve91i4cKFctW9ra+Pxxx/nrrvuOq2f32w2s23bNurr65kwYQIAKpWKu+66i7feekuu4Rw9ehSTyYROp6OxsbFfAqnq6mqUSiU33ngjS5YsoaOjA7vdzo033sj3vvc93n//fdxutxyMvLWFkJAQli5dyv3338+2bdsIDQ0lKiqKuXPnotfr6e3tZc+ePYSFhfULqFqtlksuuYQf/vCHbN26lV27dtHV1YXBYJDTMLhcLpYsWUJubq78ugMHDjBnzhyuueYa4uLiOHjwoPyc3W7HZrOxe/dubr31Vh566CF8fX1pb29HkiQWL17MnXfeyXvvvTfIf0XhfBKB5QwUCgVtbW0cO3aM48ePk5OTw44dOyguLkav15OTk4NSqaS0tFS+aAwGA7/85S+57LLLmDRp0mltGGFhYVx33XX84Ac/4M0338TpdKJQKEhOTmbatGls2rQJt9vN6tWrueOOO4iIiOCGG27gk08+kS/22NhYenp6KC0tpaamhtDQUKxWK0eOHKG4uFgOWJs3b6a4uFi+PVGpVKjVapRKpTxl3u12U15eTmJiIvfeey8TJkzg/fff7xdYLBYLJSUlfPXVVwQGBpKYmEhwcDDTpk3j+9//PlOnTj3j8fNmfDt69CiVlZVkZGTIA9i8Y0/S0tLYs2cPhw8fxmq1olaraWpqorS0lC+++IKZM2cO7h9VOK/EyNszaG9vp6KiAh8fH/kiOHDgAE1NTVx22WVERUVRWlpKe3s7paWlSJLELbfcQnl5OWq1GrVaTX5+PlqtFkCe15KTk4PH46G0tJSMjAyqq6vJysoC4NChQyQnJ9PR0UF2drYcBLyvUygUSJJETU0NX3zxBfn5+cyYMYOqqiq2b99Obm4uM2bMwG63s3nzZvz9/QkPDycrK4u6ujoyMjLweDxUV1eTnp7O0aNH6evrIyQkhKKiIhQKBTNmzJBTHDidTu655x6uvfZalEolCxcuxNfXl5KSErZv387UqVOZMWMGtbW1REdH4+/vLx8/q9XKgQMHKC8vZ968eaSnp9PU1ERUVBTHjx8nPT2djo4Ojh07RnNzM3v37uXZZ5+ltLSU/fv3k5OTw4wZM0ZNjpvxQiTTHmKSJPH+++9z/PhxAgICWLp0ab8cImOBy+XixRdf5P7775cD5GDav38/mzdvpq+vj+9973vfmCleGBlEYDkPTj5sY3GeylDv31g/fmORmCv0HZ3NSX/y45Ikybcq33aRfNs23s8+l4vt1Nd+l4t3qC92EUzGNtF4ewYHDx7kb3/7G8888wxNTU3AiTVvvHlD4ETvhtPpxOl08sc//hGTyURVVZU81qS3t1fOL+r9f3Nzs9zY6x23cfIguL6+Pl5//XX+93//F4fD0a9MDodDfh84caviHUviHb/S0tJCaWmp/P7Hjh3jjTfe4I9//CPt7e1DeswE4WSixnIGRqORyy67jNraWg4fPkxDQwO7d+/G5XJx++23s2vXLgwGA2azme9///vs3buXe+65h/b2dqxWK5988glBQUHcddddfPbZZ7S2tpKWloZGo2Hjxo1YrVaam5tpaWnB19eXu+66C51Oh06n4+abb+aOO+7oNw6mt7eXTz75hOPHj3PttdfidDrZu3cvBoOBK6+8kr6+PgoLC3G5XEycOJH8/HwUCgXp6ek4HA6qqqpobW0lIiJiGI+qMJ6IGssZOJ1OPv30UzZv3kxkZCQffvghfn5++Pv7c/ToUVpbW7nzzjvJyMhgw4YNwImekGPHjvHll19y66238sADD+Dj44PRaESr1fL3v/+dhIQE5s2bx+TJk/n444+Jjo6ms7NTXjNHqVSesSekp6dHXrJh8+bNVFZWEh8fz4033shbb73F+++/z4033sicOXP63WKcnKpgKBpgBeHriMByBn5+flx//fU89NBDrFu3jsTERBQKBVFRUWRmZmIymdi4cSPV1dVMnz6dwMBAfHx80Ol0REdHs3nzZr744gtqa2spKSkhJCSEgIAA/Pz8KCkp4dixYyQnJ+Pj40N+fr68rKXb7ebAgQNYrVaKi4vlWktZWRktLS2Ehobi6+uLr68vgYGBqNVqdDodEydO5KuvvuLgwYNy7hJJkigsLKShoeFr85kIwlARvUJn4HA48PHxQalUYjab0el0mEwmedW4NWvWcNVVV6HVauVyBAQE4HA4UKvVdHV1ARAaGorJZJLXxQkODqa7u1sesGaxWOQFo7wrFXZ3d+N0OuX3VigUuN1uOjs7UalU+Pr6ypnVfHx8sFqtcl5ZrVaLXq9Ho9EgSRJGoxGHw4FKpSI0NFQ0mArfiehuHkKSJGEymeSLXhDGC9HdPIQUCgXBwcHDXQxBGNFEG8spampqBpyLRRDGIpfLRUtLyzmlsBCB5RTr1q1jz549w10MQRgWkiRhtVqpqqri2LFj6HS6c3qfcXMrdKZRqGJYuSCc4B1kWVNTg8lkIjs7m9TU1HO+LsZNYLFYLHz11VfExsYydepUFAoFZWVl1NbWEhYWxowZM0RwEcYdj8eDxWKhqakJjUZDeno6Wq32O18L3+lWaOXKlSgUCpYtWyY/JkkSTzzxBLGxsfj6+jJ//nx5mLmX3W7n4YcfJjw8HD8/P5YsWUJjY+N3Kcq38i6C1d3dLedKaWxsZMGCBezdu5fe3l62bt3K7t27USrFHaJwfkmSRGtrK62trac97vF46Orq4ujRo3It2+Px4HA4KC8vp729nbq6Ovr6+njppZfYsmULRqORpqYmtmzZwsqVK3nnnXcoLy9n165dcvbB5557jq1bt9LQ0EBqairp6enodLpB+YI95xrL/v37efXVV+XkQl5PPfUUzzzzDKtWrSIzM5Pf//73XHrppZSXlxMQEADAsmXLWLduHWvXriUsLIxHH32Uq666qt8Ar8GmUqmYPn26fNC9I1Lr6uoICQlBrVaTl5dHVlaWyLcqDAtvYq7Q0FCMRiP+/v5yUq/o6GhWrFjBc889R2ZmJh9//DF/+9vfmDJlCt3d3axfv565c+eydetWWlpaSElJwePxUFtbi1qtxuPxkJCQQGNjI9nZ2QQFBXHo0CGee+455s+fP/i1dekcWCwWKSMjQ9q8ebM0b9486cc//rEkSZLk8Xik6Oho6cknn5S3tdlsUlBQkPTKK69IkiRJRqNRUqvV0tq1a+VtmpqaJKVSKW3YsOGsPt9kMkmAZDKZzrrMXV1d0tGjR6XKykrJZrNJzc3NktlslkpLS6W2tjZ5u+eee0766KOPzvp9haHncrmk9vZ2qa+vT37MbrdLBoNBMhgMUm9vr+TxeKSuri7JaDQOY0kHpqKiQqqpqTnt8cOHD0vLli2Tjh49Kv3v//6vdPjwYam9vV266qqrpJ/85CfSf/7zH+kXv/iFFBISIk2YMEGKjo6WAgMDpdtvv12aOXOmpFAopJiYGGn27NkSIAGSj4+P5OfnJykUCkmhUEiApNfrpV27dp11eQdy3Z1Tnf/BBx/kyiuvZOHChf0er6mpobW1lUWLFsmPabVa5s2bx65du4ATM4edTme/bWJjY8nPz5e3OZXdbsdsNvf7GaiQkBBycnJIS0tDq9USExNDQEAAubm5REZGDvj9hHMn/bfnwbvqwbdtu2bNGtauXcuzzz4rzwa3WCwcOHCAF154gU8//ZR9+/bx4osv8pe//EVOJj6SSJLEhx9+yNGjR+XHnE5nvxSmBoOBDRs2kJCQwN13301hYSGHDx+mr68PnU7H7NmzmTBhAsuXL+ePf/wjQUFBPP7448ydO5fAwEA2bNjA3r170Wg0uN1uJEkiPDycwMBA3G43vb29REVFERERgY+PDxqNhvr6+iHZ3wEHlrVr13Lo0CFWrlx52nPe+8OoqKh+j0dFRcnPtba2otFoCAkJ+dptTrVy5UqCgoLkn4SEhIEWWxhB7HY7P/jBD9i2bZt8YR07dow1a9bIP96xRN7q/A9/+EPMZjPHjx8HTuQQvuyyy1CpVMyePZutW7eydOlSJk6cyGeffTZs+/ZN5s6dS3Jysvx7Y2Njv3O+t7eX//znPwQGBpKfn09UVBRTp06VJ8L+7W9/IyUlhZ/97GfcdtttJCYm8swzz8g9Od5pHxqNBoPBwO7du+no6KC3t1d+PDIykiuvvBKVSkVPT488T22wDSiwNDQ08OMf/5g333zzG/u3T71fk75jAqTHHnsMk8kk/zQ0NAyk2MII4/F46Ovr45ZbbpHPI+/Mbu88qpPPhc7OTjkJuE6nk2eDl5eX4+PjQ0xMDG63m76+PhwOB6GhocOyX99EoVAQERGBXq+XH1uwYAGzZs2Sf09KSuIvf/kLbrcbi8XCjBkzCAsL45lnnqG2thar1cqf//xn7HY7DoeDnTt30t7eLufmiY+Px8fHR26nTE1NlT87ISGBadOm0dDQwJtvvomvry9Llixhzpw5Q7K/AwosBw8exGAwMHXqVPkk2Lp1K3/9619RqVRyTeXUmofBYJCfi46OxuFwyCfHmbY5lVarJTAwsN+PMHp4b32k/zaK+/j4MH369H7BIzExkYULF7Jw4UIWLFgg12iVSiXTp0/nX//6FzExMeh0Oj744APgxFIpixYtore3l2uuuYYdO3ZQU1PDpZdeev538hycvFSLJEm0tLTQ1NREe3s7H374IXv27OGZZ54hPj6euXPnsnDhQiRJ4quvviIjIwM4cUv4q1/9iptvvpnu7m6ysrIwmUwA1NfXk5+fj9vtpquri3379pGVlUVWVhY2m43CwkKam5uHZN8GFFgWLFjAkSNHKCoqkn+mTZvG7bffTlFREampqXLaAC+Hw8HWrVuZPXs2AFOnTkWtVvfbpqWlhZKSEnkbYWyQJAmXy0V3dzePPPKI3Kag0Wj4+c9/DiCnhtDpdISEhMg/arUaOPFte+ONN2K1WnG5XOzevZt77rkHgJkzZ3LgwAHWrFlDWFgYF198MUeOHOm3CuVI4z0mkiSxdetW9u3bB5wIMvv376erq4uYmBjuvvtudu7cidvtZu/evfz2t78lKCiIn/zkJ2RlZREdHU1oaCgej4eVK1eydetW7HY7BoOBsLAwlEolWq2WyZMno1QqueyyywgNDcXPz4/29nYCAgIICwsjLi5uSPZzQN3NAQEB5Ofn93vMz8+PsLAw+fFly5axYsUKMjIyyMjIYMWKFej1em677TYAgoKCuPfee3n00UcJCwsjNDSUn/70pxQUFJzWGCyMbseOHeO5557jiiuuoKysjLfffhuDwcDVV1/N22+/TWRkJImJiVx++eUA7Nq1S67tarVaLr30UrRaLQaDgerqap599lnuv/9+Fi9ejK+vL++88w49PT1kZGTgdrvJzMwkNjb2vK9LPRDeVTR//etfs2DBAqD/qFeFQkFMTAxqtZq7776b6667jtraWjlL4Pvvvy8v5/LEE09QU1PDxx9/THZ2Nq2trajVatrb25k9ezZxcXFkZmaiVqv5+OOP0Wg0NDQ0cOuttzJ16lR27NiBzWbrt3TLYBn0kWA///nPWbZsGQ888ADTpk2jqamJTZs2yWNYAP7yl79w7bXXcvPNNzNnzhz0ej3r1q0TCYnGmL6+PgICArjwwguJiIjgtttu49ChQ5SWlrJw4UIuv/xyPvroI3l7b54ZhULRb5CiUqmUF1nzZtOz2+3Y7XamTp2KTqfj7bffll87kvn7+/P0008THR2NUqnE4XDwzDPPoFar+clPfkJ6ejo9PT189tlnJCQk0NnZiSRJ2Gw2pk6dSnt7O1FRUSQlJREfH09DQwO33HIL1dXV3H777ZjNZv7nf/4HnU6HxWJhy5YtXHvttSQmJjJhwgSWLFlCSEgIu3fv5g9/+MOQLVsj8rGc4q9//StJSUlcc801g/q+41F5eTlffPEFd9xxB08//TRPPPEEy5cv5/rrr+c///kPer2eSy65hIKCAgC6u7ux2+3AiWASERGBUqnE7Xbz8ssvA5CSkkJsbCwWi4WMjAw+/vhjzGYzc+bMISwsjBdeeIEJEyZw0003ndbzOFJUVlYSGBhIZGQkHo8Hk8mEXq+npKSEgoICNBoNcKI7+te//jURERFy429+fj7l5eVMnDiRnp4ewsLC+Pjjj9m9ezcajYaqqipmzpzJ4sWLSUhI4Nlnn0WlUlFXV8ebb75JS0sLb7/9NldddRWTJ08GkG87v41I9PQdiMAyeE4+tby9fif/C5x1LcPbFnOmHseTHx/o+w6HzZs3k52d3W/YhNPpZMuWLVx88cXyqPDu7m6qq6vx9fVFrVYTGhpKT08PgYGB8rQZ7y3gvHnz2L17N/PmzaO2tpZf//rXxMbGolKpaGxs5Oqrr+auu+7i/fffp62tTe510uv1XHjhhWdVbpHoSRgRTk3sfaZ/z9bXzd869X1GajA52Zl6rdRqNYsWLcLpdFJXV4fJZCI0NJTJkydjs9mwWq2EhoayY8cOLr/8crKysigtLeUXv/gF69evx2g0yl+GkyZNYv78+fT29vLQQw/R1taGUqlEo9Fw6623UlZWRmRk5JB2y4vZdoIwDCRJ4oUXXpAH/LlcLhoaGigpKcHtdpObm0tCQgI+Pj40NDSwZ88eFAoF1157LVqtFkmS+MEPfkBiYiKSJPUbH9Pb20tRURFRUVH8/ve/Z82aNSQlJWEwGKitrSU3N5fw8HCUSuWQTbgVNRZBGCY333wzOp2O48ePYzQa2bVrFz/4wQ/Q6/X9al7esSfex3p6eti+fTspKSlMnjyZG264oV97Um9vL7feeiuXX345LpeL9vZ2tFotR48exeVykZKSQn19PWFhYfj5+Q3JvonAIgjDoK+vj66uLsxmM8nJySQkJJxx1DGcfnvn7+/PypUrqaqqAiA4OLjfNhEREVx55ZXyulLetpypU6cCJ2pLO3bsYOHChSKwCMJo5+0BamlpQa1Wk5yc3C+pkreX5lSNjY00NTXJycgUCgWpqanykP3t27ej1WqZO3cu8O3tTAqFQh5XNlREYBGEIebxeOjs7KSurg6VSkVmZma/NpFv09vb+41rb8+dO3fEjQETgUUQhoD038xvBoOBjo4OQkNDmTJlyjl1g3vbWL7OSFw+d9wElr6+PioqKoiMjCQmJgaFQiGPEwgPDycpKWm4iyiMEQ6Hg4aGBpqamkhOTiY3N3fE1SiG2rgJLFarlYSEBIqLi4mKisLHx4fm5mYMBgPx8fHAiS6/kTzPRBi5JEnC4XDQ1NSE1WolLi7uO2W5H+3GzTiWkJAQOWeFUqlEkiTS0tLIzs6msLAQSZKorq6mrq5u3J4MwsBJkoTZbObAgQMUFxcTFhZGbm7uaT014824qbE0Nzdjs9nkXBR1dXUEBATgdDrlJVMzMzNJTU0VybSFb+UNKN5lMwoKCgZl2YyxYtwEFu/tDoCvry/Z2dly9TUjI0OcEMJZcbvdtLS0UF1dTWxsLOnp6fKkQeH/GTeB5UwUCsWIbFEXRhZJkpAkiba2Njo6OggLC2P27Nn4+PiIL6SvMa4DiyB8G6vVSkVFBQ6Hg6SkJPLy8sSCdmdBBBZBOIX3FrmhoQGHw0FCQgLBwcEioAzAmAgs3d3dWK1WPB4Pra2tckOaIAyEJEl0dnZSXV1NQEAAiYmJQzaXZqwb9YHFYrHw1FNPERwcTEVFBREREVRUVLB06VJx/yt8K28PoMlkorGxEa1WS25uLn5+fuL8+Q5GfWDp6ekhNTWVhQsXcvToUWbNmsXKlSu59dZbxYkhfCOn00l9fT0mk4nw8HCys7Pl7G3CdzPqj2JERAQXX3wxycnJpKSkYLPZeOCBB0RQEb6Wt8u4q6tLXnVQdBkPrlEfWFQqFenp6fLvOp2OlJSUYSyRMBJJkkRvby/19fW43W5iY2PlJN7iS2jwjfrA4lVfX8+hQ4e45JJL6OrqIikpSZwwAnBiAmpjYyN9fX3ExMTI2f+FoTMmjq7VauUf//gHTU1NGI1GXnvtteEukjDMPB4PHR0dlJSU0NjYSEJCApMmTSIqKkoElfNgTNRYfHx8UKlUtLS0sG3btm/MXSGMbd4u46amJgDS09NPyyErDL0xEVgAJk6cSFxcHG63m8DAQHEijSPe9ZDb2tro6uoiNDSU/Px8lEqlOA+GyZioExYWFvLaa6/R2dlJc3MzFRUVw10k4TzxdhkfOHAAh8NBbm4u8fHxYh7PMBsTNZaUlBT+v//v/yM1NRWPx0NUVNRwF0kYQt61jJubm7FarcTHx5OYmCgCyQgyJgJLUFAQR44cYc+ePXLqyRtvvFGcaGOMJEn09PRQU1NDb28v2dnZ4z6h0kg1JgKLN1lTcnIy4eHhbNu2jRtvvPG07VwuV7/V3yRJwul0nnEtF2Hk8Hg8WCwWOalSVlYWGo1G/M1GsDERWPR6PTfffDOAvJrcqbq7u9m/fz+SJHHRRRfh6+vLsWPH6OrqQpIk5syZI07UEcbtdtPe3k5lZSWhoaGkpaWJyaWjxJhovO3r62Pt2rWEhIRw4YUX8tFHH522jXct28jISAwGA3BiUN20adMoKiqit7eX8vJyqqqqRIAZRt5lM1paWjh69Cgej4dZs2aRk5MjgsooMiZqLGq1mo6ODl577TWUSiWRkZGnbRMZGYnZbKanp4ewsDBKSkrw9fWlubmZ2NhYtFot6enpJCcni5y3w8Rms1FdXU1XVxfp6ekiqdIoNuoDi8fjwePx8KMf/Yht27YRFRUlLwx1spycHKqrq0lOTiYkJASHw8Hs2bOpqKjgkksuQa1WA4y79V+GmzepUmNjI3a7ndjYWLKzs0VAGeVGfWCx2+0UFxdjNBpRKpW0t7djMBhOm4ioVCr7TVb0/j83N/e8llc4QZIkuru7OXbsGP7+/qSmpoocKGPIqA8sAAaDAafTOdzFEM6CJEmYTCaamprQ6XRMmjQJX19fEVDGmFEfWLRaLZdccgmtra0UFRWhUCgIDAwc7mIJp3C5XNTX12MwGIiOjiYzM1O+/RTGnlF/I6tUKlGpVLz55pvk5eURGRnJSy+9JBpgRwBJknC73TQ1NVFWVoZGo2HKlCkkJyeLoDLGjfoaC5xowDUajWzatAmLxUJvby9ffvkl8+fPF42xw0CSJPr6+qisrAQgLi6O/Px8cbszjoyJwKLRaLjuuuvo6OggODgYSZIICgoSJ/IwsFqtNDY24nA4iI2NJSwsTPTwjENj4i/udrs5cuQI1dXVZGVlUVxczNSpU8UJfZ54PB4MBgNFRUXU19cTHx9PXl6eyNQ2jo2JGovL5aK9vV1eAkT0EJ0f3i7jpqYmfHx8SEtLw9/fX9QUhbERWPR6PXfddRerV69m//793HLLLeLkHkJOp5OmpibMZjMhISHk5uaKpEpCP6M+sLhcLl5//XV5Kcyuri48Hs9wF2tMcjqdtLS00NHRQWBgIDk5OaJ3RzijUR9Y2tvb2b17N88++yyvvvoqDz74oJisNogkSerXIBsXF0dCQoKonQjfaNQHFr1eT15eHuvWrUOn0/HBBx8wc+bMfsP3hYHzrsPT0NCA2WwmNTWV8PBwEVCEszLqA0tQUBCPPvrocBdjzPB4PJhMJlpaWlCr1aSmpoqkSsKAjfrAIgwO7zo8tbW16HQ6MjMz0el0w10sYZQSgWUc8yZVamtro7Ozk7CwMKZNm4ZCoRA1FOE7EYFlnLLb7TQ0NNDc3Exqaiq5ubli+oMwaMZNYOnq6qKlpQWtVktaWhoKhQKz2Ux9fT0hISHExcUNdxGHnCRJ2O32fstmeI+FIAymcTPeWqfTERoaSmlpqTzzub6+noaGBnmb9vZ2Ojs7x9yF5s2Bsn//fkpKSggPDyc3N1fMpxKGzJivsRiNRo4dO0ZOTg7Hjh0jPj4ehUKBJEmkpqai0+koKysjNjaWsrIyKisrmTZt2nAXe1BIkoTZbKaxsRGdTsfEiRNFD49wXoz5wBIQEMCECRNoaWmhsrKSvLw8LBYLzc3NSJJES0uLnHz7oosuoqioaNSP3HW5XDQ3N1NXV0dsbKxIqiScd2M+sPj4+KDX60lLSyMtLU1+3JtlLicnZ7iKNqi8PTwGg4GOjg7CwsKYNWsWKtWY/xMLI5A460Y575B776zu5ORkkVRJGHYisIxS3h6exsZGnE4niYmJBAcHi/wnwoggAsso4/F46OzspKamhoCAABITE/Hz8xvuYglCPyKwjBKSJGE0GmlqakKr1ZKbmyvW4RFGLBFYRjin00l9fT0mk4nw8HBycnLECFlhxBOBZQTyLpvR0tKC0WgkICCAgoIC0WUsjBoisIwg3hwo9fX1uN1u4uLiiI+PBxC3PMKoIgLLCOBdh6exsRGr1UpsbCwREREimAijlggsw8jj8dDV1UVraysajYakpCS0Wq0IKMKoN+BBD01NTXzve98jLCwMvV7PpEmTOHjwoPy8JEk88cQTxMbG4uvry/z58yktLe33Hna7nYcffpjw8HD8/PxYsmQJjY2N331vRgmPx0N7ezuHDx+mqamJ1NRUObGSCCrCWDCgwNLd3c2cOXNQq9WsX7+eo0eP8uc//5ng4GB5m6eeeopnnnmGF154gf379xMdHc2ll16KxWKRt1m2bBkffvgha9euZceOHfT09HDVVVfhdrsHbcdGGkmScLlcNDY2UlJSgsPhYMKECUyYMAG9Xj/cxROEwSUNwC9+8Qtp7ty5X/u8x+ORoqOjpSeffFJ+zGazSUFBQdIrr7wiSZIkGY1GSa1WS2vXrpW3aWpqkpRKpbRhw4azKofJZJIAyWQyDaT4Z+W5556TPvroo0F9T7vdLlVXV0s7d+6UampqJJfLNajvLwjnw0CuuwHVWD755BOmTZvGTTfdRGRkJJMnT+bvf/+7/HxNTQ2tra0sWrRIfkyr1TJv3jx27doFwMGDB3E6nf22iY2NJT8/X97mVHa7HbPZ3O9npJP+O4enqqqK48ePExoayqxZs0hOThbjUIQxb0CBpbq6mpdffpmMjAw2btzI/fffz//8z//wxhtvANDa2gpAVFRUv9dFRUXJz3kbKkNCQr52m1OtXLmSoKAg+SchIWEgxT6vpP/mQCkuLqaoqIiwsDCRVEkYdwbUK+TxeJg2bRorVqwAYPLkyZSWlvLyyy9z5513ytudegFJkvStF9U3bfPYY4/xyCOPyL+bzeYRF1yk/2Zpa25uRqvVkpOTI5IqCePWgGosMTEx5Obm9nssJyeH+vp6AKKjowFOq3kYDAa5FhMdHY3D4aC7u/trtzmVVqslMDCw389AeW9NbDabnJpSkiRsNtt3ajR2u900NzezY8cO2traSE9PJy0tTXQbC+PagALLnDlzKC8v7/dYRUUFSUlJAKSkpBAdHc3mzZvl5x0OB1u3bmX27NkATJ06FbVa3W+blpYWSkpK5G2GgtFoZP369VRUVMiPVVdXs2/fPg4cODCgrHHSSUPujx49ikKhYPbs2WRmZqLRaIai+IIwqgwosPzkJz9hz549rFixgsrKSt566y15vWQ4cQu0bNkyVqxYwYcffkhJSQl33303er2e2267DTixcuG9997Lo48+yhdffEFhYSHf+973KCgoYOHChYO+g3a7nba2NhQKBQEBAZjNZlwuFwCVlZVccMEF7Nu3j76+PiorK6mpqfnamoa3hlNaWsqePXtQKpXk5eURExODj4+PqKEIwn8NqI1l+vTpfPjhhzz22GMsX76clJQUnn32WW6//XZ5m5///OdYrVYeeOABuru7mTlzJps2bSIgIEDe5i9/+QsqlYqbb74Zq9XKggULWLVq1ZD0ljidTtrb2wkNDWXGjBl8/PHHpKWl4XQ68fHxoaqqivDwcLRaLYmJicTHx8u3Sl7Sf5MqNTU14XA4SEhIIDc3VyRVEoSvoZBOvYpGAbPZTFBQECaT6azbWxwOB62trQQGBhIQEIDNZkOj0dDa2kpISAj+/v4A/PWvfyUpKYlrrrkGSZLo6uqioqICf39/UlJS5O0EYbwZyHU3buYKaTQaEhMT5d+9Wde+rnfJm1TJu2yGr6+vuNURhLM0bgLL2YqLi+OTTz6hqamJ+Ph4fHx8OHbsWL9tXC7XqGtT8Xg8o+rWze12o1AoRlWZXS7XqFoVwXuz8k3nsVqt5qKLLkKn0w3ovUfPUThPrrnmGubPn39aO8vJVq1axdy5c0lPTz+PJTt3kiSxdetW5s+fP9xFOWtffPEFKpWKefPmDXdRzkpfXx+rVq3iRz/60aj5wqmsrMTpdH7jEjgKheKcejrHTRvLdyVJkhxszGYzer0etVqN0+nE7XaPyJnJkiTR09ODTqfD4XDg5+fXbz9GWm1A+m+iK71ej9VqBU4sjWu1Wkdkfl9vo75CoUCtVmMymQgODu5XExiJZfaWz+VyIUkSKpXqrI6xaGMZImvWrCEsLIzZs2ej0Wiw2Wzs2rWLiIgIoqOjiYiIGO4i9tPW1obRaKS6ulqem3X06FFKSkrIz88nLy9vmEv4/0iSxPbt2wGw2WwsXLgQj8fDgQMHUKvV+Pr6jrjF5dxuNwaDgcOHDzNt2jSio6MxGAx89dVXZGZmMnHixOEu4mmOHTvG4cOHyc3NpaCgAEmS2LVrF1FRUXg8HjIzMwflc0bWV9YI1dDQQFtbG8nJycTFxckLydvtdvR6PWFhYXR0dAxzKf+f3t5eDh8+jJ+fH5GRkTgcDvmbKCQkhAkTJtDZ2TnMpTxdQ0MDU6ZMwWKx4HK5cLlcOBwOEhMTqamp+cbb0+GgUqkICgrCarXK+Yg1Gg3Z2dk0NDTgdDqHuYSnCw4OZsKECXR1dcmP2Ww2EhMTaWtrG7TPEYHlLAQHBxMQEEBycjJWqxWz2UxdXR0AdXV1lJaWjqgRtxqNhpiYGHp7e9m2bRu5ublYrVa2bdsml7Onp2eYS3m62NhYqqqq0Gq1VFdXU1tbS19fH8eOHSMxMXHE3VY4HA7eeOMNEhIS0Gg0FBcXy7cUbrd7ROYX8s5f6+npoaOjA6PRiNFo5MCBA4MauEUby1mSJInW1lacTicxMTE4HA58fX3p7OzE6XQSHR094tosenp6aG9vR6PREBkZSXd3NyqVCrPZTERExIhb6Mw7TSIyMhK73Q6caKfo7u4mNjZ2xKWb8Hg8NDY24vF45JqhdzKqNzviSAuGXV1d8t9fpVKhUCiw2+10dXURExPzjV+QA7nuRGARhp0kSTgcDvn/A+3aFM4P0XgrDDtvAjCr1crNN98sf6P7+PigUqlwuVy43W5UKhVut5uPP/6YmJgYrFYrCxcuxOFwoFQqUalUctDRaDSsX7+emJgYJk6cKKf7VKvVcm3Gbrfz0UcfERAQwBVXXDFs+z/eicAiDIldu3Zx++23s3XrVo4cOUJDQwMOh4P29nZ+9KMfsXz5cqZPn053dzd33HEHNpsNi8WCxWLhP//5D/X19aSnpzNp0iS+/PJLWltbmTZtGp999hmBgYH4+PjwySefEBsbi9Pp5L777pPHXCxevJgPPvhguA/BuDayGgWEMUOSJOrq6jh27BgRERH8+9//RqfT4e/vT0tLC06nk1tvvRWNRkN7eztwoqrd0dHBRx99xD333MNll12G3W6nr68PpVJJTU0NSUlJzJs3D7PZTGFhISqVSt4GRubYkfFI1FiEIaHX61m4cCHTp0/nyy+/5MorryQ2Npbg4GA5Deknn3xCc3Mz4eHhqFQqVCoVvr6+XHrppXz55ZfExsZit9vp7e2VB3NlZGTQ2trKpEmTmDZtGsnJyfT09MgrHbhcLg4ePEh1dTXl5eVkZWUN85EYn0TjrTAk2tvbCQ4ORqVSYbFY8PPzo7m5GTgxlubFF19k6dKlhIaG4ufnR09PDwqFAo/HQ0BAAC0tLSgUCqKjo2lubsbPzw+1Wo1er6e1tRV/f3/UajUtLS34+fnJK0e63W6amppwuVz4+/sTGRk5zEdi7BC9QsKI5vF4aGtrIzo6Wty2jCKiV0gY0ZRKJTExMcNdDGEIicZbQRAGnQgsgiAMOhFYBpF3WZNR2Gw15KxWKx6PB7PZPKAVEYTRSQSWMzhw4ABffPFFv8e8M21PXZPI+/vBgwd55pln+NnPfsZrr73Wb2Zrb28vTzzxBG+88Qa/+93vqKurw+Px8Pvf/57PPvsMSZJYv349R44cAaC2tpYVK1bQ1NR0Wtm8OUC83a/e37/tYnW73XL5Ozo6eP311+X3c7lc/fbl5M/yzjL2PufxeLDb7f229Xg8p83kdbvd/bZ78803MZvNrFq1ip6eHrncJ0/U8x5TEXhGP9F4ewZ1dXW0trayYMEC4ESmrVdeeYW+vj6uueYaLr30Ut555x3+85//kJ+fz7Rp07jggguYMGECCoWCpUuXcuONN8rLyNrtdo4ePcqvfvUrXn75ZXbs2MGtt97K9u3bKS8vZ9KkSRw5cgSFQkFeXh7vvPMOISEhrFu3Th5RCidWHFi7di3bt29n4sSJ/OAHP+Df//43O3bsIC8vj3vvvZfOzk4iIiIICgrivffeY9GiRaxfvx6TycSePXv44Q9/iNvt5o033sDPz4/MzEw+/PBDjh8/zqOPPsrUqVOBE4Fh9erVNDQ0YLVa+f73v09AQABr1qyhsLCQuXPnctddd7Fp0yaqqqpQKBQ8/PDDADQ3N7Nq1SoqKytZsmQJV199NX19fUiSJE/W3Lx5M//+979JT08nOjqau+66i7fffpvNmzczY8YM7r77btHjN4qJGstZ+Mc//sE111zDU089xdtvv822bdvYuXMnr776KpmZmWzduhV/f3+USiUbN25k7ty5/ZY7ASgrK+O2227j/fffly/egIAAbrvtNl577TX5m7uxsZHi4mIuvvhiCgsLsVgs8nscPXqUAwcO8Nxzz/HDH/6QxsZG9u7dy4svvkheXh6vvfYaVVVVGI1GAD744APMZjMff/wx1157LY8//jjvvvsu0dHRJCcnc80111BZWYlCoeC+++7rl2rT4/Hw2Wef8aMf/YilS5fy1ltvsX79eqqrq7nyyivZvXs3Bw8eZPPmzVxwwQXcf//98ms3btxIdnY2L774IgcPHqSsrEx+7sMPP6S7u5tVq1bxpz/9iTvvvJN169bR1NTEoUOH+Mc//kF6ejpvvfXWoP8dhfNHBJavYbfbsVgs9PT0YLVa5Sz9brcbHx8feUaud3p/X18fv/vd7ygqKuLOO+88LYVCRkYGr7zyCkuXLmXHjh3y4zNnzkSr1bJz504kSeLNN9+ku7ub3//+91RWVvLJJ5/ItxNOp1Meoer93cfHR06RqVAocLlcNDQ0YDabMZlMAPj6+hISEkJYWBi9vb3yhD2n08nixYu57rrrKCsr4/nnn+93GyJJEkqlEq1Wi0ajQZIkcnJyuOSSS/jd737H5MmTUSgU6HQ6OdERnOhO9qY81Gq1p91ieZNke4+fx+ORa2U+Pj4jKreNcG7ErdAZhISE8Morr7Bz504CAgJ44IEH2LBhA59//jlXXXUVc+bMobOzk//5n/8hLCyMgIAAdu/ezcaNG4mMjKSyspKnn36a8PBw4MTFkpmZSUhICHfccQdPPvkkFouF1NRU1Go1d9xxB1988QU+Pj709vby/vvvo9PpMBqN/OlPf5JnAefn55Obm8vLL7+MRqPhzjvvZM6cOTz77LP09fVx3333YTQaeeWVVygtLSU8PByNRkNycjIKhQIfHx8SExOJiooiNzeXH/zgB9x0003y6o+zZ8/uN2BNoVDIge6uu+5Cr9fz1ltv8d5779Hb28uDDz5IfHw8vr6+/Y7fFVdcwUsvvURDQwOpqank5uZSUlKCj48PKSkp6PV6fvjDH/L444+TkJCAXq8nLi6O/Px8XnjhBXnBO2H0EiNvz8Dj8ci3Jt4L0uPxIEmSXFvZvHkzZrOZ0tJSbrrpJnJyck57jfcilSRJXn7DW+tRKpX9luTwPuatJSgUitNe530v7+d4ax7e13pf7106Azjtc7z/P/m9vbWUk8vsdDr5zW9+w29+8xu0Wq38nPfYeD/v1PJ5ud1uPB6PnEzIWyvxbv/ZZ5/R09NDeXk5M2bM4LLLLpPL5D1+wsgihvQPMW/vhfdWRK/Xj7mh6d5eG61WOyT7Zrfb5Qz3fn5+Iy77nnA6MaR/iCkUCnx9fU+7BRhLvG0nQ0Wr1aLVaofs/YXhJb4mzqCiooLVq1fz6aefynk+vo7b7ea9996jp6fnWzPJOxwOent7v/Z5l8vFl19+yQcffCDPBB4Io9EoJ/mWJIljx46xZs0aNm/eLGdhE4TzQdRYzqC+vp7p06dTVFTEjh07yM/Pp7CwEICLLrqImpoaOUn11KlTWb16NZdccgl2u52Ojg7279+PSqVi7ty5FBYWYjQaSU1Npb29na1bt7JkyRIAmpqa0Ov1XHjhhXLbTU5ODna7ndWrV/Ozn/0MODFqdceOHdhsNqZNm4bdbqesrAy32820adNwOp0UFxfjcrkICgoiKSkJgIiICGbOnMnTTz9NZGTkiFznRhibRI3lDJxOJ5999hlbtmwhNjaWV199lebmZsrKytixYwebNm0iKSmJsrIytmzZAoDFYmH37t2sWbMGf39/pkyZgkqloq+vj+7ublasWEFQUBCpqanExMTwzDPPYLVa+fTTT+X1XNRqNeHh4WzcuLHf4lxutxur1Up7ezvvvfceu3btorGxkeDgYF5//XVef/11EhISkCRJHgHrXZ3vqaeeor6+nqioqPN/IIVxSwSWM9BoNNxwww3ce++97NixAx8fH/z8/Jg1axb5+fk4HA75FkmtVssNjwqFAq1Wi8PhwOl0UlZWxtq1a/H398disaDVajGZTFgsFnlsyZVXXimP0HW5XLz00kt0dXUxadIkuTwHDhygsrKSwMBAHA6H3IUcGhpKS0sLPT09uFwueUwNIC9DsWLFCubPn09ra+v5O4DCuCd6hc7AaDTi6+uLRqOhrq4Of39/NmzYgEqlYsGCBaxevZrk5GTUajULFiygtraWlJQUTCYTvr6+bNq0id7eXq699lo2bNiAQqEgJyeH1NRUeXwJnBihmpSUxIIFC9BoNLhcLoqLi/F4PERERJCcnAycuBVavXo1SUlJFBQUyIPkNBoNra2tBAYGsmnTJlJSUsjOziYkJARJkigtLWX37t3k5eUxY8YMeWCdIJwL0d08hCRJoqamhuTkZNFFKowrort5CCkUClJTU4e7GIIwoomv3FOYzeZv7WIWhPFAkiSsVisul2vArxWB5RSrV6+We3oEYTySJInOzk6Kioqora09p8Xtxa3QKZxO5zlFaEEY7TweD0ajUV56JT09/bT0H2dr3ASWvr4+9u3bR3BwMBMnTkShUGCz2SguLiYxMVEsRSGMS96Jn01NTXR3dxMcHExWVla/CannYtwEFpvNRldXlzwqFWD79u1MnjyZjRs3cuONN8oD0QThfPF4PDgcjn7zsnp7e1Gr1XJeGrvdjkqlwmAwyEMVent7CQoK6jfrvbu7Wx4gGRoaSlVVFSEhIfj5+WG1WgkICJBz9MCJjgjvaPGAgADy8vIGbUjCuAksgYGBLFiwgI0bNxIYGEhgYCAulwudTofFYsHtdtPe3k5nZ6eouQjnTV1dHV999RV33XWXPHxh8+bNpKenk5+fj8fj4Ve/+hXXXHMNn332GQsXLmTv3r1s3bqV5cuXc+zYMXbs2MEtt9zCG2+8weeff47T6WTWrFmUlpYyceJEnE4nR48e5fHHH2fFihWYTCY0Gg1xcXF89NFHTJo0adDP+XEzjqWjo4P29na6u7spKCigra0NnU5HQ0MDfn5+FBQUoFAo+Otf/0pSUhLXXHPNEO+FMBDePC0nXwDeZOInf2uPtkXhvbciFouFTz/9lFtuuYXOzk70ej0BAQFIkkRzczO+vr6sXr2a9evX89BDD3H48GF2796Nx+OhtLSUzMxMXC4X9fX1HD9+HEAexOn9XafTyXlywsLCcLvdvPPOO1xyySVnVVYxjuUMwsLCCA0NlU887x8tNjZ21J2M44kkSRw8eJB3332X6dOnc/3116NUKnG5XHz66ad88cUX/PjHP0aSJN544w3Cw8NHVSLuvr4+WltbiYuLk1N97ty5k/T0dDk5e3R0NBs3bsTX1xeHw8Fjjz2GxWLB6XTicDhwOBwcP36c733ve2zbtg04MQG1sbGxX0eEzWaTz3ODwUBYWBgRERFDsl/jprvZm2f11NSLZ8p+Jgwtu93Oz3/+c9566y15flN9fT1ffvklX375JVu2bKG7uxs4UVN57733+MlPfkJhYaG8JMqRI0f4z3/+w8UXXyxn9EtKSsLtdp9TyonhYjabKS4uRqvVkpubK+cwPjmDntFo5Omnn0aSJMrKynA6ndx444309fXR09NDQECAvKSLtxenvb2dyMhI9Hp9v7w33loegMlkkifADrZxE1iEkcPtdlNRUcH1118vn/QOhwOLxYLFYsFsNvcbOxEXF0d4eDh2u11egcBsNhMYGEhiYiLPP/+8fHG1tbVRX18/HLt1TqKjo7nuuusAqKqqwuFwcNVVV5Gbm0t9fT1//OMf2bdvHx6Ph6effpqFCxdit9t55513uOGGG/jBD36An58fZrNZTvDu1dbWhlqtxuFwoNfrUSqVqFQqQkND8ff3Z/ny5UyZMmVI9ksEFmHIeb8lvd+U3ur9yXOtgoKCSElJISUlhdTUVLmXRKFQ0NXVxb///W/8/f2JiYnh4MGDxMXF4ePjw+HDh+Xb2cDAQAICAkZ8vlyPx0NrayttbW39Vig4cOCAfLtSVlbGgQMHCA4OpqqqioaGBiZPnsz69evJy8sjLi6OrKws1Go1NpuNjIwMmpub6enp6fdZbrcbSZKYNm0aWq0Wl8uFUqkkMTGRu+++m9DQ0CHZRxFYhCHX29vLihUr5MCi0+l49tln+y0ZEhERwYQJE5gwYQIFBQX4+/sDJ5KB//KXvyQuLo4pU6bQ1NREfn4+aWlpPPjgg/j7+zNx4kSWLFnC1KlT5Tw2p67MOJK4XC7ef/991q5dS1dXFxaLhdWrV2M0GgkMDKS0tBS73U5ycjIpKSksXryYr776itdff50pU6Zw44030tbWRmhoKD09PdjtdmJiYuS2Qq1WS3R0NFOmTMHhcBATE8OOHTvkWfZms5nu7m4++OADOjs7h2QfRWARhkxdXR3Lly/niy++YN26dbz22mu8/fbbtLe38+qrr/Liiy/Kty3eVJp79uxhz549HDx4UA4O3d3dfPjhh0yYMIFXX30VOFGTaWhooL6+Xh7zsWrVKiIiIlizZs0Zl6cdKTQaDQ8++CAXXXQR69atQ6vVEhoayg033ADA/v37cbvdpKen4/F4KCkp4dlnn6Wuro7Ozk6qq6uZPXs2q1atYv/+/eh0Ovr6+oiMjCQ0NBSn04lSqaShoQGNRoPFYmHKlCn85Cc/kcfNFBQU4Ovr+42pUr8LEViEIdPR0YHZbObCCy8kKiqKO+64g02bNrFz504SExNZuHAhTz75pFyTMRqNtLW10dbWRnt7u/y4y+WS1x5qaWmRA86UKVO46qqrOHToEFu3biU0NJTDhw/T19c3KhJ1FxQUsHTpUjQaDVdccQV+fn68++673H777UydOpXjx4/T09PDrFmziIyM5Mknn+SHP/whX331FUuWLOHyyy9n+fLlBAYGolKpCAkJ4ZZbbmHKlCnceuutBAcHc+utt+J0OmlsbOTjjz8mNTWVpKQk5s6dyz333ENiYuKQ7Nu46W4Wzj+1Wk1gYCA+Pj5ERkai0WhQq9WEhYXx1Vdf0draKi83CxAcHNzvtd62B+/I0TVr1jBr1ix6enowGo309fVx4MABPB4Per0eh8OBr68vgYGB/bLpjVQnr2rp/T03N1deiyk1NZW4uDjCwsKYMmUKubm5JCUlMXHiRN5//32ysrIoLCzkzjvvlAd49vb28qtf/Yru7m4qKipISEjguuuuo66ujri4OI4fP85TTz1FXV0db731FhdddBHx8fGDvm/jZoDc2RID5AaPN12mt8rt7+9PT08Per2elpYW6uvrmTp16lnVLrq7uyksLGTmzJmo1Wp5YbYdO3bIDb4Oh4N9+/aRnZ1NZGTkqB5G4Ha7eeCBB9ixYwf33Xcf1dXVLF68mPLycqKjo9m+fTv3338/69atw2g0smXLFv75z3+SmJjIa6+9RnNzM/v372f27Nnk5OTw4osvUllZSUJCAllZWSQkJNDd3c2TTz4pTxP4NiKD3HcgAoswnCRJorW1ldbWVmpqakhPTycyMpKQkBDeffddgoODmTVrFh6Ph/Xr11NQUMDOnTtZvHgx//znP/n+979PWloaJpMJl8slD+cvLy8nPDycyZMnyyN0Fy1aBHDWAXgg151oYxGEYeKd5+PtirdarVRXV7N8+XJaWlq4/PLLKSgooK6ujoaGBm677TZ5qd8//OEPxMXFkZGRgVqt5sUXX8RsNmOz2Th8+DDBwcGEh4cTHR1NTk4Ot99+OxMmTKCxsZH4+Hh5YOhQ1epEYBGEYaLT6dBoNHR3d3PgwAGqqqqIiopi2bJlaLVaXn/9dQD51s/pdLJkyRICAwN5/PHHmT9/Pmq1moSEBO6++25uu+02fH195dG3kiSxfft2/Pz8eP3119Hr9Vx88cX4+PgQFhY2pPsmboVOIW6FhPNBkiQ5qZJGoyEqKqpfUiW3243T6ZQHCj7//PNMmTKFOXPmsHPnTqxWKwsXLvzWz6iqqiI+Pp7e3l6Cg4PlwYPems9ABhOKSYiCMEK53W4aGhro6OggPDyczMzMMyZV8vHx6XfRP/jgg/I2s2fPPu197XY7kiT1y+vizQIHnLYO9969ezEajSxevHjQ9u1kIrAIwnngcrlobW3FaDTi5+fHxIkT+408/jYnT384U7vIvn37cDgcLFiw4Kzeb/r06d+4zvh3JQKLIAwRSZKw2WzU1tbidDqJiYkhLy9vSBpML7zwwgFtP9SL142bwCJJEr29veh0OrnqeXKioNE85kEYWSRJoq+vj6amJvr6+oiJiRn142oGatwElqamJnp7e6muruayyy5DoVBQXV1NYWEheXl5/RZhF4Rz4V02o62tDZVKRUJCAr6+vuMqoHiNm8ASHx9Pe3s7R44ckR8LDg4mJyeHzs5OeRLc8ePH5TWTBeFseDweOjs7aWpqQqFQkJGRgV6vH+5iDatxE1ja29vZvn07BQUFWCwWSktLycjIQKVSyTM8s7OzycjIGNJGLWHs8DbIepfNKCgoEBkJ/2vcBBaNRsPEiRPRaDRotVrS0tKAE4OPLrzwQnEyCGfFO1CtqamJ1tZWuUH25F4bYRwFlqCgIIKCguTfvf36Q5VMWBhbvEPu29ra6OvrIzY2lqSkJBFQvsaAjorL5eLXv/41KSkp+Pr6kpqayvLly/F4PPI2kiTxxBNPEBsbi6+vL/Pnz6e0tLTf+9jtdh5++GHCw8Px8/NjyZIlNDY2Ds4eCcIgkiQJs9nM4cOH5Tk4ubm5hISEiKDyDQZ0ZP74xz/yyiuv8MILL1BWVsZTTz3F008/zfPPPy9v89RTT/HMM8/wwgsvsH//fqKjo7n00kuxWCzyNsuWLePDDz9k7dq17Nixg56eHq666qpzWnxaEIaCd8h9eXk5BoOBrKwsZs6cSUhIiLhtPhvSAFx55ZXSPffc0++x66+/Xvre974nSZIkeTweKTo6WnryySfl5202mxQUFCS98sorkiRJktFolNRqtbR27Vp5m6amJkmpVEobNmw4q3KYTCYJkEwmU7/HPR6P5Ha7B7JLp3nuueekjz766Du9hzB6uVwuqbm5Wdq+fbt09OhRyWazDXeRRoyvu+7OZEBtLHPnzuWVV16hoqKCzMxMiouL2bFjB88++ywANTU1tLa2ynkeALRaLfPmzWPXrl3cd999ci7Tk7eJjY0lPz+fXbt2cdlll532uXa7vV9GMLPZLP/fZrPxr3/9C71eT3t7O3q9nry8PObOnSu+WYSz5nK56OzspKOjg5CQEDmhlHBuBnQr9Itf/IKlS5eSnZ2NWq1m8uTJLFu2jKVLlwLQ2toKQFRUVL/XRUVFyc+1trai0WgICQn52m1OtXLlSrnxNSgoiISEBPm5zs5O/Pz8mDRpEv7+/tx00028++67ostY+FaSJGG32ykrK2PPnj14PB5ycnKIjY0VQeU7GlCN5e233+bNN9/krbfeIi8vj6KiIpYtW0ZsbCx33XWXvN2pNQXppPV1v843bfPYY4/xyCOPyL+bzWY5uPj7+xMYGEheXh65ubnYbDZuvPFGUVsRvpb03zk8LS0t2O12oqOjycrKEo2xg2hAgeVnP/sZv/zlL7n11lsB5OxWK1eu5K677iI6OhpA7t/3MhgMci0mOjoah8NBd3d3v1qLwWA443RwOHE79XV5UYOCgliyZIn8u5+fHxdddNFAdksYJ6T/NshWVFTIY5n8/f3Fl9AQGFCI7uvrOy2q+/j4yN3NKSkpREdHs3nzZvl5h8PB1q1b5aAxdepU1Gp1v21aWlooKSn52sByNtxuN42Njbjd7n7d34LgDSjHjh2ju7ubvLw8Jk6cSEBAgAgqQ2RANZarr76aP/zhDyQmJpKXl0dhYSHPPPMM99xzD3DiFmjZsmWsWLGCjIwMMjIyWLFiBXq9nttuuw04UcO49957efTRRwkLCyM0NJSf/vSnFBQUfGtGrK/jcDh49tlnaWtr49FHH2XDhg18//vfFyfNOOdNqtTc3Ex0dDTp6eny0hrC0BpQYHn++ef5zW9+wwMPPIDBYCA2Npb77ruPxx9/XN7m5z//OVarlQceeIDu7m5mzpzJpk2b+qXd+8tf/oJKpeLmm2/GarWyYMECVq1adc5r7rpcLiRJIj8/n+rqaqqrq8/pfYSxweVyYTAY6O7uJjAwkGnTpqHRaIa7WOPKmMh5K0kShw8fpqKiApVKxUUXXXTOyYJFztvRydsgW1VVhcPhID4+noiICFE7GUTjbvmP6upq3njjDQoKCkhKSqKwsHC4iyScJ9J/kypVVlZSU1NDZGQkkydPHneJlUaaMTEJ0WazERMTQ1dXFwB5eXnDXCJhqEn/TarU2NiITqcjISEBvV4vgskIMSYCS3Z2NocOHZIXf/KOTRAn2djj8Xjo7u6mtbVVTqrk5+c33MUSTjEmAktfXx/Nzc2kpqaSl5fHu+++y/z584e7WMIgcrlcNDU1YTQaCQkJISsr64zLZggjw5gILHq9nunTpxMbG8uePXuYNWvWadvY7XaKi4tRKpVMmTIFpVKJwWCgsrKSmJgYUlJShqHkwrdxOp00NzfT2dlJcHAw+fn559x7KJw/Y6Lx1uFw0NLSQmZmJjfccEO/CYteRUVFAJhMJgwGAwCHDh0iPT2dL774AqvVSlVVFXV1deJbcJhJ/11RobKykoqKCgICApg8eTKpqakiqIwSYyKweDweKioqMJvN1NbWnrFXyOl0olar0el0uFwueXSur68vTqcTSZKwWCz09PSc7+IL/yVJEj09PZSVlVFSUkJISAi5ubmEhoaKYD/KjPpbIUmS0Ov13HDDDbz//vsA/PCHPzxtuylTprB3716USiUhISEcP36cnJwc9u7dy7x58/D19WXSpEnk5eWJmdHnmSRJdHV1YTAYUKlUpKSkoNPpRDAZxUZ9YLFarezdu5euri55dTeLxdJvEiScaIe5+OKL5d+zsrIASEpKOn+FFfrxeDy0t7dTU1ODv78/6enpp60xLIxOoz6wqFQqAgIC8PX1lR8LDg4evgIJ38rpdNLR0UFXVxchISFMmzZtyJf8FM6vUf/XVCgUBAUF4XK5qKurQ61Wi3khI5AkSTgcDurr62lpaSE1NZWcnByRA2WMGvWBRa1Wk5SUxEsvvURTUxNKpZK2tjaWLl0q7tFHAOmkZTO8I6TT0tJEQBnjRn1ggf+3It0jjzxCZ2cn69atw2q1jtt1c0cC6b/LZlRUVKBUKklPTycwMFD8PcaJMRFY1Go1eXl5fP7553g8HqKjo9m+fTsLFy4U4x7OM4/Hg8Viobm5Ga1WS35+vujhGYfGRH3Ux8cHf39/KioquPTSS4mOjmbRokUiqJxH3qRKO3bsoL29nbS0NHlhOxFUxp8xEVj6+vo4ePAgOTk52Gw2tmzZIsainCcul4uWlhaOHTuGSqXiggsuID09HY1GIwLKODYmboX0ej1Tp05l+/bt1NbWctNNN4mTegh5Z5AfP36cvr4+kpOTRQ+P0M+oDyySJFFcXExnZydXX301hw4dErNeh4g3S1tTUxMul4uYmBjCwsLEsRZOM+q/Yjo7O3n++eeZO3cub7zxBjfeeCOTJk0a7mKNKd6kSvv376e6ulpehyc8PFwEFeGMRn1g8d7jr169GqfTyT/+8Q8OHTo03MUaEzweD11dXZSVldHV1UVOTg55eXliLR7hW436W6GIiAhef/31fo21QUFBw1ii0c/tdlNfX09nZyfh4eFkZmaK20thQEZ9YPHx8ZFXYBS+G6fTSWtrK0ajkYCAACZNmiTm8AjnRJw145w3y319fb3cIBsXFyd6eITvRASWccqbpa2pqQmr1UpcXJxojBUGzbgJLG63m+7ubvR6PXq9HjiR0tJoNOLr69tvpcaxTJIkOjo6MBgMqNVqEhISxOhYYdCNm8DS1tZGR0cHnZ2dzJs3D6VSSWVlJXV1deTl5Y35wOLxeOjo6KCpqQmVSkV6enq/HDaCMJjGfGDp7e2lubmZ+Ph4/Pz8cDgc8nNRUVG0t7dTW1tLQkICx44d4/jx4yQnJw9fgQeZ0+mkra0Nk8lEUFAQBQUFokFWGHJj/gxTKpVotVpaW1sxGAxMnToVh8NBQ0MDAQEBZGVlyRn8s7OzSU9PH/XzjCRJwul00tjYSGtrK3FxcWLIvXBejfnA4uvrS2JiIm63m8TERHx8fPDx8SElJQVJkuju7mbBggVyG8Nobmvw9vAYDAasVisxMTEkJyeLgCKcd2M+sHidnEJBoVDItwORkZHDVaRB4126pKqqCqfTSWZmJsnJyaM6SAqj27gJLGORJEkYjUba2tpQq9Xk5OSg1WpFQBGGnQgso5Db7aa1tZWqqioiIyNJTU1FrVaLgCKMGCKwjCInL5sRFhbGrFmzUKvVw10sQTiNCCwjnHfZjKqqKrq7u8WyGcKoIALLCOVNqtTc3IzT6SQ2Npbs7GwRUIRRQQSWEcbbIFteXo5eryclJUXkPxFGHRFYRgiPx4PJZKKlpQVfX1/y8/Px9/cf7mIJwjkRgWWYeZfNaGlpITo6moyMDFQqlaihCKOaCCzDxLt6o3cOz7Rp00QPjzBmiMByHnnXMa6srMTtdhMfHy96eIQxSQSW88AbUBobG3E6nURHRxMRESFud4QxSwSWIeRNqtTY2IherxdJlYRxQwSWIeDxeOjs7KS1tRUfHx+ysrLkrHWCMB6IwDKIXC4XjY2NmM1mQkJCyMnJEUmVhHFpXJ/1Jyd0OtfbE0mScLlcNDU10dnZSWhoKHl5ef3SNAjCeDNuAovdbqe0tBQ/Pz8yMzNRKBR0dnZSV1dHUFAQaWlpAwou3iz3ra2t2O12YmJiSEpKEu0ngsAYWGL123g8HhwOBz09PTQ2NmK1WuWayr59+0hOTmbLli3YbDY5feU3BQdvUqXS0lLKysoIDQ0lNzeX0NBQEVQE4b/GfI3FZDJRUlJCbm4uixYt4t133yU6OhqtVouPjw9qtVpuB9FqtWi12jPmvPUujN7R0SFnuRdJlQThzMZ8YAkJCeHCCy+krq6O4uJiJk2ahE6no7i4mFmzZlFUVMT8+fPR6XT4+vqetu6zx+PBYDBQW1tLQEAA6enpaDQaEVAE4RuM+cDilZSURFJSkvz7vHnzALjooovOuL3D4aCzs5Ouri5CQ0OZNm2a6OERhLMkrpRTKBQKdu3aRWNjI6mpqcTExGAwGDAYDMNdNEE477zjsAY6j00EllNcf/317Nu3DzhRa6mrqzttm40bN5Kfn09cXNz5Lt45kSRJvg0cLQ4dOoSPjw8TJ04c7qKcFZvNxsaNG1myZMmouU1uamrC5XL1q8mfSq1Wk5aWNuDAopBG4epcZrOZoKAgTCYTgYGB5/3zvUmsR8uyrJIk0dbWRnR09HAX5ay1tLSgVCqJiooa7qKcFZfLRXV1NRkZGaMmsFgsFjwez2ntil9nINedqLGcJUmS2Lx5MxqNhilTphAQEIDb7aaoqIjOzk5mzZo14gJNa2srZWVl+Pj4MGfOHABqa2spLy8nPj6evLy8YS7h/yNJEjU1NXLQnjBhApIksWPHDtxuN5MmTTrrC+B8cblc7Nu3D6PRyNy5c8nMzKS7u5t9+/YRHR3NhAkTRlyQqauro6ysjPj4ePLz85EkifLychoaGsjNzSU2NnZQPmfMj2MZDJIkIUkSzc3NwInkTHBiXWiHw0F6ejoNDQ3DWcR+vOWNiopi5syZtLe3y13oJpMJu90+4oIgwMGDB5k+fTpHjx7F5XLhcDjo6+sjJyeHQ4cOjbilb318fJg5cyZKpRKTyQScGIhps9mw2WzDXLozM5vNp/39GxoamDVrFsePHx+0zxGB5SwcO3aMuro6rr32WlJSUqioqMDpdKJQKHC73ZjN5hGVU8VkMvHVV1/R3NzM22+/LVfPrVYr6enpzJo1i6qqquEu5mnUajUWiwU4sdSJy+Wir6+Pvr4+/Pz8hrl0p/N4PBQWFpKQkEB0dDR2u52goCBmz55NeXk5vb29w13E06SlpTF79myqqqpwu914PB6AQT+Hxa3QWcjJyZGrjL29veTl5dHT00NAQADR0dGYTCZSU1OHu5iy4OBgLr74Ytrb20lNTUWpVOJ0OmlpaUGlUmEwGMjJyRnuYvajUCiYPXs2tbW1zJo1i76+PuDEse/u7h6RtxWSJOF2u+nr66OnpwebzYbH46GlpYXJkyePyGDY2dlJW1sbOTk52Gw2lEolubm5NDY2MmHChEH7HNF4Kwy7U0/BkRZAhBNE460w7L744gssFgs+Pj5ccskl3/jt7XQ62bJlCwEBAVitVhYsWPC12xYXFxMaGkpCQsIZn29qauLgwYN4PB5mzJgxaI2RwsCIwCIMidWrV/PjH/+Y//znP+h0OqKjoykqKiIyMpLZs2fzzjvvoFKpCA4OZtGiRbS1taHT6XA4HFRWVrJhwwaCg4NZsmQJH3/8MSaTiZtuuolXXnkFf39/7r33XlpaWmhoaCAnJ4dp06ahUCjkNo7Kyko++OADHnrooeE+FOPSyGlxFMYUl8vFl19+SWtrK9nZ2Tz55JNkZGRw4MABiouL2b59O9deey3Hjh2jpaUFODF2pba2lhdffJHrrruOa6+9FpfLhcVioa2tjY0bN5KSksKCBQuw2+28++675OTk8NZbb8mNvn5+fqjVaj777DMWLVo0nIdgXBM1FmFIqNVqrr76akpKStiwYQNqtZrjx48zYcIEkpOTcblcdHR0YDKZTuuN0Ol0NDY24uPjw/bt2/H19WX69Ol0dHSg1+tpb28nMjISk8lEY2MjV155JTqdDjgxdufvf/87F1xwAaGhocOx6wKi8VYYIkVFRWRmZqJUKqmqqiIiIoKSkhK0Wi35+fk8/fTTLFiwgICAACZNmiSPtHW5XAQFBVFYWEhQUBBJSUkcPnyYgIAAYmNj8fPz49ChQ2RlZWEymWhrayM0NJSCggIUCgVtbW0UFxej0WiIiooacb1fo9lArjsRWITzzuVyceTIESZNmiR6gEYR0SskjGgqlYrJkycPdzGEITQqA4u3kmU2m4e5JIIwfnivt7O5yRmVgcXbA/B1YxkEQRg6FovlWyeEjso2Fo/HQ3l5Obm5uTQ0NIh2lq9hNptJSEgQx+hbiON0dryJ5GNjY791XtGorLEolUo5yVJgYKA4Gb6FOEZnRxynb3e2qSvEADlBEAadCCyCIAy6URtYtFot//d//4dWqx3uooxY4hidHXGcBt+obLwVBGFkG7U1FkEQRi4RWARBGHQisAiCMOhEYBEEYdCJwCIIwqAblYHlpZdeIiUlBZ1Ox9SpU9m+fftwF+m8WblyJdOnTycgIIDIyEiuvfZaysvL+20jSRJPPPEEsbGx+Pr6Mn/+fEpLS/ttY7fbefjhhwkPD8fPz48lS5bQ2Nh4PnflvFq5ciUKhYJly5bJj4njNISkUWbt2rWSWq2W/v73v0tHjx6VfvzjH0t+fn5SXV3dcBftvLjsssuk119/XSopKZGKioqkK6+8UkpMTJR6enrkbZ588kkpICBAev/996UjR45It9xyixQTEyOZzWZ5m/vvv1+Ki4uTNm/eLB06dEi6+OKLpYkTJ0oul2s4dmtI7du3T0pOTpYmTJgg/fjHP5YfF8dp6Iy6wDJjxgzp/vvv7/dYdna29Mtf/nKYSjS8DAaDBEhbt26VJEmSPB6PFB0dLT355JPyNjabTQoKCpJeeeUVSZIkyWg0Smq1Wlq7dq28TVNTk6RUKqUNGzac3x0YYhaLRcrIyJA2b94szZs3Tw4s4jgNrVF1K+RwODh48OBpSZIXLVrErl27hqlUw8u7tKc3v2tNTQ2tra39jtH/3879u6YOxVEAP2AaKR2kpWAoQtHZqelUSgXrVv+B4l8gRiz9B7p2LnQsrp0cOtbiD3Arihjp3OqgiOCmoLTfNzW8VN8bSqJEzwey3FwkOcMJycXr9/sRi8WsjOr1OmazmW3OwcEBotHo2uWYyWRwcXGBRCJhG2dO7vLUv5uHwyE+Pz8RDAZt48FgEP1+f0VXtToiguvra5yeniIajQKAlcOijD4+Pqw5qqpid3d3bs465fj4+IhGo4HX19e5c8zJXZ4qlm8/90kVkY3cO9UwDLRaLdRqtblzv8lonXLsdrvI5XJ4fn62dvBfZNNzcounXoX29/fh8/nmnhaDwWDuybPustksnp6eUC6XEQqFrHFN0wDgvxlpmobpdIrRaPTPOV5Xr9cxGAyg6zoURYGiKKhWq7i7u4OiKNZ9bnpObvFUsaiqCl3XUSwWbePFYhEnJycruqrlEhEYhoFCoYBSqYRwOGw7Hw6HoWmaLaPpdIpqtWplpOs6tra2bHN6vR7a7fba5Hh+fg7TNNFsNq3j+PgYqVQKzWYTkUiEOblphR+Of+V7ufnh4UHe3t7k6upKdnZ25P39fdWXthTpdFoCgYBUKhXp9XrWMR6PrTm3t7cSCASkUCiIaZpyeXm5cBk1FArJy8uLNBoNicfja7+M+veqkAhzcpPnikVE5P7+Xg4PD0VVVTk6OrKWWjcBgIVHPp+35nx9fcnNzY1omiZ+v1/Ozs7ENE3b70wmEzEMQ/b29mR7e1uSyaR0Op0l381y/SwW5uQe7sdCRI7z1DcWIvIGFgsROY7FQkSOY7EQkeNYLETkOBYLETmOxUJEjmOxEJHjWCxE5DgWCxE5jsVCRI77A9PFy9d6oNFRAAAAAElFTkSuQmCC",
            "text/plain": [
              "<Figure size 640x480 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "rna_pairwise = cv2.imread(f'{results_dir}/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_RNA_pairwise_minThreshold.png')\n",
        "\n",
        "rna_pairwise_no_thresh_resize = cv2.resize(rna_pairwise, (600, 1000))\n",
        "\n",
        "plt.imshow(rna_pairwise_no_thresh_resize)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "In the following two figures you can see that this trend continues for a threshold of 10 barcodes per insert.\n",
        "- DNA correlation: 0.31 - 0.39\n",
        "- RNA correlation: 0.79 - 0.80"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAARYAAAGiCAYAAAAiFuksAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/TGe4hAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB2BElEQVR4nO3dd5hb1Z34/7c0KiNN77335l6wcQODCc10CCSEBJKlJ16STcJmNyF8sxjMLwRCiZcSG4LB9Bpwobkb97FnPMXTq6aPNEVd5/eHo7seF7DNjDXlvJ5nHnuurqRz71x9dO4pn6MSQggkSZKGkdrXBZAkafyRgUWSpGEnA4skScNOBhZJkoadDCySJA07GVgkSRp2MrBIkjTsZGCRJGnYycAiSdKwk4FFkqRh59PA8txzz5GWloa/vz/Tp09ny5YtviyOJEnDxGeB5Y033mDZsmX87ne/Y//+/cyfP59LL72UhoYGXxVJkqRhovLVJMTZs2czbdo0/va3vynb8vLyuPrqq1m+fLkviiRJ0jDR+OJNHQ4He/fu5be//e2Q7UuWLGH79u0n7G+327Hb7crvHo+H7u5uIiIiUKlUI15eSZJACEFfXx/x8fGo1d98s+OTwNLZ2Ynb7SYmJmbI9piYGEwm0wn7L1++nD/+8Y/nqniSJH2DxsZGEhMTv3EfnwQWr+NrG0KIk9ZAHnzwQR544AHld7PZTHJyMo2NjQQHBw97uWpra7FareTn55/2c44cOUJxcTE6nY7AwEAWLVqkRHWbzcZHH32E0WjE5XIxc+ZM4uLiKCsro7a2lu9973sMDAxw8OBB5s2bhxCCAwcOIIRg6tSpI1Irq6+vR6/XExsbO+yvfTIlJSXs2bOHa665hs8//5xrrrlG1jbHGIvFQlJSEkFBQd+6r08CS2RkJH5+fifUTtrb20+oxQDo9Xr0ev0J24ODg0cksFRVVWEymTjvvPOAo7duX3/9NS6Xi4yMDJKSkqiqqqKpqYmAgACCg4OJjIzkvPPOIzg4mDvvvJNFixYpZXO5XPzzn//kxRdfxGQysXLlSn7yk5+wd+9ePvjgAwIDAyksLOSDDz7gsssuw2KxsHLlSgIDA5k5cyYBAQFK2SwWC3v27EGtVpOamkpkZCS7d+9GCEFqaiqJiYkcPHiQ6dOn43A46OnpwWg0Ul5ejt1ux8/Pj0mTJvHJJ59gMBi48sorAairq8PPz4+pU6cqF05VVRV9fX2YzWZiYmLIzc2lra2NsrIyALKysggICKCqqgqr1UpWVhZxcXEANDc3U1FRgUajYdasWQgh6OjoICoqiqlTpxIYGMiuXbuw2Ww4HA7OO+88dDodu3btwu12k5ubS1xcnAw+o9Dp/E180iuk0+mYPn06GzduHLJ948aNzJ071xdF+kYvvvgiLS0txMfHs3z5cioqKnj++edJTk5m//79rFmzhpSUFPLy8ujp6SE3Nxd/f/8hr6HRaAgNDSU/P58LL7yQdevWAXDTTTexfv16enp6gKO1to8//pjFixczbdo0NmzYgLd93ePx8Nxzz1FfX09aWhqhoaGsWbOG2tpawsLCePDBB+no6OCxxx5DCIHZbGbnzp3U19fz2GOPkZCQwGuvvUZVVRVRUVHEx8cTGhrKH/7wB4QQxMbGDrlo3n33Xb788ktSUlJ44YUXOHz4MI899hg6nY6AgACeeeYZysvLefzxx4d8k/X19bF8+XJiY2Pp7u7m2WefVV6zr6+Pd999l5KSEj7++GOSk5N57rnnqK6u5o033sBkMhEdHc3DDz/MwMDAiP5dpZHjs+7mBx54gBdffJG///3vlJWV8e///u80NDRw1113+apIp1RSUsKUKVPIzs7G6XRSVVWF2+0mLS2NwsJC4GhAOHToEG+++Sb//d//fdIalne/mpoaQkJCAAgLC+Paa6/lb3/7G0II7HY7H3zwAXA0GL311lvYbDYA3G43Bw8eZOnSpSQnJxMcHExpaSlLlixh0qRJREVF4XQ6lfcaGBjA7XYDkJiYSHp6Ov7+/ng8HiIiIoiKiiI0NJSf//znVFZW8re//Y3e3t4h5c3JySE1NZWQkBDKyso4ePAgR44cobKykhkzZiCEICsri9TUVAIDAwHo7+/HbDaTl5fH1KlTKS0tPeE8NDQ0cMEFF5Cenk5ubi4AO3bsYObMmRQUFNDX16cctzT2+KyN5aabbqKrq4uHH36Y1tZWCgsL+eSTT0hJSfFVkRQ6nY7a2lo2btyIVqvl0ksv5aOPPiIrK4vExEQWLFjAF198wUsvvURdXR1qtZqdO3fypz/9iZ/85CccOHCA8847T6m1qFQqnE4nGzduZMuWLQQEBHD77bfz5ptvotPpWLBgAZ999hmdnZ2Ul5dTUFDAjTfeiEqloqenh0OHDjFr1iw0Gg3XX389Tz31FPPmzSMpKYlLLrmEP//5z8yePZvo6GiioqLIz8/n7bffZteuXcyYMQM/Pz+MRiMARqMRPz8/wsPD2bp1KwkJCdTU1JCens7hw4dPqCVs3rwZPz8/rFYr8+fPx2Qy0dvbq3z4tVotBoNhSE0nIiKC7Oxs3n33XZqamrjqqqvQaDTKfgaDgenTp/P4449jtVo5cuQIcPSaeOmll5g6dSoFBQUjcpsrnRs+G8fyXVgsFkJCQjCbzSNy8VmtVuWbW6VSERkZSXd3N263m9DQUPz9/SkrK+PIkSPs3r2befPmsWDBAsxmMwBqtVppR4KjtzAdHR14PB7UajVRUVGo1WoGBgbw8/PD39+f/v5+nE4nRqMRh8Oh3FYMDg4qH0Y4WuPp6urC6XQSGBhIQECA0ssWERGBTqdjcHAQs9mMv78/BoMBPz8/BgcHCQkJwWKxKNs6OjqU93M4HGg0GiIjI5Ug8ec//5nMzExmzZpFYGAgQUFBeDwe5f38/f0JCAjAZrOd8Hew2Wz09PQox+tyubDb7RiNRgYHBwkMDGTv3r00NTWxdu1a/vjHP5KVlaWcp/Dw8FPW+iTfOJPPnQwsZ0EIQWtrK2azGZ1OR2pqqhJExpPOzk78/f2VW5zhJISgvr4eq9WKXq8nLS1NNtSOcmfyufNpd/NYpVKpiI+PJz4+3tdFGVGRkZEj9toqlYrU1NQRe33Jt+Ts5pNwuVwMDAwMafw8FY/Hw4EDB7BarTQ3N/NNFUCPx4PL5Trl40IIBgcHsVqt3/g6p2K1Wunu7lZey2azUVlZSV9f31m9niSdLRlYTmLbtm28/vrr/PnPf+brr7/G7XZTWVlJdXU1brebzs5OqqqqqK+vx+l08vDDD2OxWKiqqgKgurqaw4cP43A4qKqqoqysDIfDwcGDB3n11VdpampicHCQ0tJSWltblQ+9w+Fg/fr1/OxnP8PhcCjl8Xg81NTUUFZWhs1mo7e3l4aGBsrLy3E4HLjdbsrLy9m/fz8HDx5Unvf111+zadMm/vCHP1BXV3dOz6E0sclboZMYGBigvb0dl8uFy+XizTffpLq6msDAQCZNmsTevXuZPXs2X331FVdccYXSTVxXV4fD4aC0tJQFCxbgdrupq6ujtbWVzz77jJkzZ9Lb24vVauVPf/oTmZmZ1NbWct999xETE4Ner+fSSy/llVdeGVLDGBgYoLi4mO7ubnbu3InBYKCkpITCwkIOHDiA0+lU3mv27NnA0VuNhQsXMmPGDB5++GG0Wq2vTqc0Ackay0mo1Wqio6PRaDT09vZSVlZGZmYm+fn5pKenExQUxOzZs8nJyVFqCEIIPB4PlZWVXHbZZUydOpWWlha+/PJLDAYDGzduJDQ0lJSUFBISEmhvbyc+Pp558+Z9a+Noa2srfX19GAwGuru78fPzY/78+RQVFbFt2zaKi4u58MILWbBgARqNRilPW1sbr776Ktdeey0JCQkjft4kyUsGlpOIiIhg6dKl/OY3v6G2tpZrrrmGmpoaGhoaaGtro6+vT6nFXHbZZWRlZaHX64mMjOTKK6/kzTff5I033lDmPlksFjIzM4mKiqKkpIR9+/Zx5ZVXcuTIEWU4PBxt2/nyyy/x9/dn27ZteDweAAwGA729vTQ3NxMbG0t4eDihoaEYDAZSUlL40Y9+xOuvv87OnTuHzOP43//9X2w2G1VVVXR2dvrkXEoTk+xuPoljT4k3OHg/5ABvvfUW1157LRqNRnlMrVafsK9arVb+r1KplMeO/b93P5VKpdR6vE623fvcU5Xv215Lks6W7G7+jo79AHr/7x2nIoTguuuuU4LKsY8d//vx//+2x1Qq1UnHw5xq+6ne89ueI0kjTQaWM6RSqWRDqCR9C9nGcpydO3dSXl7u62JIkk95PB5aWlqGDF84EzKwHGfXrl1UVFT4uhiS5BNut5vW1lYOHz6MSqUiLy/vrF5nwtwKeceaqFQqdDqd0uDZ09NDUFAQOp3O10WUJJ+x2+1KYq/U1FTy8vK+UxvdhAksvb29fPLJJ+Tm5jJt2jQAtm7dSmBgIHV1dVx11VVKb4okjSfH9xB6G/yFEDidTpqamnA6ncTGxpKfnz8svYcTJrD4+/sza9Ys9u/fT3Z2NoGBgdhsNubMmcOWLVuwWq00NTVRXV1NWlqar4srSWelurqa/v5+Jk+eDBytifz973/n448/xmw2ExQUxPnnn4/dbufQoUMsXbqUG264gcDAwGEdjjBhAktPTw8DAwMEBQXR39/P119/TXx8PPv27WPy5MkYjUZyc3PJzMyUtRZpzDIajUqA8Hg8DAwMEBISQnZ2NosXL+bhhx/m0UcfxWazkZiYyJ/+9KfTSo59piZMYAkLC0On05GTk4NOpyMoKIiAgAB6e3sJDAz81nVSJN8QQmAymWhubiY+Pl5JsN3X10dlZSVqtZqCggLsdjsVFRXK7xM1SZQ3mTkcnWO2evVqfvzjH9Pe3k55eTmpqalMnz6dwMBA3n//fZqampT0qsNpwnyaDAYDkZGRSvY0b9UvLCxMjks5x5xOJ6+//jqff/65MotbCHHCj3f78uXLsdlsPP3001gsFuDoSGK9Xk9xcTFfffUVf/vb36iurmbLli3U19f77NhGCyEERqOR6667jv379xMcHMyNN97IwMAA0dHRpKSkcNlll3H++eePyPtPmBqLNHo4nU5Wr17N6tWrlaC+d+9eNm/erOxzww03kJSUhBCCjIwMzjvvPN577z3q6uqYPHkyAQEBuFwuSktLSUhIIDMzE5PJRGVlJTfeeKOvDm1UGBgYoL6+Ho/HQ2xsLBdccAFqtRq3283vfvc7Ghsb6e3tJSEhQWl3GW4ysEg+YTQahyyRGx4eTk5OzpDH4WgPRmNjIzU1NQQHBxMREUF5eTlxcXGkpKRw0003sX37dvr7+zn//PPxeDzs2rWLq666yifH5StCCKxWK42NjVitVhISEobkL4ajk1y3bdvG1VdfjVqtJiYm5oRlaoaLDCzSiHO73bS0tJCYmKhMibj55puHjJNIT08nPT39hOeqVCoeeOAB3n77bfLy8lCr1VRVVREYGMg777yDy+ViyZIlREdHs3r1apxOJwsXLsRms43Yh2Y08S4E19nZiUajITExcUgD7rG0Wi033HADcXFxI94GNWHaWKRzr6Ojg08//ZTNmzezbNkyPv30U6qqqlCr1SQkJPDRRx8pbSanolKpCA4Opre3F4PBwOrVq/ne975HYmIi119/PfPnz+f9999ncHCQK664gkWLFvHkk0+O+zWJhBB0dnZy4MAB2traSE1NVVamPFW3sXf1zHPRsC0DizRiGhoaWLNmDenp6fT395OZmcmjjz7K1q1b+frrrwkLC+PVV19VGms//vhjnn32WZ599ln+/ve/K3lqent76ezs5Hvf+x579+7FbrcDEB8fT09PDw0NDeh0OgoKCnC5XOTk5CgLwo03TqeTxsZGDh8+jN1up6ioiKKiolPWUnxF3gpJIyouLk5ZYzsjI4OBgQE6OzuJiooiPDx8yLyssLAwJWjo9XplCIBWq2VgYIDu7m5CQ0MZGBjAZrOhUqm44IILaGpqYtOmTVx//fW88cYb/OY3vxlVH7Lh4HK5aGxsxGQyERsbS05OjpItcDQavSWTxrzo6Gjmz5+PXq/n8ssvR6VScemllzJnzhx27tzJ9u3b+e1vf6sEgVN1fUZHR/PjH/+Yd955h2XLltHT00N/fz/9/f00NDTg5+fHNddcw+DgIPPnzx8Vq2kOB++qDW1tbdhsNqXBeiyMuZIZ5I7z17/+lZSUlAnXqyCNHkIIBgYGqKmpYXBwkOzsbMLCwnxeC5MZ5CRpDPJ4PFgsFkwmE1qtlqysLPz9/X0eUM6GDCyS5GPeFRWqq6sJCQkhMzMTvV4/JgOKlwwskuQjLpeLjo4Ourq6iIyMZPbs2aO6QfZMjI+jkKQxxOFwUFtbS3t7O+np6d85qdJoJAOLJJ0DQggcDgdNTU04HA7i4uLIzs4e07c730QGFkkaYb29vVRUVKDX65WVNMdrQPGSgUWSRoAQArPZTEtLC/7+/hQVFWEwGMZ9QPGaMIGlubmZI0eOoFKpmD9/Pmq1mpaWFkpLS0lOTh4ys1aSzpbH46GxsVFZDjczMxOtVjthAorXhAks8fHxhIWF8eWXXyrb+vr6cDgc+Pv7I4Rg//79HDp0iNTUVN8VVBpzvMmqW1tb6enpISwsjFmzZuHn5zfhAorXhAksg4ODvP3228yePRshBDabjaSkJMLCwjhw4ADJycmEhYWN28lr0siwWq0cOXIEj8dDfHw8BQUFY2LI/UibMIHFYrGQnJyMw+HAZrPR3t4OHJ3aX1BQgEqlIi0tjeTkZJlMW/pG3i+mxsZGHA4H8fHxQ5JWSRMosMTFxQ1JNOxd4kMu9SGdLm9SpZaWFgwGwzcmVZroJkxgkaSzJYSgu7ublpYWNBoN2dnZE6qH52zIwCJJp+BdJbC/v5/Q0FByc3Plig6nSQYWSTqGEAK3201zczOdnZ2EhYUpQ+5lDeX0jfnAIoTg8OHDFBcXc+mll2IwGDCZTLLLWDpjfX19tLW14XA4iI2NJSkpSfbwnKUxf9a6u7t5/fXXCQoK4oknnqC0tJSDBw/6uljSGOFNqnT48GHKy8sJDQ0lLy+P8PBwGVS+gzFfY/F4PGRkZHDFFVcwf/583nzzTTkWRfpWHo+Hnp4eOjo60Gg0pKWljdmkSqPRmA8skZGR3HrrrQCEhoZy++23j/ulH6Sz5+0yrqurQ6/Xk52dLQPKCBjzgUWlUg1JjqPRaAgMDPRhiaTRyOFw0NXVRVdXFxEREUybNm3cJFUajcbFTaTb7eazzz7j+eefp6uri82bN8vRsxJwtMu4qqqKr7/+GqfTSV5eHnFxcTKojLBxcXatViv79u0jNTWV/v5+3n33XebNmyertxOUd8h9S0uLMuQ+IyNDXg/n0LgILP7+/sTFxfHWW2+xY8cObrjhBnkRTVAWi4XKyko8Hg/Z2dmEhITIa8EHxkVgGRwcxGq18tRTT2G32+nu7pYX0wTiXTbDO4enoKBANsj62LgILKtWreJ///d/2bFjB0II7rvvPl8XSToHvDlQampqiImJISMjA51OJwPKKDAuAsvdd99NUVERNTU1OJ1OQkNDfV0kaQS53W5MJhM9PT1EREQwZ84cOeR+lBk3vUJbtmzBbrcTFxfH119/LXuFxiG73c6hQ4fYs2cPGo1mSA+PDCqjy7iosfj5+VFYWIhWq2Xbtm0kJiaesI8QgtbWVgwGA6GhoahUKiU/aWRkJAEBAT4oufRthBDY7XYaGxtxuVwkJCSMinWMpW82LgKLWq0mOjqauXPnkpubS39//wn7VFdX09rayuDgILNnzyY0NJTdu3cTGBhISUkJl1xyybhbNGos8+ZAqaqqIigoiOTkZAICAmRAGSPGxa2QzWZj06ZNCCEIDg5m1apVJ9wKNTQ0EBISQmxsLL29vQCYzWays7Oprq7GZrNRXl6uZPKXfMMbUEpLS+nu7qaoqIi8vDwCAwPl32UMGRc1FqPRyKxZs1i9ejUWi4XbbrvthIuwoKCAvXv34nK5mDNnDuXl5URERLBnzx5yc3MxGo3k5eWRlZUl22d8wO12U1dXh9lsJjIykpycHJlUaQwb84HFYrHwz3/+k9bWVqUmolKpTggs0dHRXHjhhWg0Gvz8/IiMjASOjtr19/eXU+R9wLtsRnNzMz09PYSGhjJp0iTZwzMOjPnAotVqiY2NxWg0YrfbAQgLCzthP5VKhb+//wnbjUbjiJdROlF/fz8NDQ14PB7i4uJITEyUwX0cGfOBxWazsXbtWgYGBpRtUVFRpKen+7BU0skIIbBarTQ0NGC1WklKSpIJlcapMR9YQkNDee655/jggw8YHBzE5XKRkZGBEEJWp0cJj8dDR0cHXV1daLVakpOTZZb7cW7Mf1WoVCpsNhvV1dXceOONXHDBBaxcuRJANsL6kBBCSap08OBB2traSE1NJTMzU67FMwGM+RoLHG1n0el0vPbaa9hsNgICAnj//fe58sorZd4NH3A4HLS3t9Pb20t4eDiFhYXy7zDBnFGNZfny5cycOZOgoCCio6O5+uqrqaioGLKPEIKHHnqI+Ph4DAYDixYtorS0dMg+drud+++/XxnxunTpUpqams76ILRaLVdccQUqlYof/vCH3HnnnVx99dXyYj7HXC4XtbW17Nu3D5fLRW5uLvHx8fLvMAGdUWDZtGkT9957Lzt37mTjxo24XC6WLFkypOF0xYoVPPHEEzzzzDPs3r2b2NhYLr74Yvr6+pR9li1bxnvvvcfatWvZunUr/f39XHHFFbjd7rM6iMHBQVatWkVfXx/d3d38z//8j7wNOke8We5ramqorKwkLCyM2bNnk5qaKgPKRCa+g/b2dgGITZs2CSGE8Hg8IjY2Vjz66KPKPjabTYSEhIiVK1cKIYTo7e0VWq1WrF27VtmnublZqNVqsW7dutN6X7PZLABhNpuFEEI4HA7xzDPPiIceekg8/fTT4u233xYej+esjumpp54S77///lk9d6KxWCyiuLhYbN++XXR1dZ31OZfGhuM/d9/kO32lmM1mAMLDwwGora3FZDKxZMkSZR+9Xs/ChQvZvn07d955J3v37sXpdA7ZJz4+nsLCQrZv384ll1xywvvY7XZljAocHRR3fDmKiopITU2lpKSEyMhI2Tg4QjweD2azGZPJpGS51+v18nxLQ5x1r5AQggceeIB58+ZRWFgIgMlkAiAmJmbIvjExMcpjJpMJnU53wiC2Y/c53vLlywkJCVF+kpKSlMf6+vr44x//yO7du1mxYgX9/f3k5OSc7WFJpyD+NTt8+/bttLS0kJaWJtfikU7prAPLfffdx8GDB3n99ddPeOz4C02cxpiSb9rnwQcfxGw2Kz+NjY3KYxaLhdraWmbMmEFoaCjx8fFYrdazOCLpZFwuFy0tLZSWlqJSqTjvvPPIz8+XAUX6Rmd1K3T//ffz4Ycfsnnz5iG5T2JjY4GjtZK4uDhle3t7u1KLiY2NxeFw0NPTM6TW0t7ezty5c0/6fnq9Hr1ef9LHwsPDueuuu+jv72fmzJmYzWYcDsfZHJZ0DIfDQXV1NZ2dnWRkZCgLo0vS6TijwCKE4P777+e9997jq6++Ii0tbcjjaWlpxMbGsnHjRqZOnQocvUA3bdrEY489BsD06dPRarVs3LiRG2+8EYDW1lZKSkpYsWLFGR+AwWDgiiuuOOPnSScS/0qq1NTUhNPpJC4ujtzcXFkzkc7YGQWWe++9l9dee40PPviAoKAgpU0kJCREGaK9bNkyHnnkEbKyssjKyuKRRx7BaDRyyy23KPvecccd/PKXvyQiIoLw8HB+9atfUVRUxEUXXTT8Ryh9KyEEvb29VFRUYDAYSE9Pl/lPpO/kjALL3/72NwAWLVo0ZPuqVav48Y9/DMCvf/1rrFYr99xzDz09PcyePZsNGzYQFBSk7P+Xv/wFjUbDjTfeiNVqZfHixaxevVpWtc8xb0DxLpsxefJk2XYiDQuVEGNvJJnFYiEkJASz2UxwcPBpP8/j8Zw0V8ux/vrXv5KSksJVV101HEUdldxuNw0NDUpbWHx8PDqdztfFkka5M/ncTZihkS0tLXR0dGC1Wpk9ezYqlYq2tjYOHz5MYmIiWVlZvi7iiBL/SqrU0tKizOGZOXOmTKokjYhxH1icTicDAwNERkYSGhrKp59+qnRt22w2NBoNLpcLIQR1dXU0NjaSmprq62IPq8HBQY4cOQJAQkICCQkJMgeKNKLGfWCxWq3U1NSQkpLCoUOHKCoqUmbfxsXFERYWxubNm8nNzSU2NpaoqKhxMc9I/Gth9IaGBpxOJ4mJiYSFhcmAIp0T4z6wBAcHM23aNJqamvB4PPT19WGz2ejo6MBut2MymZSucYPBcNL0lWOJEIL29nZaW1sxGo0kJibK/CfSOTfuA4tXYmLikMF806dPBxg3bStCCLq6umhpaUGj0ZCdnS2ztEk+M2ECy3jlcDhobm6mv7+f8PBw8vPzZboCyefkFThGuVwumpqa6OzsJCIigry8PBlQpFFDXoljiBCC/v5+TCaTMuQ+JSVF3u5Io44MLGOEdx2e/v5+MjIyCA8PlwFFGrVkYBnFPB4PPT09tLe3o9PpSE9Pl0mVpDFBBpZRyNtlXF9fj7+/P1lZWXIOjzSmyMAyijgcDjo7O+nu7iYyMpLp06ejVqtlQJHGHBlYRgGHw0FDQ4OS8lEmVZLGOhlYfOT4pEoJCQlkZGTI2ok0LsjAco4JIbBYLFRWVgKQnZ1NcHCwDCjSuCIDyzni8XiwWCw0NzdjNBopKiqSPTzSuCUDywjzeDw0NzdTW1tLfHw8mZmZp0wMLknjhQwsI8CbVKm1tZWenh6ioqI4//zzZQ+PNGHIwDLMbDYbFRUVOBwOUlJSKCgokDlQpAlHBpZh4O3haWhowO12k5ycTEhIiAwo0oQ1YQKLy+Wira0NjUZDdHS0kpqyo6ODkJCQM0rK7SWEoLOzk5qaGmXpV5lUSZK+wxKrY43ZbGbnzp10dHQo27Zt20Z7ezvr1q1TcuMODg5+62t5kyqVlJRgNpuZNGkSOTk5BAQEyKAiSUyAwNLf309JSQlqtZr09HQOHDhAf38/cHQZjKKiIpqamrDb7WzdupVdu3ad8hbG6XRSVVXFgQMHGBwcJDc3l8zMTJmpTZKOM+5vhbRaLdHR0cqC8jqdDpfLRUVFBREREXz22WdMnToVo9HIkiVLqKiowOPxKM/39vA0NTXR09NDeHg4RUVFctkMSfoG4z6w6PV6oqOj8Xg8JCUlAaBWqwkJCUGlUimLmB1fS/EmVWpoaAAgLi6O5ORkGUwk6TSM+8DidXzg8AaIk032s1gs7N69G5vNRlJSkpJUqa+v75yUVZJGC5VKRUBAwBn3cE6YwHK6Zs2axRNPPMH69etPuRSI2WzGaDSi1WrPcenO3uDgIEaj0dfFOG3eRvSxUmbvOthhYWG+LsppczqdCCG+cXldo9HIww8/TGho6Bm99oRau/l0CCGUn1P5xz/+wdy5c8nIyBjW9x4pQgi2bNnCggULfF2U0/bll1+i0WiYP3++r4tyWqxWK6+88gp33nmnr4ty2qqrq3E6neTm5p5yH+9a5yqV6ow+dzKwnCYhBAcPHsTtdpORkUFQUBAqlYqKigrsdjt5eXmjbmF1q9VKRUUF4eHhxMfHo9Fo6OjooLa2lsjISNLT031dxCG8o5bT0tIICAgAoKOjg5aWFvLz80fdYnJut5vS0lKMRiOpqakIIXA4HBw+fJjQ0FAyMzNHXZucd9xVREQEaWlpwNF1zVtaWsjJySEkJOSUzz2Tz924724eLkIItm/fjtVqRaPRoFarGRwcpLKyEqfTSU1Nja+LeAK73U5ycjLV1dXKtrq6Otra2kbdh1QIwYYNGwgLC2PdunUIIXC73Rw6dAh/f38OHTo06pa+VavVpKamUlVVRVNTE1qtlt7eXlpaWjAYDL4u3kk1NDTQ2tqKv7+/Mndtz549hIeHs2/fvmF7HxlYTkNFRQWNjY3ccMMNBAUFsX//foQQuFwuoqOjiYuLG1UXvdlsZvPmzbjdbsxmMwkJCfj5+SGEoLCwkKKiomG9iIZLX18foaGh+Pn54fF48Hg8Sq9eT0+Pr4t3Un19fVitVuLj4xFCEB0dzaxZs9ixY8dpDbY81/Ly8pgyZQr79u1TbvmDgoJISkoa1nWpZOPtacjMzASgtrYWtVpNdHQ0nZ2dBAYG4nK56O7uJjU11beFPEZwcDBz5syhvb2dgYEBCgoKcDgcytQDp9NJZGSkr4t5gilTprBnzx6SkpLo7OxEpVIRHh5ObW0t2dnZo+62wu12Y7VaufLKK1Gr1TQ0NKDT6bBYLISHh4+6W2OAnp4eHA4HkZGR9Pf3o1KpiIiIoKqqalhvjWUbyxlwu914PB40Go1ykXu/WcfKKoRCCJxOJ1qtdtR9UL21wGPPr3fbWOmBG83nF05evpOd95M5k8/d2Pg0jBJ+fn4njHtRq9VjahazSqU6J9+k1dXVWCwWIiIiSEpK+sYL1uPx0N3drdwCRUREDCnvsUHFOyr6VOdcCEFDQwMajYaEhIRhOprTd67O79k6WfmOP8fDYex8IqQx5ZFHHsFkMvH0009TVlZGf38/dXV1tLe34/F46OjoUBqSHQ4Hn376KYODg/T19eF0Oqmrq6OlpQW73U5dXR319fU4nU5WrlzJBx98QF9fn9LDZbFYlDYuh8PBwYMH+ec//+njMzCxyRqLNCLcbjednZ2Eh4fjdDp59tlnyc7Opr6+nksvvZTHH3+cW265hY8++ogHHngAIQQ7d+6kt7eX/v5+EhMTSU9Px2AwUFxcTH19PampqdTU1DAwMEBKSgqffvop+fn51NfXc/fdd6PX69Hr9SxatIh33nnH16dgQpM1FmlEqNVqsrOz0el09Pb28vXXXyu9aAA6nY7Zs2cTHx+PzWYDjo4EtdlsHDp0iAsuuICCggK2bt1KdXU1Ho+Hzs5OYmNjyc3Nxel0cuTIEdxuN3FxcbjdbgBlLInL5cLpdPrs+Cc6WWORRsTSpUvJzs4mOzub/fv3c//997N9+3bi4uKIjo5Gr9fzzDPPEB8fT2JiIpMmTcLPzw+73c7ChQt58cUXSUlJYerUqRw4cICcnBxyc3MJDAxk06ZNZGVlcdlll7Fv3z4uvPBCZdyI0+lk8+bNuFwu9uzZw5w5c3x8JiYm2SsknRPeAW9qtRqXy8Vnn33GxRdffNKeCG+qCu9Qco/Hc9JE5KfbmyEND9krJI06KpVK6ZLX6XRcdtll37jvsb1vp1pudiR6M6ThIdtYJEkadjKwSJI07GRgkSRp2MnAchLd3d20tbWd0XMcDgf79+9n37599Pb2DpmU6HK52L9/P7t27aKqqgqXy4UQgra2NioqKpR1iWprawGUxzo6OkZscqN3KsK50tfXR1NTE06nk6amplE1aVMafjKwnMRXX301ZICVEILm5mZl9KcQgoGBAWpra2lvb6e7u5v6+nq2bdtGX18fd91115A0lhaLhbvvvpt3332X1157jZ/+9Kd0dnbyzjvvcNVVV1FbW0tLSwsrVqwAjnaZLlu2jF/96le4XK4hZRNC0NraSk1NDT09PUoQqq2txW6343a7aWtrU3phBgYGcDgctLW1UVdXh8lkwul08s477/Dpp5/S09NDf38/tbW11NXVYbfblffq7e2lvb19yHt5PB4aGxtpaGjA4/Fgs9lob2+nvr5+yDG73W4aGhpobGzE4/Fw6NAhXnjhBaxWKzt27FDOS11dHR0dHdjtdoQQdHR0UFNTg9VqHZG/rXRuyF6hkzg+g9xHH31EcXExsbGxNDQ08Otf/5oHH3yQ2bNnU1NTg0ql4ve//z1ZWVl0dXXR399/Qm0gOTmZRx55BJVKxapVq3j77bcRQnDHHXfwxBNP8POf/1x5zsGDB5k6dSpWq3XIWAwhBJs3b2bbtm3MnTuXpqYm/Pz82LBhA1OnTmXNmjXceeed/OpXv+Lll1+mu7ubnTt3kp6ezq9+9StuueUW3n//ff74xz9SV1eHwWCgoKCAp556ioULF+JwOJg7dy6JiYkArFq1iq1bt3LTTTexdetWfv/73/Pll1/S0dGBn58fQUFBZGZmct999/Hzn/+cgoICpk6ditvt5tFHHyU8PByLxUJqaiqJiYl4PB7sdjuVlZU0Nzfz+OOPc8UVV/Dyyy/zq1/9is7OTjZs2MC8efNYvXo1v/vd79Dr9efory4NJ1ljOQ3r1q3jxhtv5I477qC5uZlNmzah1+v54Q9/yEUXXaSM8Gxra+P3v/89t9122yn7+VUqlTJlHSA9PZ2pU6cqc1vcbjdvvfUWcXFxZGRk8NZbbymjSl0uF6tXr+buu+9m0aJFzJs3j3fffZef/exnXHnllTQ1NTEwMKCMZHW73bhcLjweD9nZ2dx6662kpaXhcDjIyclh8uTJyvrStbW1J4xPcDqd3Hrrrdxwww1ERESwbds2Xn75Zfz9/dHpdHR2duJyuZg/fz4/+MEPmDp1KnA0S1l5eTn/9m//xve//33Wr18/5Bw4HA727dvH0qVLufjii5Wg8+abb3LPPfdw5ZVXUlVVJZOXj2EysJyESqWira2N8vJyjhw5Ql5eHlu3bqW4uBij0UheXh5qtZrS0lL27t0LQHt7O7/97W+55JJLmDJlihIMvLxpF7/88ks2btzIddddBxwd+n7TTTfxxRdfYLVaqa2tpaenh+nTpzNt2jT6+/tpamoCjo7nKCws5KuvvqKsrIzu7m4mTZrEV199RWlpKWazmYCAAIxGI3v37uWzzz5TAph3AJn3X6fTSUlJCZ2dnSxZsoSFCxeydevWEzLhFRcXc/jwYTo6OpTRr9nZ2cyePZtrr70WtVqNv7//kAFqRqORsLAwDhw4wJYtW8jLyzuhDBkZGaxfv55Dhw5RXFwMwLRp0/j666/Zt28fg4ODYyYVhXQiOfL2JDo6OqisrMTPzw+tVktGRgZ79uyhubmZSy65hJiYGEpLS+no6KC0tBQhBDfddBMVFRVotVq0Wi2FhYVKNd7pdCoByOPxkJqaSlxcHC0tLeh0OiIjI6mvr8dqtRIXF0dvb6+SOKqmpobIyEjlOPv7+9m9ezeVlZVcfvnlSk2itbWV8847j8zMTCorK9myZQuzZ88mNjYWo9FIa2srmZmZVFdXEx0djdVq5ciRIyQnJ2M2m/niiy+YOXMmM2fOVD7QK1asICAggKCgIGbOnElubi6dnZ3s27ePpqYm5s+fT2xsLF1dXUr+1GPP4fr164mMjGTu3LkIIeju7iYhIYGWlhaSkpLYsmULHo+HlStX8l//9V/k5OSwbds2GhsbmTVrFrm5uXJE7Sgik2mPMCEE77zzDkeOHCEoKIibb755SA6R8eKtt94iLS2NGTNmDPtrCyF4/vnn6e3tJScnh8svv1yOoh3lZGA5B449beP1W9V7jCNxfMdfduP1HI4ncq7Qd3Q6QePY7UIIVCqV8u+3vfY37fNdPszHP/e7Br+R/LDLQDK+ycbbk9i7dy//+7//yxNPPEFzczNwtPHV29sCR5fWcDqdOJ1OHnvsMcxmM9XV1cpgt4GBASW/qPf/LS0tSluLd23oY8epDA4OsmrVKv7zP/8Th8MxpEwOh0N5HTjaQ9TX16d0jQ8ODtLa2kppaany+uXl5bzyyis89thjdHR0jOg5k6RjyRrLSfT29nLJJZdQV1fHwYMHaWxsZMeOHbhcLn7wgx+wfft22tvbsVgs/OQnP+Hrr7/m9ttvp6OjA6vVyocffkhISAi33XYbn3zyCSaTiYyMDHQ6HevXr8dqtdLS0kJraysGg4HbbrsNf39//P39ufHGG7n11luHjIMZGBjgww8/5MiRI1x99dU4nU6+/vpr2tvbufzyyxkcHGT//v24XC4mT55MYWEhKpWKzMxMHA4H1dXVmEwmoqKifHhWpYlE1lhOwul08s9//pONGzcSHR3Ne++9R0BAAIGBgRw+fBiTycSPfvQjsrKyWLduHXB01cHy8nK++OILvv/973PPPffg5+dHb28ver2eF154gaSkJBYuXMjUqVP54IMPlB4V75o5arX6pF2s/f39ypINGzdupKqqisTERK6//npee+013nnnHa6//nrOP//8IbcYx6YqkAPNpHNJBpaTCAgI4Nprr+W+++7jo48+Ijk5GZVKRUxMDNnZ2ZjNZtavX09NTQ0zZ84kODgYPz8//P39iY2NZePGjXz++efU1dVRUlJCWFgYQUFBBAQEUFJSQnl5Oampqcq4FO+ylm63mz179mC1WikuLlZqLWVlZbS2thIeHo7BYMBgMBAcHIxWq8Xf35/Jkyfz1VdfsXfvXiV3iRCC/fv309jYeMp8JpI0UmSv0Ek4HA78/PxQq9VYLBb8/f0xm83KqnGvv/46V1xxBXq9XilHUFAQDocDrVZLd3c3AOHh4ZjNZmVdnNDQUHp6etBoNGi1Wvr6+pQFo7wrFfb09OB0OpXXVqlUuN1uurq60Gg0GAwGJbOan58fVqtVySur1+sxGo3odDqEEPT29uJwONBoNISHh8sGU+k7kd3NI0gIgdlsVj70kjRRyO7mEaRSqQgNDfV1MSRpVJNtLMepra0941wskjQeuVwuWltbzyp3jgwsx/noo4/YuXOnr4shST4hhMBqtVJdXU15eTn+/v5n9ToT5lboZKNQJ8KwfEk6Hd5Blt70Gbm5uaSnp5/152LCBJa+vj6++uor4uPjmT59OiqVirKyMurq6oiIiGDWrFkyuEgTjsfjoa+vj+bmZnQ6HZmZmej1+u/8WfhOt0LLly9HpVKxbNkyZZsQgoceeoj4+HgMBgOLFi1Shpl72e127r//fiIjIwkICGDp0qVKzpGR4k2r2NPTo+RKaWpqYvHixXz99dcMDAywadMmduzYgVot7xClsc3j8dDe3k51dTV9fX243W4GBwcpLS2lqamJbdu2sWfPHrZu3UpjYyPp6elkZmaekFvnbJ11jWX37t08//zzTJo0acj2FStW8MQTT7B69Wqys7P505/+xMUXX0xFRQVBQUEALFu2jI8++oi1a9cSERHBL3/5S6644oohA7yGm0ajYebMmXz55Zf09vYqI1Lr6+sJCwtDq9VSUFBATk6OTPQsjVnea7e3t5c1a9ZQWVnJrbfeSnt7O2+99RY6nY729na2bt2KWq3mvffeo6CgYNhr62f11dzf388PfvADXnjhBcLCwpTtQgiefPJJfve733HttddSWFjIyy+/zODgIK+99hoAZrOZl156iT//+c9cdNFFTJ06lVdffZVDhw7x2WefDc9RnYTD4cBisTBnzhyCgoKwWq3MmTMHl8vFJZdcgl6vJzIykvDwcBlYRhm3201nZ+eQBNsOh4OOjg46OzvxeDzK4MJjk4FPRHv37mXdunW43W40Gg333XcfkydPJi8vD7PZjNvtJicnh6lTp3L55ZfjdDpHpAngrALLvffey+WXX85FF100ZHttbS0mk4klS5Yo2/R6PQsXLmT79u3A0QN3Op1D9omPj6ewsFDZ53h2ux2LxTLk50yFhYWRl5dHRkYGer2euLg4goKCyM/PJzo6+oxfTzp73p4H76oH37bv66+/ztq1a3nyySeV2eC1tbV88cUXvPrqq+zcuZOKigpuv/12vvjii3NxCKNWUVERCxcuJDIyEpVKRVRUFCaTCbfbzerVq0lOTqasrIyGhgauueYaLrjgghEpxxkHlrVr17Jv3z6WL19+wmMmkwmAmJiYIdtjYmKUx0wmEzqdbkhN5/h9jrd8+XJCQkKUn6SkpDMttjSK2O12fvrTn7J582alvau8vJzXX39d+fGOJfJ4PNTV1fGzn/0Mi8XCkSNHAMjOziY/P5+uri5cLhc5OTksWrTohHQTE41er8dgMGA2m5k+fTqtra1ER0eTl5dHZGQkDz/8MIsWLeJnP/sZixYtGrmmhzPZubGxkV/84hds2LDhG/u3j69afdcESA8++CAPPPCA8rvFYpHBZQzzeDwMDg5y0003odPpgKEzu71zoby6urpwOp3KpMuenh7CwsLIz8/nqquu4u2332b+/PkTvldPCIHFYqGpqQm9Xs+UKVNOaIxVqVTcc889GAyGEU1WfkavvHfvXtrb25k+fbqyze12s3nzZp555hkqKiqAo7WSuLg4ZZ/29nalFhMbG4vD4VAujmP3mTt37knfV6/Xy2n/Y5gQApvNplzkfn5+zJw5c8gFn5ycPCRfTGBgIHA04MycOZOXX36ZuLg4/P39effdd7n88stZt24dDQ0N3HDDDfT09NDV1YXZbGbevHnjMgfxqbhcLpqammhtbSUuLo6srCwlYB9PpVIpnSgj6YxuhRYvXsyhQ4c4cOCA8jNjxgx+8IMfcODAAdLT05W0AV4Oh4NNmzYpQWP69Olotdoh+7S2tlJSUnLKwCKNTUIIXC4XPT09PPDAAzidTtxuNzqdjl//+tcASmoIf39/wsLClB9vYm2VSsX111+P1WrF5XKxY8cObr/9dmJiYsjKyiI2NpaPP/4Yp9PJzTffTHl5+YS4HfIOn2hpaaGsrAyNRsOMGTNISUk5ZVA5l86oxhIUFERhYeGQbQEBAURERCjbly1bxiOPPEJWVhZZWVk88sgjGI1GbrnlFgBCQkK44447+OUvf0lERATh4eH86le/oqio6ITGYGlsKy8v56mnnuKyyy6jrKyMN954g/b2dq688kreeOMNoqOjSU5O5nvf+x4A27dvV9rZ9Ho9F198MXq9Xlnm9cknn+Suu+7i0ksvxWAwcN555+F0OpXUFNnZ2cTHx5+wptN44m34rqqqwuPxkJCQoGQMHE2GfSTYr3/9a5YtW8Y999zDjBkzaG5uZsOGDUOqX3/5y1+4+uqrufHGGzn//PMxGo189NFHMiHRODM4OEhQUBDz588nKiqKW265hX379lFaWspFF13E9773Pd5//31lf2/bikqlGjJIUa1Wo9VqcbvdSjY9u92u3CaFhISwdevWE9pmxhubzUZVVRU1NTXExcUxadIkoqKiRuUxf+fWm6+++mrI7yqVioceeoiHHnrolM/x9/fn6aef5umnn/6uby+NYoGBgaSlpaHRaMjLy0OlUpGVlUVGRgYff/wxRqORe++9V9k/NzdXWfjMG0wA5bbnpZde4rbbbqOmpoa+vj5CQkI4ePAgNpuNCy64gIqKCgYHB/nkk0+44YYbTuh5HIs8Hg+dnZ00NzdjNBpJSkrCaDT6uljfSiZ6Os5f//pXUlJSuOqqq4b1dSeiYy+t45dIOXapktP5xvW2xZysx/HY7Wf6uqOR9xh6enpoaWlBpVKRnJxMYGCgT49JJnqSRoXjuzlP9u/pOtX8reNfZ6wGEy+n00lzczNms5nw8HDy8vLGZBOBDCySNAp4kyp1dnYSHBxMfn7+mF5yVgYWSfIRbw6UlpYW7HY78fHxJCYmAmO/5iUDiyT5wMDAAI2NjVgsFlJTU4mMjBxX6TpkYJGkc8Tj8WA2m2ltbUWr1ZKamjosSZVGIxlYJGmEeTweurq6qK+vR6PRkJ2dPSa6jL8LGVgkaQR4h9y3t7fT2dlJeHg406ZNG9Pd4GdiwgSWwcFBKisriY6OJi4uDpVKRU9PDzU1NURGRpKSkuLrIkrjhMPhoLGxkebmZlJTU8nPzx+TXcbfxYQJLFarlaSkJIqLi4mJicHPz4+Wlhba29uVlniXyzWu55lII0cIgcPhoLm5GavVSkJCwnfKcj/WjZ9m6G8RFhZGX18fiYmJqNVqhBBkZGSQm5vL/v37EUJQU1NDfX39hL0YpDPnzYGyZ88eiouLiYiIID8/n9DQ0Al9HU2YGktLSws2m42cnBxsNhv19fUEBQXhdDqVJVOzs7NJT0+XOW+lb+UNKN5lM4qKisZtD8/ZmDCBxXu7A2AwGMjNzVWqr1lZWfKCkE6L2+2mtbWVmpoa4uPjyczMHBX5T0abCRNYTkalUsnMdNK3EkIghKCtrY3Ozk4iIiKYO3cufn5+8gvpFCZ0YJGkb2O1WqmsrMThcJCSkkJBQcG4GiE7UmRgkaTjeG+RGxsbcTgcJCUlERoaKgPKGRgXgaWnpwer1YrH48FkMikNaZJ0JoQQdHV1UVNTQ1BQEMnJyQQEBPi6WGPSmA8sfX19rFixgtDQUCorK4mKiqKyspKbb75Z3v9K38rbA2g2m5VlM/Lz8wkICJDXz3cw5gNLf38/6enpXHTRRRw+fJg5c+awfPlyvv/978sLQ/pGTqeThoYGzGYzkZGR5ObmjuhaOxPJmD+LUVFRXHDBBaSmppKWlobNZuOee+6RQUU6JW+XcXd3N8HBwRQWFsou42E25gOLRqMhMzNT+d3f319JyCxJXkIIBgYGaGhowO12Ex8fT1FRETD2kyqNRmM+sHg1NDSwb98+LrzwQrq7u0lJSZEXjAQcnYDa1NTE4OAgcXFxREVFyR6eETYuzq7VauWll16iubmZ3t5eXnzxRV8XSfIx77IZJSUlNDU1kZSUxJQpU4iJiZFB5RwYFzUWPz8/NBoNra2tbN68mZycHF8XSfIRb5dxc3MzAJmZmRiNRll7PcfGRWABmDx5MgkJCbjdboKDg+WFNIF414hua2uju7ub8PBwCgsLUavV8jrwkXFRJ9y/fz8vvvgiXV1dtLS0UFlZ6esiSeeIt8t4z549OBwO8vPzSUxMlPN4fGxc1FjS0tL4t3/7N9LT0/F4PMTExPi6SNIIEkJgs9loaWnBarWSmJhIcnKyDCSjyLgILCEhIRw6dIidO3cqqSevv/56eaGNM0II+vv7qa2tZWBggNzc3AmfUGm0GheBxZusybs+y+bNm7n++utP2M/lcqFWq5VeASEETqcTrVYrL85RzOPx0NfXpyRVysnJQafTyb/ZKDYuAovRaOTGG28EYPv27fz0pz89YZ+enh52796NEIIFCxZgMBgoLy+nu7sbIQTnn3++vFBHGbfbTUdHB1VVVYSHh5ORkSEnl44R46LxdnBwkLVr1xIWFsb8+fN5//33T9jnwIEDxMTEEB0dTXt7O3B0UN2MGTM4cOAAAwMDVFRUUF1dLQOMD3mXzWhtbeXw4cN4PB7mzJlDXl6eDCpjyLiosWi1Wjo7O3nxxRdRq9VER0efsE90dDQWi4X+/n4iIiIoKSnBYDDQ0tJCfHw8er2ezMxMUlNTZc5bH7HZbNTU1NDd3U1mZqZMqjSGjfnA4vF48Hg83H333WzevJmYmBhlYahj5eXlUVNTQ2pqKmFhYTgcDubOnUtlZSUXXnghWq0WYMKt/+Jr3qRKTU1NysLoubm5MqCMcWM+sNjtdoqLi+nt7UWtVtPR0UF7e/sJExHVavWQyYre/+fn55/T8kpHCSHo6emhvLycwMBA0tPTZQ6UcWTMBxaA9vZ2nE6nr4shnQYhBGazmebmZvz9/ZkyZQoGg0EGlHFmzAcWvV7PhRdeiMlk4sCBA6hUKoKDg31dLOk4LpeLhoYG2tvbiY2NJTs7W7n9lMafMX8jq1ar0Wg0vPrqqxQUFBAdHc1zzz0nG2BHASEEbreb5uZmysrK0Ol0TJs2jdTUVBlUxrkxX2OBow24vb29bNiwgb6+PgYGBvjiiy9YtGiRbIz1ASEEg4ODVFVVAZCQkEBhYaG83ZlAxkVg0el0XHPNNXR2dhIaGooQgpCQEHkh+4DVaqWpqQmHw0F8fDwRERGyh2cCGhd/cbfbzaFDh6ipqSEnJ4fi4mKmT58uL+hzxOPx0N7ezoEDB2hoaCAxMZGCggKZqW0CGxc1FpfLRUdHh7IEiOwhOje8XcbNzc34+fmRkZFBYGCgrClK4yOwGI1GbrvtNtasWcPu3bu56aab5MU9gpxOJ83NzVgsFsLCwsjPz5dJlaQhxnxgcblcrFq1SlkKs7u7G4/H4+tijUtOp5PW1lY6OzsJDg4mLy9P9u5IJzXmA0tHRwc7duzgySef5Pnnn+fee++Vk9WGkRBiSINsQkICSUlJsnYifaMxH1iMRiMFBQV89NFH+Pv78+677zJ79uwhw/elM+ddh6exsRGLxUJ6ejqRkZEyoEinZcwHlpCQEH75y1/6uhjjhsfjwWw209railarJT09XSZVks7YmA8s0vDwrsNTV1eHv78/2dnZ+Pv7+7pY0hglA8sE5k2q1NbWRldXFxEREcyYMQOVSiVrKNJ3IgPLBGW322lsbKSlpYX09HTy8/Pl9Adp2EyYwNLd3U1rayt6vZ6MjAxUKhUWi4WGhgbCwsJISEjwdRFHnBACu90+ZNkM77mQpOE0YcZb+/v7Ex4eTmlpqTLzuaGhgcbGRmWfjo4Ourq6xt0HzZsDZffu3ZSUlBAZGUl+fr6cTyWNmHFfY+nt7aW8vJy8vDzKy8tJTExEpVIhhCA9PR1/f3/KysqIj4+nrKyMqqoqZsyY4etiDwshBBaLhaamJvz9/Zk8ebLs4ZHOiXEfWIKCgpg0aRKtra1UVVVRUFBAX18fLS0tCCFobW1Vkm8vWLCAAwcOjPmRuy6Xi5aWFurr64mPj5dJlaRzbtwHFj8/P4xGIxkZGWRkZCjbvVnm8vLyfFW0YeXt4Wlvb6ezs5OIiAjmzJmDRjPu/8TSKCSvujHOO+TeO6s7NTVVJlWSfE4GljHK28PT1NSE0+kkOTmZ0NBQmf9EGhVkYBljPB4PXV1d1NbWEhQURHJyMgEBAb4uliQNIQPLGCGEoLe3l+bmZvR6Pfn5+XIdHmnUkoFllHM6nTQ0NGA2m4mMjCQvL0+OkJVGPRlYRiHvshmtra309vYSFBREUVGR7DKWxgwZWEYRbw6UhoYG3G43CQkJJCYmAshbHmlMkYFlFPCuw9PU1ITVaiU+Pp6oqCgZTKQxSwYWH/J4PHR3d2MymdDpdKSkpKDX62VAkca8Mx700NzczA9/+EMiIiIwGo1MmTKFvXv3Ko8LIXjooYeIj4/HYDCwaNEiSktLh7yG3W7n/vvvJzIykoCAAJYuXUpTU9N3P5oxwuPx0NHRwcGDB2lubiY9PV1JrCSDijQenFFg6enp4fzzz0er1fLpp59y+PBh/vznPxMaGqrss2LFCp544gmeeeYZdu/eTWxsLBdffDF9fX3KPsuWLeO9995j7dq1bN26lf7+fq644grcbvewHdhoI4TA5XLR1NRESUkJDoeDSZMmMWnSJIxGo6+LJ0nDS5yB3/zmN2LevHmnfNzj8YjY2Fjx6KOPKttsNpsICQkRK1euFEII0dvbK7RarVi7dq2yT3Nzs1Cr1WLdunWnVQ6z2SwAYTabz6T4p+Wpp54S77///rC+pt1uFzU1NWLbtm2itrZWuFyuYX19SToXzuRzd0Y1lg8//JAZM2Zwww03EB0dzdSpU3nhhReUx2trazGZTCxZskTZptfrWbhwIdu3bwdg7969OJ3OIfvEx8dTWFio7HM8u92OxWIZ8jPaiX/N4amurubIkSOEh4czZ84cUlNT5TgUadw7o8BSU1PD3/72N7Kysli/fj133XUXP//5z3nllVcAMJlMAMTExAx5XkxMjPKYt6EyLCzslPscb/ny5YSEhCg/SUlJZ1Lsc0r8KwdKcXExBw4cICIiQiZVkiacM+oV8ng8zJgxg0ceeQSAqVOnUlpayt/+9jd+9KMfKfsd/wESQnzrh+qb9nnwwQd54IEHlN8tFsuoCy7iX1naWlpa0Ov15OXlyaRK0oR1RjWWuLg48vPzh2zLy8ujoaEBgNjYWIATah7t7e1KLSY2NhaHw0FPT88p9zmeXq8nODh4yM+Z8t6a2Gw2JTWlEAKbzfadGo3dbjctLS1s3bqVtrY2MjMzycjIkN3G0oR2RoHl/PPPp6KiYsi2yspKUlJSAEhLSyM2NpaNGzcqjzscDjZt2sTcuXMBmD59Olqtdsg+ra2tlJSUKPuMhN7eXj799FMqKyuVbTU1NezatYs9e/acUdY4ccyQ+8OHD6NSqZg7dy7Z2dnodLqRKL4kjSlnFFj+/d//nZ07d/LII49QVVXFa6+9pqyXDEdvgZYtW8YjjzzCe++9R0lJCT/+8Y8xGo3ccsstwNGVC++44w5++ctf8vnnn7N//35++MMfUlRUxEUXXTTsB2i322lra0OlUhEUFITFYsHlcgFQVVXFeeedx65duxgcHKSqqora2tpT1jS8NZzS0lJ27tyJWq2moKCAuLg4/Pz8ZA1Fkv7ljNpYZs6cyXvvvceDDz7Iww8/TFpaGk8++SQ/+MEPlH1+/etfY7Vaueeee+jp6WH27Nls2LCBoKAgZZ+//OUvaDQabrzxRqxWK4sXL2b16tUj0lvidDrp6OggPDycWbNm8cEHH5CRkYHT6cTPz4/q6moiIyPR6/UkJyeTmJio3Cp5iX8lVWpubsbhcJCUlER+fr5MqiRJp6ASx3+KxgCLxUJISAhms/m021scDgcmk4ng4GCCgoKw2WzodDpMJhNhYWEEBgYC8Ne//pWUlBSuuuoqhBB0d3dTWVlJYGAgaWlpyn6SNNGcyeduwswV0ul0JCcnK797s66dqnfJm1TJu2yGwWCQtzqSdJomTGA5XQkJCXz44Yc0NzeTmJiIn58f5eXlQ/ZxuVxjrk3F4/GMqVs3t9uNSqUaU2V2uVxjalUE783KN13HWq2WBQsW4O/vf0avPXbOwjly1VVXsWjRohPaWY61evVq5s2bR2Zm5jks2dkTQrBp0yYWLVrk66Kcts8//xyNRsPChQt9XZTTMjg4yOrVq7n77rvHzBdOVVUVTqfzG5fAUalUZ9XTOWHaWL4rIYQSbCwWC0ajEa1Wi9PpxO12j8qZyUII+vv78ff3x+FwEBAQMOQ4RlttQPwr0ZXRaMRqtQJHl8a1Wq2jMr+vt1FfpVKh1Woxm82EhoYOqQmMxjJ7y+dyuRBCoNFoTuscyzaWEfL6668TERHB3Llz0el02Gw2tm/fTlRUFLGxsURFRfm6iEO0tbXR29tLTU2NMjfr8OHDlJSUUFhYSEFBgY9L+H+EEGzZsgUAm83GRRddhMfjYc+ePWi1WgwGw6hbXM7tdtPe3s7BgweZMWMGsbGxtLe389VXX5Gdnc3kyZN9XcQTlJeXc/DgQfLz8ykqKkIIwfbt24mJicHj8ZCdnT0s7zO6vrJGqcbGRtra2khNTSUhIUFZSN5ut2M0GomIiKCzs9PHpfw/AwMDHDx4kICAAKKjo3E4HMo3UVhYGJMmTaKrq8vHpTxRY2Mj06ZNo6+vD5fLhcvlwuFwkJycTG1t7TfenvqCRqMhJCQEq9Wq5CPW6XTk5ubS2NiI0+n0cQlPFBoayqRJk+ju7la22Ww2kpOTaWtrG7b3kYHlNISGhhIUFERqaipWqxWLxUJ9fT0A9fX1lJaWjqoRtzqdjri4OAYGBti8eTP5+flYrVY2b96slLO/v9/HpTxRfHw81dXV6PV6ampqqKurY3BwkPLycpKTk0fdbYXD4eCVV14hKSkJnU5HcXGxckvhdrtHZX4h7/y1/v5+Ojs76e3tpbe3lz179gxr4JZtLKdJCIHJZMLpdBIXF4fD4cBgMNDV1YXT6SQ2NnbUtVn09/fT0dGBTqcjOjqanp4eNBoNFouFqKioUbfQmXeaRHR0NHa7HTjaTtHT00N8fPyoSzfh8XhoamrC4/EoNUPvZFRvdsTRFgy7u7uVv79Go0GlUmG32+nu7iYuLu4bvyDP5HMnA4vkc0IIHA6H8v8z7dqUzg3ZeCv5nDcBmNVq5cYbb1S+0f38/NBoNLhcLtxuNxqNBrfbzQcffEBcXBxWq5WLLroIh8OBWq1Go9EoQUen0/Hpp58SFxfH5MmTlXSfWq1Wqc3Y7Xbef/99goKCuOyyy3x2/BOdDCzSiNi+fTs/+MEP2LRpE4cOHaKxsRGHw0FHRwd33303Dz/8MDNnzqSnp4dbb70Vm81GX18ffX19fPzxxzQ0NJCZmcmUKVP44osvMJlMzJgxg08++YTg4GD8/Pz48MMPiY+Px+l0cueddypjLi699FLeffddX5+CCW10NQpI44YQgvr6esrLy4mKiuIf//gH/v7+BAYG0traitPp5Pvf/z46nY6Ojg7gaFW7s7OT999/n9tvv51LLrkEu93O4OAgarWa2tpaUlJSWLhwIRaLhf3796PRaJR9YHSOHZmIZI1FGhFGo5GLLrqImTNn8sUXX3D55ZcTHx9PaGiokob0ww8/pKWlhcjISDQaDRqNBoPBwMUXX8wXX3xBfHw8drudgYEBZTBXVlYWJpOJKVOmMGPGDFJTU+nv71dWOnC5XOzdu5eamhoqKirIycnx8ZmYmGTjrTQiOjo6CA0NRaPR0NfXR0BAAC0tLcDRsTTPPvssN998M+Hh4QQEBNDf349KpcLj8RAUFERraysqlYrY2FhaWloICAhAq9ViNBoxmUwEBgai1WppbW0lICBAWTnS7XbT3NyMy+UiMDCQ6OhoH5+J8UP2Ckmjmsfjoa2tjdjYWHnbMobIXiFpVFOr1cTFxfm6GNIIko23kiQNOxlYJEkadjKwDCPvsiZjsNlqxDmdTgYGBvB4PAwMDMhzNM7JwHISe/bs4fPPPx+yzTvT9vg1iby/7927lyeeeIL/+I//4MUXXxwys3VgYICHHnqIV155hZdeeok333wTm83G559/zu9//3v6+vro6Ojg5ZdfVl579erVPP300yf9ALrdbmw2m7Jkidvtxm63f+uH1eFwKCsU7N+/X1kKxZtXxOl0njSR+LHvBUeDxLHH5x2S733tk23bv38/zzzzDP39/XzyySfA0UbcY8/pmRyLNLrJxtuTqK+vx2QysXjxYuBopq2VK1cyODjIVVddxcUXX8ybb77Jxx9/TGFhITNmzOC8885j0qRJqFQqbr75Zq6//nplGVm73c7hw4f57//+bzweD4899hgbNmygsbGRxsZGXn75ZS677DK2b9/ObbfdRnt7OyUlJcqkvPj4eOD/Bp29/PLLlJWV8cADD5CYmMjq1aupqqpi6dKlXHzxxXz88cfceOONDA4O0tDQQHh4OH//+99pbm4mPDyc+++/nzfffBO9Xs+1117LoUOHWLduHenp6Sxbtkwp95YtW9iyZQulpaXMnDmTe++9l4MHD/Laa68hhOD73/8+cXFxfPDBB1RUVHD11VezePFiJcfHG2+8gRCC22+/HafTSX9/P35+fkRERGCz2Vi9ejXl5eVkZ2dzzTXXoNVqWbVqFSUlJVx++eVcd911YyrVo/R/ZI3lNLz00ktcddVVrFixgjfeeIPNmzezbds2nn/+ebKzs9m0aROBgYGo1WrWr1/PvHnzhix34uXNNDZz5kwOHz4MwPe+9z3q6+spKSkBjgaPf/zjH0oinnfffVf59vZ4PDz77LPccsstvPzyy0ybNo01a9Zw6aWX8swzz/Dee+/R2dnJ22+/rWRjq6iooL29nbKyMlasWMHAwACNjY3MmDGDBQsWkJeXx9atW5k1axY333zzkHLv2LGDuLg4XnzxRZqbm9mxYwdPPfUUU6dOZerUqXz88cc0NTVx6NAhVqxYoaS+7Orq4oUXXuBPf/oTP/rRj5R5Q3A0heOmTZvYtWsXBoOBFStWUF1djclk4s9//jNLlixh5cqVvPPOO5jN5pH6k0ojTH4dnILdbqevrw+VSoXValWy9Lvdbvz8/JSqvnd6/+DgICtWrECr1XLPPfeckELB7XbT39+Py+Xi008/5aqrruLw4cNotVruv/9+7r33XiIjI2lra+Pzzz8nJCQEOJr64OabbyYiIgIAPz8/ZSKf2+3G5XKhVqtRq9VK8mmNRoPNZuPIkSNKisewsDCMRqMybkStVmOxWPB4PPy///f/KCkp4f/7//4/li1bRmFhoVJu78RBtVqNx+MhMDCQOXPmKCkYjxw5QmRkJAaDQXmOd19AyU1yPD8/PxwOBw6HQ3n82PMraypjm/zrnURYWBgrV65k27ZtBAUFcc8997Bu3To+++wzrrjiCs4//3y6urr4+c9/TkREBEFBQezYsYP169cTHR1NVVUVjz/+OJGRkcDRD5HT6eQnP/kJc+fO5YILLmDevHnK6NSkpCRuvfVWDh06xN69e/nZz37GNddcA8D777/P7t27+d73vodareb+++/n2WefJTk5malTp/KTn/yEF154AX9/f6644gri4uK48sor+c///E+KiopITU3FYDAoy5wkJiZiNBrJy8vjL3/5izL5D44mWgoPDx9yLnbs2EF/fz9JSUnMnTuXiIgIVq1aRXJyMtHR0aSnpyu3aseev+uuu45nnnkGt9vN/fffj8vlIj4+Ho1GQ1xcHDNnzuTIkSP8/ve/p6mpCZVKxV133cXq1avZsGED1157LaGhoSP8l5ZGihx5exIej2fIt6ifnx8ejwchhFJb2bhxIxaLhdLSUm644Qby8vJOeI63duBd69n7fG9txvutrlarEUIovx+77IV3uzctgPd372t5v+WP/d2bTsD7HO9QeT8/P9xu9wnv7/3/sWUDWLlyJTk5OcyfP3/Iax//fseWz8u737HH4l2CxPvvJ598gsVioaKiggceeIDg4OATXlsaPeSQ/hHm7SnxLtN67C3GeHJsTpThJoRgcHBQWeFgNKX2lE5ODukfYSqVCoPBMKRdYTwayQ+7SqUadakxpeEje4VOorKykjVr1vDPf/5TyfNxKm63m7fffpv+/v5vzSTvcDgYGBg45eMul4svvviCd999V5kJfCZ6e3uVJN9CCMrLy3n99dfZuHGjkoVNks4FWWM5iYaGBmbOnMmBAwfYunUrhYWF7N+/H4AFCxZQW1urJKmePn06a9as4cILL8Rut9PZ2cnu3bvRaDTMmzeP/fv309vbS3p6Oh0dHWzatImlS5cC0NzcjNFoVNowhBDk5eVht9tZs2YN//Ef/wEcTe+4detWbDYbM2bMwG63U1ZWhtvtZsaMGTidToqLi3G5XISEhJCSkgJAVFQUs2fP5vHHHyc6OnpUrnMjjU+yxnISTqeTTz75hC+//JL4+Hief/55WlpaKCsrY+vWrWzYsIGUlBTKysr48ssvAejr62PHjh28/vrrBAYGMm3aNDQaDYODg/T09PDII48QEhJCeno6cXFxPPHEE1itVv75z38q67lotVoiIyNZv379kMW53G43VquVjo4O3n77bbZv305TUxOhoaGsWrWKVatWkZSUhBBCGRHrHTOzYsUKGhoaiImJOfcnUpqwZGA5CZ1Ox3XXXccdd9zB1q1b8fPzIyAggDlz5lBYWIjD4VBukbRardLroVKp0Ov1OBwOnE4nZWVlrF27lsDAQPr6+tDr9ZjNZvr6+jAYDISFhXH55ZcrI11dLhfPPfcc3d3dTJkyRSnPnj17qKqqIjg4GIfDgU6nIzU1lfDwcFpbW5XxMd4xNYCyDMUjjzzCokWLMJlM5+4EShOe7BU6id7eXgwGAzqdjvr6egIDA1m3bh0ajYbFixezZs0aUlNT0Wq1LF68mLq6OtLS0jCbzRgMBjZs2MDAwABXX30169atQ6VSkZeXR3p6OqWlpcr4lvXr15OSksLixYvR6XS4XC6Ki4vxeDxERUWRmpoKHL0VWrNmDSkpKRQVFSlpHHU6HSaTieDgYDZs2EBaWhq5ubmEhYUhhKC0tJQdO3ZQUFDArFmz5KAz6TuR3c0jSAhBbW0tqampo26BMkkaSbK7eQSpVCrS09N9XQxJGtXkV+5xLBbLt3YxS9JEIITAarUOSYdxumRgOc6aNWuUnh5JmoiEEHR1dXHgwAHq6urOanF7eSt0HKfTeVYRWpLGOo/HQ29vr7L0SmZm5knTf5yOCRNYBgcH2bVrF6GhoUyePBmVSoXNZqO4uJjk5GS5FIU0IXkntTY3N9PT00NoaCg5OTnfeRLohLkVstlsdHd3K3lO4GiGtIyMDL744gtlbIo3f4kkjRculwuLxYLdbqetrQ2z2YzVaqWzs5OmpiYOHjyIw+GgoKCAlJQUNBrNd/6SnTA1luDgYBYvXsz69esJDg4mODgYl8uFv78/fX19uN1uOjo66OrqkjUXacxrbW3lvffew2w2c8011/Daa68REhLCxo0bCQsLw+l0Yjab+eMf/8icOXOG/ZqfMDWW3t5eWlpaSExMVAa+FRUVcejQIebOnYvBYCAlJYXExESZyHkU8uagOX6bN6fMsXlqJIiMjOSaa64hNzcXu92OyWTitddeIyAggLa2NrKysoiKiqK4uHhEvkgnTI0lIiKC8PBwVCoVKpWKoKAghBDEx8cr26TRRwjB3r17eeutt5g5cybXXnstarWazs5Onn/+efr7+7n99tux2Wy8+OKLZGdn8+Mf/1hZJH6i0mq1xMXFcfXVV9PT08PPf/5zhBD09/fT1tbG8uXLiY6ORqvVjsj7T5jAcrLgIQOKb9jtdv77v/+bKVOmcN1116HX62loaKCqqgo4+neZMmUKYWFheDwe3n77bf793/+dp59+mtmzZ5OUlERoaCg//elP2bhxI0eOHGHRokU89NBD/PrXv6a3t3fCBxaPx0NnZyf19fXo9Xqys7Px9/cH4MiRI0yaNImHHnpIyaU83CZMYJFGD7fbTWVlJQ8//DB6vR44mqvGm3vXu49XQkICkZGR2O12ent7SUpKQqPR0NXVhclkYnBwEL1ez8cff8ysWbOIjo4+58c0WrhcLjo7O+ns7CQ0NJRp06adkDY0KyuL559/fkS/VGVgkUbcse0e3lpibGzskLlWISEhpKWlKft4v11VKhXd3d384x//IDAwkLi4OPbu3UtycjKlpaWEh4ej1WrZsmUL77//Ptdddx0Oh2NCTbj0pstoaGigtbWV1NRU8vPzv3Eu20jX1OUkxOP89a9/JSUlhauuumpYX3ci6+/v56mnnuLBBx9UEofb7Xb0ev1pXeB2u51NmzZhs9lISUkhNzcXjUZDSUkJ1dXV6HQ6zjvvPJqamqipqSE8PJwFCxaM+0mi3tzLbW1tDA4OEhcXR0hIyIgdt5yEKI0K3lUbJ0+ezEcffURUVBQhISFccMEFrF27FrVazdKlS0lOTkYIQUVFBb29vcDRxsdJkyah1Wrp6enhvffe4ze/+Q2PP/44TzzxBH5+fmRkZBAaGsqHH35IYmIir776KnfddRcvvPACKSkpSg1ovBFCYLFYlCVys7OzSUlJGVXtheM7pEs+1dnZicViYf78+cTExHDrrbeyYcMGtm3bRnJyMhdddBGPPvqocqvU29tLW1sbbW1tdHR0KNtdLhdGo5GEhARaW1uVLHkBAQFUVVXR2NiIw+Ggrq6O6OjoE9prxgtv8q7y8nI6OzuV5X1DQkJGVVABWWORRpBWqyU4OBg/Pz+io6PR6XRotVoiIiL46quvMJlMTJ8+Xdn/2AXKtFqt8mEJCAjAarXy+uuvM2fOHPr7++nt7WVwcJCOjg5CQkJobm7m/PPP56233sLhcChZ+cYD7xretbW1xMTEkJGRMeqXS5FtLMeRbSzDx5su02AwMDAwQGBgIP39/RiNRlpbW2loaGD69OlKz9A36enpYf/+/cyePRutVqssiLZlyxaSkpLIycnB4XCwa9cucnNziY6OHnXf4mfK28PT1dVFWFgYkZGRPg0oso1FGhW8KTQBZZas99/ExEQSExNP+7XCwsK48MILT9h+ySWXKP83GAwsXLjwuxTZ57wN20eOHGFwcJCUlBTy8/PHXJCUgUWSRgFvD09LSwtOp5O4uDjCw8PHbM+WDCyS5ENCCHp6eqiursZgMJCamkpAQMCYq6EcTwYWSfIBIYSSVEmn05Gbm3vWSZVGIxlYJOkccrvdNDY20tnZSWRkJNnZ2d85qdJoJAOLJJ0DLpcLk8lEb28vAQEBTJ48ecRmFo8GMrBI0gjxNsjW1dUpDbIFBQXjrnZyMhMmsAghGBgYwN/fX6l6CiEQQsj0CdKwEkIwODhIc3OzModnPIyrORMTJrA0NzczMDBATU0Nl1xyCSqVipqaGvbv309BQcGQRdgl6Wx4l81oa2tDo9GQlJSEwWCYUAHFa8IElsTERDo6Ojh06JCyLTQ0lLy8PLq6uhBCUF5ezpEjR5Q1kyXpdHg8Hrq6umhubkalUpGVlTXhE01NmMDS0dHBli1bKCoqoq+vj9LSUrKystBoNAwMDACQm5tLVlaWzJsqnRZvg6x32YyioiLUavWErKEcb8IEFp1Ox+TJk9HpdOj1ejIyMoCjk93mz58vLwbptHiTKjU3N2MymZQG2bE6QnakTJjAEhISMmRNIW+GsqioKF8VSRpDvOsYe5MqxcfHk5KSIgPKKZzRWXG5XPzXf/0XaWlpGAwG0tPTefjhh5UlGODoH+Chhx4iPj4eg8HAokWLKC0tHfI6drud+++/n8jISAICAli6dClNTU3Dc0SSNIy8SZUOHjzIwYMHCQ0NJT8/n7CwMBlUvsEZnZnHHnuMlStX8swzz1BWVsaKFSt4/PHHefrpp5V9VqxYwRNPPMEzzzzD7t27iY2N5eKLLx6SeGfZsmW89957rF27lq1bt9Lf388VV1xxVotPS9JI8A65r6iooL29nZycHGbPnk1YWJi8bT4d4gxcfvnl4vbbbx+y7dprrxU//OEPhRBCeDweERsbKx599FHlcZvNJkJCQsTKlSuFEEL09vYKrVYr1q5dq+zT3Nws1Gq1WLdu3WmVw2w2C0CYzeYh2z0ej3C73WdySCd46qmnxPvvv/+dXkMau1wul2hpaRFbtmwRhw8fFjabzddFGjVO9bk7mTNqY5k3bx4rV66ksrKS7OxsiouL2bp1K08++SQAtbW1mEwmlixZojxHr9ezcOFCtm/fzp133snevXtxOp1D9omPj6ewsJDt27cPya/hZbfbsdvtyu8Wi0X5v81m4+WXX8ZoNNLR0YHRaKSgoIB58+bJbxbptLlcLrq6uujs7CQsLExJKCWdnTO6FfrNb37DzTffTG5uLlqtlqlTp7Js2TJuvvlmAEwmEwAxMTFDnhcTE6M8ZjKZ0Ol0J6QOPHaf4y1fvlxpfA0JCSEpKUl5rKuri4CAAKZMmUJgYCA33HADb731luwylr6V+FdSpbKyMnbu3InH4yEvL4/4+HgZVL6jM6qxvPHGG7z66qu89tprFBQUcODAAZYtW0Z8fDy33Xabst/xNQXxr2Hz3+Sb9nnwwQd54IEHlN8tFosSXAIDAwkODqagoID8/HxsNhvXX3+9rK1IpyT+NYentbUVu91ObGwsOTk5sjF2GJ1RYPmP//gPfvvb3/L9738fgKKiIurr61m+fDm33XYbsbGxAEr/vld7e7tSi4mNjcXhcNDT0zOk1tLe3s7cuXNP+r56vf6UeVFDQkJYunSp8ntAQAALFiw4k8OSJgjxrwbZyspKZSxTYGCg/BIaAWcUogcHB0+I6n5+fkp3c1paGrGxsWzcuFF53OFwsGnTJiVoTJ8+Ha1WO2Sf1tZWSkpKThlYTofb7aapqQm32z2k+1uSvAGlvLycnp4eCgoKmDx5MkFBQTKojJAzqrFceeWV/M///A/JyckUFBSwf/9+nnjiCW6//Xbg6C3QsmXLeOSRR8jKyiIrK4tHHnkEo9HILbfcAhytYdxxxx388pe/JCIigvDwcH71q19RVFTERRdddFYH4XA4ePLJJ2lra+OXv/wl69at4yc/+Ym8aCY4b1KllpYWYmNjyczMRKPRyOviHDijwPL000/z3//939xzzz20t7cTHx/PnXfeye9//3tln1//+tdYrVbuueceenp6mD17Nhs2bBiSdu8vf/kLGo2GG2+8EavVyuLFi1m9evUJi1efLpfLhRCCwsJCampqqKmpOavXkcYHl8tFe3s7PT09BAcHM2PGjFG/Ds94My7WFRJCcPDgQSorK9FoNCxYsICIiIizem25rtDY5G2Qra6uxuFwkJiYSFRUlKydDKMzWVdoXDSD19TU8Morr1BUVERKSgr79+/3dZGkc0T8K6lSVVUVtbW1REdHM3Xq1AmXWGm0GReTEG02G3FxcXR3dwNQUFDg4xJJI038K6lSU1MT/v7+JCUlYTQaZTAZJcZFYMnNzWXfvn1s3bpVGfQUGxsrL7JxyOPx0NPTg8lkUpIqBQQE+LpY0nHGRWAZHBykpaWF9PR0CgoKeOutt1i0aJGviyUNI5fLRXNzM729vYSFhZGTkzMul80YL8ZFYDEajcycOZP4+Hh27tzJnDlzTtjHbrdTXFyMWq1m2rRpqNVq2tvbqaqqIi4ujrS0NB+UXPo2TqeTlpYWurq6CA0NpbCw8Kx7D6VzZ1w03jocDlpbW8nOzua6664bMmHR68CBAwCYzWba29sB2LdvH5mZmXz++edYrVaqq6upr6+X34I+Jv61okJVVRWVlZUEBQUxdepU0tPTZVAZI8ZFYPF4PFRWVmKxWKirqztpr5DT6USr1eLv74/L5VJG5xoMBpxOJ0II+vr66O/vP9fFl/5FCEF/fz9lZWWUlJQQFhZGfn4+4eHhMtiPMWP+VkgIgdFo5LrrruOdd94B4Gc/+9kJ+02bNo2vv/4atVpNWFgYR44cIS8vj6+//pqFCxdiMBiYMmUKBQUFcmb0OSaEoLu7m/b2djQaDWlpafj7+8tgMoaN+cBitVr5+uuv6e7uRqM5ejh9fX1DJkHC0XaYCy64QPk9JycHgJSUlHNXWGkIj8dDR0cHtbW1BAYGkpmZqeQilsa2MR9YNBoNQUFBGAwGZVtoaKjvCiR9K6fTSWdnJ93d3YSFhTFjxgzlS0EaH8b8X1OlUhESEoLL5aK+vh6tVivnhYxCQggcDgcNDQ20traSnp5OXl6ezIEyTo35wKLVaklJSeG5556jubkZtVpNW1sbN998s7xHHwXEMctmeEdIZ2RkyIAyzo35wAL/tyLdAw88QFdXFx999BFWq3XCrps7Goh/LZtRWVmJWq0mMzOT4OBg+feYIMZFYNFqtRQUFPDZZ5/h8XiIjY1ly5YtXHTRRXLcwznm8Xjo6+ujpaUFvV5PYWGh7OGZgMZFfdTPz4/AwEAqKyu5+OKLiY2NZcmSJTKonEPepEpbt26lo6ODjIwMZWE7GVQmnnERWAYHB9m7dy95eXnYbDa+/PJLORblHHG5XLS2tlJeXo5Go+G8884jMzMTnU4nA8oENi5uhYxGI9OnT2fLli3U1dVxww03yIt6BHlnkB85coTBwUFSU1NlD480xJgPLEIIiouL6erq4sorr2Tfvn1y1usI8WZpa25uxuVyERcXR0REhDzX0gnG/FdMV1cXTz/9NPPmzeOVV17h+uuvZ8qUKb4u1rjiTaq0e/duampqlHV4IiMjZVCRTmrMBxbvPf6aNWtwOp289NJL7Nu3z9fFGhc8Hg/d3d2UlZXR3d1NXl4eBQUFci0e6VuN+VuhqKgoVq1aNaSxNiQkxIclGvvcbjcNDQ10dXURGRlJdna2vL2UzsiYDyx+fn7KCozSd+N0OjGZTPT29hIUFMSUKVPkHB7prMirZoLzZrlvaGhQGmQTEhJkD4/0ncjAMkF5s7Q1NzdjtVpJSEiQjbHSsJkwgcXtdtPT04PRaMRoNAJHU1r29vZiMBiGrNQ4ngkh6OzspL29Ha1WS1JSkhwdKw27CRNY2tra6OzspKuri4ULF6JWq6mqqqK+vp6CgoJxH1g8Hg+dnZ00Nzej0WjIzMwcksNGkobTuA8sAwMDtLS0kJiYSEBAAA6HQ3ksJiaGjo4O6urqSEpKory8nCNHjpCamuq7Ag8zp9NJW1sbZrOZkJAQioqKZIOsNOLG/RWmVqvR6/WYTCba29uZPn06DoeDxsZGgoKCyMnJUTL45+bmkpmZOebnGQkhcDqdNDU1YTKZSEhIkEPupXNq3AcWg8FAcnIybreb5ORk/Pz88PPzIy0tDSEEPT09LF68WGljGMttDd4envb2dqxWK3FxcaSmpsqAIp1z4z6weB2bQkGlUim3A9HR0b4q0rDxLl1SXV2N0+kkOzub1NTUMR0kpbFtwgSW8UgIQW9vL21tbWi1WvLy8tDr9TKgSD4nA8sY5Ha7MZlMVFdXEx0dTXp6OlqtVgYUadSQgWUMOXbZjIiICObMmYNWq/V1sSTpBDKwjHLeZTOqq6vp6emRy2ZIY4IMLKOUN6lSS0sLTqeT+Ph4cnNzZUCRxgQZWEYZb4NsRUUFRqORtLQ0mf9EGnNkYBklPB4PZrOZ1tZWDAYDhYWFBAYG+rpYknRWZGDxMe+yGa2trcTGxpKVlYVGo5E1FGlMk4HFR7yrN3rn8MyYMUP28Ejjhgws55B3HeOqqircbjeJiYmyh0cal2RgOQe8AaWpqQmn00lsbCxRUVHydkcat2RgGUHepEpNTU0YjUaZVEmaMGRgGQEej4euri5MJhN+fn7k5OQoWeskaSKQgWUYuVwumpqasFgshIWFkZeXJ5MqSRPShL7qj03odLa3J0IIXC4Xzc3NdHV1ER4eTkFBwZA0DZI00UyYwGK32yktLSUgIIDs7GxUKhVdXV3U19cTEhJCRkbGGQUXb5Z7k8mE3W4nLi6OlJQU2X4iSYyDJVa/jcfjweFw0N/fT1NTE1arVamp7Nq1i9TUVL788ktsNpuSvvKbgoM3qVJpaSllZWWEh4eTn59PeHi4DCqS9C/jvsZiNpspKSkhPz+fJUuW8NZbbxEbG4ter8fPzw+tVqu0g+j1evR6/Ulz3noXRu/s7FSy3MukSpJ0cuM+sISFhTF//nzq6+spLi5mypQp+Pv7U1xczJw5czhw4ACLFi3C398fg8FwwrrPHo+H9vZ26urqCAoKIjMzE51OJwOKJH2DcR9YvFJSUkhJSVF+X7hwIQALFiw46f4Oh4Ouri66u7sJDw9nxowZsodHkk6T/KQcR6VSsX37dpqamkhPTycuLo729nba29t9XTRJOue847DOdB6bDCzHufbaa9m1axdwtNZSX19/wj7r16+nsLCQhISEc128syKEUG4Dx4p9+/bh5+fH5MmTfV2U02Kz2Vi/fj1Lly4dM7fJzc3NuFyuITX542m1WjIyMs44sKjEGFydy2KxEBISgtlsJjg4+Jy/vzeJ9VhZllUIQVtbG7Gxsb4uymlrbW1FrVYTExPj66KcFpfLRU1NDVlZWWMmsPT19eHxeE5oVzyVM/ncyRrLaRJCsHHjRnQ6HdOmTSMoKAi3282BAwfo6upizpw5oy7QmEwmysrK8PPz4/zzzwegrq6OiooKEhMTKSgo8HEJ/48QgtraWiVoT5o0CSEEW7duxe12M2XKlNP+AJwrLpeLXbt20dvby7x588jOzqanp4ddu3YRGxvLpEmTRl2Qqa+vp6ysjMTERAoLCxFCUFFRQWNjI/n5+cTHxw/L+4z7cSzDQQiBEIKWlhbgaHImOLoutMPhIDMzk8bGRl8WcQhveWNiYpg9ezYdHR1KF7rZbMZut4+6IAiwd+9eZs6cyeHDh3G5XDgcDgYHB8nLy2Pfvn2jbulbPz8/Zs+ejVqtxmw2A0cHYtpsNmw2m49Ld3IWi+WEv39jYyNz5szhyJEjw/Y+MrCchvLycurr67n66qtJS0ujsrISp9OJSqXC7XZjsVhGVU4Vs9nMV199RUtLC2+88YZSPbdarWRmZjJnzhyqq6t9XcwTaLVa+vr6gKNLnbhcLgYHBxkcHCQgIMDHpTuRx+Nh//79JCUlERsbi91uJyQkhLlz51JRUcHAwICvi3iCjIwM5s6dS3V1NW63G4/HAzDs17C8FToNeXl5SpVxYGCAgoIC+vv7CQoKIjY2FrPZTHp6uq+LqQgNDeWCCy6go6OD9PR01Go1TqeT1tZWNBoN7e3t5OXl+bqYQ6hUKubOnUtdXR1z5sxhcHAQOHrue3p6RuVthRACt9vN4OAg/f392Gw2PB4Pra2tTJ06dVQGw66uLtra2sjLy8Nms6FWq8nPz6epqYlJkyYN2/vIxlvJ546/BEdbAJGOko23ks99/vnn9PX14efnx4UXXviN395Op5Mvv/ySoKAgrFYrixcvPuW+xcXFhIeHk5SUdNLHm5ub2bt3Lx6Ph1mzZg1bY6R0ZmRgkUbEmjVr+MUvfsHHH3+Mv78/sbGxHDhwgOjoaObOncubb76JRqMhNDSUJUuW0NbWhr+/Pw6Hg6qqKtatW0doaChLly7lgw8+wGw2c8MNN7By5UoCAwO54447aG1tpbGxkby8PGbMmIFKpVLaOKqqqnj33Xe57777fH0qJqTR0+IojSsul4svvvgCk8lEbm4ujz76KFlZWezZs4fi4mK2bNnC1VdfTXl5Oa2trcDRsSt1dXU8++yzXHPNNVx99dW4XC76+vpoa2tj/fr1pKWlsXjxYux2O2+99RZ5eXm89tprSqNvQEAAWq2WTz75hCVLlvjyFExossYijQitVsuVV15JSUkJ69atQ6vVcuTIESZNmkRqaioul4vOzk7MZvMJvRH+/v40NTXh5+fHli1bMBgMzJw5k87OToxGIx0dHURHR2M2m2lqauLyyy/H398fODp254UXXuC8884jPDzcF4cuIRtvpRFy4MABsrOzUavVVFdXExUVRUlJCXq9nsLCQh5//HEWL15MUFAQU6ZMUUbaulwuQkJC2L9/PyEhIaSkpHDw4EGCgoKIj48nICCAffv2kZOTg9lspq2tjfDwcIqKilCpVLS1tVFcXIxOpyMmJmbU9X6NZWfyuZOBRTrnXC4Xhw4dYsqUKbIHaAyRvULSqKbRaJg6daqviyGNoDEZWLyVLIvF4uOSSNLE4f28nc5NzpgMLN4egFONZZAkaeT09fV964TQMdnG4vF4qKioID8/n8bGRtnOcgoWi4WkpCR5jr6FPE+nx5tIPj4+/lvnFY3JGotarVaSLAUHB8uL4VvIc3R65Hn6dqebukIOkJMkadjJwCJJ0rAbs4FFr9fzhz/8Ab1e7+uijFryHJ0eeZ6G35hsvJUkaXQbszUWSZJGLxlYJEkadjKwSJI07GRgkSRp2MnAIknSsBuTgeW5554jLS0Nf39/pk+fzpYtW3xdpHNm+fLlzJw5k6CgIKKjo7n66qupqKgYso8Qgoceeoj4+HgMBgOLFi2itLR0yD52u53777+fyMhIAgICWLp0KU1NTefyUM6p5cuXo1KpWLZsmbJNnqcRJMaYtWvXCq1WK1544QVx+PBh8Ytf/EIEBASI+vp6XxftnLjkkkvEqlWrRElJiThw4IC4/PLLRXJysujv71f2efTRR0VQUJB45513xKFDh8RNN90k4uLihMViUfa56667REJCgti4caPYt2+fuOCCC8TkyZOFy+XyxWGNqF27donU1FQxadIk8Ytf/ELZLs/TyBlzgWXWrFnirrvuGrItNzdX/Pa3v/VRiXyrvb1dAGLTpk1CCCE8Ho+IjY0Vjz76qLKPzWYTISEhYuXKlUIIIXp7e4VWqxVr165V9mlubhZqtVqsW7fu3B7ACOvr6xNZWVli48aNYuHChUpgkedpZI2pWyGHw8HevXtPSJK8ZMkStm/f7qNS+ZZ3aU9vftfa2lpMJtOQc6TX61m4cKFyjvbu3YvT6RyyT3x8PIWFhePuPN57771cfvnlXHTRRUO2y/M0ssbU7ObOzk7cbjcxMTFDtsfExGAymXxUKt8RQvDAAw8wb948CgsLAZTzcLJzVF9fr+yj0+kICws7YZ/xdB7Xrl3Lvn372L179wmPyfM0ssZUYPE6Pk+qEGJC5k697777OHjwIFu3bj3hsbM5R+PpPDY2NvKLX/yCDRs2KBn8T2ain6eRMqZuhSIjI/Hz8zvh26K9vf2Eb57x7v777+fDDz/kyy+/JDExUdkeGxsL8I3nKDY2FofDQU9Pzyn3Gev27t1Le3s706dPR6PRoNFo2LRpE3/961/RaDTKcU708zRSxlRg0el0TJ8+nY0bNw7ZvnHjRubOneujUp1bQgjuu+8+3n33Xb744gvS0tKGPJ6WlkZsbOyQc+RwONi0aZNyjqZPn45Wqx2yT2trKyUlJePmPC5evJhDhw5x4MAB5WfGjBn84Ac/4MCBA6Snp8vzNJJ82HB8VrzdzS+99JI4fPiwWLZsmQgICBB1dXW+Lto5cffdd4uQkBDx1VdfidbWVuVncHBQ2efRRx8VISEh4t133xWHDh0SN99880m7URMTE8Vnn30m9u3bJy688MJx3416bK+QEPI8jaQxF1iEEOLZZ58VKSkpQqfTiWnTpildrRMBcNKfVatWKft4PB7xhz/8QcTGxgq9Xi8WLFggDh06NOR1rFaruO+++0R4eLgwGAziiiuuEA0NDef4aM6t4wOLPE8jR+ZjkSRp2I2pNhZJksYGGVgkSRp2MrBIkjTsZGCRJGnYycAiSdKwk4FFkqRhJwOLJEnDTgYWSZKGnQwskiQNOxlYJEkadjKwSJI07P5/XuV1s0cUZw8AAAAASUVORK5CYII=",
            "text/plain": [
              "<Figure size 640x480 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "dna_pairwise_no_thresh = cv2.imread(f'{results_dir}/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_DNA_pairwise_minThreshold.png')\n",
        "\n",
        "dna_pairwise_no_thresh_resize = cv2.resize(dna_pairwise_no_thresh, (600, 1000))\n",
        "\n",
        "plt.imshow(dna_pairwise_no_thresh_resize)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAARYAAAGiCAYAAAAiFuksAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjkuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/TGe4hAAAACXBIWXMAAA9hAAAPYQGoP6dpAAB+3klEQVR4nO3dd3xb5b348Y9kDct7r3iveGbvTQaUFQIFWlahUH7sSy60BS63LQVKKNyy12U0FMhgBggjgwDZ0yux49jxXrLlKcm2ts7vj1Tn4gxIgh2v5/16+ZX46Eh6zvHRV895xvdRSJIkIQiC0I+Ug10AQRBGHhFYBEHodyKwCILQ70RgEQSh34nAIghCvxOBRRCEficCiyAI/U4EFkEQ+p0ILIIg9DsRWARB6HeDGlheeeUVkpKS8Pb2ZvLkyWzfvn0wiyMIQj8ZtMDy/vvvs3z5ch5++GEKCgqYO3cuF154IXV1dYNVJEEQ+olisCYhTp8+nUmTJvHqq6/K2zIzM1m2bBkrVqwYjCIJgtBPVIPxpna7nby8PB588ME+288//3x27dp1wv42mw2bzSb/7na76ejoIDQ0FIVCMeDlFQQBJEnCbDYTExODUvnjNzuDElja2tpwuVxERkb22R4ZGUlzc/MJ+69YsYK//vWv56p4giD8iPr6emJjY390n0EJLB7H1zYkSTppDeShhx7ivvvuk383Go3Ex8dTX19PQEBAv5eruroai8VCVlbWaT/n6NGjFBUVodFo8PPzY8GCBXJUt1qtrF+/Hh8fHxQKBfPmzcPX15f169eTk5NDcnIyRUVFhISEEBcXR2dnJ5s2bWLp0qXodLp+Pz6TyURhYSHz5s3r99c+lU8++YQlS5bw7rvvct111xEYGHjO3lvoHyaTibi4OPz9/X9y30EJLGFhYXh5eZ1QOzEYDCfUYgC0Wi1arfaE7QEBAQMSWCoqKmhubmbGjBnAsVu3vXv34nQ6SUlJIS4ujoqKChoaGvD19SUgIICwsDBmzJhBQEAAt912GwsWLJDL5nQ6+fLLL3njjTd48cUXUSqV/PKXv+Stt94iOzubp59+mt27dzNu3DiysrJ4//332bRpE/Hx8Zx//vlysJUkifLycpqamvDz82PixInU1tZSV1eHRqNh+vTpdHR04Ofnh06nIz8/n6ysLEpLS3E6nfT09DB+/Hj0ej2vvvoqAQEBpKamUlZWhkKhIDk5mfj4eODY7eaePXuwWCwATJo0icDAQAoLCzGZTISGhpKTk0NxcTEtLS2MGTOGzMxM+Xzl5eVhtVoJCwsjJycHo9GIj48Ps2fPJjw8nN7eXgoLCwkICMDtdjNlyhTKy8vR6/V4eXkxY8YMNBpNv/9thZ/vdJofBqVXSKPRMHnyZDZv3txn++bNm5k1a9ZgFOlHvfnmmzQ1NRETE8OKFSsoKyvj9ddfJz4+noKCAlatWkVCQgKZmZl0dnaSkZGBt7d3n9ew2WwcOnSIhoYG0tPTAQgNDSUhIYHPP/8cTxt6V1cXu3fv5vHHH+fjjz/GbrfLr1FdXc0rr7xCQkIC0dHRNDc388wzzxAXF0d5eTnvvfce+fn5GAwGAP7nf/6HtrY2/vrXv+J2u/Hy8uKf//wnOp0Of39/EhMT2bx5M3l5ecTExPSpRbhcLv7617/i7e1Na2srK1euZNu2bXz55ZfExsayevVqiouLeeONN6irqyM8PFx+7saNG9m8eTNxcXE899xzfb5AXnnlFdrb2/nrX/9KQEAADoeDP//5zzQ3N/PEE08QExPDrl272LRpU///IYVzZtC6m++77z7efPNN/vnPf1JaWsp//ud/UldXx+233z5YRTql4uJiJkyYQHp6Og6Hg4qKClwuF0lJSeTk5ADHahOHDh3igw8+4E9/+tNJa1iBgYGMGTNGrgUA3HDDDWzbto2GhgYANm3ahEKh4NChQ7S0tPD999/LQaelpQU/Pz+SkpIYM2YMRqMRpVJJSkoKubm5HD58GEmSsFqtfd43NDSUiRMnMmHCBMrLy9HpdPj6+hIXF8dVV12Ft7c369evZ+fOnfywkzAsLIyJEyeyePFiuru7KS8vp6Wlhb179xIZGYlCocDLy4spU6YQFhYmP6+1tZUpU6aQkpJCeno6paWlfcrjdDppaGhg3LhxpKWl4e3tTW9vLxEREaSnp7NkyRJ27tz5M/9qwmAatDaWX/3qV7S3t/Poo4+i1+vJycnhq6++IiEhYbCKJNNoNFRXV7N582bUajUXXngh69evJy0tjdjYWObNm8e3337LW2+9RU1NDUqlkj179vD444/z29/+lsLCQmbMmCHXWhQKBZGRkYwfP56oqChWrFhBTk6OfMty2223cd1113HRRRexZ88e/uu//ouIiAiysrL45z//yeLFi/Hy8iI9PR2Xy8WGDRvw8fEhKysLf39/vvzyS4qKirj88stxu918+OGHTJw4EbPZjFKpxNfXF4VCgUKhwMfHB19fX1QqFVu2bKG3t5fU1FSamprQ6/V92rlaWlp4//33KS8vJzs7m2nTplFXV0dERAT+/v6kpqai0+nw8vLqc/6mTZvGmjVr6OnpobS0lLvuuovy8nIUCgW+vr7odDoWLFjAv/71L+rq6jCbzYSHh6PRaPjss8/4/PPPueeee875313oP4M2juXnMJlMBAYGYjQaB6SNxWKx0NXVBRwLCmFhYXR0dOByuQgKCsLb25vS0lKOHj3K/v37mTNnDvPmzcNoNAKgVCrldiQ41l5hNBoJCgpCoVDQ1dWFn58fZrOZwMBAFAoFbW1t+Pn5YbVa5f3cbjddXV0EBwfLH/bu7m7MZjMqlYrQ0FCsVitGoxGVSkVYWBiSJNHa2opCoUClUhEYGCi/jyRJ8rnr7OzEZrPh7e2N1WqVj1OlOvZd43A4uPvuu3nooYfw9vaWH+vs7MRqtaJUKgkPD8dsNuPj44Nare5zDj37+fn54e/vT09PDzqdTv6b1dXVcfDgQVpaWrBardx9993YbDa6urpQq9ViKMEQdCafOxFYzoIkSej1eoxGIxqNhsTExBO+tYc7zzFGRUX95JiFs2EwGGhvb0epVJKUlCQaaocBEVgEQeh3Z/K5E7ObT8LTNdvT04PL5frRfd1uN4WFhVgsFhobG/mxOO12u3E6nad8XJIkent7sVgsP/o6p2KxWOjo6JBfy2q1Ul5ejtlsPqvXE4SzJQLLSezcuZM1a9bwj3/8g7179+JyuSgvL6eyshKXy0VbWxsVFRXU1tbicDh49NFHMZlMVFRUAFBZWcnhw4ex2+1UVFRQWlqK3W7n4MGDvPfeezQ0NNDb20tJSYncYArHxn9s3LiRW2+9tU83s9vtpqqqitLSUqxWK11dXdTV1XHkyBHsdjsul4sjR45QUFDAwYMH5eft3buXrVu38pe//IWamppzeg6F0W1QR94OVT09PRgMBpxOJ06nkw8++IDKykr8/PwYN24ceXl5TJ8+ne+//55LLrkESZKw2WzU1NRgt9spKSlh3rx5uFwuampq0Ov1fPPNN0ydOpWuri4sFguPP/44qampVFdXc/fddxMZGYlWq+XCCy/knXfe6VPD6OnpoaioiI6ODvbs2YNOp6O4uJicnBwKCwtxOBzye02fPh041ug8f/58pkyZwqOPPnpC46ogDCRRYzkJpVJJREQEKpWKrq4uSktLSU1NJSsri+TkZPz9/Zk+fTpjx46VawiSJOF2uykvL+eiiy5i4sSJNDU18d1336HT6di8eTNBQUEkJCQwZswYDAYDMTExzJkzBz8/vx8tj16vx2w2o9Pp6OjowMvLi7lz55Kbm8vOnTspKipi4cKFzJs3T+7VkSSJlpYW3nvvPa644grGjBkz4OdNEDxEYDmJ0NBQli5dygMPPEB1dTWXX345VVVV1NXV0dLSgtlslmsxF110EWlpaWi1WsLCwrj00kv54IMPeP/99+UxISaTidTUVMLDwykuLiY/P59LL72Uo0ePUlFRIQ+YczqdfPfdd3h7e7Nz507cbjcAOp2Orq4uGhsbiYqKIiQkhKCgIHQ6HQkJCfzmN79hzZo17Nmzp888jv/93//FarVSUVFBW1vboJxLYXQSvUIn8cNT4gkOng85wIcffsgVV1yBSqWSH1MqlSfsq1Qq5f97Bqi53e4+//fsp1Ao5FqPx8m2e557qvL91GsJwtk6k8+daGM5iR9+AD3/94xTkSSJX/7yl3JQ+eFjx/9+/P9/6jHPEPmTledU42RO9p4/9RxBGGgisJwhhUIhGkIF4SeINpbj7NmzhyNHjgx2MQRhULndbpqamvoMXzgTIrAcZ9++fZSVlQ12MQRhULhcLvR6PYcPH0ahUMg5ds7UqLkV8ow1USgUaDQaucGzs7MTf39/MVdFGNVsNhsVFRWYzWYSExPJzMz8WW10oyawdHV18dVXX5GRkcGkSZMA2LFjB35+ftTU1HDZZZfJvSmCMNR5rtPje/o8vY4//L/nuvb8K0mSPLHUZrPR0NCAw+EgKiqKrKysfuk9HDWBxdvbm2nTplFQUEB6erqcomDmzJls374di8VCQ0MDlZWVJCUlDXZxBaEPSZLo6enB19cXOJZtcfr06QQEBKBQKLDb7Xz55Zd8/fXXpKamolQq+fTTT7niiivo7e2lrKyMuXPnYrVa+eabb5g+fTpZWVmkpaWRkJCAn59fvw5HGDWBpbOzk56eHvz9/enu7mbv3r3ExMSQn5/P+PHj8fHxISMjg9TUVFFrEQadzWZj3759TJs2Da1Wi8Ph4O233+b6668nMDCQ6Oho1q1bR01NDb///e954IEH2LhxI3AssXtBQQFarZY//elP9Pb2ArBlyxYSEhKor68nKCiI9evXs379+tNKjn2mRk1gCQ4ORqPRMHbsWDQaDf7+/vj6+spJlwYi54jw80mSRHNzM42NjcTExBAdHY1CocBsNlNeXo7b7SY2NhY/Pz/KysoICAggLS1t2A8GVCgU8sx6l8uFWq3mzjvvRKFQoNfrefbZZwkLCyMsLAyr1UpUVBRarZbo6Gh27dqFw+Ggu7u7z+x8vV7P+PHj6e3tZePGjWi1Wt5++23+8Ic/9Hv5R82nSafTERYWJqdS9FT9goODxbiUc8zhcLBmzRq2bNkiz+L23Pv/8MezfcWKFVitVl588UVMJhNwbCSxVqvlk08+YdeuXaxatYqOjg4+/vjjETFcQKPREBISwkMPPcQLL7yAw+FAoVBw4MABKioqePjhh1mxYgXFxcU89NBDvPXWWxgMBgoKCrBYLISGhnLnnXeiUqkICgpCrVYTHR3NjBkzSEpKYuzYsUiSJK9E0d9GTY1FGDo81fq3335bDup5eXls27ZN3ueqq64iLi4OSZJISUlhxowZctV//Pjx+Pr6kpCQQFtbG4sXL+aLL75Ar9dTUVHB999/f9bdpENJdnY2f/vb3+jt7cXlcvH444/Lc9JycnIoKCjgV7/6FY899hheXl54eXnJNTWDwcCGDRsICQkhPDwcpVKJ3W5n06ZNBAcHY7PZ+MUvfjFggWXU1FiEocXHx6dPXtuQkBDGjh0r//j4+ADHbgnq6+upqqoiICCA0NBQuUayd+9eFixYQGBgIEuWLGH69OlERETIqSOGu2+++YbVq1ej1WpRKpVcdNFFfPrpp3z33XfU1tayZcsW1q1bR2NjIzqdDrvdTlhYGNOnTycwMJD29nbuvfdegoKCKCsro6mpicbGRgwGA/Hx8YwdO1Zuf+lvosYiDDiXy0VTUxOxsbHylIhrrrmmzziJ5ORkkpOTT3iuQqHgvvvu46OPPiIzMxOlUklFRQUZGRm4XC5SU1M5fPgwPT09FBcXk5ubS2Ji4ilX1RyqJEmis7MTOBZkAWbOnMmqVavYuHEjR48e5Q9/+AN/+MMfmDp1KmPGjKGuro6vv/6anJwc9Ho9CQkJdHR0sHDhQmJiYtDr9bz77rv4+fnxj3/8g6amJiIjIzEYDHzwwQekp6cPSMMtiBqLMIBaW1v5+uuv2bZtG8uXL+frr7+moqICpVLJmDFjWL9+vdxmcioKhYKAgAC6urrQ6XS8/fbb/OIXvwCQ13R6+umnUavVdHR0oFAoePrpp2lvbx/w4/u5ent72bx5s5yutKqqqs+o74CAAO644w6WLVvGr371K1auXMmVV15JUlIShw8f5t1332XKlCk0NDRwxRVXYLPZWLJkCbGxsZSUlJCTk8OECROoq6ujsbGRzs5Ovv32W2699VZyc3OZN2/egAVfUWMRBkxdXR2rVq3ib3/7G93d3aSmpvLkk09yww03kJeXx+TJk3nvvfe44447APjyyy+pra0FjjW2X3PNNXIumra2Nn7xi1/w9ttvY7PZUKlUjBkzBo1GQ3x8PFlZWTz22GOsXLmSffv20dTU1GcRtaHIM/rb6XSiUqmYMmUKcGxgm91ul9el8vLyIjQ0lGXLltHW1sbatWtRq9UsWbKE/Px83nrrLXbv3s1f//pXLBYLnZ2dzJs3D41GQ0dHBwsWLOCmm27CaDRSVFSEy+Xi3XffRafTicAiDE/R0dHyGtspKSn09PTQ1tZGeHg4ISEhfb6hPY2KgNyuAKBWq+np6aGjo4OgoCB6enqwWq2EhISwbt06Zs+ejZeXF4GBgXR0dOBwOE5Y4nYo0ul0XHDBBSdsb25uZv369dx2223yNs9yxBEREWRnZxMdHQ0cW6Vz7NixJCUlcfDgQRYsWIDNZkOSJDo6Oti4cSORkZFs2LABjUbD559/Tnt7O5dffjljx44dsGMTgUUYMBEREcydOxetVsvFF1+MQqHgwgsvZObMmezZs4ddu3bx4IMPyt+as2fPPuXr3HTTTXz88ccsX76czs5Ouru7CQkJwdvbm1mzZqFUKvnv//5vvv76ay6++GJSU1PP5aH2q+joaH73u9/Jqza0tLSQkZEhr6+9efNmTCYTXl5e3HvvvQQEBCBJEocPH2bixIlyG43b7aalpYWbbrpJHs6fmppKUFAQkZGRA3oMIoPccV544QUSEhK47LLL+vV1BeF0eYbvV1VV0dvbS3p6urwa5uHDh9m3bx833XTTCc+x2WxotVo5UFssFp599lnuvfdeeSrAzyEyyAnCMOR2uzGZTDQ3N6NWq0lLS8Pb27tPO0h8fDxRUVEnPFehUMi3f263G7fbjbe3N8uXL0en052zY/AQgUUQBplnRYXKykoCAwNJTU3tU/P4oZ9a0QGgoqKCffv2cf3118vjgc41EVgEYZA4nU5aW1tpb2+XB7Z5lm/5OVJTUwd9hr4ILIJwjtntdqqrqzEYDCQnJ//spErHUyqVgz6pVgQWQTgHJEnCbrfT0NCA3W4nOjqa9PT0YTU6+EyIwCIIA6yrq4uysjK0Wq28kuZIDSgeIrAIwgCQJAmj0UhTUxPe3t7k5uYO6EjXoWbUBJbGxkaOHj2KQqFg7ty5KJVKmpqaKCkpkWd6CsLP5Xa7qa+vl5fDTU1NRa1Wj5qA4jFqAktMTAzBwcF899138jaz2SzPyZAkiYKCAg4dOkRiYuLgFVQYdjzL2er1ejo7OwkODmbatGl98qOMNqMmsPT29vLRRx8xffp0JEnCarUSFxdHcHAwhYWFxMfHExwcTGBg4GAXVRhGLBYLR48exe12ExMTQ3Z29qD3yAwFoyawmEwm4uPjsdvtWK1WDAYDcGxqf3Z2NgqFgqSkJOLj40UybeFHeb6Y6uvrsdvtxMTE9ElaJYyiwBIdHS3PCAXkAUSDPZBIGD4kSaK1tZWmpiZ0Oh2xsbH4+PiIgHISoyawCMLZ8qQgaGpqQqVSkZ6ePqp6eM6GCCyCcAoOh4OGhga6u7sJCgoiIyNDrOhwmkRgEYQfkCQJl8tFY2MjbW1tBAcHy0PuRQ3l9A37wOJJcFNUVMSFF16ITqejublZdBkLZ8xsNtPS0oLdbicqKoq4uDjRw3OWhv1Z6+joYM2aNfj7+/PMM89QUlLCwYMHB7tYwjDhSap0+PBhjhw5QlBQEJmZmYSEhIig8jMM+xqL2+0mJSWFSy65hLlz5/LBBx+IsSjCT3K73XR2dtLa2opKpSIpKemEpErC2Rv2gSUsLIwbbrgBgKCgIG6++WasVusgl0oYqjxdxjU1NWi1WtLT00VAGQDDPrAoFIo+yXFUKtVpZdkSRhe73U57ezvt7e2EhoYyadKkfkmqJJzciLiJdLlcfPPNN7z++uu0t7ezbds2MXpWAI51GVdUVLB3714cDgeZmZlER0eLoDLARsTZtVgs5Ofnk5iYSHd3N5988glz5swR1dtRyjPkvqmpSR5yn5KSIq6Hc2hEBBZvb2+io6P58MMP2b17N1dddZW4iEYpk8lEeXk5breb9PR0AgMDxbUwCEZEYOnt7cVisfD8889js9nkNXyF0cGzbIZnDk92drZokB1kIyKwrFy5kv/93/9l9+7dSJLE3XffPdhFEs4BTw6UqqoqIiMjSUlJQaPRiIAyBIyIwHLHHXeQm5tLVVUVDoeDoKCgwS6SMIBcLhfNzc10dnYSGhrKzJkzxZD7IWbE9Apt374dm81GdHQ0e/fuFb1CI5DNZuPQoUMcOHAAlUrVp4dHBJWhZUTUWLy8vMjJyUGtVrNz505iY2NP2EeSJPR6PTqdjqCgIBQKhZyfNCwsrF/WthX6n2dN4vr6epxOJ2PGjJHXMRaGrhERWJRKJREREcyaNYuMjAy6u7tP2KeyshK9Xk9vby/Tp08nKCiI/fv34+fnR3FxMRdccEG/Lhol/DyeHCgVFRX4+/sTHx+Pr6+vCCjDxIi4FbJarWzduhVJkggICGDlypUn3ArV1dURGBhIVFQUXV1dABiNRtLT06msrMRqtXLkyBE5k78wODwBpaSkhI6ODnJzc8nMzMTPz0/8XYaREVFj8fHxYdq0abz99tuYTCZuvPHGEy7C7Oxs8vLycDqdzJw5kyNHjhAaGsqBAwfIyMjAx8eHzMxM0tLSRPvMIHC5XNTU1GA0GgkLC2Ps2LEiqdIwNuwDi8lk4ssvv0Sv18s1EYVCcUJgiYiIYOHChahUKry8vAgLCwOOjdr19vYWU+QHgWfZjMbGRjo7OwkKCmLcuHGih2cEGPaBRa1WExUVhY+PDzabDYDg4OAT9lMoFHh7e5+w3cfHZ8DLKJyou7uburo63G430dHRxMbGiuA+ggz7wGK1Wlm7di09PT3ytvDwcJKTkwexVMLJSJKExWKhrq4Oi8VCXFycSKg0Qg37wBIUFMQrr7zCZ599Rm9vL06nk5SUFCRJEtXpIcLtdtPa2kp7eztqtZr4+HiR5X6EG/ZfFQqFAqvVSmVlJVdffTXnnXcer732GoBohB1EkiTJSZUOHjxIS0sLiYmJpKamirV4RoFhX2OBY+0sGo2G1atXY7Va8fX15dNPP+XSSy8VeTcGgd1ux2Aw0NXVRUhICDk5OeLvMMqcUY1lxYoVTJ06FX9/fyIiIli2bBllZWV99pEkiUceeYSYmBh0Oh0LFiygpKSkzz42m4177rlHHvG6dOlSGhoazvog1Go1l1xyCQqFguuvv57bbruNZcuWiYv5HHM6nVRXV5Ofn4/T6SQjI4OYmBjxdxiFziiwbN26lbvuuos9e/awefNmnE4n559/fp+G06eeeopnnnmGl156if379xMVFcWSJUswm83yPsuXL2fdunWsXbuWHTt20N3dzSWXXILL5Tqrg+jt7WXlypWYzWY6Ojr429/+Jm6DzhFPlvuqqirKy8sJDg5m+vTpJCYmioAymkk/g8FgkABp69atkiRJktvtlqKioqQnn3xS3sdqtUqBgYHSa6+9JkmSJHV1dUlqtVpau3atvE9jY6OkVCqlDRs2nNb7Go1GCZCMRqMkSZJkt9ull156SXrkkUekF198Ufroo48kt9t9Vsf0/PPPS59++ulZPXe0MZlMUlFRkbRr1y6pvb39rM+5MDwc/7n7MT/rK8VoNAIQEhICQHV1Nc3NzZx//vnyPlqtlvnz57Nr1y5uu+028vLycDgcffaJiYkhJyeHXbt2ccEFF5zwPjabTR6jAscGxR1fjtzcXBITEykuLiYsLEw0Dg4Qt9uN0WikublZznKv1WrF+Rb6OOteIUmSuO+++5gzZw45OTkANDc3AxAZGdln38jISPmx5uZmNBrNCYPYfrjP8VasWEFgYKD8ExcXJz9mNpv561//yv79+3nqqafo7u5m7NixZ3tYwilI/54dvmvXLpqamkhKShJr8QindNaB5e677+bgwYOsWbPmhMeOv9Ck0xhT8mP7PPTQQxiNRvmnvr5efsxkMlFdXc2UKVMICgoiJiYGi8VyFkcknIzT6aSpqYmSkhIUCgUzZswgKytLBBThR53VrdA999zD559/zrZt2/rkPomKigKO1Uqio6Pl7QaDQa7FREVFYbfb6ezs7FNrMRgMzJo166Tvp9Vq0Wq1J30sJCSE22+/ne7ubqZOnYrRaMRut5/NYQk/YLfbqayspK2tjZSUFHlhdEE4HWcUWCRJ4p577mHdunV8//33JCUl9Xk8KSmJqKgoNm/ezMSJE4FjF+jWrVv5+9//DsDkyZNRq9Vs3ryZq6++GgC9Xk9xcTFPPfXUGR+ATqfjkksuOePnCSeS/p1UqaGhAYfDQXR0NBkZGaJmIpyxMwosd911F6tXr+azzz7D399fbhMJDAyUh2gvX76cJ554grS0NNLS0njiiSfw8fHh2muvlfe95ZZbuP/++wkNDSUkJITf//735Obmsnjx4v4/QuEnSZJEV1cXZWVl6HQ6kpOTRf4T4Wc5o8Dy6quvArBgwYI+21euXMlNN90EwB//+EcsFgt33nknnZ2dTJ8+nU2bNuHv7y/v/+yzz6JSqbj66quxWCwsWrSIt99+W1S1zzFPQPEsmzF+/HjRdiL0C4UkDb+RZCaTicDAQIxGIwEBAaf9PLfbfdJcLT/0wgsvkJCQwGWXXdYfRR2SXC4XdXV1cltYTEwMGo1msIslDHFn8rkbNUMjm5qaaG1txWKxMH36dBQKBS0tLRw+fJjY2FjS0tIGu4gDSvp3UqWmpiZ5Ds/UqVNFUiVhQIz4wOJwOOjp6SEsLIygoCC+/vpruWvbarWiUqlwOp1IkkRNTQ319fUkJiYOdrH7VW9vL0ePHgVgzJgxjBkzRuRAEQbUiA8sFouFqqoqEhISOHToELm5ufLs2+joaIKDg9m2bRsZGRlERUURHh4+IuYZSf9eGL2urg6Hw0FsbCzBwcEioAjnxIgPLAEBAUyaNImGhgbcbjdmsxmr1Uprays2m43m5ma5a1yn0500feVwIkkSBoMBvV6Pj48PsbGxIv+JcM6N+MDiERsb22cw3+TJkwFGTNuKJEm0t7fT1NSESqUiPT1dZGkTBs2oCSwjld1up7Gxke7ubkJCQsjKyhLpCoRBJ67AYcrpdNLQ0EBbWxuhoaFkZmaKgCIMGeJKHEYkSaK7u5vm5mZ5yH1CQoK43RGGHBFYhgnPOjzd3d2kpKQQEhIiAoowZInAMoS53W46OzsxGAxoNBqSk5NFUiVhWBCBZQjydBnX1tbi7e1NWlqamMMjDCsisAwhdrudtrY2Ojo6CAsLY/LkySiVShFQhGFHBJYhwG63U1dXJ6d8FEmVhOFOBJZBcnxSpTFjxpCSkiJqJ8KIIALLOSZJEiaTifLycgDS09MJCAgQAUUYUURgOUfcbjcmk4nGxkZ8fHzIzc0VPTzCiCUCywBzu900NjZSXV1NTEwMqampp0wMLggjhQgsA8CTVEmv19PZ2Ul4eDizZ88WPTzCqCECSz+zWq2UlZVht9tJSEggOztb5EARRh0RWPqBp4enrq4Ol8tFfHw8gYGBIqAIo9aoCSxOp5OWlhZUKhURERFyasrW1lYCAwPPKCm3hyRJtLW1UVVVJS/9KpIqCcLPWGJ1uDEajezZs4fW1lZ5286dOzEYDGzYsEHOjdvb2/uTr+VJqlRcXIzRaGTcuHGMHTsWX19fEVQEgVEQWLq7uykuLkapVJKcnExhYSHd3d3AsWUwcnNzaWhowGazsWPHDvbt23fKWxiHw0FFRQWFhYX09vaSkZFBamqqyNQmCMcZ8bdCarWaiIgIeUF5jUaD0+mkrKyM0NBQvvnmGyZOnIiPjw/nn38+ZWVluN1u+fmeHp6GhgY6OzsJCQkhNzdXLJshCD9ixAcWrVZLREQEbrebuLg4AJRKJYGBgSgUCnkRs+NrKZ6kSnV1dQBER0cTHx8vgokgnIYRH1g8jg8cngBxssl+JpOJ/fv3Y7VaiYuLk5Mqmc3mc1JWQRgqFAoFvr6+Z9zDOWoCy+maNm0azzzzDBs3bjzlUiBGoxEfHx/UavU5Lt3Z6+3txcfHZ7CLcdo8jejDpcyedbCDg4MHuyinzeFwIEnSjy6v6+Pjw6OPPkpQUNAZvfaoWrv5dEiSJP+cyrvvvsusWbNISUnp1/ceKJIksX37dubNmzfYRTlt3333HSqVirlz5w52UU6LxWLhnXfe4bbbbhvsopy2yspKHA4HGRkZp9zHs9a5QqE4o8+dCCynSZIkDh48iMvlIiUlBX9/fxQKBWVlZdhsNjIzM4fcwuoWi4WysjJCQkKIiYlBpVLR2tpKdXU1YWFhJCcnD3YR+/CMWk5KSsLX1xeA1tZWmpqayMrKGnKLyblcLkpKSvDx8SExMRFJkrDb7Rw+fJigoCBSU1OHXJucZ9xVaGgoSUlJwLF1zZuamhg7diyBgYGnfO6ZfO5GfHdzf5EkiV27dmGxWFCpVCiVSnp7eykvL8fhcFBVVTXYRTyBzWYjPj6eyspKeVtNTQ0tLS1D7kMqSRKbNm0iODiYDRs2IEkSLpeLQ4cO4e3tzaFDh4bc0rdKpZLExEQqKipoaGhArVbT1dVFU1MTOp1usIt3UnV1dej1ery9veW5awcOHCAkJIT8/Px+ex8RWE5DWVkZ9fX1XHXVVfj7+1NQUIAkSTidTiIiIoiOjh5SF73RaGTbtm24XC6MRiNjxozBy8sLSZLIyckhNze3Xy+i/mI2mwkKCsLLywu3243b7ZZ79To7Owe7eCdlNpuxWCzExMQgSRIRERFMmzaN3bt3n9Zgy3MtMzOTCRMmkJ+fL9/y+/v7ExcX16/rUonG29OQmpoKQHV1NUqlkoiICNra2vDz88PpdNLR0UFiYuLgFvIHAgICmDlzJgaDgZ6eHrKzs7Hb7fLUA4fDQVhY2GAX8wQTJkzgwIEDxMXF0dbWhkKhICQkhOrqatLT04fcbYXL5cJisXDppZeiVCqpq6tDo9FgMpkICQkZcrfGAJ2dndjtdsLCwuju7kahUBAaGkpFRUW/3hqLNpYz4HK5cLvdqFQq+SL3fLMOl1UIJUnC4XCgVquH3AfVUwv84fn1bBsuPXBD+fzCyct3svN+MmfyuRsen4YhwsvL64RxL0qlcljNYlYoFOfkm7SyshKTyURoaChxcXE/esG63W46OjrkW6DQ0NA+5f1hUPGMij7VOZckibq6OlQqFWPGjOmnozl95+r8nq2Tle/4c9wfhs8nQhhWnnjiCZqbm3nxxRcpLS2lu7ubmpoaDAYDbreb1tZWuSHZbrfz9ddf09vbi9lsxuFwUFNTQ1NTEzabjZqaGmpra3E4HLz22mt89tlnmM1muYfLZDLJbVx2u52DBw/y5ZdfDvIZGN1EjUUYEC6Xi7a2NkJCQnA4HLz88sukp6dTW1vLhRdeyNNPP821117L+vXrue+++5AkiT179tDV1UV3dzexsbEkJyej0+koKiqitraWxMREqqqq6OnpISEhga+//pqsrCxqa2u544470Gq1aLVaFixYwMcffzzYp2BUEzUWYUAolUrS09PRaDR0dXWxd+9euRcNQKPRMH36dGJiYrBarcCxkaBWq5VDhw5x3nnnkZ2dzY4dO6isrMTtdtPW1kZUVBQZGRk4HA6OHj2Ky+UiOjoal8sFII8lcTqdOByOQTv+0U7UWIQBsXTpUtLT00lPT6egoIB77rmHXbt2ER0dTUREBFqtlpdeeomYmBhiY2MZN24cXl5e2Gw25s+fz5tvvklCQgITJ06ksLCQsWPHkpGRgZ+fH1u3biUtLY2LLrqI/Px8Fi5cKI8bcTgcbNu2DafTyYEDB5g5c+Ygn4nRSfQKCeeEZ8CbUqnE6XTyzTffsGTJkpP2RHhSVXiGkrvd7pMmIj/d3gyhf4heIWHIUSgUcpe8RqPhoosu+tF9f9j7dqrlZgeiN0PoH6KNRRCEficCiyAI/U4EFkEQ+p0ILCfR0dFBS0vLGT3HbrdTUFBAfn4+XV1dfSYlOp1OCgoK2LdvH/X19fLkr8OHD9PV1QWAXq+X/2+xWCgsLMTpdPbXIfXhaUg9l5qbm3E6nVRUVIhu4FFABJaT+P777/sMsJIkicbGRnn0pyRJ9PT0UF1djcFgoKOjg9raWnbu3InZbOb222/vk8bSZDJxxx138Mknn7BixQry8/Nxu93cdtttvPrqq7jdbt577z327NmDJEl888033HvvvRQVFfUpl2dhtOrqahobG3G73XR3d1NdXU1HRweSJNHb2yuX0WAw4HQ6aW1tpauri5qaGnmlgQceeIDq6mrsdjt6vZ6qqio5sHneq7W1Fb1eT1NTkxwMOzo6qK6uxmw2y8ugGAwGDAZDn+e2tLRQXV2NzWZDkiQ++OADzGYzX375JRaLBbvdTnV1NU1NTbS0tMg5hquqquRjEYYvEVhO4vgMcuvXr+ef//wnmzZt4tFHH6W7u5sHHniAHTt28Oqrr/LSSy+RlpbG3XffTU5ODt3d3X0y/QPExMRw//33ExERIeduCQ0NxWAwyFPYAbkr9k9/+hOrVq3qU7NobW3lL3/5CwcOHCAvL4/W1lYeeeQRDhw4wJ/+9Cfy8/PZuXMner0egD/+8Y80Nzdzyy238NFHH/H++++zbt06urq6OHLkCI2NjWzfvp1XX32V2tpaiouL5fdyOp3ceeed5OXl8a9//YutW7dSUlLCs88+S1lZGf/4xz9oa2vjySef5PXXX6etrU1+bnFxMS+99BJFRUU89NBD9Pb2ysdXUFCAxWLh+eefZ/v27bzxxhvceOONmEwmHn/8ccrKynjsscf6lEUYfkRgOQ0bNmzg6quv5pZbbqGxsZGtW7ei1Wq5/vrrWbx4sVy1b2lp4c9//jM33njjCf38JpOJbdu2UVdXx7hx4wBQqVTceOONrF69Wq7hHD58GKPRiLe3Nw0NDX0SSFVVVaFUKrnyyitZunQpbW1t2Gw2rrzySq6//no+/vhjXC6XHIw8tYXg4GCuueYabr/9drZt20ZISAiRkZHMmTMHHx8fenp62LNnD6GhoX0CqlarZeHChdx6661s3bqVXbt20dHRgcFgkNMwOJ1Oli5dSlZWlvy8AwcOMHv2bC677DLGjBlDXl6e/JjNZsNqtbJ7925+/etfc/fdd6PT6WhtbUWSJC688EJ+85vf8NFHH/XzX1E4l0RgOQmFQkFLSwtHjhzh6NGjZGZmsmPHDoqKivDx8SEzMxOlUklJSYn8oTEYDDz44INccMEFTJgw4YQ2jNDQUC6//HJ+97vf8d577+FwOFAoFCQmJjJlyhQ2bdqEy+Vi1apV3HDDDYSHh/PLX/6Szz//XP6wx8TE0N3dTUlJCdXV1YSEhGCxWDh06BBFRUVywNq8eTNFRUXy7YlKpUKtVqNUKuUp8y6Xi7KyMuLj47nlllsYN24cH3/8cZ/AYjabKS4u5vvvvycgIID4+HiCgoKYMmUKv/3tb5k8efJJz58n49vhw4epqKggLS1NHsDmGXuSkpLCnj17OHjwIBaLBbVaTWNjIyUlJWzZsoXp06f37x9VOKfEyNuTaG1tpby8HC8vL/lDcODAARobG7nggguIjIykpKSE1tZWSkpKkCSJX/3qV5SVlaFWq1Gr1eTk5KDVagHkeS2ZmZm43W5KSkpIS0ujqqqKsWPHApCfn09iYiJtbW1kZGTIQcDzPIVCgSRJVFdXs2XLFnJycpg2bRqVlZVs376drKwspk2bhs1mY/Pmzfj5+REWFsbYsWOpra0lLS0Nt9tNVVUVqampHD58mN7eXoKDgyksLEShUDBt2jQ5xYHD4eDmm29m2bJlKJVKFi9ejE6no7i4mO3btzN58mSmTZtGTU0NUVFR+Pn5yefPYrFw4MABysrKmD9/PqmpqTQ2NhIZGcnRo0dJTU2lra2NI0eO0NTUxN69e3nuuecoKSlh//79ZGZmMm3atGGT42a0EMm0B5gkSXz88cccPXoUf39/rrnmmj45REYCp9PJyy+/zO233y4HyP60f/9+Nm/eTG9vL9dff/2PZooXhgYRWM6BH562kThPZaCPb6Sfv5FIzBX6mU7nov/hdkmS5FuVn/qQ/NQ+nvc+mw/b8c/9OR/egf6wi2AysonG25PIy8vjf//3f3nmmWdobGwEjq1548kbAsd6NxwOBw6Hg7///e8YjUYqKyvlsSY9PT1yflHP/5uamuTGXs+4jR8Oguvt7WXlypX813/9F3a7vU+Z7Ha7/Dpw7FbFM5bEM35Fr9dTUlIiv/6RI0d45513+Pvf/05ra+uAnjNB+CFRYzmJrq4uLrjgAmpqajh48CD19fXs3r0bp9PJddddx65duzAYDJhMJn7729+yd+9ebr75ZlpbW7FYLHz++ecEBgZy44038tVXX9Hc3ExKSgoajYaNGzdisVhoampCr9ej0+m48cYb8fb2xtvbm6uvvpobbrihzziYnp4ePv/8c44ePcqyZctwOBzs3bsXg8HAxRdfTG9vLwUFBTidTsaPH09OTg4KhYLU1FTsdjuVlZU0NzcTHh4+iGdVGE1EjeUkHA4HX375JZs3byYiIoJ169bh6+uLn58fhw8fprm5md/85jekpaWxYcMG4FhPyJEjR/j222/59a9/zZ133omXlxddXV1otVreeOMN4uLimD9/PhMnTuSzzz4jKiqK9vZ2ec0cpVJ50p6Q7u5uecmGzZs3U1FRQWxsLFdeeSWrV6/m448/5sorr2T27Nl9bjF+mKpgIBpgBeFURGA5CV9fX6644gruvvtu1q9fT3x8PAqFgsjISNLT0zEajWzcuJGqqiqmTp1KQEAAXl5eeHt7ExUVxebNm9myZQs1NTUUFxcTHByMv78/vr6+FBcXc+TIERITE/Hy8iInJ0de1tLlcnHgwAEsFgtFRUVyraW0tBS9Xk9ISAg6nQ6dTkdAQABqtRpvb2/Gjx/P999/T15enpy7RJIkCgoKqK+vP2U+E0EYKKJX6CTsdjteXl4olUpMJhPe3t4YjUZ51bg1a9ZwySWXoNVq5XL4+/tjt9tRq9V0dHQAEBISgtFolNfFCQoKorOzUx6wZjab5QWjPCsVdnZ24nA45NdWKBS4XC7a29tRqVTodDo5s5qXlxcWi0XOK6vVavHx8UGj0SBJEl1dXdjtdlQqFSEhIaLBVPhZRHfzAJIkCaPRKH/oBWG0EN3NA0ihUBAUFDTYxRCEIU20sRynurr6jHOxCMJI5HQ60ev1Z5XCQgSW46xfv549e/YMdjEEYVBIkoTFYqGyspIjR47g7e19Vq8zam6FTjYKVQwrF4RjPIMsq6urMRqNZGRkkJycfNafi1ETWMxmM99//z0xMTFMnjwZhUJBaWkpNTU1hIaGMm3aNBFchFHH7XZjNptpbGxEo9GQmpqKVqv92Z+Fn3UrtGLFChQKBcuXL5e3SZLEI488QkxMDDqdjgULFsjDzD1sNhv33HMPYWFh+Pr6snTpUhoaGn5OUX6SZxGszs5OOVdKQ0MDixYtYu/evfT09LB161Z2796NUinuEIWhT5IkysrK5ERjnjzK1dXVOJ1Ompub2b59O6WlpfJs/F27dpGXl0dXVxcFBQVs3bqV+vp6kpOTSU1Nxdvbu1++YM/6E7R//35ef/11ObmQx1NPPcUzzzzDSy+9xP79+4mKimLJkiV9csAuX76cdevWsXbtWnbs2EF3dzeXXHLJgCZ4VqlUTJ06lZaWFrq6uuT8rrW1tQQHB6NWq8nOzmbs2LEi36owJPT29srJ11taWli9erU8aFKSJHbu3MmOHTuw2+1IkkRbWxsPPfQQN9xwA9deey3PPvss5513HgsWLOC2227js88+45JLLuH111/nl7/8JfPmzaOqqors7Oyzbks5lbMKLN3d3Vx33XW88cYbBAcHy9slSeK5557j4Ycf5oorriAnJ4d//etf9Pb2snr1agCMRiNvvfUW//jHP1i8eDETJ07kvffe49ChQ3zzzTf9c1QnYbfbMZlMzJw5E39/fywWCzNnzsTpdHLBBReg1WoJCwsjJCREBJYhxuVy0dbWhsVikbfZ7XZaW1tpbW2lt7dX3sdoNA5iSftXbW0t7777Lna7HW9vb8aOHYtCoaC6upqysjKOHj3KwoULWb9+PX/961/56KOP5JpIbW0te/bsweVyERISQkBAAF988QV2u51vv/2W7777DrfbTVZW1oA0AZxVG8tdd93FxRdfzOLFi3n88cfl7dXV1TQ3N3P++efL27RaLfPnz2fXrl3cdttt5OXl4XA4+uwTExNDTk4Ou3bt4oILLjjh/Ww2GzabTf7dZDKdcZmDg4P7BMHo6GiAPrlahXNDkiSsVisGg4GYmJgfXSZVkiTWrFlDV1cXZrOZP/zhD6hUKsxmMwcOHGDPnj3k5ORgtVqx2+00NDRw5513jogJlykpKdTV1dHc3Ex8fDw5OTkAHDx4kM7OTkpLS0lKSuKTTz6hpqYGl8tFfn4+ERERKJVKxo8fz+HDh3E6ndTW1lJbW0t2djYmk4mYmJhTzk3rD2dcY1m7di35+fmsWLHihMeam5sBiIyM7LM9MjJSfqy5uRmNRtPnQ378PsdbsWIFgYGB8k9cXNyZFlsYQmw2G7/73e/Ytm2bfPt75MgR1qxZI/94xhK53W5qamq49dZbMZlMHD16FDiWQ/iCCy5ApVIxa9Ys6urqyM3NJT4+ni+//HLQjq0/KZVKUlJS0Gq19PT08B//8R/YbDZcLhe7d++mrKyMhx9+mNraWnp6ejh8+DC+vr5YLBYKCgp47733aGtrkzsotFotEydOZNGiRbS3txMXF0dMTMyAlP2MwlV9fT333nsvmzZt+tF7suOrVj83AdJDDz3EfffdJ/9uMplEcBnG3G43vb29/OpXv0Kj0QB9Z3Z75kJ5tLe3y0nAvb296ezsJDg4mLKyMry8vIiOjubaa6/l22+/pbKyktmzZw/KcfUXTw6foKAgbrjhBlauXElsbCxwbDG9Cy64gIMHD6LT6SgtLSUgIACdTseSJUvIz8+Xa28hISHyzyWXXILD4cBoNLJu3TqioqJ4/fXXByywnFGNJS8vD4PBwOTJk1GpVKhUKrZu3coLL7yASqWSayrH1zwMBoP8WFRUFHa7XU4VcLJ9jqfVagkICOjzIwwfnkFXnrYrLy8vpk6d2id4xMfHs3jxYhYvXsyiRYvkGq1SqWTq1Kn861//Ijo6Gm9vbz755BPg2FIp559/Pj09PVitVry8vPD19WXOnDnn/iB/Js+ieDU1NbS2trJu3Try8/P505/+REJCAjNmzGD58uWsWrWKkpISvL292bx5MwAPP/ww5513npzs3eVyERAQQGtrK5MmTSIwMJBPPvmEN998k/DwcC699FKCg4PZtm2bXAPsb2cUWBYtWsShQ4coLCyUf6ZMmcJ1111HYWEhycnJctoAD7vdztatW5k1axYAkydPRq1W99lHr9dTXFws7yOMDJIk4XQ66ezs5L777sPhcOByudBoNPzxj38EkHs5vL295XYwTy8dHKu9XHnllVgsFpxOJ7t37+bmm28GYPr06Rw4cIA1a9YQHByMn58fOp2OF154gZ6ensE56J+hq6sLk8lEdHQ0N910E/7+/jz44IMEBwfjdrtRq9Xcf//9tLa2EhwcTGRkJLfffjtffPEF7e3taLVampqaiI2NxeVyERcXx4wZM9ixYwfTpk0jJCSEgwcPkpCQwO9+9zt+8YtfyDWh/nZGt0L+/v5yA5KHr68voaGh8vbly5fzxBNPkJaWRlpaGk888QQ+Pj5ce+21AAQGBnLLLbdw//33ExoaSkhICL///e/Jzc1l8eLF/XRYwlBw5MgRnn/+eS666CJKS0t5//33MRgMXHrppbz//vtEREQQHx/PL37xCwB27dol13a1Wi1LlixBq9ViMBioqqriueee4/bbb+fCCy9Ep9PxwQcf0N3dTVpaGg6Hg48//phnnnmGv/zlL1RWVp4wFGIoUygUZGdnA//Xg5mWlgZAQkICNTU1fPTRRzgcDrq7u5k7dy6PPfYY+/fvZ/fu3SQlJXHZZZeRnZ1Nfn6+vHBdeHg42dnZVFdX4+Pjg81m4/zzz8dms/HZZ59xww034OPj0+/H0+8jwf74xz+yfPly7rzzTqZMmUJjYyObNm3C399f3ufZZ59l2bJlXH311cyePRsfHx/Wr18vEhKNML29vfj7+zN37lzCw8O59tpryc/Pp6SkhMWLF/OLX/yCTz/9VN7f07aiUCj6DFJUKpXyImuebHqensLJkyfj7e3NRx99xP3338+aNWvIz88f1tdSR0cHX331lfy7y+Vi7969XHfddfIo8enTpzNjxgy8vLy4/vrrmT9/PmFhYfzzn//kggsuICsri+7ubrZs2cKbb76Jt7c306dPZ/Xq1ezevZs1a9Zw/fXXExISMiDHIPKxHOeFF14gISGByy67rF9fdzQqKytjy5Yt3HDDDTz99NM88sgjPProo1xxxRV88cUX+Pj4sHDhQnJzcwHo7OyUhxUolUrCw8NRKpW4XC5effVVAJKSkoiJicFsNpOWlsZnn32GyWRi9uzZhIaGsmvXLmw2G7fccovcMDycud1u2traaGxsxMfHh7i4OLmG4UnmdfDgQZRKJZMmTeKFF17g7rvvpr6+nuDgYD755BMmT55MXl4eN998MzqdTn5tz4jdH+vu/yGR6OlnEIGl//zw0jp+iZQfLlVyOgO0PG0xJ+tx/OF2z+sP53lfnmPq7OykqakJhUJBfHw8fn5+1NbWEhcX16dG5nK55FqeZ+rKgw8+yNKlS0lKSgKOLbv729/+ts/7bNy4ER8fH+bOnXta5RKJnoQh4fjE3if793Sdav7W8a8znAMKHKtFNDY2YjQaCQkJITMzE6vVisViwdfXl8LCQqKjo/sElh/+35OydMWKFTidTqxWK2q1+qRd8EuWLBmw4xCz7QRhCHA6ndTX11NcXIzL5SIrK0uumdTX17Nnzx4UCgXLli37yRUXLBYLDQ0NdHV18d577+Hj40N6evoJ+ymVygGbcCtqLIIwSDw5UJqamrDZbMTExMjdvz+seY0dO1aeJ3Q6zGYzNTU1zJ8/n7vuumtAyv5TRGARhEHQ09NDfX09JpOJxMREwsLCTvt276eEh4czf/78Qb0tFIFFEM4Rt9uN0WhEr9ejVqtJTEzsl6RKxxsK7UwisAjCAHO73bS3t1NbW4tKpSI9PX1ABqUNJSKwCMIA8HT7GgwG2traCAkJYdKkScO+K/x0jZrA0tvbS3l5OREREURHR6NQKOjs7KSqqoqwsDASEhIGu4jCCGG326mvr6exsZHExESysrKG9UjgszFqAovFYiEuLo6ioiIiIyPx8vKiqakJg8Egt8Q7nc4BTY8pjFySJGG322lsbMRisTBmzJifleV+uBs141iCg4Mxm83ExsbKIxRTUlLIyMigoKAASZKoqqqitrZ21F4MwpmTJAmTycSBAwcoKioiNDSUrKwsgoKCRvV1NGpqLE1NTVitVsaOHYvVaqW2thZ/f38cDoe8ZGp6ejrJycki563wkzwBxbNsRm5u7oD08AxXoyaw/DDvhE6nIyMjQ66+pqWliQtCOC0ulwu9Xk9VVRUxMTGkpqaOiMmO/W3UBJaTUSgUPzk8WhAkSZKX4GhrayM0NJRZs2bh5eUlvpBOYVQHFkH4KRaLhfLycux2OwkJCWRnZ4sF7U6DCCyCcBzPLXJ9fT12u524uDiCgoJEQDkDIyKwdHZ2YrFYcLvdNDc3yw1pgnAmJEmivb2dqqoq/P39iY+Px9fXd7CLNSwN+8BiNpt56qmnCAoKory8nPDwcMrLy7nmmmvE/a/wkzw9gEajkYaGBrRaLVlZWfj6+orr52cY9oGlu7ub5ORkFi9ezOHDh5k5cyYrVqzg17/+tbgwhB/lcDioq6vDaDQSFhZGRkbGgK0MONoM+7MYHh7OeeedR2JiIklJSVitVu68804RVIRT8nQZd3R0EBAQQE5Ojugy7mfDPrCoVCpSU1Pl3729veU8n4LgIUkSPT091NXV4XK5iImJkZN4iy+h/jfsA4tHXV0d+fn5LFy4kI6ODhISEsQFIwDHJqA2NDTQ29tLdHS0nP1fGDgj4uxaLBbeeustGhsb6erq4s033xzsIgmDzLNsRnFxMQ0NDcTFxTFhwgQiIyNFUDkHRkSNxcvLC5VKhV6vZ9u2bYwdO3awiyQMEk+XcWNjIwCpqan4+PiI2us5NiICC8D48eMZM2aMvCC2uJBGD88a0S0tLXR0dBASEkJOTg5KpVJcB4NkRNQJCwoKePPNN2lvb6epqYny8vLBLpJwjni6jA8cOIDdbicrK4vY2Fgxj2eQjYgaS1JSEv/v//0/kpOTcbvdREZGDnaRhAEkSRJWq5WmpiYsFguxsbHEx8eLQDKEjIjAEhgYyKFDh9izZ4+cevLKK68UF9oII0kS3d3dVFdX09PTQ0ZGxqhPqDRUjYjA4knW5FmfZdu2bVx55ZUn7Od0Ovus/iZJEg6HA7VaLS7OIcztdmM2m+WkSmPHjkWj0Yi/2RA2IgKLj48PV199NQC7du3id7/73Qn7dHZ2sn//fiRJYt68eeh0Oo4cOUJHRweSJDF79mxxoQ4xLpeL1tZWKioqCAkJISUlRUwuHSZGRONtb28va9euJTg4mLlz5/Lpp5+esE9hYSGRkZFERERgMBiAY4PqpkyZQmFhIT09PZSVlVFZWSkCzCDyLJuh1+s5fPgwbrebmTNnkpmZKYLKMDIiaixqtZq2tjbefPNNlEolERERJ+wTERGByWSiu7ub0NBQiouL0el0NDU1ERMTg1arJTU1lcTERJHzdpBYrVaqqqro6OggNTVVJFUaxoZ9YHG73bjdbu644w62bdtGZGSkvDDUD2VmZlJVVUViYiLBwcHY7XZmzZpFeXk5CxcuRK1WA4y69V8GmyepUkNDg7wwekZGhggow9ywDyw2m42ioiK6urpQKpW0trZiMBhOmIioVCr7TFb0/D8rK+ucllc4RpIkOjs7OXLkCH5+fiQnJ4scKCPIsA8sAAaDAYfDMdjFEE6DJEkYjUYaGxvx9vZmwoQJ6HQ6EVBGmGEfWLRaLQsXLqS5uZnCwkIUCgUBAQGDXSzhOE6nk7q6OgwGA1FRUaSnp8u3n8LIM+xvZJVKJSqVivfee4/s7GwiIiJ45ZVXRAPsECBJEi6Xi8bGRkpLS9FoNEyaNInExEQRVEa4YV9jgWMNuF1dXWzatAmz2UxPTw/ffvstCxYsEI2xg0CSJHp7e6moqABgzJgx5OTkiNudUWREBBaNRsPll19OW1sbQUFBSJJEYGCguJAHgcVioaGhAbvdTkxMDKGhoaKHZxQaEX9xl8vFoUOHqKqqYuzYsRQVFTF58mRxQZ8jbrcbg8FAYWEhdXV1xMbGkp2dLTK1jWIjosbidDppbW2VlwARPUTnhqfLuLGxES8vL1JSUvDz8xM1RWFkBBYfHx9uvPFGVq1axf79+/nVr34lLu4B5HA4aGxsxGQyERwcTFZWlkiqJPQx7AOL0+lk5cqV8lKYHR0duN3uwS7WiORwONDr9bS1tREQEEBmZqbo3RFOatgHltbWVnbv3s1zzz3H66+/zl133SUmq/UjSZL6NMiOGTOGuLg4UTsRftSwDyw+Pj5kZ2ezfv16vL29+eSTT5g+fXqf4fvCmfOsw1NfX4/JZCI5OZmwsDARUITTMuwDS2BgIPfff/9gF2PEcLvdGI1G9Ho9arWa5ORkkVRJOGPDPrAI/cOzDk9NTQ3e3t6kp6fj7e092MUShikRWEYxT1KllpYW2tvbCQ0NZcqUKSgUClFDEX4WEVhGKZvNRn19PU1NTSQnJ5OVlSWmPwj9ZtQElo6ODvR6PVqtlpSUFBQKBSaTibq6OoKDgxkzZsxgF3HASZKEzWbrs2yG51wIQn8aNeOtvb29CQkJoaSkRJ75XFdXR319vbxPa2sr7e3tI+6D5smBsn//foqLiwkLCyMrK0vMpxIGzIivsXR1dXHkyBEyMzM5cuQIsbGxKBQKJEkiOTkZb29vSktLiYmJobS0lIqKCqZMmTLYxe4XkiRhMploaGjA29ub8ePHix4e4ZwY8YHF39+fcePGodfrqaioIDs7G7PZTFNTE5Ikodfr5eTb8+bNo7CwcNiP3HU6nTQ1NVFbW0tMTIxIqiSccyM+sHh5eeHj40NKSgopKSnydk+WuczMzMEqWr/y9PAYDAba2toIDQ1l5syZqFQj/k8sDEHiqhvmPEPuPbO6ExMTRVIlYdCJwDJMeXp4GhoacDgcxMfHExQUJPKfCEOCCCzDjNvtpr29nerqavz9/YmPj8fX13ewiyUIfYjAMkxIkkRXVxeNjY1otVqysrLEOjzCkCUCyxDncDioq6vDaDQSFhZGZmamGCErDHkisAxBnmUz9Ho9XV1d+Pv7k5ubK7qMhWFDBJYhxJMDpa6uDpfLxZgxY4iNjQUQtzzCsCICyxDgWYenoaEBi8VCTEwM4eHhIpgIw5YILIPI7XbT0dFBc3MzGo2GhIQEtFqtCCjCsHfGgx4aGxu5/vrrCQ0NxcfHhwkTJpCXlyc/LkkSjzzyCDExMeh0OhYsWEBJSUmf17DZbNxzzz2EhYXh6+vL0qVLaWho+PlHM0y43W5aW1s5ePAgjY2NJCcny4mVRFARRoIzCiydnZ3Mnj0btVrN119/zeHDh/nHP/5BUFCQvM9TTz3FM888w0svvcT+/fuJiopiyZIlmM1meZ/ly5ezbt061q5dy44dO+ju7uaSSy7B5XL124ENNZIk4XQ6aWhooLi4GLvdzrhx4xg3bhw+Pj6DXTxB6F/SGXjggQekOXPmnPJxt9stRUVFSU8++aS8zWq1SoGBgdJrr70mSZIkdXV1SWq1Wlq7dq28T2Njo6RUKqUNGzacVjmMRqMESEaj8UyKf1qef/556dNPP+3X17TZbFJVVZW0c+dOqbq6WnI6nf36+oJwLpzJ5+6Maiyff/45U6ZM4aqrriIiIoKJEyfyxhtvyI9XV1fT3NzM+eefL2/TarXMnz+fXbt2AZCXl4fD4eizT0xMDDk5OfI+x7PZbJhMpj4/Q5307zk8lZWVHD16lJCQEGbOnEliYqIYhyKMeGcUWKqqqnj11VdJS0tj48aN3H777fzHf/wH77zzDgDNzc0AREZG9nleZGSk/JinoTI4OPiU+xxvxYoVBAYGyj9xcXFnUuxzSvp3DpSioiIKCwsJDQ0VSZWEUeeMeoXcbjdTpkzhiSeeAGDixImUlJTw6quv8pvf/Ebe7/gPkCRJP/mh+rF9HnroIe677z75d5PJNOSCi/TvLG1NTU1otVoyMzNFUiVh1DqjGkt0dDRZWVl9tmVmZlJXVwdAVFQUwAk1D4PBINdioqKisNvtdHZ2nnKf42m1WgICAvr8nCnPrYnVapVTU0qShNVq/VmNxi6Xi6amJnbs2EFLSwupqamkpKSIbmNhVDujwDJ79mzKysr6bCsvLychIQGApKQkoqKi2Lx5s/y43W5n69atzJo1C4DJkyejVqv77KPX6ykuLpb3GQhdXV18/fXXlJeXy9uqqqrYt28fBw4cOKOscdIPhtwfPnwYhULBrFmzSE9PR6PRDETxBWFYOaPA8p//+Z/s2bOHJ554goqKClavXi2vlwzHboGWL1/OE088wbp16yguLuamm27Cx8eHa6+9Fji2cuEtt9zC/fffz5YtWygoKOD6668nNzeXxYsX9/sB2mw2WlpaUCgU+Pv7YzKZcDqdAFRUVDBjxgz27dtHb28vFRUVVFdXn7Km4anhlJSUsGfPHpRKJdnZ2URHR+Pl5SVqKILwb2fUxjJ16lTWrVvHQw89xKOPPkpSUhLPPfcc1113nbzPH//4RywWC3feeSednZ1Mnz6dTZs24e/vL+/z7LPPolKpuPrqq7FYLCxatIi33357QHpLHA4Hra2thISEMG3aND777DNSUlJwOBx4eXlRWVlJWFgYWq2W+Ph4YmNj5VslD+nfSZUaGxux2+3ExcWRlZUlkioJwikopOM/RcOAyWQiMDAQo9F42u0tdrud5uZmAgIC8Pf3x2q1otFoaG5uJjg4GD8/PwBeeOEFEhISuOyyy5AkiY6ODsrLy/Hz8yMpKUneTxBGmzP53I2auUIajYb4+Hj5d0/WtVP1LnmSKnmWzdDpdOJWRxBO06gJLKdrzJgxfP755zQ2NhIbG4uXlxdHjhzps4/T6Rx2bSput3tY3bq5XC4UCsWwKrPT6RxWqyJ4blZ+7DpWq9XMmzcPb2/vM3rt4XMWzpHLLruMBQsWnNDO8kNvv/02c+bMITU19RyW7OxJksTWrVtZsGDBYBfltG3ZsgWVSsX8+fMHuyinpbe3l7fffps77rhj2HzhVFRU4HA4fnQJHIVCcVY9naOmjeXnkiRJDjYmkwkfHx/UajUOhwOXyzUkZyZLkkR3dzfe3t7Y7XZ8fX37HMdQqw1I/0505ePjg8ViAY4tjWuxWIZkfl9Po75CoUCtVmM0GgkKCupTExiKZfaUz+l0IkkSKpXqtM6xaGMZIGvWrCE0NJRZs2ah0WiwWq3s2rWL8PBwoqKiCA8PH+wi9tHS0kJXVxdVVVXy3KzDhw9TXFxMTk4O2dnZg1zC/yNJEtu3bwfAarWyePFi3G43Bw4cQK1Wo9Pphtzici6XC4PBwMGDB5kyZQpRUVEYDAa+//570tPTGT9+/GAX8QRHjhzh4MGDZGVlkZubiyRJ7Nq1i8jISNxuN+np6f3yPkPrK2uIqq+vp6WlhcTERMaMGSMvJG+z2fDx8SE0NJS2trZBLuX/6enp4eDBg/j6+hIREYHdbpe/iYKDgxk3bhzt7e2DXMoT1dfXM2nSJMxmM06nE6fTid1uJz4+nurq6h+9PR0MKpWKwMBALBaLnI9Yo9GQkZFBfX09DodjkEt4oqCgIMaNG0dHR4e8zWq1Eh8fT0tLS7+9jwgspyEoKAh/f38SExOxWCyYTCZqa2sBqK2tpaSkZEiNuNVoNERHR9PT08O2bdvIysrCYrGwbds2uZzd3d2DXMoTxcTEUFlZiVarpaqqipqaGnp7ezly5Ajx8fFD7rbCbrfzzjvvEBcXh0ajoaioSL6lcLlcQzK/kGf+Wnd3N21tbXR1ddHV1cWBAwf6NXCLNpbTJEkSzc3NOBwOoqOjsdvt6HQ62tvbcTgcREVFDbk2i+7ublpbW9FoNERERNDZ2YlKpcJkMhEeHj7kFjrzTJOIiIjAZrMBx9opOjs7iYmJGXLpJtxuNw0NDbjdbrlm6JmM6smOONSCYUdHh/z3V6lUKBQKbDYbHR0dREdH/+gX5Jl87kRgEQadJEnY7Xb5/2fatSmcG6LxVhh0ngRgFouFq6++Wv5G9/LyQqVS4XQ6cblcqFQqXC4Xn332GdHR0VgsFhYvXozdbkepVKJSqeSgo9Fo+Prrr4mOjmb8+PFyuk+1Wi3XZmw2G59++in+/v5cdNFFg3b8o50ILMKA2LVrF9dddx1bt27l0KFD1NfXY7fbaW1t5Y477uDRRx9l6tSpdHZ2csMNN2C1WjGbzZjNZr744gvq6upITU1lwoQJfPvttzQ3NzNlyhS++uorAgIC8PLy4vPPPycmJgaHw8Ftt90mj7m48MIL+eSTTwb7FIxqQ6tRQBgxJEmitraWI0eOEB4ezrvvvou3tzd+fn7o9XocDge//vWv0Wg0tLa2Aseq2m1tbXz66afcfPPNXHDBBdhsNnp7e1EqlVRXV5OQkMD8+fMxmUwUFBSgUqnkfWBojh0ZjUSNRRgQPj4+LF68mKlTp/Ltt99y8cUXExMTQ1BQkJyG9PPPP6epqYmwsDBUKhUqlQqdTseSJUv49ttviYmJwWaz0dPTIw/mSktLo7m5mQkTJjBlyhQSExPp7u6WVzpwOp3k5eVRVVVFWVkZY8eOHeQzMTqJxlthQLS2thIUFIRKpcJsNuPr60tTUxNwbCzNyy+/zDXXXENISAi+vr50d3ejUChwu934+/uj1+tRKBRERUXR1NSEr68varUaHx8fmpub8fPzQ61Wo9fr8fX1lVeOdLlcNDY24nQ68fPzIyIiYpDPxMgheoWEIc3tdtPS0kJUVJS4bRlGRK+QMKQplUqio6MHuxjCABKNt4Ig9DsRWARB6HcisPQjz7Imw7DZasBZLBbcbjcmk+mMVkQQhicRWE7iwIEDbNmypc82z0zb49ck8vyel5fHM888wx/+8AfefPPNPjNbe3p6eOSRR3jnnXd47LHHqK2txe128/jjj/PVV18hSRJff/01hw4dAqCmpoYnnniCxsbGE8rmyQHi6X71/P5TH1aXyyWXv62tjZUrV8qv53Q6+xzLD9/LM8vY85jb7cZms/XZ1+12nzCT1+Vy9dnvvffew2Qy8fbbb9Pd3S2X+4cT9TznVASe4U803p5EbW0tzc3NLFq0CDiWaeu1116jt7eXyy67jCVLlvDBBx/wxRdfkJOTw5QpU5gxYwbjxo1DoVBwzTXXcOWVV8rLyNpsNg4fPszDDz/Mq6++yo4dO/j1r3/N9u3bKSsrY8KECRw6dAiFQkF2djYffPABwcHBrF+/Xh5RCsdWHFi7di3bt29n/Pjx/O53v+Pdd99lx44dZGdnc8stt9De3k54eDiBgYF89NFHnH/++Xz99dcYjUb27NnDrbfeisvl4p133sHX15f09HTWrVvH0aNHuf/++5k8eTJwLDCsWrWK+vp6LBYLv/3tb/H392fNmjUUFBQwZ84cbrzxRjZt2kRlZSUKhYJ77rkHgKamJt5++20qKipYunQpl156Kb29vUiSJE/W3Lx5M++++y6pqalERUVx44038v7777N582amTZvGTTfdJHr8hjFRYzkNb731FpdddhlPPfUU77//Ptu2bWPnzp28/vrrpKens3XrVvz8/FAqlWzcuJE5c+b0We4EoLS0lGuvvZaPP/5Y/vD6+/tz7bXX8uabb8rf3A0NDRQVFXHeeedRUFCA2WyWX+Pw4cMcOHCA559/nltvvZWGhgb27t3Lyy+/THZ2Nm+++SaVlZV0dXUB8Mknn2Aymfjss89YtmwZf/7zn/nwww+JiooiMTGRyy67jIqKChQKBbfddlufVJtut5uvvvqKO+64g2uuuYbVq1fz9ddfU1VVxcUXX8zu3bvJy8tj8+bNzJgxg9tvv11+7saNG8nIyODll18mLy+P0tJS+bF169bR2dnJ22+/zf/8z//wm9/8hvXr19PY2Eh+fj5vvfUWqamprF69ut//jsK5IwLLKdhsNsxmM93d3VgsFjlLv8vlwsvLS56R65ne39vby2OPPUZhYSG/+c1vTkihkJaWxmuvvcY111zDjh075O3Tp09Hq9Wyc+dOJEnivffeo7Ozk8cff5yKigo+//xz+XbC4XDII1Q9v3t5eckpMhUKBU6nk/r6ekwmE0ajEQCdTkdwcDChoaH09PTIE/YcDgcXXnghl19+OaWlpbz44ot9bkMkSUKpVKLVatFoNEiSRGZmJgsXLuSxxx5j4sSJKBQKvL295URHcKw72ZPyUKvVnnCL5UmS7Tl/brdbrpV5eXkNqdw2wtkRt0InERwczGuvvcbOnTvx9/fnzjvvZMOGDXzzzTdccsklzJ49m/b2dv7jP/6D0NBQ/P392b17Nxs3biQiIoKKigqefvppwsLCgGMflvT0dIKDg7nhhht48sknMZvNJCcno1arueGGG9iyZQteXl709PTw8ccf4+3tTVdXF//zP/8jzwLOyckhKyuLV199FY1Gw29+8xtmz57Nc889R29vL7fddhtdXV289tprlJSUEBYWhkajITExEYVCgZeXF/Hx8URGRpKVlcXvfvc7rrrqKnn1x1mzZvUZsKZQKORAd+ONN+Lj48Pq1av56KOP6Onp4a677iI2NhadTtfn/F100UW88sor1NfXk5ycTFZWFsXFxXh5eZGUlISPjw+33norf/7zn4mLi8PHx4cxY8aQk5PDSy+9JC94JwxfYuTtSbjdbvnWxPOBdLvdSJIk11Y2b96MyWSipKSEq666iszMzBOe4/mQSpIkL7/hqfUolco+S3J4tnlqCQqF4oTneV7L8z6emofnuZ7ne5bOAE54H8//f/janlrKD8vscDj405/+xJ/+9Ce0Wq38mOfceN7v+PJ5uFwu3G63nEzIUyvx7P/VV1/R3d1NWVkZ06ZN44ILLpDL5Dl/wtAihvQPME/vhedWxMfHZ8QNTff02mi12gE5NpvNJme49/X1HXLZ94QTiSH9A0yhUKDT6U64BRhJPG0nA0Wr1aLVagfs9YXBJb4mTqK8vJxVq1bx5Zdfynk+TsXlcvHRRx/R3d39k5nk7XY7PT09p3zc6XTy7bff8sknn8gzgc9EV1eXnORbkiSOHDnCmjVr2Lx5s5yFTRDOBVFjOYm6ujqmTp1KYWEhO3bsICcnh4KCAgDmzZtHdXW1nKR68uTJrFq1ioULF2Kz2Whra2P//v2oVCrmzJlDQUEBXV1dJCcn09raytatW1m6dCkAjY2N+Pj4MHfuXLntJjMzE5vNxqpVq/jDH/4AHBu1umPHDqxWK1OmTMFms1FaWorL5WLKlCk4HA6KiopwOp0EBgaSkJAAQHh4ONOnT+fpp58mIiJiSK5zI4xMosZyEg6Hg6+++orvvvuOmJgYXn/9dZqamigtLWXHjh1s2rSJhIQESktL+e677wAwm83s3r2bNWvW4Ofnx6RJk1CpVPT29tLZ2ckTTzxBYGAgycnJREdH88wzz2CxWPjyyy/l9VzUajVhYWFs3Lixz+JcLpcLi8VCa2srH330Ebt27aKhoYGgoCBWrlzJypUriYuLQ5IkeQSsZ3W+p556irq6OiIjI8/9iRRGLRFYTkKj0fDLX/6SW265hR07duDl5YWvry8zZ84kJycHu90u3yKp1Wq54VGhUKDVarHb7TgcDkpLS1m7di1+fn6YzWa0Wi1GoxGz2SyPLbn44ovlEbpOp5NXXnmFjo4OJkyYIJfnwIEDVFRUEBAQgN1ul7uQQ0JC0Ov1dHd343Q65TE1gLwMxRNPPMGCBQtobm4+dydQGPVEr9BJdHV1odPp0Gg01NbW4ufnx4YNG1CpVCxatIhVq1aRmJiIWq1m0aJF1NTUkJSUhNFoRKfTsWnTJnp6eli2bBkbNmxAoVCQmZlJcnKyPL4Ejo1QTUhIYNGiRWg0GpxOJ0VFRbjdbsLDw0lMTASO3QqtWrWKhIQEcnNz5UFyGo2G5uZmAgIC2LRpE0lJSWRkZBAcHIwkSZSUlLB7926ys7OZNm2aPLBOEM6G6G4eQJIkUV1dTWJiougiFUYV0d08gBQKBcnJyYNdDEEY0sRX7nFMJtNPdjELwmggSRIWiwWn03nGzxWB5TirVq2Se3oEYTSSJIn29nYKCwupqak5q8Xtxa3QcRwOx1lFaEEY7txuN11dXfLSK6mpqSek/zhdoyaw9Pb2sm/fPoKCghg/fjwKhQKr1UpRURHx8fFiKQphVPJM/GxsbKSzs5OgoCDGjh3bZ0Lq2Rg1gcVqtdLR0SGPSgXYvn07EydOZOPGjVx55ZXyQDRBGOqcTucJM+iNRiP+/v50dnYSGBhIdXU18fHxaLVaTCaTPI7KQ6FQyKPF/f39yc7O7rchCaMmsAQEBLBo0SI2btxIQEAAAQEBOJ1OvL29MZvNuFwuWltbaW9vFzUXYUhzOp289tprXHzxxXKO4J6eHv71r3/h4+NDXl4e/v7+eHt7M3/+fLKysvjHP/7Bddddx2uvvYbBYMDtdhMSEsKnn37KhAkT+v2aHzWBpauri9bWVmJjY+WBb7m5uRw6dIhZs2ah0+lISEggNjZWZNkfgjx5Wn74AfAkE/9hrpnRsCi8Jy/QP//5T3Q6HWazmfr6elpbW+ns7ESr1fLFF18wd+5c1q1bx0MPPYTZbGbnzp3ExsbS0dHB7NmzKSgooLOzc0CWoR01gSU0NJSQkBD5wvP390eSJGJiYkbFxThcSZJEXl4eH374IVOnTuWKK65AqVRiNptZtWoVtbW1/Pa3v8Xb25t3330Xf39/br75Zvz8/Aa76Kelp6eH5uZmkpOTf/IadLvdbNiwgSlTpvDuu+/S3t7OwoULWbZsGWvXruXqq69m7969rFu3joyMDFwuF+Xl5fIqC+PGjWPXrl1oNBq2bdtGUFDQgKXGGDXdzZ48q8enXjxZ9jNhYNlsNv74xz+yevVqeX5TXV0d3377Ld9++y3fffcdnZ2dwLEP00cffcR//ud/UlBQIC+JUltby6FDh1iwYAH79u1j9erVLFmyhIiICDZt2jRox3amTCYTRUVFP7qP2+3GbrfT3NxMS0sLa9as4Z133kGr1bJu3Tq++eYbjh49yr/+9S9yc3MJDAzkyJEjVFdXo1arWbp0KQqFgl27dgHHpoh0dnai1+uprKwckOMaNYFFGDo836RXXHGFnOzJbrdjNpsxm82YTKY+YyfGjBlDWFgYNptNXoHAM/mzqqqKqqoqfH19iYuLQ6VSDasJl1FRUVx++eU/+uV26NAh/v73v3PLLbfQ0tLCzp07efnll1mxYgXXXnstBw8epLy8HF9fXx599FE0Gg2pqamYzWa559PTbXzttdcyd+5cdDodDz/8cJ/Jrv1JBBZhwHnaQjxtVwqFQl5fyCMwMJCkpCSSkpJITk6Wq+gKhYKOjg7effdd/Pz8iI6OJi8vD7fbLVfltVotYWFhfPDBB+zfv58lS5YMynGejdO5Dc/NzeX3v/89F110Ea+//jo5OTlkZmYybdo0ZsyYQVBQEBMnTkSv12MymVAoFPT09KBWq8nNzQXgn//8JykpKWzZsoXDhw8THx/PTTfdREhIyIAc16hpYxEGT09PD88//zwPPfSQnPLyueee67NkSHh4OOHh4Sc8V6lU8uCDD7J161ZCQ0NpbGwkJycHjUbDVVddRUVFBb6+vkycOJHe3l4qKytpbGwkJSVlxEwSVSqV6HQ67rjjDux2OxdddBF/+9vfGD9+PPv27cPb25uwsDDKysrQaDQEBwdjt9sZO3Ysl1xyCUePHmXOnDlERUXR3d0t5wj6/vvv+fWvfz0gTQEisAgDpra2ln/961+MHz+e9evXyys0nnfeeaxduxalUsnSpUuJj49HkiTKysr63OqMGzcOtVpNZ2cn69at44EHHuDpp5/mmWeekSeDBgUF8cADDzB58mTeffddbr/9dt544w0SEhJISkoa3BPwM7lcLnl9pnfffZdp06bR3t7OZ599hlKp5OGHH8ZqtdLW1kZiYiK33nqrnCpDo9GwbNkyIiMjsdvtqNVqXnjhBcrLy+Vz7OvrO2BlHxkhXRiS2traMJlMzJ07l8jISG644QY2bdrEzp07iY+PZ/HixTz55JPyLVJXVxctLS20tLTQ2toqb3c6nfLaQ3q9Xs6S5+vri16vJysri4SEBGpqaoiIiJDba4a7wsJCPvvsMwDmzJnDq6++yvjx42lsbOTOO+/k008/5bHHHiMmJgadTkdpaSnR0dHcf//9bNy4kRkzZqBUKmlpaeEvf/kLH3/8MUlJScTGxnLLLbfIjboDQdRYhAGjVqsJCAjAy8uLiIgINBoNarWa0NBQvv/+e5qbm+XlZgGCgoL6PNdz0fv6+mKxWFizZg0zZ86ku7ubrq4uxowZw969e7n22mvx8vJi9uzZfPjhh9jtdjkr33A2YcIExo8fLy9Yd8kllxAXF8fll18urxY5Y8YM7rrrLhYtWkRPTw/vv/8+EydOlG9/Jk2ahJeXF1lZWYSHh6NSqbBarVgsFnbv3k16ejqxsbH9XnaR6Ok4L7zwAgkJCVx22WX9+rqjkSddpk6no6enBz8/P7q7u/Hx8UGv11NXV8fkyZNPaxmQzs5OCgoKmD59Omq1GkmS0Gg0dHd3y+sSWSwW9u3bR0ZGBhEREcN+GIHT6aStrY329naCg4PllS1/yGw2s2fPHhYuXCjf8iiVSnbu3EldXR0XX3wxQUFBHD16lM8++4ybb76ZLVu2EBsby7vvvsujjz4qZzT8KSLRkzAk/HCdaU93p+ff2NjYM/qmDA4OZuHChSds/+HsW51Ox/z5839OkQedZ6G4o0eP0tvbS0JCAllZWSeMOLbZbHzwwQfk5ubS0dGBJEk89dRTXHfddaSmpjJnzhxsNpvcQG6xWJg6dSrBwcFceeWVwLHazkARgUUQhgDP6ppNTU04HA4iIyPx8/PDx8fnhH0tFgv/9V//xaWXXsq4ceMYP348a9euJS0tDUmSKCoqYvz48X1G1Xq6nc9VLU4EFkEYRJIk0dnZSWVlJTqdjsTERHx9fTl06BDbt2/nzjvvPCEYqFQq/vznPxMYGCivce2Z6tDS0nLCuteSJPH8889zxRVXEB8ff06OSwQWQRgEkiTJSZU0Gg0ZGRl9buuys7NJT08/aQ1j//79WCwWFi9eLG/z1E7i4uJO+n433HBDn8bxgSYCiyCcQy6Xi/r6etra2ggLCyM9Pf2kSZW8vLxOqHl4zJo164zeU6FQEBoaetZlPhsisAjCOeB0OmlubqarqwtfX1/Gjx/fZ+TxmRgOvV0isAjCAPE0yNbU1OBwOIiOjiY7O3tYBIafa9QEFkmS6OnpwdvbW656eibGiXwsQn+SJIne3l4aGxvp7e0lOjp6RIyrOROjJrA0NjbS09NDVVUVF1xwAQqFgqqqKgoKCsjOzu6zCLsgnA3PshktLS2oVCri4uLQ6XSjKqB4jJrAEhsbS2trK4cOHZK3BQUFkZmZSXt7O5IkceTIEY4ePSqvmSwIp8PtdtPe3k5jYyMKhYK0tLSTjj8ZTUZNYGltbWX79u3k5uZiNpspKSkhLS0NlUpFT08PABkZGfIgI0H4KZ4GWc+yGbm5uSIj4b+NmsCi0WgYP348Go0GrVZLSkoKcGyy29y5c8XFIJwWSZJwOBw0NjbS3NwsN8iOlNwv/WXUBJbAwEACAwPl3z0Dik6WXEgQjudZx7ilpYXe3l5iYmJISEgQAeUUzuisOJ1O/vu//5ukpCR0Oh3Jyck8+uijuN1ueR9JknjkkUfkHBELFiygpKSkz+vYbDbuuecewsLC8PX1ZenSpTQ0NPTPEQlCP5IkCZPJxMGDBzl48CBBQUFkZWURHBwsgsqPOKMz8/e//53XXnuNl156idLSUp566imefvppXnzxRXmfp556imeeeYaXXnqJ/fv3ExUVxZIlS/ok3lm+fDnr1q1j7dq17Nixg+7ubi655JKzWnxaEAaCZ8h9WVkZBoOBsWPHMn36dIKDg8Vt8+mQzsDFF18s3XzzzX22XXHFFdL1118vSZIkud1uKSoqSnryySflx61WqxQYGCi99tprkiRJUldXl6RWq6W1a9fK+zQ2NkpKpVLasGHDaZXDaDRKgGQ0Gvtsd7vdksvlOpNDOsHzzz8vffrppz/rNYThy+l0Sk1NTdL27dulw4cPS1ardbCLNGSc6nN3MmfUxjJnzhxee+01ysvLSU9Pp6ioiB07dvDcc88BUF1dTXNzM+eff778HK1Wy/z589m1axe33XYbeXl5OByOPvvExMSQk5PDrl27uOCCC054X5vNJq8/A8cSznhYrVZ5acnW1lZ8fHzIzs5mzpw54ptFOG1Op5P29nba2toIDg6WE0oJZ+eMboUeeOABrrnmGjIyMlCr1UycOJHly5dzzTXXAMjruURGRvZ5XmRkpPxYc3OznEn8VPscb8WKFXLja2BgYJ8ZnO3t7fj6+jJhwgT8/Py46qqr+PDDD0WXsfCTpH8nTCotLWXPnj243W4yMzOJiYkRQeVnOqMay/vvv897773H6tWryc7OprCwkOXLlxMTE8ONN94o73d8TUH697D5H/Nj+zz00EPcd9998u8mk0kOLn5+fgQEBJCdnU1WVhZWq5Urr7xS1FaEU5L+PYdHr9djs9mIiopi7NixojG2H51RYPnDH/7Agw8+yK9//WvgWFaq2tpaVqxYwY033khUVBSA3L/vYTAY5FpMVFQUdrudzs7OPrUWg8FwyungWq32lHlRAwMDWbp0qfy7r68v8+bNO5PDEkYJ6d8NsuXl5fJYJj8/P/ElNADOKET39vaeENW9vLzk7uakpCSioqLYvHmz/Ljdbmfr1q1y0Jg8eTJqtbrPPnq9nuLi4jPOM/FDLpeLhoYGXC5Xn+5vQfAElCNHjtDZ2Ul2djbjx4/H399fBJUBckY1lksvvZS//e1vxMfHk52dTUFBAc888ww333wzcOwWaPny5TzxxBOkpaWRlpbGE088gY+PD9deey1wrIZxyy23cP/99xMaGkpISAi///3vyc3N7ZMR60zY7Xaee+45WlpauP/++9mwYQO//e1vxUUzynmSKjU1NREVFUVqaioqlUpcF+fAGQWWF198kT/96U/ceeedGAwGYmJiuO222/jzn/8s7/PHP/4Ri8XCnXfeSWdnJ9OnT2fTpk190u49++yzqFQqrr76aiwWC4sWLeLtt98+Zcasn+J0OpEkiZycHHmRcGH0cjqdGAwGOjs7CQgIYMqUKScsmyEMrBGxrpAkSRw8eJDy8nJUKhXz5s0761R8Yl2h4cnTIFtZWYndbic2Npbw8HBRO+lHZ7Ku0IhoBq+qquKdd94hNzeXhIQECgoKBrtIwjki/TupUkVFBdXV1URERDBx4sRRl1hpqBkRkxCtVivR0dF0dHQAxzKcCyOb9O+kSg0NDXh7exMXF4ePj48IJkPEiAgsGRkZ5Ofns2PHDnnQU1RUlLjIRiC3201nZyfNzc1yUiVfX9/BLpZwnBERWHp7e2lqaiI5OZns7Gw+/PBDFixYMNjFEvqR0+mksbGRrq4ugoODGTt27EmXzRCGhhERWHx8fJg6dSoxMTHs2bOHmTNnnrCPzWajqKgIpVLJpEmTUCqVGAwGKioqiI6OJikpaRBKLvwUh8NBU1MT7e3tBAUFkZOTc9a9h8K5MyIab+12O3q9nvT0dH75y1/2mbDoUVhYCIDRaMRgMACQn59PamoqW7ZswWKxUFlZSW1trfgWHGTSv1dUqKiooLy8HH9/fyZOnEhycrIIKsPEiAgsbreb8vJyTCYTNTU1J+0VcjgcqNVqvL29cTqd8uhcnU6Hw+FAkiTMZjPd3d3nuvjCv0mSRHd3N6WlpRQXFxMcHExWVhYhISEi2A8zw/5WSJIkfHx8+OUvf8nHH38MwK233nrCfpMmTWLv3r0olUqCg4M5evQomZmZ7N27l/nz56PT6ZgwYQLZ2dliZvQ5JkkSHR0dGAwGVCoVSUlJeHt7i2AyjA37wGKxWNi7dy8dHR2oVMcOx2w295kECcfaYc477zz597FjxwKQkJBw7gor9OF2u2ltbaW6uho/Pz9SU1PlXMTC8DbsA4tKpcLf3x+dTidvCwoKGrwCCT/J4XDQ1tZGR0cHwcHBTJkyRf5SEEaGYf/XVCgUBAYG4nQ6qa2tRa1Wi3khQ5AkSdjtdurq6tDr9SQnJ5OZmSlyoIxQwz6wqNVqEhISeOWVV2hsbESpVNLS0sI111wj7tGHAOkHy2Z4RkinpKSIgDLCDfvAAv+3It19991He3s769evx2KxjNp1c4cC6d/LZpSXl6NUKklNTSUgIED8PUaJERFY1Go12dnZfPPNN7jdbqKioti+fTuLFy8W4x7OMbfbjdlspqmpCa1WS05OjujhGYVGRH3Uy8sLPz8/ysvLWbJkCVFRUZx//vkiqJxDnqRKO3bsoLW1lZSUFHlhOxFURp8REVh6e3vJy8sjMzMTq9XKd999J8ainCNOpxO9Xs+RI0dQqVTMmDGD1NRUNBqNCCij2Ii4FfLx8WHy5Mls376dmpoarrrqKnFRDyDPDPKjR4/S29tLYmKi6OER+hj2gUWSJIqKimhvb+fSSy8lPz9fzHodIJ4sbY2NjTidTqKjowkNDRXnWjjBsP+KaW9v58UXX2TOnDm88847XHnllUyYMGGwizWieJIq7d+/n6qqKnkdnrCwMBFUhJMa9oHFc4+/atUqHA4Hb731Fvn5+YNdrBHB7XbT0dFBaWkpHR0dZGZmkp2dLdbiEX7SsL8VCg8PZ+XKlX0aawMDAwexRMOfy+Wirq6O9vZ2wsLCSE9PF7eXwhkZ9oHFy8tLXoFR+HkcDgfNzc10dXXh7+/PhAkTxBwe4ayIq2aU82S5r6urkxtkx4wZI3p4hJ9FBJZRypOlrbGxEYvFwpgxY0RjrNBvRk1gcblcdHZ24uPjg4+PD3AspWVXVxc6na7PSo0jmSRJtLW1YTAYUKvVxMXFidGxQr8bNYGlpaWFtrY22tvbmT9/PkqlkoqKCmpra8nOzh7xgcXtdtPW1kZjYyMqlYrU1NQ+OWwEoT+N+MDS09NDU1MTsbGx+Pr6Yrfb5cciIyNpbW2lpqaGuLg4jhw5wtGjR0lMTBy8Avczh8NBS0sLRqORwMBAcnNzRYOsMOBG/BWmVCrRarU0NzdjMBiYPHkydrud+vp6/P39GTt2rJzBPyMjg9TU1GE/z0iSJBwOBw0NDTQ3NzNmzBgx5F44p0Z8YNHpdMTHx+NyuYiPj8fLywsvLy+SkpKQJInOzk4WLVoktzEM57YGTw+PwWDAYrEQHR1NYmKiCCjCOTfiA4vHD1MoKBQK+XYgIiJisIrUbzxLl1RWVuJwOEhPTycxMXFYB0lheBs1gWUkkiSJrq4uWlpaUKvVZGZmotVqRUARBp0ILMOQy+WiubmZyspKIiIiSE5ORq1Wi4AiDBkisAwjP1w2IzQ0lJkzZ6JWqwe7WIJwAhFYhjjPshmVlZV0dnaKZTOEYUEEliHKk1SpqakJh8NBTEwMGRkZIqAIw4IILEOMp0G2rKwMHx8fkpKSRP4TYdgRgWWIcLvdGI1G9Ho9Op2OnJwc/Pz8BrtYgnBWRGAZZJ5lM/R6PVFRUaSlpaFSqUQNRRjWRGAZJJ7VGz1zeKZMmSJ6eIQRQwSWc8izjnFFRQUul4vY2FjRwyOMSCKwnAOegNLQ0IDD4SAqKorw8HBxuyOMWCKwDCBPUqWGhgZ8fHxEUiVh1BCBZQC43W7a29tpbm7Gy8uLsWPHylnrBGE0EIGlHzmdThoaGjCZTAQHB5OZmSmSKgmj0qi+6n+Y0Olsb08kScLpdNLY2Eh7ezshISFkZ2f3SdMgCKPNqAksNpuNkpISfH19SU9PR6FQ0N7eTm1tLYGBgaSkpJxRcPFkuW9ubsZmsxEdHU1CQoJoPxEERsASqz/F7XZjt9vp7u6moaEBi8Ui11T27dtHYmIi3333HVarVU5f+WPBwZNUqaSkhNLSUkJCQsjKyiIkJEQEFUH4txFfYzEajRQXF5OVlcX555/Phx9+SFRUFFqtFi8vL9RqtdwOotVq0Wq1J81561kYva2tTc5yL5IqCcLJjfjAEhwczNy5c6mtraWoqIgJEybg7e1NUVERM2fOpLCwkAULFuDt7Y1Opzth3We3243BYKCmpgZ/f39SU1PRaDQioAjCjxjxgcUjISGBhIQE+ff58+cDMG/evJPub7fbaW9vp6Ojg5CQEKZMmSJ6eAThNIlPynEUCgW7du2ioaGB5ORkoqOjMRgMGAyGwS6aIJxznnFYZzqPTQSW41xxxRXs27cPOFZrqa2tPWGfjRs3kpOTw5gxY8518c6KJEnybeBwkZ+fj5eXF+PHjx/sopwWq9XKxo0bWbp06bC5TW5sbMTpdPapyR9PrVaTkpJyxoFFIQ3D1blMJhOBgYEYjUYCAgLO+ft7klgPl2VZJUmipaWFqKiowS7KadPr9SiVSiIjIwe7KKfF6XRSVVVFWlrasAksZrMZt9t9QrviqZzJ507UWE6TJEls3rwZjUbDpEmT8Pf3x+VyUVhYSHt7OzNnzhxygaa5uZnS0lK8vLyYPXs2ADU1NZSVlREbG0t2dvYgl/D/SJJEdXW1HLTHjRuHJEns2LEDl8vFhAkTTvsDcK44nU727dtHV1cXc+bMIT09nc7OTvbt20dUVBTjxo0bckGmtraW0tJSYmNjycnJQZIkysrKqK+vJysri5iYmH55nxE/jqU/SJKEJEk0NTUBx5IzwbF1oe12O6mpqdTX1w9mEfvwlDcyMpLp06fT2toqd6EbjUZsNtuQC4IAeXl5TJ06lcOHD+N0OrHb7fT29pKZmUl+fv6QW/rWy8uL6dOno1QqMRqNwLGBmFarFavVOsilOzmTyXTC37++vp6ZM2dy9OjRfnsfEVhOw5EjR6itrWXZsmUkJSVRXl6Ow+FAoVDgcrkwmUxDKqeK0Wjk+++/p6mpiffff1+unlssFlJTU5k5cyaVlZWDXcwTqNVqzGYzcGypE6fTSW9vL729vfj6+g5y6U7kdrspKCggLi6OqKgobDYbgYGBzJo1i7KyMnp6ega7iCdISUlh1qxZVFZW4nK5cLvdAP1+DYtbodOQmZkpVxl7enrIzs6mu7sbf39/oqKiMBqNJCcnD3YxZUFBQZx33nm0traSnJyMUqnE4XCg1+tRqVQYDAYyMzMHu5h9KBQKZs2aRU1NDTNnzqS3txc4du47OzuH5G2FJEm4XC56e3vp7u7GarXidrvR6/VMnDhxSAbD9vZ2WlpayMzMxGq1olQqycrKoqGhgXHjxvXb+4jGW2HQHX8JDrUAIhwjGm+FQbdlyxbMZjNeXl4sXLjwR7+9HQ4H3333Hf7+/lgsFhYtWnTKfYuKiggJCSEuLu6kjzc2NpKXl4fb7WbatGn91hgpnBkRWIQBsWrVKu69916++OILvL29iYqKorCwkIiICGbNmsUHH3yASqUiKCiI888/n5aWFry9vbHb7VRUVLBhwwaCgoJYunQpn332GUajkauuuorXXnsNPz8/brnlFvR6PfX19WRmZjJlyhQUCoXcxlFRUcEnn3zC3XffPdinYlQaOi2OwojidDr59ttvaW5uJiMjgyeffJK0tDQOHDhAUVER27dvZ9myZRw5cgS9Xg8cG7tSU1PDyy+/zOWXX86yZctwOp2YzWZaWlrYuHEjSUlJLFq0CJvNxocffkhmZiarV6+WG319fX1Rq9V89dVXnH/++YN5CkY1UWMRBoRarebSSy+luLiYDRs2oFarOXr0KOPGjSMxMRGn00lbWxtGo/GE3ghvb28aGhrw8vJi+/bt6HQ6pk6dSltbGz4+PrS2thIREYHRaKShoYGLL74Yb29v4NjYnTfeeIMZM2YQEhIyGIcuIBpvhQFSWFhIeno6SqWSyspKwsPDKS4uRqvVkpOTw9NPP82iRYvw9/dnwoQJ8khbp9NJYGAgBQUFBAYGkpCQwMGDB/H39ycmJgZfX1/y8/MZO3YsRqORlpYWQkJCyM3NRaFQ0NLSQlFRERqNhsjIyCHX+zWcncnnTgQW4ZxzOp0cOnSICRMmiB6gYUT0CglDmkqlYuLEiYNdDGEADcvA4qlkmUymQS6JIIwens/b6dzkDMvA4ukBONVYBkEQBo7ZbP7JCaHDso3F7XZTVlZGVlYW9fX1op3lFEwmE3FxceIc/QRxnk6PJ5F8TEzMT84rGpY1FqVSKSdZCggIEBfDTxDn6PSI8/TTTjd1hRggJwhCvxOBRRCEfjdsA4tWq+Uvf/kLWq12sIsyZIlzdHrEeep/w7LxVhCEoW3Y1lgEQRi6RGARBKHficAiCEK/E4FFEIR+JwKLIAj9blgGlldeeYWkpCS8vb2ZPHky27dvH+winTMrVqxg6tSp+Pv7ExERwbJlyygrK+uzjyRJPPLII8TExKDT6ViwYAElJSV99rHZbNxzzz2EhYXh6+vL0qVLaWhoOJeHck6tWLEChULB8uXL5W3iPA0gaZhZu3atpFarpTfeeEM6fPiwdO+990q+vr5SbW3tYBftnLjggguklStXSsXFxVJhYaF08cUXS/Hx8VJ3d7e8z5NPPin5+/tLH3/8sXTo0CHpV7/6lRQdHS2ZTCZ5n9tvv10aM2aMtHnzZik/P18677zzpPHjx0tOp3MwDmtA7du3T0pMTJTGjRsn3XvvvfJ2cZ4GzrALLNOmTZNuv/32PtsyMjKkBx98cJBKNLgMBoMESFu3bpUkSZLcbrcUFRUlPfnkk/I+VqtVCgwMlF577TVJkiSpq6tLUqvV0tq1a+V9GhsbJaVSKW3YsOHcHsAAM5vNUlpamrR582Zp/vz5cmAR52lgDatbIbvdTl5e3glJks8//3x27do1SKUaXJ6lPT35Xaurq2lubu5zjrRaLfPnz5fPUV5eHg6Ho88+MTEx5OTkjLjzeNddd3HxxRezePHiPtvFeRpYw2p2c1tbGy6Xi8jIyD7bIyMjaW5uHqRSDR5JkrjvvvuYM2cOOTk5APJ5ONk5qq2tlffRaDQEBwefsM9IOo9r164lPz+f/fv3n/CYOE8Da1gFFo/j86RKkjQqc6fefffdHDx4kB07dpzw2Nmco5F0Huvr67n33nvZtGmTnMH/ZEb7eRoow+pWKCwsDC8vrxO+LQwGwwnfPCPdPffcw+eff853331HbGysvD0qKgrgR89RVFQUdrudzs7OU+4z3OXl5WEwGJg8eTIqlQqVSsXWrVt54YUXUKlU8nGO9vM0UIZVYNFoNEyePJnNmzf32b5582ZmzZo1SKU6tyRJ4u677+aTTz7h22+/JSkpqc/jSUlJREVF9TlHdrudrVu3yudo8uTJqNXqPvvo9XqKi4tHzHlctGgRhw4dorCwUP6ZMmUK1113HYWFhSQnJ4vzNJAGseH4rHi6m9966y3p8OHD0vLlyyVfX1+ppqZmsIt2Ttxxxx1SYGCg9P3330t6vV7+6e3tlfd58sknpcDAQOmTTz6RDh06JF1zzTUn7UaNjY2VvvnmGyk/P19auHDhiO9G/WGvkCSJ8zSQhl1gkSRJevnll6WEhARJo9FIkyZNkrtaRwPgpD8rV66U93G73dJf/vIXKSoqStJqtdK8efOkQ4cO9Xkdi8Ui3X333VJISIik0+mkSy65RKqrqzvHR3NuHR9YxHkaOCIfiyAI/W5YtbEIgjA8iMAiCEK/E4FFEIR+JwKLIAj9TgQWQRD6nQgsgiD0OxFYBEHodyKwCILQ70RgEQSh34nAIghCvxOBRRCEfvf/AaWqWmZ4JeGrAAAAAElFTkSuQmCC",
            "text/plain": [
              "<Figure size 640x480 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "rna_pairwise_no_thresh = cv2.imread(f'{results_dir}/experimentMPRAworkshop/statistic/assigned_counts/MPRAworkshop/default/HepG2_RNA_pairwise_minThreshold.png')\n",
        "\n",
        "rna_pairwise_no_thresh_resize = cv2.resize(rna_pairwise_no_thresh, (600, 1000))\n",
        "\n",
        "plt.imshow(rna_pairwise_no_thresh_resize)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Even though it is a small example, the minimal barcode count threshold shows a clear effect on correlation. However, we also note that we are losing a proportion of oligos with these strict filters and thereby limit the conclusions that we can draw from the data."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Now we want to go through the final output of MPRAsnakeflow: the count table of sequences and barcodes together with their aggregated log2 values. Note that there are no statistics calculated at this step yet, the table doesn't say anything about certainty/variance of the activity. The log2 value refers to the log2 ratio of RNA counts over DNA counts for each oligo, which is a measurement of the activity of the target sequence. More on this in the statistical analysis tutorial."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "replicate\toligo_name\tdna_counts\trna_counts\tdna_normalized\trna_normalized\tlog2FoldChange\tn_bc\n",
            "1\toligo_002500\t23\t41\t47.3898\t30.0382\t-0.6578\t8\n",
            "1\toligo_002501:oligo_002502\t12\t26\t28.2573\t21.7699\t-0.3763\t7\n",
            "1\toligo_002503\t14\t23\t32.9669\t19.2579\t-0.7756\t7\n",
            "1\toligo_002504\t1\t2\t16.4834\t11.7222\t-0.4918\t1\n",
            "1\toligo_002505\t24\t41\t49.4503\t30.0382\t-0.7192\t8\n",
            "1\toligo_002506\t7\t15\t28.846\t21.9792\t-0.3922\t4\n",
            "1\toligo_002507\t11\t21\t30.2196\t20.5139\t-0.5589\t6\n",
            "1\toligo_002508\t11\t29\t36.2635\t33.9945\t-0.0932\t5\n",
            "1\toligo_002510\t15\t31\t49.4503\t36.3389\t-0.4445\t5\n"
          ]
        }
      ],
      "source": [
        "!zcat {results_dir}/experimentMPRAworkshop/assigned_counts/MPRAworkshop/tutorialConfig/HepG2_allreps_merged.tsv.gz | awk 'NR <= 10 {{print}}'"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### QC report\n",
        "\n",
        "We have a QC report for the assignment as well as for the experiment workflow. For our run you can find it here:\n",
        "\n",
        "- [MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html](MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html) - 10 BC threshold\n",
        "\n",
        "- [MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html](MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html) - 5 BC threshold\n",
        "\n",
        "The QC report gives a nice overview of the experiment run and might be the first thing you will look at. Because we have two configurations you can compare both."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "\n",
              "        <iframe\n",
              "            width=\"700\"\n",
              "            height=\"600\"\n",
              "            src=\"./MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html\"\n",
              "            frameborder=\"0\"\n",
              "            allowfullscreen\n",
              "            \n",
              "        ></iframe>\n",
              "        "
            ],
            "text/plain": [
              "<IPython.lib.display.IFrame at 0x7f2e12d2ab10>"
            ]
          },
          "execution_count": 22,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "from IPython.display import IFrame\n",
        "\n",
        "IFrame(src='./MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.default.html', width=700, height=600)\n",
        "\n",
        "IFrame(src='./MPRAsnakeflow_tutorial/results/experiments/experimentMPRAworkshop/qc_report.HepG2.MPRAworkshop.tutorialConfig.html', width=700, height=600)"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [
        "32PHp-vU99x0",
        "6GilAsqZ4t9r"
      ],
      "include_colab_link": true,
      "provenance": []
    },
    "kernelspec": {
      "display_name": "jupyter",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.12.4"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
